{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 285,
   "id": "a5af7905-9b65-4d41-b747-0c795477d895",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "### Check out correlations between ligand, receptors and targets and combine with nichenet data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7ad3b8fc-2638-401d-8299-974e2612bdfd",
   "metadata": {
    "tags": []
   },
   "source": [
    "#############################################\n",
    "# Prerequisites - Load Libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 286,
   "id": "8bc0aa68-a56c-4954-b7f1-c940eed22535",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in read.table(file = file, header = header, sep = sep, quote = quote, :\n",
      "“incomplete final line found by readTableHeader on '../conda_environment/Environment_Configs.csv'”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/home/icb/corinna.losert/miniconda3/envs/stark_stemi_R_Env_4_1//lib/R/library\"\n"
     ]
    }
   ],
   "source": [
    "source('MS0_Libraries.r')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 287,
   "id": "a1fe13f0-3991-4bcf-8a72-1e57f5de3639",
   "metadata": {},
   "outputs": [],
   "source": [
    "source('MS4_Plot_Config.r')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c0d8f19c-38da-490a-904c-ead6cbd8fada",
   "metadata": {
    "tags": []
   },
   "source": [
    "###############################################\n",
    "# Preqrequisites Configurations & Parameters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "id": "32a08c89-f24e-4022-bcf8-70120b317e3b",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_path = '../data/current'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "id": "41b26d19-d26f-4917-bf53-88d39efe056c",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_path = '../results/current'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "id": "61a059d8-eb1d-4224-84dd-a61292c1b6c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'../data/current'"
      ],
      "text/latex": [
       "'../data/current'"
      ],
      "text/markdown": [
       "'../data/current'"
      ],
      "text/plain": [
       "[1] \"../data/current\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "data_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 291,
   "id": "434926b9-50d3-42e5-8d6c-2e98586db31b",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Version of the pre-processed data to load"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "id": "55ae9b4e-d653-4f50-82ce-c45e566e4bf3",
   "metadata": {},
   "outputs": [],
   "source": [
    "name = 'V_FINAL'  \n",
    "# of dataset to load, prepared and combined data of E1 - skip E2 step "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "id": "98f9d1fb-733a-41a7-8432-96ca94d08820",
   "metadata": {},
   "outputs": [],
   "source": [
    "filter_samples = 'all'  # decide whether to select only a subset of samples; 'all'  = not filtering\n",
    "#filter_samples = 'acs_w_o_infection_ccs'\n",
    "#filter_samples = 'no_ccs_ccs'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4d78b773-c356-43bd-a1b7-988fd621712b",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Load Data "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "88d6ff01-805b-4e59-bfbc-122177418d3c",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Sample Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 294,
   "id": "d3c14685-c0dd-444c-ab8a-dccd9cfbddbc",
   "metadata": {},
   "outputs": [],
   "source": [
    "## sample meta and clinical data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "id": "b5d4992b-aefd-4b17-8114-bc765b0700f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_data = read.csv(paste0(result_path, '/00_Data_Overview/Merged_Sample_Meta_Data.csv'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "248e225a-e54f-4535-b981-a3e7955c616c",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Prepared combined data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "id": "f9d4e05a-0aa9-444b-887e-484ba2d665af",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Select and load pre-processed and combined data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "id": "ae3feb85-f2b2-4b90-a67c-151d3cf87723",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"../results/current/E-Analysis/Combined_Data_V_FINAL.csv\"\n",
      "[1] \"2024-01-03 14:22:52 CET\"\n"
     ]
    }
   ],
   "source": [
    "path = paste0(result_path, '/E-Analysis/Combined_Data_',name,'.csv')\n",
    "print(path)\n",
    "data_long = read.csv(path)\n",
    "print(file.info(path)$mtime)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 298,
   "id": "126a3bfb-1256-4c6b-b778-aeb0c6f1a343",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>sample_id</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>value</th><th scope=col>config</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>k10</td><td>single_cell</td><td>X0_T.cell.CD4__SSU72</td><td>0.5896835</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>2</td><td>k10</td><td>single_cell</td><td>X0_T.cell.CD4__PARK7</td><td>0.8587511</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & X & sample\\_id & type & variable & value & config\\\\\n",
       "  & <int> & <chr> & <chr> & <chr> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & k10 & single\\_cell & X0\\_T.cell.CD4\\_\\_SSU72 & 0.5896835 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\\\\\n",
       "\t2 & 2 & k10 & single\\_cell & X0\\_T.cell.CD4\\_\\_PARK7 & 0.8587511 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 6\n",
       "\n",
       "| <!--/--> | X &lt;int&gt; | sample_id &lt;chr&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | k10 | single_cell | X0_T.cell.CD4__SSU72 | 0.5896835 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- |\n",
       "| 2 | 2 | k10 | single_cell | X0_T.cell.CD4__PARK7 | 0.8587511 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- |\n",
       "\n"
      ],
      "text/plain": [
       "  X sample_id type        variable             value    \n",
       "1 1 k10       single_cell X0_T.cell.CD4__SSU72 0.5896835\n",
       "2 2 k10       single_cell X0_T.cell.CD4__PARK7 0.8587511\n",
       "  config                          \n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE-"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 299,
   "id": "5fc8233d-b29a-41cc-b9e7-06ffddb55a22",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1557939"
      ],
      "text/latex": [
       "1557939"
      ],
      "text/markdown": [
       "1557939"
      ],
      "text/plain": [
       "[1] 1557939"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(data_long)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 300,
   "id": "db789d09-2390-4ab9-aba9-d0e2730752dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "13289"
      ],
      "text/latex": [
       "13289"
      ],
      "text/markdown": [
       "13289"
      ],
      "text/plain": [
       "[1] 13289"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique(data_long$variable))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0a077569-b065-469d-9d21-df4d73ecf422",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Ligand Receptor Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "id": "7c7fb68c-e0da-4d4a-b15a-cf9c6c97e7b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Load ligand receptor interactions from nichenet "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "id": "9699012a-a3f6-4e2b-86a1-0c98b182a8e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "lr_network = readRDS(url(\"https://zenodo.org/record/3260758/files/lr_network.rds\")) # ligand receptor interactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "id": "0439aa90-4c14-4714-81de-86753c5ba3cf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 2 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>from</th><th scope=col>to</th><th scope=col>source</th><th scope=col>database</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>CXCL1</td><td>CXCR2</td><td>kegg_cytokines</td><td>kegg</td></tr>\n",
       "\t<tr><td>CXCL2</td><td>CXCR2</td><td>kegg_cytokines</td><td>kegg</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 2 × 4\n",
       "\\begin{tabular}{llll}\n",
       " from & to & source & database\\\\\n",
       " <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t CXCL1 & CXCR2 & kegg\\_cytokines & kegg\\\\\n",
       "\t CXCL2 & CXCR2 & kegg\\_cytokines & kegg\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 2 × 4\n",
       "\n",
       "| from &lt;chr&gt; | to &lt;chr&gt; | source &lt;chr&gt; | database &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| CXCL1 | CXCR2 | kegg_cytokines | kegg |\n",
       "| CXCL2 | CXCR2 | kegg_cytokines | kegg |\n",
       "\n"
      ],
      "text/plain": [
       "  from  to    source         database\n",
       "1 CXCL1 CXCR2 kegg_cytokines kegg    \n",
       "2 CXCL2 CXCR2 kegg_cytokines kegg    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(lr_network,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "id": "76be70e6-58f7-4229-9096-1b6e36acccf7",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Summarise to have only ligand or receptor assignment not double"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 305,
   "id": "9372b74d-318f-4838-867d-328b7941248a",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_receptor_interactions = lr_network %>% group_by(from,to) %>% count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 306,
   "id": "33767995-6fba-4c90-9e4f-4bb8b237c489",
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames(ligand_receptor_interactions) = c('Ligand', 'Receptor', 'amount_sources')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 307,
   "id": "f6174978-a76f-40d8-9c71-2698e688f7b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Ligand</th><th scope=col>Receptor</th><th scope=col>amount_sources</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>A2M</td><td>LRP1    </td><td>1</td></tr>\n",
       "\t<tr><td>A2M</td><td>TNFRSF14</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 2 × 3\n",
       "\\begin{tabular}{lll}\n",
       " Ligand & Receptor & amount\\_sources\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t A2M & LRP1     & 1\\\\\n",
       "\t A2M & TNFRSF14 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 2 × 3\n",
       "\n",
       "| Ligand &lt;chr&gt; | Receptor &lt;chr&gt; | amount_sources &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| A2M | LRP1     | 1 |\n",
       "| A2M | TNFRSF14 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand Receptor amount_sources\n",
       "1 A2M    LRP1     1             \n",
       "2 A2M    TNFRSF14 1             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_receptor_interactions,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 308,
   "id": "9eb4c6a2-e588-408f-b6fc-0f7c86abac85",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Aggregate to contain only unique ligand-receptor interactions across different data sources"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "id": "e3ffd5fa-677c-476d-9082-babe75735e35",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_amounts = ligand_receptor_interactions %>% group_by(Ligand) %>% summarise(amount_l = sum(amount_sources))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 310,
   "id": "a7aa8dff-aaae-4410-bef7-2aaae9366e09",
   "metadata": {},
   "outputs": [],
   "source": [
    "receptor_amounts = ligand_receptor_interactions %>% group_by(Receptor) %>% summarise(amount_r = sum(amount_sources))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "id": "4ea1be21-be31-4f73-b456-37169478be6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "l_r_merged = merge(ligand_amounts, receptor_amounts, by.x = 'Ligand', by.y = 'Receptor')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 312,
   "id": "8a79dc18-b5dd-4e67-aa55-b6136397c473",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand = l_r_merged$Ligand[l_r_merged$amount_l >= l_r_merged$amount_r]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 313,
   "id": "2e77c390-00da-4e46-bbcd-7250f7f686b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "receptor = l_r_merged$Ligand[l_r_merged$amount_l < l_r_merged$amount_r]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "id": "379fd38a-a70e-44db-a833-73b96112fb9f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "115"
      ],
      "text/latex": [
       "115"
      ],
      "text/markdown": [
       "115"
      ],
      "text/plain": [
       "[1] 115"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(l_r_merged)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 315,
   "id": "07eeea0f-f1d5-4888-9262-7e8328f6db06",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Ligand</th><th scope=col>amount_l</th><th scope=col>amount_r</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ALCAM </td><td>3</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AMIGO1</td><td>2</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & Ligand & amount\\_l & amount\\_r\\\\\n",
       "  & <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & ALCAM  & 3 & 2\\\\\n",
       "\t2 & AMIGO1 & 2 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 3\n",
       "\n",
       "| <!--/--> | Ligand &lt;chr&gt; | amount_l &lt;int&gt; | amount_r &lt;int&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | ALCAM  | 3 | 2 |\n",
       "| 2 | AMIGO1 | 2 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand amount_l amount_r\n",
       "1 ALCAM  3        2       \n",
       "2 AMIGO1 2        1       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(l_r_merged,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "id": "677019a5-f8a9-4b83-9e27-2aeab665e012",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "84"
      ],
      "text/latex": [
       "84"
      ],
      "text/markdown": [
       "84"
      ],
      "text/plain": [
       "[1] 84"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(ligand)  # amount ligands"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 317,
   "id": "bb368b72-cce0-458a-a4e4-dd2e6224193b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "31"
      ],
      "text/latex": [
       "31"
      ],
      "text/markdown": [
       "31"
      ],
      "text/plain": [
       "[1] 31"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(receptor)  # amount receptors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 318,
   "id": "40ff416e-fea0-4356-a36a-b4c7eb8a5dd3",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Filter out and decide whether it is a receptor or ligand"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 319,
   "id": "cbdb333d-637b-42fe-b26d-5add4a7b28cb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Ligand</th><th scope=col>Receptor</th><th scope=col>amount_sources</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>A2M</td><td>LRP1    </td><td>1</td></tr>\n",
       "\t<tr><td>A2M</td><td>TNFRSF14</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 2 × 3\n",
       "\\begin{tabular}{lll}\n",
       " Ligand & Receptor & amount\\_sources\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t A2M & LRP1     & 1\\\\\n",
       "\t A2M & TNFRSF14 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 2 × 3\n",
       "\n",
       "| Ligand &lt;chr&gt; | Receptor &lt;chr&gt; | amount_sources &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| A2M | LRP1     | 1 |\n",
       "| A2M | TNFRSF14 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand Receptor amount_sources\n",
       "1 A2M    LRP1     1             \n",
       "2 A2M    TNFRSF14 1             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_receptor_interactions,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 320,
   "id": "3b097b9b-9cbe-4b26-85db-7cf5c6fb1c1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_receptor_interactions = ligand_receptor_interactions[!ligand_receptor_interactions$Receptor %in% ligand,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 321,
   "id": "8c6a6b28-81aa-4e80-85cf-9352224bc7d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_receptor_interactions = ligand_receptor_interactions[!ligand_receptor_interactions$Ligand %in% receptor,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 322,
   "id": "40e083b4-3b56-4061-970b-4d1588665a55",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "11639"
      ],
      "text/latex": [
       "11639"
      ],
      "text/markdown": [
       "11639"
      ],
      "text/plain": [
       "[1] 11639"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_receptor_interactions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 323,
   "id": "bbe47f2a-c4dc-4b6d-9b0a-695feae2a384",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Generate a dataset specifying ligand and receptor genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 324,
   "id": "8c36e108-4550-47f3-a239-bac3a03f6e34",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "ligand_receptor_vars = rbind(data.frame(gene = ligand_receptor_interactions$Ligand, interaction_type = 'Ligand'), data.frame(gene = ligand_receptor_interactions$Receptor, interaction_type = 'Receptor'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 325,
   "id": "56fdf875-b311-4d7e-97c9-dc1f7ebd52d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_receptor_vars = unique(ligand_receptor_vars)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 326,
   "id": "33fd1c26-aa15-4f7f-b93c-40e17576d4b8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>interaction_type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A2M  </td><td>Ligand</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>AANAT</td><td>Ligand</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & gene & interaction\\_type\\\\\n",
       "  & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & A2M   & Ligand\\\\\n",
       "\t4 & AANAT & Ligand\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 2\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | interaction_type &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| 1 | A2M   | Ligand |\n",
       "| 4 | AANAT | Ligand |\n",
       "\n"
      ],
      "text/plain": [
       "  gene  interaction_type\n",
       "1 A2M   Ligand          \n",
       "4 AANAT Ligand          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_receptor_vars,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 327,
   "id": "c9129614-bfe8-447b-8aff-fd43415ac148",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1414"
      ],
      "text/latex": [
       "1414"
      ],
      "text/markdown": [
       "1414"
      ],
      "text/plain": [
       "[1] 1414"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_receptor_vars)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 328,
   "id": "d91a51dd-0332-4e24-b8d6-d60b4d1bf653",
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove empty spaces\n",
    "ligand_receptor_vars$gene = str_replace(ligand_receptor_vars$gene, ' ', '')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f977280-9d12-4b5b-b3c9-4c715443bbac",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Available ligand receptor potential + transcription factor datasets from  nichenet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 329,
   "id": "e489e0dc-7a6b-48c0-a6d4-b87dd5456fdb",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Weighted networks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 330,
   "id": "b5b53376-5c56-43fd-85b9-694972d38e02",
   "metadata": {},
   "outputs": [],
   "source": [
    "weighted_networks = readRDS(url(\"https://zenodo.org/record/3260758/files/weighted_networks.rds\")) # integrated weighted ligand-signaling and gene regulatory network"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 331,
   "id": "98be6e73-1cc4-443f-834b-46408c899c8c",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Prepared influential matrices"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 332,
   "id": "218857a4-9abf-4155-b1d0-b220a9a84f6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_matrix = readRDS(url(\"https://zenodo.org/record/3260758/files/ligand_target_matrix.rds\"))  # gene-gene regulatory scores\n",
    "# gives the regulatory potential of a target gene based on the ligand-tf & tf- target-gene links and weights of the signaling and gr network (multiply PPR of each ligand -tf  pair with the weights of the gr network)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 333,
   "id": "fb475f82-0f33-43cb-a614-ea5428142561",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Generate data in long format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 334,
   "id": "fb089af0-96e9-469c-9e64-debf11e69b6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_matrix_long = melt(ligand_target_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 335,
   "id": "a27dddb7-4c9d-4f91-b8a9-522d1a2a9aac",
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames(ligand_target_matrix_long) = c('target', 'ligand', 'regulatory_potential')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 336,
   "id": "80e5fe18-967f-403e-a487-ea20378bc381",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>regulatory_potential</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A1BG    </td><td>CXCL1</td><td>0.0003534343</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A1BG-AS1</td><td>CXCL1</td><td>0.0001650894</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & target & ligand & regulatory\\_potential\\\\\n",
       "  & <fct> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & A1BG     & CXCL1 & 0.0003534343\\\\\n",
       "\t2 & A1BG-AS1 & CXCL1 & 0.0001650894\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 3\n",
       "\n",
       "| <!--/--> | target &lt;fct&gt; | ligand &lt;fct&gt; | regulatory_potential &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | A1BG     | CXCL1 | 0.0003534343 |\n",
       "| 2 | A1BG-AS1 | CXCL1 | 0.0001650894 |\n",
       "\n"
      ],
      "text/plain": [
       "  target   ligand regulatory_potential\n",
       "1 A1BG     CXCL1  0.0003534343        \n",
       "2 A1BG-AS1 CXCL1  0.0001650894        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_target_matrix_long,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 337,
   "id": "d871bfeb-4931-4970-b0c6-bf0808c4965a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "688"
      ],
      "text/latex": [
       "688"
      ],
      "text/markdown": [
       "688"
      ],
      "text/plain": [
       "[1] 688"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique(ligand_target_matrix_long$ligand))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e6ec2a5a-ee0f-498c-b2a3-217cbef10b92",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Cell Gene Expression Data (percentage of cells expressing gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 338,
   "id": "1ba1c9b9-bc3b-43ef-9480-0e73a4667e93",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Load information about how many cells express a gene generated in C0 Script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 339,
   "id": "66aff71b-4ad6-4196-a424-8d8ccf873e23",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"2024-01-03 14:10:30 CET\"\n"
     ]
    }
   ],
   "source": [
    "path = paste0(result_path, '/C-Analysis/C0_Filter_Genes_Input_Correlations_Perc_Values.csv')\n",
    "cell_perc_cluster = read.csv( path) # cluster alternative\n",
    "print(file.info(path)$mtime)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 340,
   "id": "637602d5-8dfa-4a03-a4ab-04b53cb37679",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Filter on relevant cell types (clusters with low amount of cells per sample; similar to MOFA analysis)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 341,
   "id": "e248ad84-cbca-4ca3-bf2f-8e4b2f894b1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "cell_perc_cluster = cell_perc_cluster[!(cell_perc_cluster$cluster %in% c('16_Plasma Blast', '14_Other', '17_Progenitor', '15_Plasma Blast', '18_Megakaryocytes' ,'Plasma Blast', 'Other', 'Progenitor', 'Plasma Blast', 'Megakaryocytes' )),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 342,
   "id": "6cb18288-4fb6-4865-8158-81ab077ad6a5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'8_B-cell'</li><li>'9_Monocytes - CD16_FCGR3A'</li><li>'5_T-cell-CD4'</li><li>'0_T-cell-CD4'</li><li>'1_T-cell-CD8'</li><li>'6_Monocytes - CD14'</li><li>'2_T-cell-CD4'</li><li>'4_Monocytes - CD14'</li><li>'11_T-cell-CD4'</li><li>'3_NK'</li><li>'7_Monocytes - CD14'</li><li>'13_Dendritic'</li><li>'10_B-cell'</li><li>'12_Monocytes - CD16_FCGR3A'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '8\\_B-cell'\n",
       "\\item '9\\_Monocytes - CD16\\_FCGR3A'\n",
       "\\item '5\\_T-cell-CD4'\n",
       "\\item '0\\_T-cell-CD4'\n",
       "\\item '1\\_T-cell-CD8'\n",
       "\\item '6\\_Monocytes - CD14'\n",
       "\\item '2\\_T-cell-CD4'\n",
       "\\item '4\\_Monocytes - CD14'\n",
       "\\item '11\\_T-cell-CD4'\n",
       "\\item '3\\_NK'\n",
       "\\item '7\\_Monocytes - CD14'\n",
       "\\item '13\\_Dendritic'\n",
       "\\item '10\\_B-cell'\n",
       "\\item '12\\_Monocytes - CD16\\_FCGR3A'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '8_B-cell'\n",
       "2. '9_Monocytes - CD16_FCGR3A'\n",
       "3. '5_T-cell-CD4'\n",
       "4. '0_T-cell-CD4'\n",
       "5. '1_T-cell-CD8'\n",
       "6. '6_Monocytes - CD14'\n",
       "7. '2_T-cell-CD4'\n",
       "8. '4_Monocytes - CD14'\n",
       "9. '11_T-cell-CD4'\n",
       "10. '3_NK'\n",
       "11. '7_Monocytes - CD14'\n",
       "12. '13_Dendritic'\n",
       "13. '10_B-cell'\n",
       "14. '12_Monocytes - CD16_FCGR3A'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"8_B-cell\"                   \"9_Monocytes - CD16_FCGR3A\" \n",
       " [3] \"5_T-cell-CD4\"               \"0_T-cell-CD4\"              \n",
       " [5] \"1_T-cell-CD8\"               \"6_Monocytes - CD14\"        \n",
       " [7] \"2_T-cell-CD4\"               \"4_Monocytes - CD14\"        \n",
       " [9] \"11_T-cell-CD4\"              \"3_NK\"                      \n",
       "[11] \"7_Monocytes - CD14\"         \"13_Dendritic\"              \n",
       "[13] \"10_B-cell\"                  \"12_Monocytes - CD16_FCGR3A\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(cell_perc_cluster$cluster)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 343,
   "id": "ab758f80-4568-4fc0-b6ab-8ba007523410",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th><th scope=col>gene</th><th scope=col>cluster</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AL627309.1</td><td>0.24327612</td><td>18</td><td>AL627309.1</td><td>8_B-cell</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AL627309.4</td><td>0.05406136</td><td> 4</td><td>AL627309.4</td><td>8_B-cell</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & X & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene & gene & cluster\\\\\n",
       "  & <chr> & <dbl> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & AL627309.1 & 0.24327612 & 18 & AL627309.1 & 8\\_B-cell\\\\\n",
       "\t2 & AL627309.4 & 0.05406136 &  4 & AL627309.4 & 8\\_B-cell\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 5\n",
       "\n",
       "| <!--/--> | X &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; | gene &lt;chr&gt; | cluster &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | AL627309.1 | 0.24327612 | 18 | AL627309.1 | 8_B-cell |\n",
       "| 2 | AL627309.4 | 0.05406136 |  4 | AL627309.4 | 8_B-cell |\n",
       "\n"
      ],
      "text/plain": [
       "  X          perc_cells_expressing_gene total_amount_cells_expressing_gene\n",
       "1 AL627309.1 0.24327612                 18                                \n",
       "2 AL627309.4 0.05406136                  4                                \n",
       "  gene       cluster \n",
       "1 AL627309.1 8_B-cell\n",
       "2 AL627309.4 8_B-cell"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cell_perc_cluster,2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "663e9dbc-c818-4c80-bc91-5757eb1da480",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Data Preparations"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f795ec1-31e8-4b45-815b-81fe4e0c0853",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Prepare list of cytokines in data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 344,
   "id": "9b4a14f3-2995-443e-9f9b-8d7be93fa2fe",
   "metadata": {},
   "outputs": [],
   "source": [
    "## determine all cytokines in input data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 345,
   "id": "057902db-f598-4038-b934-a1c665ac62d8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>sample_id</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>value</th><th scope=col>config</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>k10</td><td>single_cell</td><td>X0_T.cell.CD4__SSU72</td><td>0.5896835</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>2</td><td>k10</td><td>single_cell</td><td>X0_T.cell.CD4__PARK7</td><td>0.8587511</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & X & sample\\_id & type & variable & value & config\\\\\n",
       "  & <int> & <chr> & <chr> & <chr> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & k10 & single\\_cell & X0\\_T.cell.CD4\\_\\_SSU72 & 0.5896835 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\\\\\n",
       "\t2 & 2 & k10 & single\\_cell & X0\\_T.cell.CD4\\_\\_PARK7 & 0.8587511 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 6\n",
       "\n",
       "| <!--/--> | X &lt;int&gt; | sample_id &lt;chr&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | k10 | single_cell | X0_T.cell.CD4__SSU72 | 0.5896835 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- |\n",
       "| 2 | 2 | k10 | single_cell | X0_T.cell.CD4__PARK7 | 0.8587511 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- |\n",
       "\n"
      ],
      "text/plain": [
       "  X sample_id type        variable             value    \n",
       "1 1 k10       single_cell X0_T.cell.CD4__SSU72 0.5896835\n",
       "2 2 k10       single_cell X0_T.cell.CD4__PARK7 0.8587511\n",
       "  config                          \n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE-"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 346,
   "id": "15f87b86-6183-4aa3-94dd-dfc3cd2eef43",
   "metadata": {},
   "outputs": [],
   "source": [
    "cytokines = unique(data_long$variable[data_long$type == 'cytokine'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 347,
   "id": "9a0e272a-a40f-440a-ba1a-a2dd078806e4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'BCA1__CXCL13'</li><li>'CTACK__CCL27'</li><li>'EGF__EGF'</li><li>'ENA78__CXCL5'</li><li>'Eotaxin__CCL11'</li><li>'Eotaxin2__CCL24'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'BCA1\\_\\_CXCL13'\n",
       "\\item 'CTACK\\_\\_CCL27'\n",
       "\\item 'EGF\\_\\_EGF'\n",
       "\\item 'ENA78\\_\\_CXCL5'\n",
       "\\item 'Eotaxin\\_\\_CCL11'\n",
       "\\item 'Eotaxin2\\_\\_CCL24'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'BCA1__CXCL13'\n",
       "2. 'CTACK__CCL27'\n",
       "3. 'EGF__EGF'\n",
       "4. 'ENA78__CXCL5'\n",
       "5. 'Eotaxin__CCL11'\n",
       "6. 'Eotaxin2__CCL24'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"BCA1__CXCL13\"    \"CTACK__CCL27\"    \"EGF__EGF\"        \"ENA78__CXCL5\"   \n",
       "[5] \"Eotaxin__CCL11\"  \"Eotaxin2__CCL24\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cytokines)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 348,
   "id": "796d030c-fed8-49eb-ab76-8ffaeeeb1d94",
   "metadata": {},
   "outputs": [],
   "source": [
    "cytokines = str_replace(cytokines, '.*__', '') # extract gene code from cytokines"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 349,
   "id": "33a0a3b2-876e-4bd5-89a3-6161b21d414e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'CXCL13'</li><li>'CCL27'</li><li>'EGF'</li><li>'CXCL5'</li><li>'CCL11'</li><li>'CCL24'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'CXCL13'\n",
       "\\item 'CCL27'\n",
       "\\item 'EGF'\n",
       "\\item 'CXCL5'\n",
       "\\item 'CCL11'\n",
       "\\item 'CCL24'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'CXCL13'\n",
       "2. 'CCL27'\n",
       "3. 'EGF'\n",
       "4. 'CXCL5'\n",
       "5. 'CCL11'\n",
       "6. 'CCL24'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"CXCL13\" \"CCL27\"  \"EGF\"    \"CXCL5\"  \"CCL11\"  \"CCL24\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cytokines)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 350,
   "id": "63d66c8c-9338-4670-98b9-5c8e3da23694",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1236"
      ],
      "text/latex": [
       "1236"
      ],
      "text/markdown": [
       "1236"
      ],
      "text/plain": [
       "[1] 1236"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_receptor_interactions[ligand_receptor_interactions$Ligand %in% cytokines,])  # check how many ligand receptor interactions contain cytokine ligand"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "45e5085a-e03e-4220-b5ac-e24bb59ce434",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Adjustments on input data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a89f9812-3b9f-49a4-8377-967f37c2d635",
   "metadata": {},
   "source": [
    "### Filter samples (optional)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 351,
   "id": "c66b8840-471e-426a-9427-a97d3b6f28cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "## filter out some samples if specified in the beginning"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 352,
   "id": "c02dca7f-62b6-450f-948c-bbe80f51499e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>sample_id</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>value</th><th scope=col>config</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>k10</td><td>single_cell</td><td>X0_T.cell.CD4__SSU72</td><td>0.5896835</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>2</td><td>k10</td><td>single_cell</td><td>X0_T.cell.CD4__PARK7</td><td>0.8587511</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & X & sample\\_id & type & variable & value & config\\\\\n",
       "  & <int> & <chr> & <chr> & <chr> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & k10 & single\\_cell & X0\\_T.cell.CD4\\_\\_SSU72 & 0.5896835 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\\\\\n",
       "\t2 & 2 & k10 & single\\_cell & X0\\_T.cell.CD4\\_\\_PARK7 & 0.8587511 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 6\n",
       "\n",
       "| <!--/--> | X &lt;int&gt; | sample_id &lt;chr&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | k10 | single_cell | X0_T.cell.CD4__SSU72 | 0.5896835 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- |\n",
       "| 2 | 2 | k10 | single_cell | X0_T.cell.CD4__PARK7 | 0.8587511 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- |\n",
       "\n"
      ],
      "text/plain": [
       "  X sample_id type        variable             value    \n",
       "1 1 k10       single_cell X0_T.cell.CD4__SSU72 0.5896835\n",
       "2 2 k10       single_cell X0_T.cell.CD4__PARK7 0.8587511\n",
       "  config                          \n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE-"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 353,
   "id": "cde8fd1c-8200-49d2-bf72-e01d33bd77e9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'vollstaendiger_ausschluss'</li><li>'ccs'</li><li>'koronarsklerose'</li><li>'acs_subacute'</li><li>'acs_w_infection'</li><li>'acs_w_o_infection'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'vollstaendiger\\_ausschluss'\n",
       "\\item 'ccs'\n",
       "\\item 'koronarsklerose'\n",
       "\\item 'acs\\_subacute'\n",
       "\\item 'acs\\_w\\_infection'\n",
       "\\item 'acs\\_w\\_o\\_infection'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'vollstaendiger_ausschluss'\n",
       "2. 'ccs'\n",
       "3. 'koronarsklerose'\n",
       "4. 'acs_subacute'\n",
       "5. 'acs_w_infection'\n",
       "6. 'acs_w_o_infection'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"vollstaendiger_ausschluss\" \"ccs\"                      \n",
       "[3] \"koronarsklerose\"           \"acs_subacute\"             \n",
       "[5] \"acs_w_infection\"           \"acs_w_o_infection\"        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(sample_data$classification)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 354,
   "id": "7c5c9e60-e52a-46ad-8e93-5ad018306922",
   "metadata": {},
   "outputs": [],
   "source": [
    "if(filter_samples == 'all'){\n",
    "    filter = unique(sample_data$sample_id)\n",
    "    }\n",
    "\n",
    "\n",
    "if(filter_samples == 'acs_w_o_infection_ccs'){\n",
    "    filter =unique(sample_data$sample_id[sample_data$classification %in% c('ccs', 'acs_w_o_infection')])\n",
    "    \n",
    "    }\n",
    "\n",
    "if(filter_samples == 'no_ccs_ccs'){\n",
    "    filter =unique(sample_data$sample_id[sample_data$classification %in% c('vollstaendiger_ausschluss','ccs','koronarsklerose')])\n",
    "    \n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 355,
   "id": "6715853e-8f7b-4b25-a8fb-b9458ee0bc21",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "146"
      ],
      "text/latex": [
       "146"
      ],
      "text/markdown": [
       "146"
      ],
      "text/plain": [
       "[1] 146"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(filter)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 356,
   "id": "a2e20c9f-7c75-4119-a9f3-b430b6a2d72c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 357,
   "id": "9242c888-c0b6-4a94-ab4d-dd9415b698d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1557939"
      ],
      "text/latex": [
       "1557939"
      ],
      "text/markdown": [
       "1557939"
      ],
      "text/plain": [
       "[1] 1557939"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(data_long)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 358,
   "id": "a9378624-840d-4fc5-b03b-c73cc06873c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long = data_long[data_long$sample_id %in% filter,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "id": "bd1bedae-c7fe-4315-a6cb-c7ccab913490",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>sample_id</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>value</th><th scope=col>config</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>k10</td><td>single_cell</td><td>X0_T.cell.CD4__SSU72</td><td>0.5896835</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>2</td><td>k10</td><td>single_cell</td><td>X0_T.cell.CD4__PARK7</td><td>0.8587511</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & X & sample\\_id & type & variable & value & config\\\\\n",
       "  & <int> & <chr> & <chr> & <chr> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & k10 & single\\_cell & X0\\_T.cell.CD4\\_\\_SSU72 & 0.5896835 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\\\\\n",
       "\t2 & 2 & k10 & single\\_cell & X0\\_T.cell.CD4\\_\\_PARK7 & 0.8587511 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 6\n",
       "\n",
       "| <!--/--> | X &lt;int&gt; | sample_id &lt;chr&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | k10 | single_cell | X0_T.cell.CD4__SSU72 | 0.5896835 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- |\n",
       "| 2 | 2 | k10 | single_cell | X0_T.cell.CD4__PARK7 | 0.8587511 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- |\n",
       "\n"
      ],
      "text/plain": [
       "  X sample_id type        variable             value    \n",
       "1 1 k10       single_cell X0_T.cell.CD4__SSU72 0.5896835\n",
       "2 2 k10       single_cell X0_T.cell.CD4__PARK7 0.8587511\n",
       "  config                          \n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE-"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 360,
   "id": "cb94911c-51a9-481b-9f4d-fc62e6cd83e4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1557939"
      ],
      "text/latex": [
       "1557939"
      ],
      "text/markdown": [
       "1557939"
      ],
      "text/plain": [
       "[1] 1557939"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(data_long)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2489aedb-b0e5-4551-824d-32022ccce2af",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Filter on expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 361,
   "id": "e5a1f65c-ac3f-4a86-b1c6-35bc90759e34",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter out features with low expression (as in E2 script)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 362,
   "id": "3e943b19-a354-4993-89f4-cfa3c4808fd9",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long$expression = TRUE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 363,
   "id": "d26209ca-574b-44e5-a3d3-9e4fb09b7479",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long$expression[data_long$value == 0] = FALSE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 364,
   "id": "bdb68a11-5c3f-4f8a-9df7-df078e36f206",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'type'. You can override using the\n",
      "`.groups` argument.\n"
     ]
    }
   ],
   "source": [
    "expression_filter = data_long %>% group_by(type, variable) %>% summarise(perc_expression = sum(expression)  )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 365,
   "id": "e956950f-2cf0-4ce5-b4a1-50d07c754ec2",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_filter$perc_expression = expression_filter$perc_expression / length(unique(data_long$sample_id))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 366,
   "id": "ebe460c4-3b93-481e-b49c-464a01495ccb",
   "metadata": {},
   "outputs": [],
   "source": [
    "filtered_out = expression_filter[expression_filter$perc_expression <= 0.2,]  # threshold: minimum expression in 20% of samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 367,
   "id": "617706f1-e0b5-4967-9701-ab15051148fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_filter = expression_filter[expression_filter$perc_expression > 0.2,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 368,
   "id": "ea3e17bf-f2e1-4d5f-9386-4b03edc590aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "13283"
      ],
      "text/latex": [
       "13283"
      ],
      "text/markdown": [
       "13283"
      ],
      "text/plain": [
       "[1] 13283"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(expression_filter)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 369,
   "id": "d2049aa5-cbd2-4b79-9d5c-856755e7921e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>type</th><th scope=col>variable</th><th scope=col>perc_expression</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cytokine</td><td>IL16__IL16 </td><td>0.0890411</td></tr>\n",
       "\t<tr><td>cytokine</td><td>IL21__IL21 </td><td>0.1575342</td></tr>\n",
       "\t<tr><td>cytokine</td><td>IL23__IL23 </td><td>0.1506849</td></tr>\n",
       "\t<tr><td>cytokine</td><td>IL28A__IL28</td><td>0.1027397</td></tr>\n",
       "\t<tr><td>cytokine</td><td>IL33__IL33 </td><td>0.1643836</td></tr>\n",
       "\t<tr><td>cytokine</td><td>TSLP__TSLP </td><td>0.1780822</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 6 × 3\n",
       "\\begin{tabular}{lll}\n",
       " type & variable & perc\\_expression\\\\\n",
       " <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t cytokine & IL16\\_\\_IL16  & 0.0890411\\\\\n",
       "\t cytokine & IL21\\_\\_IL21  & 0.1575342\\\\\n",
       "\t cytokine & IL23\\_\\_IL23  & 0.1506849\\\\\n",
       "\t cytokine & IL28A\\_\\_IL28 & 0.1027397\\\\\n",
       "\t cytokine & IL33\\_\\_IL33  & 0.1643836\\\\\n",
       "\t cytokine & TSLP\\_\\_TSLP  & 0.1780822\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 6 × 3\n",
       "\n",
       "| type &lt;chr&gt; | variable &lt;chr&gt; | perc_expression &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| cytokine | IL16__IL16  | 0.0890411 |\n",
       "| cytokine | IL21__IL21  | 0.1575342 |\n",
       "| cytokine | IL23__IL23  | 0.1506849 |\n",
       "| cytokine | IL28A__IL28 | 0.1027397 |\n",
       "| cytokine | IL33__IL33  | 0.1643836 |\n",
       "| cytokine | TSLP__TSLP  | 0.1780822 |\n",
       "\n"
      ],
      "text/plain": [
       "  type     variable    perc_expression\n",
       "1 cytokine IL16__IL16  0.0890411      \n",
       "2 cytokine IL21__IL21  0.1575342      \n",
       "3 cytokine IL23__IL23  0.1506849      \n",
       "4 cytokine IL28A__IL28 0.1027397      \n",
       "5 cytokine IL33__IL33  0.1643836      \n",
       "6 cytokine TSLP__TSLP  0.1780822      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "filtered_out   # features that are filtered out because of too low expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 370,
   "id": "89c6e156-d4f6-4a7b-8456-ee97d5e43dcd",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# filter the data\n",
    "data_long = merge(data_long, expression_filter[,c('type', 'variable')], by.x = c('type', 'variable'), by.y = c('type', 'variable'))   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 371,
   "id": "4d945e56-3ccf-43bc-9d3d-e1d910234fe0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'clinical_data'</li><li>'cytokine'</li><li>'neutrophil'</li><li>'proteomics'</li><li>'single_cell'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'clinical\\_data'\n",
       "\\item 'cytokine'\n",
       "\\item 'neutrophil'\n",
       "\\item 'proteomics'\n",
       "\\item 'single\\_cell'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'clinical_data'\n",
       "2. 'cytokine'\n",
       "3. 'neutrophil'\n",
       "4. 'proteomics'\n",
       "5. 'single_cell'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"clinical_data\" \"cytokine\"      \"neutrophil\"    \"proteomics\"   \n",
       "[5] \"single_cell\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(data_long$type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 372,
   "id": "1d60252e-15ac-4313-ab12-aab0dcf2ceae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1557183"
      ],
      "text/latex": [
       "1557183"
      ],
      "text/markdown": [
       "1557183"
      ],
      "text/plain": [
       "[1] 1557183"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(data_long)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 374,
   "id": "6e3a9573-dd28-4c93-b328-c640a59028af",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long$ident = paste0(data_long$type, '_0_', data_long$variable)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d35c4176-5c09-480e-b03f-de43b6b3ad75",
   "metadata": {},
   "source": [
    "### Extract gene format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 381,
   "id": "d4507413-08ee-437d-a8c0-a269dd01cec8",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Extract gene codes from variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 382,
   "id": "e0c25d9c-f1e4-4a37-b0c0-51995cb353ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long$gene = str_replace(data_long$variable, '.*__', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 383,
   "id": "f8f45fe2-b433-4337-b511-a54e75450808",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long$gene[data_long$type == 'proteomics'] = str_replace(data_long$gene[data_long$type == 'proteomics'], '_.*', '')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0dd83fb4-7d0a-43d8-bf7b-fb2a46e35d52",
   "metadata": {},
   "source": [
    "### Data filtered for ligands and receptors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 398,
   "id": "ebfcee7e-5738-4cab-b43d-f09621524114",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Filter dataset on ligand and receptor variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 399,
   "id": "0adc55fc-0a2b-4364-9d61-db00ccf0ad8e",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long_lig_rec = merge(data_long, ligand_receptor_vars)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 400,
   "id": "77a7905c-4a37-4438-9c4b-2299987a87ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 7 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>type</th><th scope=col>interaction_type</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cytokine   </td><td>Ligand  </td><td>61</td></tr>\n",
       "\t<tr><td>neutrophil </td><td>Ligand  </td><td>24</td></tr>\n",
       "\t<tr><td>neutrophil </td><td>Receptor</td><td>43</td></tr>\n",
       "\t<tr><td>proteomics </td><td>Ligand  </td><td>53</td></tr>\n",
       "\t<tr><td>proteomics </td><td>Receptor</td><td>19</td></tr>\n",
       "\t<tr><td>single_cell</td><td>Ligand  </td><td>53</td></tr>\n",
       "\t<tr><td>single_cell</td><td>Receptor</td><td>75</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 7 × 3\n",
       "\\begin{tabular}{lll}\n",
       " type & interaction\\_type & n\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t cytokine    & Ligand   & 61\\\\\n",
       "\t neutrophil  & Ligand   & 24\\\\\n",
       "\t neutrophil  & Receptor & 43\\\\\n",
       "\t proteomics  & Ligand   & 53\\\\\n",
       "\t proteomics  & Receptor & 19\\\\\n",
       "\t single\\_cell & Ligand   & 53\\\\\n",
       "\t single\\_cell & Receptor & 75\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 7 × 3\n",
       "\n",
       "| type &lt;chr&gt; | interaction_type &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| cytokine    | Ligand   | 61 |\n",
       "| neutrophil  | Ligand   | 24 |\n",
       "| neutrophil  | Receptor | 43 |\n",
       "| proteomics  | Ligand   | 53 |\n",
       "| proteomics  | Receptor | 19 |\n",
       "| single_cell | Ligand   | 53 |\n",
       "| single_cell | Receptor | 75 |\n",
       "\n"
      ],
      "text/plain": [
       "  type        interaction_type n \n",
       "1 cytokine    Ligand           61\n",
       "2 neutrophil  Ligand           24\n",
       "3 neutrophil  Receptor         43\n",
       "4 proteomics  Ligand           53\n",
       "5 proteomics  Receptor         19\n",
       "6 single_cell Ligand           53\n",
       "7 single_cell Receptor         75"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# overview amount of ligands and receptors measured per dimension \n",
    "unique(data_long_lig_rec[,c('type', 'interaction_type', 'gene')]) %>% group_by(type, interaction_type) %>% count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 401,
   "id": "a4785708-163c-402c-8d8a-08d8d5555c0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "## generate a variable defining type: cell-type cluster or view"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 402,
   "id": "2e9fcade-c699-4646-a6e3-e6bccfb23b72",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long_lig_rec$cluster = data_long_lig_rec$type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 403,
   "id": "dd28dd33-1a4f-441c-9ff0-84b7271e1bfc",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long_lig_rec$cluster[data_long_lig_rec$type == 'single_cell'] = str_replace(data_long_lig_rec$variable[data_long_lig_rec$type == 'single_cell'] , '__.*', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 404,
   "id": "7b95d801-9a91-46c2-82f7-962036fe2254",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'proteomics'</li><li>'neutrophil'</li><li>'X6_Monocytes...CD14'</li><li>'X4_Monocytes...CD14'</li><li>'X7_Monocytes...CD14'</li><li>'X3_NK'</li><li>'X10_B.cell'</li><li>'X0_T.cell.CD4'</li><li>'X13_Dendritic'</li><li>'X1_T.cell.CD8'</li><li>'X11_T.cell.CD4'</li><li>'X5_T.cell.CD4'</li><li>'X12_Monocytes...CD16_FCGR3A'</li><li>'X2_T.cell.CD4'</li><li>'X9_Monocytes...CD16_FCGR3A'</li><li>'X8_B.cell'</li><li>'cytokine'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'proteomics'\n",
       "\\item 'neutrophil'\n",
       "\\item 'X6\\_Monocytes...CD14'\n",
       "\\item 'X4\\_Monocytes...CD14'\n",
       "\\item 'X7\\_Monocytes...CD14'\n",
       "\\item 'X3\\_NK'\n",
       "\\item 'X10\\_B.cell'\n",
       "\\item 'X0\\_T.cell.CD4'\n",
       "\\item 'X13\\_Dendritic'\n",
       "\\item 'X1\\_T.cell.CD8'\n",
       "\\item 'X11\\_T.cell.CD4'\n",
       "\\item 'X5\\_T.cell.CD4'\n",
       "\\item 'X12\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'X2\\_T.cell.CD4'\n",
       "\\item 'X9\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'X8\\_B.cell'\n",
       "\\item 'cytokine'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'proteomics'\n",
       "2. 'neutrophil'\n",
       "3. 'X6_Monocytes...CD14'\n",
       "4. 'X4_Monocytes...CD14'\n",
       "5. 'X7_Monocytes...CD14'\n",
       "6. 'X3_NK'\n",
       "7. 'X10_B.cell'\n",
       "8. 'X0_T.cell.CD4'\n",
       "9. 'X13_Dendritic'\n",
       "10. 'X1_T.cell.CD8'\n",
       "11. 'X11_T.cell.CD4'\n",
       "12. 'X5_T.cell.CD4'\n",
       "13. 'X12_Monocytes...CD16_FCGR3A'\n",
       "14. 'X2_T.cell.CD4'\n",
       "15. 'X9_Monocytes...CD16_FCGR3A'\n",
       "16. 'X8_B.cell'\n",
       "17. 'cytokine'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"proteomics\"                  \"neutrophil\"                 \n",
       " [3] \"X6_Monocytes...CD14\"         \"X4_Monocytes...CD14\"        \n",
       " [5] \"X7_Monocytes...CD14\"         \"X3_NK\"                      \n",
       " [7] \"X10_B.cell\"                  \"X0_T.cell.CD4\"              \n",
       " [9] \"X13_Dendritic\"               \"X1_T.cell.CD8\"              \n",
       "[11] \"X11_T.cell.CD4\"              \"X5_T.cell.CD4\"              \n",
       "[13] \"X12_Monocytes...CD16_FCGR3A\" \"X2_T.cell.CD4\"              \n",
       "[15] \"X9_Monocytes...CD16_FCGR3A\"  \"X8_B.cell\"                  \n",
       "[17] \"cytokine\"                   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(data_long_lig_rec$cluster)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 405,
   "id": "9a581d85-11e0-4a7b-a3f0-532f858c6ca4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'proteomics'</li><li>'neutrophil'</li><li>'single_cell'</li><li>'cytokine'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'proteomics'\n",
       "\\item 'neutrophil'\n",
       "\\item 'single\\_cell'\n",
       "\\item 'cytokine'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'proteomics'\n",
       "2. 'neutrophil'\n",
       "3. 'single_cell'\n",
       "4. 'cytokine'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"proteomics\"  \"neutrophil\"  \"single_cell\" \"cytokine\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(data_long_lig_rec$type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 406,
   "id": "be0fb4a4-1639-4844-b8ab-ef1d08374d84",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>X</th><th scope=col>sample_id</th><th scope=col>value</th><th scope=col>config</th><th scope=col>expression</th><th scope=col>ident</th><th scope=col>interaction_type</th><th scope=col>cluster</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A2M</td><td>proteomics</td><td>A2M_P01023</td><td>1439494</td><td>k27</td><td>9.138038</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>proteomics_0_A2M_P01023</td><td>Ligand</td><td>proteomics</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A2M</td><td>proteomics</td><td>A2M_P01023</td><td>1439557</td><td>k21</td><td>9.022400</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>proteomics_0_A2M_P01023</td><td>Ligand</td><td>proteomics</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & gene & type & variable & X & sample\\_id & value & config & expression & ident & interaction\\_type & cluster\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <chr> & <dbl> & <chr> & <lgl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & A2M & proteomics & A2M\\_P01023 & 1439494 & k27 & 9.138038 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & proteomics\\_0\\_A2M\\_P01023 & Ligand & proteomics\\\\\n",
       "\t2 & A2M & proteomics & A2M\\_P01023 & 1439557 & k21 & 9.022400 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & proteomics\\_0\\_A2M\\_P01023 & Ligand & proteomics\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 11\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | X &lt;int&gt; | sample_id &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; | expression &lt;lgl&gt; | ident &lt;chr&gt; | interaction_type &lt;chr&gt; | cluster &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | A2M | proteomics | A2M_P01023 | 1439494 | k27 | 9.138038 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | proteomics_0_A2M_P01023 | Ligand | proteomics |\n",
       "| 2 | A2M | proteomics | A2M_P01023 | 1439557 | k21 | 9.022400 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | proteomics_0_A2M_P01023 | Ligand | proteomics |\n",
       "\n"
      ],
      "text/plain": [
       "  gene type       variable   X       sample_id value   \n",
       "1 A2M  proteomics A2M_P01023 1439494 k27       9.138038\n",
       "2 A2M  proteomics A2M_P01023 1439557 k21       9.022400\n",
       "  config                           expression ident                  \n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       proteomics_0_A2M_P01023\n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       proteomics_0_A2M_P01023\n",
       "  interaction_type cluster   \n",
       "1 Ligand           proteomics\n",
       "2 Ligand           proteomics"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long_lig_rec,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2153557b-7a0a-4586-976a-6195d0f0c140",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "1ac46b83-0218-4cee-8740-526b4ab7b4f1",
   "metadata": {},
   "source": [
    "#### Add information about percentage of cells expressing gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 407,
   "id": "c7ddcd66-db2e-4d0c-b227-a18a949c66a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "## add to the ligand receptor dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 408,
   "id": "869492eb-1f28-4c31-8635-6779d74dfb42",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long_lig_rec$X = NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 409,
   "id": "1935e668-890d-4234-8cf4-a56b5e797af0",
   "metadata": {},
   "outputs": [],
   "source": [
    "cell_perc_cluster$cluster = make.names(cell_perc_cluster$cluster)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 410,
   "id": "4fdcb4c4-abd1-4f64-905f-302dc0e91506",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "95246"
      ],
      "text/latex": [
       "95246"
      ],
      "text/markdown": [
       "95246"
      ],
      "text/plain": [
       "[1] 95246"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(data_long_lig_rec)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 411,
   "id": "4227a815-0b5a-4cb5-8c4f-56b58704c83c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th><th scope=col>gene</th><th scope=col>cluster</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AL627309.1</td><td>0.24327612</td><td>18</td><td>AL627309.1</td><td>X8_B.cell</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AL627309.4</td><td>0.05406136</td><td> 4</td><td>AL627309.4</td><td>X8_B.cell</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & X & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene & gene & cluster\\\\\n",
       "  & <chr> & <dbl> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & AL627309.1 & 0.24327612 & 18 & AL627309.1 & X8\\_B.cell\\\\\n",
       "\t2 & AL627309.4 & 0.05406136 &  4 & AL627309.4 & X8\\_B.cell\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 5\n",
       "\n",
       "| <!--/--> | X &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; | gene &lt;chr&gt; | cluster &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | AL627309.1 | 0.24327612 | 18 | AL627309.1 | X8_B.cell |\n",
       "| 2 | AL627309.4 | 0.05406136 |  4 | AL627309.4 | X8_B.cell |\n",
       "\n"
      ],
      "text/plain": [
       "  X          perc_cells_expressing_gene total_amount_cells_expressing_gene\n",
       "1 AL627309.1 0.24327612                 18                                \n",
       "2 AL627309.4 0.05406136                  4                                \n",
       "  gene       cluster  \n",
       "1 AL627309.1 X8_B.cell\n",
       "2 AL627309.4 X8_B.cell"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cell_perc_cluster,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 412,
   "id": "8778b21e-f755-4555-afc0-d102aabe82b0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>sample_id</th><th scope=col>value</th><th scope=col>config</th><th scope=col>expression</th><th scope=col>ident</th><th scope=col>interaction_type</th><th scope=col>cluster</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A2M</td><td>proteomics</td><td>A2M_P01023</td><td>k27</td><td>9.138038</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>proteomics_0_A2M_P01023</td><td>Ligand</td><td>proteomics</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A2M</td><td>proteomics</td><td>A2M_P01023</td><td>k21</td><td>9.022400</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>proteomics_0_A2M_P01023</td><td>Ligand</td><td>proteomics</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 10\n",
       "\\begin{tabular}{r|llllllllll}\n",
       "  & gene & type & variable & sample\\_id & value & config & expression & ident & interaction\\_type & cluster\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <lgl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & A2M & proteomics & A2M\\_P01023 & k27 & 9.138038 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & proteomics\\_0\\_A2M\\_P01023 & Ligand & proteomics\\\\\n",
       "\t2 & A2M & proteomics & A2M\\_P01023 & k21 & 9.022400 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & proteomics\\_0\\_A2M\\_P01023 & Ligand & proteomics\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 10\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | sample_id &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; | expression &lt;lgl&gt; | ident &lt;chr&gt; | interaction_type &lt;chr&gt; | cluster &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | A2M | proteomics | A2M_P01023 | k27 | 9.138038 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | proteomics_0_A2M_P01023 | Ligand | proteomics |\n",
       "| 2 | A2M | proteomics | A2M_P01023 | k21 | 9.022400 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | proteomics_0_A2M_P01023 | Ligand | proteomics |\n",
       "\n"
      ],
      "text/plain": [
       "  gene type       variable   sample_id value   \n",
       "1 A2M  proteomics A2M_P01023 k27       9.138038\n",
       "2 A2M  proteomics A2M_P01023 k21       9.022400\n",
       "  config                           expression ident                  \n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       proteomics_0_A2M_P01023\n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       proteomics_0_A2M_P01023\n",
       "  interaction_type cluster   \n",
       "1 Ligand           proteomics\n",
       "2 Ligand           proteomics"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long_lig_rec,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 413,
   "id": "9bf63eae-93cc-48e5-9da8-b7d0bd2da112",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'proteomics'</li><li>'neutrophil'</li><li>'X6_Monocytes...CD14'</li><li>'X4_Monocytes...CD14'</li><li>'X7_Monocytes...CD14'</li><li>'X3_NK'</li><li>'X10_B.cell'</li><li>'X0_T.cell.CD4'</li><li>'X13_Dendritic'</li><li>'X1_T.cell.CD8'</li><li>'X11_T.cell.CD4'</li><li>'X5_T.cell.CD4'</li><li>'X12_Monocytes...CD16_FCGR3A'</li><li>'X2_T.cell.CD4'</li><li>'X9_Monocytes...CD16_FCGR3A'</li><li>'X8_B.cell'</li><li>'cytokine'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'proteomics'\n",
       "\\item 'neutrophil'\n",
       "\\item 'X6\\_Monocytes...CD14'\n",
       "\\item 'X4\\_Monocytes...CD14'\n",
       "\\item 'X7\\_Monocytes...CD14'\n",
       "\\item 'X3\\_NK'\n",
       "\\item 'X10\\_B.cell'\n",
       "\\item 'X0\\_T.cell.CD4'\n",
       "\\item 'X13\\_Dendritic'\n",
       "\\item 'X1\\_T.cell.CD8'\n",
       "\\item 'X11\\_T.cell.CD4'\n",
       "\\item 'X5\\_T.cell.CD4'\n",
       "\\item 'X12\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'X2\\_T.cell.CD4'\n",
       "\\item 'X9\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'X8\\_B.cell'\n",
       "\\item 'cytokine'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'proteomics'\n",
       "2. 'neutrophil'\n",
       "3. 'X6_Monocytes...CD14'\n",
       "4. 'X4_Monocytes...CD14'\n",
       "5. 'X7_Monocytes...CD14'\n",
       "6. 'X3_NK'\n",
       "7. 'X10_B.cell'\n",
       "8. 'X0_T.cell.CD4'\n",
       "9. 'X13_Dendritic'\n",
       "10. 'X1_T.cell.CD8'\n",
       "11. 'X11_T.cell.CD4'\n",
       "12. 'X5_T.cell.CD4'\n",
       "13. 'X12_Monocytes...CD16_FCGR3A'\n",
       "14. 'X2_T.cell.CD4'\n",
       "15. 'X9_Monocytes...CD16_FCGR3A'\n",
       "16. 'X8_B.cell'\n",
       "17. 'cytokine'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"proteomics\"                  \"neutrophil\"                 \n",
       " [3] \"X6_Monocytes...CD14\"         \"X4_Monocytes...CD14\"        \n",
       " [5] \"X7_Monocytes...CD14\"         \"X3_NK\"                      \n",
       " [7] \"X10_B.cell\"                  \"X0_T.cell.CD4\"              \n",
       " [9] \"X13_Dendritic\"               \"X1_T.cell.CD8\"              \n",
       "[11] \"X11_T.cell.CD4\"              \"X5_T.cell.CD4\"              \n",
       "[13] \"X12_Monocytes...CD16_FCGR3A\" \"X2_T.cell.CD4\"              \n",
       "[15] \"X9_Monocytes...CD16_FCGR3A\"  \"X8_B.cell\"                  \n",
       "[17] \"cytokine\"                   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(data_long_lig_rec$cluster)  # check to have same cluster names in both"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 414,
   "id": "b5640704-df29-4d66-9006-eb1c0debc903",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'X8_B.cell'</li><li>'X9_Monocytes...CD16_FCGR3A'</li><li>'X5_T.cell.CD4'</li><li>'X0_T.cell.CD4'</li><li>'X1_T.cell.CD8'</li><li>'X6_Monocytes...CD14'</li><li>'X2_T.cell.CD4'</li><li>'X4_Monocytes...CD14'</li><li>'X11_T.cell.CD4'</li><li>'X3_NK'</li><li>'X7_Monocytes...CD14'</li><li>'X13_Dendritic'</li><li>'X10_B.cell'</li><li>'X12_Monocytes...CD16_FCGR3A'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'X8\\_B.cell'\n",
       "\\item 'X9\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'X5\\_T.cell.CD4'\n",
       "\\item 'X0\\_T.cell.CD4'\n",
       "\\item 'X1\\_T.cell.CD8'\n",
       "\\item 'X6\\_Monocytes...CD14'\n",
       "\\item 'X2\\_T.cell.CD4'\n",
       "\\item 'X4\\_Monocytes...CD14'\n",
       "\\item 'X11\\_T.cell.CD4'\n",
       "\\item 'X3\\_NK'\n",
       "\\item 'X7\\_Monocytes...CD14'\n",
       "\\item 'X13\\_Dendritic'\n",
       "\\item 'X10\\_B.cell'\n",
       "\\item 'X12\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'X8_B.cell'\n",
       "2. 'X9_Monocytes...CD16_FCGR3A'\n",
       "3. 'X5_T.cell.CD4'\n",
       "4. 'X0_T.cell.CD4'\n",
       "5. 'X1_T.cell.CD8'\n",
       "6. 'X6_Monocytes...CD14'\n",
       "7. 'X2_T.cell.CD4'\n",
       "8. 'X4_Monocytes...CD14'\n",
       "9. 'X11_T.cell.CD4'\n",
       "10. 'X3_NK'\n",
       "11. 'X7_Monocytes...CD14'\n",
       "12. 'X13_Dendritic'\n",
       "13. 'X10_B.cell'\n",
       "14. 'X12_Monocytes...CD16_FCGR3A'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"X8_B.cell\"                   \"X9_Monocytes...CD16_FCGR3A\" \n",
       " [3] \"X5_T.cell.CD4\"               \"X0_T.cell.CD4\"              \n",
       " [5] \"X1_T.cell.CD8\"               \"X6_Monocytes...CD14\"        \n",
       " [7] \"X2_T.cell.CD4\"               \"X4_Monocytes...CD14\"        \n",
       " [9] \"X11_T.cell.CD4\"              \"X3_NK\"                      \n",
       "[11] \"X7_Monocytes...CD14\"         \"X13_Dendritic\"              \n",
       "[13] \"X10_B.cell\"                  \"X12_Monocytes...CD16_FCGR3A\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(cell_perc_cluster$cluster)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 415,
   "id": "51ee18fa-f610-44c4-8220-6a1bd90709ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "## merge the data\n",
    "data_long_lig_rec = merge(data_long_lig_rec, cell_perc_cluster, all.x = TRUE) # also keep cytokine, proteomics dimension where no cell expression is available"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 416,
   "id": "e4a5aac7-2226-4f13-9151-cf6ed7d778f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>cluster</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>sample_id</th><th scope=col>value</th><th scope=col>config</th><th scope=col>expression</th><th scope=col>ident</th><th scope=col>interaction_type</th><th scope=col>X</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A2M</td><td>proteomics</td><td>proteomics</td><td>A2M_P01023</td><td>k27</td><td>9.138038</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>proteomics_0_A2M_P01023</td><td>Ligand</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A2M</td><td>proteomics</td><td>proteomics</td><td>A2M_P01023</td><td>k21</td><td>9.022400</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>proteomics_0_A2M_P01023</td><td>Ligand</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 13\n",
       "\\begin{tabular}{r|lllllllllllll}\n",
       "  & gene & cluster & type & variable & sample\\_id & value & config & expression & ident & interaction\\_type & X & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <lgl> & <chr> & <chr> & <chr> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & A2M & proteomics & proteomics & A2M\\_P01023 & k27 & 9.138038 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & proteomics\\_0\\_A2M\\_P01023 & Ligand & NA & NA & NA\\\\\n",
       "\t2 & A2M & proteomics & proteomics & A2M\\_P01023 & k21 & 9.022400 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & proteomics\\_0\\_A2M\\_P01023 & Ligand & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 13\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | cluster &lt;chr&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | sample_id &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; | expression &lt;lgl&gt; | ident &lt;chr&gt; | interaction_type &lt;chr&gt; | X &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | A2M | proteomics | proteomics | A2M_P01023 | k27 | 9.138038 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | proteomics_0_A2M_P01023 | Ligand | NA | NA | NA |\n",
       "| 2 | A2M | proteomics | proteomics | A2M_P01023 | k21 | 9.022400 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | proteomics_0_A2M_P01023 | Ligand | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  gene cluster    type       variable   sample_id value   \n",
       "1 A2M  proteomics proteomics A2M_P01023 k27       9.138038\n",
       "2 A2M  proteomics proteomics A2M_P01023 k21       9.022400\n",
       "  config                           expression ident                  \n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       proteomics_0_A2M_P01023\n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       proteomics_0_A2M_P01023\n",
       "  interaction_type X  perc_cells_expressing_gene\n",
       "1 Ligand           NA NA                        \n",
       "2 Ligand           NA NA                        \n",
       "  total_amount_cells_expressing_gene\n",
       "1 NA                                \n",
       "2 NA                                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long_lig_rec,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 417,
   "id": "ade437c8-39b7-4f46-8b61-71720b4070d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set dummy mean expression for proteomic/ cytokine data etc. where no mean cell expression values are available\n",
    "data_long_lig_rec$perc_cells_expressing_gene[is.na(data_long_lig_rec$perc_cells_expressing_gene)] = mean(data_long_lig_rec$perc_cells_expressing_gene, na.rm = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 418,
   "id": "982485cc-c310-47b3-9a8c-7c3a4d6aaba8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'proteomics'</li><li>'neutrophil'</li><li>'single_cell'</li><li>'cytokine'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'proteomics'\n",
       "\\item 'neutrophil'\n",
       "\\item 'single\\_cell'\n",
       "\\item 'cytokine'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'proteomics'\n",
       "2. 'neutrophil'\n",
       "3. 'single_cell'\n",
       "4. 'cytokine'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"proteomics\"  \"neutrophil\"  \"single_cell\" \"cytokine\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(data_long_lig_rec$type)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f2bbce85-8549-4b83-ad5f-a2c238c7a6fc",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Aggregate expression across samples (mean expression etc.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 419,
   "id": "6ebcecc9-c345-4ca2-85ae-7dc54bfe19bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "## for ligand- receptor data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 420,
   "id": "7701529a-4b65-4741-8704-7bf7fd3539b1",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'type', 'interaction_type', 'gene',\n",
      "'cluster', 'perc_cells_expressing_gene', 'total_amount_cells_expressing_gene'.\n",
      "You can override using the `.groups` argument.\n"
     ]
    }
   ],
   "source": [
    "### Calculate variance and mean expression per cell-type across samples for single cell data\n",
    "\n",
    "analyze_expr_cluster = data_long_lig_rec %>% group_by(type, interaction_type, gene, cluster, perc_cells_expressing_gene, total_amount_cells_expressing_gene, ident) %>% \n",
    "summarise(#mean_expr_norm = mean(value_norm, na.rm = TRUE),\n",
    "          mean_expr = mean(value, na.rm = TRUE), \n",
    "          var_expression = var(value, na.rm = TRUE))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 421,
   "id": "ee80fe61-1eea-4e55-923c-3bf097e95292",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Generate cell type columns\n",
    "analyze_expr_cluster$cell_type = str_replace(analyze_expr_cluster$cluster, 'X.*_', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 422,
   "id": "6fb55d7a-38ad-463b-967b-aa8f30cf1a46",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 2 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>type</th><th scope=col>interaction_type</th><th scope=col>gene</th><th scope=col>cluster</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th><th scope=col>ident</th><th scope=col>mean_expr</th><th scope=col>var_expression</th><th scope=col>cell_type</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cytokine</td><td>Ligand</td><td>CCL1 </td><td>cytokine</td><td>69.10381</td><td>NA</td><td>cytokine_0_I309__CCL1    </td><td>1.457331</td><td>0.4209982</td><td>cytokine</td></tr>\n",
       "\t<tr><td>cytokine</td><td>Ligand</td><td>CCL11</td><td>cytokine</td><td>69.10381</td><td>NA</td><td>cytokine_0_Eotaxin__CCL11</td><td>7.408225</td><td>3.1654144</td><td>cytokine</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 2 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " type & interaction\\_type & gene & cluster & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene & ident & mean\\_expr & var\\_expression & cell\\_type\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <dbl> & <int> & <chr> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t cytokine & Ligand & CCL1  & cytokine & 69.10381 & NA & cytokine\\_0\\_I309\\_\\_CCL1     & 1.457331 & 0.4209982 & cytokine\\\\\n",
       "\t cytokine & Ligand & CCL11 & cytokine & 69.10381 & NA & cytokine\\_0\\_Eotaxin\\_\\_CCL11 & 7.408225 & 3.1654144 & cytokine\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 2 × 10\n",
       "\n",
       "| type &lt;chr&gt; | interaction_type &lt;chr&gt; | gene &lt;chr&gt; | cluster &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; | ident &lt;chr&gt; | mean_expr &lt;dbl&gt; | var_expression &lt;dbl&gt; | cell_type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| cytokine | Ligand | CCL1  | cytokine | 69.10381 | NA | cytokine_0_I309__CCL1     | 1.457331 | 0.4209982 | cytokine |\n",
       "| cytokine | Ligand | CCL11 | cytokine | 69.10381 | NA | cytokine_0_Eotaxin__CCL11 | 7.408225 | 3.1654144 | cytokine |\n",
       "\n"
      ],
      "text/plain": [
       "  type     interaction_type gene  cluster  perc_cells_expressing_gene\n",
       "1 cytokine Ligand           CCL1  cytokine 69.10381                  \n",
       "2 cytokine Ligand           CCL11 cytokine 69.10381                  \n",
       "  total_amount_cells_expressing_gene ident                     mean_expr\n",
       "1 NA                                 cytokine_0_I309__CCL1     1.457331 \n",
       "2 NA                                 cytokine_0_Eotaxin__CCL11 7.408225 \n",
       "  var_expression cell_type\n",
       "1 0.4209982      cytokine \n",
       "2 3.1654144      cytokine "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(analyze_expr_cluster,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 423,
   "id": "7f2587e7-cee6-4c31-895d-7ed39d2d7ef2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cytokine'</li><li>'neutrophil'</li><li>'proteomics'</li><li>'Monocytes...CD14'</li><li>'T.cell.CD4'</li><li>'B.cell'</li><li>'FCGR3A'</li><li>'Dendritic'</li><li>'T.cell.CD8'</li><li>'NK'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cytokine'\n",
       "\\item 'neutrophil'\n",
       "\\item 'proteomics'\n",
       "\\item 'Monocytes...CD14'\n",
       "\\item 'T.cell.CD4'\n",
       "\\item 'B.cell'\n",
       "\\item 'FCGR3A'\n",
       "\\item 'Dendritic'\n",
       "\\item 'T.cell.CD8'\n",
       "\\item 'NK'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cytokine'\n",
       "2. 'neutrophil'\n",
       "3. 'proteomics'\n",
       "4. 'Monocytes...CD14'\n",
       "5. 'T.cell.CD4'\n",
       "6. 'B.cell'\n",
       "7. 'FCGR3A'\n",
       "8. 'Dendritic'\n",
       "9. 'T.cell.CD8'\n",
       "10. 'NK'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"cytokine\"         \"neutrophil\"       \"proteomics\"       \"Monocytes...CD14\"\n",
       " [5] \"T.cell.CD4\"       \"B.cell\"           \"FCGR3A\"           \"Dendritic\"       \n",
       " [9] \"T.cell.CD8\"       \"NK\"              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(analyze_expr_cluster$cell_type)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ed5aea70-1f5c-432e-98ee-99e80c8641ce",
   "metadata": {},
   "source": [
    "#### Cluster similar receptors and ligand (expressed in same cell-types)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 424,
   "id": "8cdd93f0-b703-4cf4-a10a-44a0884192a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Cluster Receptors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 425,
   "id": "ded6a527-4718-410c-8281-b1982a1f5b3b",
   "metadata": {},
   "outputs": [],
   "source": [
    "filter_type_vis = c('cytokine', 'proteomics', 'neutrophil')  # only for sc data exclude other types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 426,
   "id": "ac7abeab-0bfc-441c-a65b-c469fd1dad26",
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_data = analyze_expr_cluster[(analyze_expr_cluster$interaction_type == 'Receptor') & (!analyze_expr_cluster$type %in% filter_type_vis),] %>% dcast(gene ~ cluster, value.var = 'mean_expr')\n",
    "# value_var = mean_expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 427,
   "id": "859e5482-9eea-4485-ae31-2f29b7334f19",
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_data[is.na(cluster_data)] = 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 428,
   "id": "070733f8-8fad-48f4-b13c-8c52c4e25060",
   "metadata": {},
   "outputs": [],
   "source": [
    "rownames(cluster_data) = cluster_data$gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 429,
   "id": "d7906ced-45c6-42bb-be39-91af494bacfe",
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_data$gene = NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 430,
   "id": "9d37fc1e-3734-4b84-bd56-4e0f224bfff0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 14</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X0_T.cell.CD4</th><th scope=col>X1_T.cell.CD8</th><th scope=col>X10_B.cell</th><th scope=col>X11_T.cell.CD4</th><th scope=col>X12_Monocytes...CD16_FCGR3A</th><th scope=col>X13_Dendritic</th><th scope=col>X2_T.cell.CD4</th><th scope=col>X3_NK</th><th scope=col>X4_Monocytes...CD14</th><th scope=col>X5_T.cell.CD4</th><th scope=col>X6_Monocytes...CD14</th><th scope=col>X7_Monocytes...CD14</th><th scope=col>X8_B.cell</th><th scope=col>X9_Monocytes...CD16_FCGR3A</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>AGTRAP</th><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1.2045378</td><td>0</td><td>1.158767</td><td>1.183921</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>ANXA6</th><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0.7138537</td><td>0</td><td>0.000000</td><td>0.000000</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 14\n",
       "\\begin{tabular}{r|llllllllllllll}\n",
       "  & X0\\_T.cell.CD4 & X1\\_T.cell.CD8 & X10\\_B.cell & X11\\_T.cell.CD4 & X12\\_Monocytes...CD16\\_FCGR3A & X13\\_Dendritic & X2\\_T.cell.CD4 & X3\\_NK & X4\\_Monocytes...CD14 & X5\\_T.cell.CD4 & X6\\_Monocytes...CD14 & X7\\_Monocytes...CD14 & X8\\_B.cell & X9\\_Monocytes...CD16\\_FCGR3A\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tAGTRAP & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1.2045378 & 0 & 1.158767 & 1.183921 & 0 & 0\\\\\n",
       "\tANXA6 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0.7138537 & 0 & 0.000000 & 0.000000 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 14\n",
       "\n",
       "| <!--/--> | X0_T.cell.CD4 &lt;dbl&gt; | X1_T.cell.CD8 &lt;dbl&gt; | X10_B.cell &lt;dbl&gt; | X11_T.cell.CD4 &lt;dbl&gt; | X12_Monocytes...CD16_FCGR3A &lt;dbl&gt; | X13_Dendritic &lt;dbl&gt; | X2_T.cell.CD4 &lt;dbl&gt; | X3_NK &lt;dbl&gt; | X4_Monocytes...CD14 &lt;dbl&gt; | X5_T.cell.CD4 &lt;dbl&gt; | X6_Monocytes...CD14 &lt;dbl&gt; | X7_Monocytes...CD14 &lt;dbl&gt; | X8_B.cell &lt;dbl&gt; | X9_Monocytes...CD16_FCGR3A &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| AGTRAP | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 1.2045378 | 0 | 1.158767 | 1.183921 | 0 | 0 |\n",
       "| ANXA6 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0.7138537 | 0 | 0.000000 | 0.000000 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "       X0_T.cell.CD4 X1_T.cell.CD8 X10_B.cell X11_T.cell.CD4\n",
       "AGTRAP 0             0             0          0             \n",
       "ANXA6  0             0             0          0             \n",
       "       X12_Monocytes...CD16_FCGR3A X13_Dendritic X2_T.cell.CD4 X3_NK\n",
       "AGTRAP 0                           0             0             0    \n",
       "ANXA6  0                           0             0             0    \n",
       "       X4_Monocytes...CD14 X5_T.cell.CD4 X6_Monocytes...CD14\n",
       "AGTRAP 1.2045378           0             1.158767           \n",
       "ANXA6  0.7138537           0             0.000000           \n",
       "       X7_Monocytes...CD14 X8_B.cell X9_Monocytes...CD16_FCGR3A\n",
       "AGTRAP 1.183921            0         0                         \n",
       "ANXA6  0.000000            0         0                         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cluster_data,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 431,
   "id": "b7af832d-95de-4015-8f5d-fc5a7529bd48",
   "metadata": {},
   "outputs": [],
   "source": [
    "# cluster the receptors based on mean expression\n",
    "cluster_result = kmeans(cluster_data, centers = 7, iter.max = 10, nstart = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 432,
   "id": "dd1ebc31-6fc7-4f29-a452-fa550c35364e",
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_result = data.frame(cluster= cluster_result$cluster, gene = names(cluster_result$cluster))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 433,
   "id": "28195c83-8107-4619-a9d5-c322b19e1a82",
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_result$interaction_type = 'Receptor'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 434,
   "id": "864601da-4b75-437b-8462-208ee56aa293",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cluster</th><th scope=col>gene</th><th scope=col>interaction_type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>AGTRAP</th><td>7</td><td>AGTRAP</td><td>Receptor</td></tr>\n",
       "\t<tr><th scope=row>ANXA6</th><td>7</td><td>ANXA6 </td><td>Receptor</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & cluster & gene & interaction\\_type\\\\\n",
       "  & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\tAGTRAP & 7 & AGTRAP & Receptor\\\\\n",
       "\tANXA6 & 7 & ANXA6  & Receptor\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 3\n",
       "\n",
       "| <!--/--> | cluster &lt;int&gt; | gene &lt;chr&gt; | interaction_type &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| AGTRAP | 7 | AGTRAP | Receptor |\n",
       "| ANXA6 | 7 | ANXA6  | Receptor |\n",
       "\n"
      ],
      "text/plain": [
       "       cluster gene   interaction_type\n",
       "AGTRAP 7       AGTRAP Receptor        \n",
       "ANXA6  7       ANXA6  Receptor        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cluster_result,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 435,
   "id": "56c0d1db-3f77-4357-97b5-3bba7019b2ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Cluster Ligands"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 436,
   "id": "b842f333-8966-4101-a319-3db57fdb4e72",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter to only do for sc data\n",
    "cluster_data = analyze_expr_cluster[analyze_expr_cluster$interaction_type == 'Ligand' & (!analyze_expr_cluster$type %in% filter_type_vis),] %>% dcast(gene ~ cluster, value.var = 'mean_expr')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 437,
   "id": "17894073-80b5-4ae0-ae58-3ad3daf27421",
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_data[is.na(cluster_data)] = 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 438,
   "id": "521010d4-2180-4b2e-b371-182f605bdb57",
   "metadata": {},
   "outputs": [],
   "source": [
    "rownames(cluster_data) = cluster_data$gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 439,
   "id": "51297eb4-f420-4d09-a0c0-92c84e5e205b",
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_data$gene = NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 440,
   "id": "38b70111-7188-48b7-8c09-37f45d268345",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 14</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X0_T.cell.CD4</th><th scope=col>X1_T.cell.CD8</th><th scope=col>X10_B.cell</th><th scope=col>X11_T.cell.CD4</th><th scope=col>X12_Monocytes...CD16_FCGR3A</th><th scope=col>X13_Dendritic</th><th scope=col>X2_T.cell.CD4</th><th scope=col>X3_NK</th><th scope=col>X4_Monocytes...CD14</th><th scope=col>X5_T.cell.CD4</th><th scope=col>X6_Monocytes...CD14</th><th scope=col>X7_Monocytes...CD14</th><th scope=col>X8_B.cell</th><th scope=col>X9_Monocytes...CD16_FCGR3A</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>ADAM17</th><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0.8383966</td><td>0</td><td>0.8149382</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>AIMP1</th><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0.7533036</td><td>0</td><td>0.0000000</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 14\n",
       "\\begin{tabular}{r|llllllllllllll}\n",
       "  & X0\\_T.cell.CD4 & X1\\_T.cell.CD8 & X10\\_B.cell & X11\\_T.cell.CD4 & X12\\_Monocytes...CD16\\_FCGR3A & X13\\_Dendritic & X2\\_T.cell.CD4 & X3\\_NK & X4\\_Monocytes...CD14 & X5\\_T.cell.CD4 & X6\\_Monocytes...CD14 & X7\\_Monocytes...CD14 & X8\\_B.cell & X9\\_Monocytes...CD16\\_FCGR3A\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tADAM17 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0.8383966 & 0 & 0.8149382 & 0 & 0 & 0\\\\\n",
       "\tAIMP1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0.7533036 & 0 & 0.0000000 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 14\n",
       "\n",
       "| <!--/--> | X0_T.cell.CD4 &lt;dbl&gt; | X1_T.cell.CD8 &lt;dbl&gt; | X10_B.cell &lt;dbl&gt; | X11_T.cell.CD4 &lt;dbl&gt; | X12_Monocytes...CD16_FCGR3A &lt;dbl&gt; | X13_Dendritic &lt;dbl&gt; | X2_T.cell.CD4 &lt;dbl&gt; | X3_NK &lt;dbl&gt; | X4_Monocytes...CD14 &lt;dbl&gt; | X5_T.cell.CD4 &lt;dbl&gt; | X6_Monocytes...CD14 &lt;dbl&gt; | X7_Monocytes...CD14 &lt;dbl&gt; | X8_B.cell &lt;dbl&gt; | X9_Monocytes...CD16_FCGR3A &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ADAM17 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0.8383966 | 0 | 0.8149382 | 0 | 0 | 0 |\n",
       "| AIMP1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0.7533036 | 0 | 0.0000000 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "       X0_T.cell.CD4 X1_T.cell.CD8 X10_B.cell X11_T.cell.CD4\n",
       "ADAM17 0             0             0          0             \n",
       "AIMP1  0             0             0          0             \n",
       "       X12_Monocytes...CD16_FCGR3A X13_Dendritic X2_T.cell.CD4 X3_NK\n",
       "ADAM17 0                           0             0             0    \n",
       "AIMP1  0                           0             0             0    \n",
       "       X4_Monocytes...CD14 X5_T.cell.CD4 X6_Monocytes...CD14\n",
       "ADAM17 0.8383966           0             0.8149382          \n",
       "AIMP1  0.7533036           0             0.0000000          \n",
       "       X7_Monocytes...CD14 X8_B.cell X9_Monocytes...CD16_FCGR3A\n",
       "ADAM17 0                   0         0                         \n",
       "AIMP1  0                   0         0                         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cluster_data,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 441,
   "id": "f544a15c-c755-4960-9ead-db78f294cc00",
   "metadata": {},
   "outputs": [],
   "source": [
    "# cluster the data\n",
    "cluster_result_l = kmeans(cluster_data, centers = 10, iter.max = 10, nstart = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 442,
   "id": "aa3722fb-4328-4a06-923d-67d2d2346023",
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_result_l = data.frame(cluster= cluster_result_l$cluster, gene = names(cluster_result_l$cluster))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 443,
   "id": "8579532e-c742-4872-8024-84706fe18628",
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_result_l$interaction_type = 'Ligand'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 444,
   "id": "c11d4605-0a1d-495c-b483-c3372dcf44f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cluster</th><th scope=col>gene</th><th scope=col>interaction_type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>ADAM17</th><td>3</td><td>ADAM17</td><td>Ligand</td></tr>\n",
       "\t<tr><th scope=row>AIMP1</th><td>3</td><td>AIMP1 </td><td>Ligand</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & cluster & gene & interaction\\_type\\\\\n",
       "  & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\tADAM17 & 3 & ADAM17 & Ligand\\\\\n",
       "\tAIMP1 & 3 & AIMP1  & Ligand\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 3\n",
       "\n",
       "| <!--/--> | cluster &lt;int&gt; | gene &lt;chr&gt; | interaction_type &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| ADAM17 | 3 | ADAM17 | Ligand |\n",
       "| AIMP1 | 3 | AIMP1  | Ligand |\n",
       "\n"
      ],
      "text/plain": [
       "       cluster gene   interaction_type\n",
       "ADAM17 3       ADAM17 Ligand          \n",
       "AIMP1  3       AIMP1  Ligand          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cluster_result_l,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 445,
   "id": "e7498d6f-f850-4910-9f06-4f60e645b8ee",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Merge both cluster results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 446,
   "id": "e5b018b1-10ed-4e29-a995-b72a912b04c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster_result = rbind(cluster_result, cluster_result_l)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 447,
   "id": "11193070-1bd5-4d94-8c66-cf158c6a3c44",
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames(cluster_result) = c('cluster_vis', 'gene', 'interaction_type')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 448,
   "id": "2af9f027-63ca-4690-b460-d156fb20042d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Add clustering result to mean expression values and save the data "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 449,
   "id": "ca176bce-d1c7-4ca9-aa61-dd4dd44c2e76",
   "metadata": {},
   "outputs": [],
   "source": [
    "analyze_expr_cluster = merge(analyze_expr_cluster, cluster_result, all.x = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 450,
   "id": "d942e4a0-9326-4822-bf16-c7509c4e4e63",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>interaction_type</th><th scope=col>gene</th><th scope=col>type</th><th scope=col>cluster</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th><th scope=col>ident</th><th scope=col>mean_expr</th><th scope=col>var_expression</th><th scope=col>cell_type</th><th scope=col>cluster_vis</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Ligand</td><td>A2M   </td><td>proteomics </td><td>proteomics         </td><td>69.10381</td><td>  NA</td><td>proteomics_0_A2M_P01023                  </td><td>8.7445302</td><td>0.32468578</td><td>proteomics      </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Ligand</td><td>ADAM17</td><td>single_cell</td><td>X4_Monocytes...CD14</td><td>49.68889</td><td>7267</td><td>single_cell_0_X4_Monocytes...CD14__ADAM17</td><td>0.8383966</td><td>0.03136488</td><td>Monocytes...CD14</td><td> 3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & interaction\\_type & gene & type & cluster & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene & ident & mean\\_expr & var\\_expression & cell\\_type & cluster\\_vis\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <dbl> & <int> & <chr> & <dbl> & <dbl> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & Ligand & A2M    & proteomics  & proteomics          & 69.10381 &   NA & proteomics\\_0\\_A2M\\_P01023                   & 8.7445302 & 0.32468578 & proteomics       & NA\\\\\n",
       "\t2 & Ligand & ADAM17 & single\\_cell & X4\\_Monocytes...CD14 & 49.68889 & 7267 & single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_ADAM17 & 0.8383966 & 0.03136488 & Monocytes...CD14 &  3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 11\n",
       "\n",
       "| <!--/--> | interaction_type &lt;chr&gt; | gene &lt;chr&gt; | type &lt;chr&gt; | cluster &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; | ident &lt;chr&gt; | mean_expr &lt;dbl&gt; | var_expression &lt;dbl&gt; | cell_type &lt;chr&gt; | cluster_vis &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | Ligand | A2M    | proteomics  | proteomics          | 69.10381 |   NA | proteomics_0_A2M_P01023                   | 8.7445302 | 0.32468578 | proteomics       | NA |\n",
       "| 2 | Ligand | ADAM17 | single_cell | X4_Monocytes...CD14 | 49.68889 | 7267 | single_cell_0_X4_Monocytes...CD14__ADAM17 | 0.8383966 | 0.03136488 | Monocytes...CD14 |  3 |\n",
       "\n"
      ],
      "text/plain": [
       "  interaction_type gene   type        cluster            \n",
       "1 Ligand           A2M    proteomics  proteomics         \n",
       "2 Ligand           ADAM17 single_cell X4_Monocytes...CD14\n",
       "  perc_cells_expressing_gene total_amount_cells_expressing_gene\n",
       "1 69.10381                     NA                              \n",
       "2 49.68889                   7267                              \n",
       "  ident                                     mean_expr var_expression\n",
       "1 proteomics_0_A2M_P01023                   8.7445302 0.32468578    \n",
       "2 single_cell_0_X4_Monocytes...CD14__ADAM17 0.8383966 0.03136488    \n",
       "  cell_type        cluster_vis\n",
       "1 proteomics       NA         \n",
       "2 Monocytes...CD14  3         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(analyze_expr_cluster,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 451,
   "id": "85b7acc6-47d6-46d6-b8d4-656d3055b4d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(analyze_expr_cluster,gzfile(paste0(result_path, '/F-Analysis/' ,'F4_Ligand_Receptor_Data_Expression', name, filter_samples,  '.gz')))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7efac83d-87b6-43d5-a4ac-37a087f4d6b5",
   "metadata": {},
   "source": [
    "# Ligand Receptor Data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4225d21f-ddaf-412c-bcf3-d3d503552069",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Calculate ligand receptor correlations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 452,
   "id": "de1533c6-c539-477d-8c80-0c29c0f10a24",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>cluster</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>sample_id</th><th scope=col>value</th><th scope=col>config</th><th scope=col>expression</th><th scope=col>ident</th><th scope=col>interaction_type</th><th scope=col>X</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A2M</td><td>proteomics</td><td>proteomics</td><td>A2M_P01023</td><td>k27</td><td>9.138038</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>proteomics_0_A2M_P01023</td><td>Ligand</td><td>NA</td><td>69.10381</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A2M</td><td>proteomics</td><td>proteomics</td><td>A2M_P01023</td><td>k21</td><td>9.022400</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>proteomics_0_A2M_P01023</td><td>Ligand</td><td>NA</td><td>69.10381</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 13\n",
       "\\begin{tabular}{r|lllllllllllll}\n",
       "  & gene & cluster & type & variable & sample\\_id & value & config & expression & ident & interaction\\_type & X & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <lgl> & <chr> & <chr> & <chr> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & A2M & proteomics & proteomics & A2M\\_P01023 & k27 & 9.138038 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & proteomics\\_0\\_A2M\\_P01023 & Ligand & NA & 69.10381 & NA\\\\\n",
       "\t2 & A2M & proteomics & proteomics & A2M\\_P01023 & k21 & 9.022400 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & proteomics\\_0\\_A2M\\_P01023 & Ligand & NA & 69.10381 & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 13\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | cluster &lt;chr&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | sample_id &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; | expression &lt;lgl&gt; | ident &lt;chr&gt; | interaction_type &lt;chr&gt; | X &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | A2M | proteomics | proteomics | A2M_P01023 | k27 | 9.138038 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | proteomics_0_A2M_P01023 | Ligand | NA | 69.10381 | NA |\n",
       "| 2 | A2M | proteomics | proteomics | A2M_P01023 | k21 | 9.022400 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | proteomics_0_A2M_P01023 | Ligand | NA | 69.10381 | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  gene cluster    type       variable   sample_id value   \n",
       "1 A2M  proteomics proteomics A2M_P01023 k27       9.138038\n",
       "2 A2M  proteomics proteomics A2M_P01023 k21       9.022400\n",
       "  config                           expression ident                  \n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       proteomics_0_A2M_P01023\n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       proteomics_0_A2M_P01023\n",
       "  interaction_type X  perc_cells_expressing_gene\n",
       "1 Ligand           NA 69.10381                  \n",
       "2 Ligand           NA 69.10381                  \n",
       "  total_amount_cells_expressing_gene\n",
       "1 NA                                \n",
       "2 NA                                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long_lig_rec,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 453,
   "id": "ac39a62d-5c25-4fb4-9776-9f901cd37568",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Transform to wide format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 454,
   "id": "cff8f016-0a38-4605-a5bf-821696b06f4c",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_receptor_expression = data_long_lig_rec %>% dcast(sample_id ~ ident, value.var = 'value')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 455,
   "id": "88f250d6-3469-4c22-97ea-dda08174f226",
   "metadata": {},
   "outputs": [],
   "source": [
    "rownames(ligand_receptor_expression) = ligand_receptor_expression$sample_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 456,
   "id": "03f6c3de-0107-477c-ab3b-b014723193a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_receptor_expression$sample_id = NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 457,
   "id": "1561fff6-70c0-4c00-b073-628333d6e68f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 812</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cytokine_0_BCA1__CXCL13</th><th scope=col>cytokine_0_CTACK__CCL27</th><th scope=col>cytokine_0_EGF__EGF</th><th scope=col>cytokine_0_ENA78__CXCL5</th><th scope=col>cytokine_0_Eotaxin__CCL11</th><th scope=col>cytokine_0_Eotaxin2__CCL24</th><th scope=col>cytokine_0_Eotaxin3__CCL26</th><th scope=col>cytokine_0_FGF2__FGF2</th><th scope=col>cytokine_0_FLT3L__FLT3LG</th><th scope=col>cytokine_0_Fractalkine__CX3CL1</th><th scope=col>⋯</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__RTN4</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__SELPLG</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__SERPINA1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__SPN</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__TGFB1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__TIMP1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFRSF14</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFRSF1B</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFSF10</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFSF13B</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>k1</th><td>5.696550</td><td>9.821662</td><td>5.045268</td><td>10.562366</td><td>7.42341</td><td>8.755388</td><td>8.820594</td><td>6.734845</td><td>5.200065</td><td>7.552054</td><td>⋯</td><td>      NA</td><td>       NA</td><td>      NA</td><td>     NA</td><td>       NA</td><td>      NA</td><td>       NA</td><td>   NA</td><td>      NA</td><td>      NA</td></tr>\n",
       "\t<tr><th scope=row>k10</th><td>4.697663</td><td>9.701220</td><td>1.922198</td><td> 7.724173</td><td>9.19842</td><td>9.728363</td><td>5.598127</td><td>5.791293</td><td>3.737687</td><td>5.874305</td><td>⋯</td><td>1.343136</td><td>0.6540153</td><td>2.677473</td><td>1.21784</td><td>0.8279207</td><td>2.885468</td><td>0.8118693</td><td>2.711</td><td>1.428577</td><td>1.190677</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 812\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & cytokine\\_0\\_BCA1\\_\\_CXCL13 & cytokine\\_0\\_CTACK\\_\\_CCL27 & cytokine\\_0\\_EGF\\_\\_EGF & cytokine\\_0\\_ENA78\\_\\_CXCL5 & cytokine\\_0\\_Eotaxin\\_\\_CCL11 & cytokine\\_0\\_Eotaxin2\\_\\_CCL24 & cytokine\\_0\\_Eotaxin3\\_\\_CCL26 & cytokine\\_0\\_FGF2\\_\\_FGF2 & cytokine\\_0\\_FLT3L\\_\\_FLT3LG & cytokine\\_0\\_Fractalkine\\_\\_CX3CL1 & ⋯ & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_RTN4 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_SELPLG & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_SERPINA1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_SPN & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_TGFB1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_TIMP1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_TNFRSF14 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_TNFRSF1B & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_TNFSF10 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_TNFSF13B\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tk1 & 5.696550 & 9.821662 & 5.045268 & 10.562366 & 7.42341 & 8.755388 & 8.820594 & 6.734845 & 5.200065 & 7.552054 & ⋯ &       NA &        NA &       NA &      NA &        NA &       NA &        NA &    NA &       NA &       NA\\\\\n",
       "\tk10 & 4.697663 & 9.701220 & 1.922198 &  7.724173 & 9.19842 & 9.728363 & 5.598127 & 5.791293 & 3.737687 & 5.874305 & ⋯ & 1.343136 & 0.6540153 & 2.677473 & 1.21784 & 0.8279207 & 2.885468 & 0.8118693 & 2.711 & 1.428577 & 1.190677\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 812\n",
       "\n",
       "| <!--/--> | cytokine_0_BCA1__CXCL13 &lt;dbl&gt; | cytokine_0_CTACK__CCL27 &lt;dbl&gt; | cytokine_0_EGF__EGF &lt;dbl&gt; | cytokine_0_ENA78__CXCL5 &lt;dbl&gt; | cytokine_0_Eotaxin__CCL11 &lt;dbl&gt; | cytokine_0_Eotaxin2__CCL24 &lt;dbl&gt; | cytokine_0_Eotaxin3__CCL26 &lt;dbl&gt; | cytokine_0_FGF2__FGF2 &lt;dbl&gt; | cytokine_0_FLT3L__FLT3LG &lt;dbl&gt; | cytokine_0_Fractalkine__CX3CL1 &lt;dbl&gt; | ⋯ ⋯ | single_cell_0_X9_Monocytes...CD16_FCGR3A__RTN4 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__SELPLG &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__SERPINA1 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__SPN &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__TGFB1 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__TIMP1 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFRSF14 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFRSF1B &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFSF10 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFSF13B &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| k1 | 5.696550 | 9.821662 | 5.045268 | 10.562366 | 7.42341 | 8.755388 | 8.820594 | 6.734845 | 5.200065 | 7.552054 | ⋯ |       NA |        NA |       NA |      NA |        NA |       NA |        NA |    NA |       NA |       NA |\n",
       "| k10 | 4.697663 | 9.701220 | 1.922198 |  7.724173 | 9.19842 | 9.728363 | 5.598127 | 5.791293 | 3.737687 | 5.874305 | ⋯ | 1.343136 | 0.6540153 | 2.677473 | 1.21784 | 0.8279207 | 2.885468 | 0.8118693 | 2.711 | 1.428577 | 1.190677 |\n",
       "\n"
      ],
      "text/plain": [
       "    cytokine_0_BCA1__CXCL13 cytokine_0_CTACK__CCL27 cytokine_0_EGF__EGF\n",
       "k1  5.696550                9.821662                5.045268           \n",
       "k10 4.697663                9.701220                1.922198           \n",
       "    cytokine_0_ENA78__CXCL5 cytokine_0_Eotaxin__CCL11\n",
       "k1  10.562366               7.42341                  \n",
       "k10  7.724173               9.19842                  \n",
       "    cytokine_0_Eotaxin2__CCL24 cytokine_0_Eotaxin3__CCL26 cytokine_0_FGF2__FGF2\n",
       "k1  8.755388                   8.820594                   6.734845             \n",
       "k10 9.728363                   5.598127                   5.791293             \n",
       "    cytokine_0_FLT3L__FLT3LG cytokine_0_Fractalkine__CX3CL1 ⋯\n",
       "k1  5.200065                 7.552054                       ⋯\n",
       "k10 3.737687                 5.874305                       ⋯\n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__RTN4\n",
       "k1        NA                                      \n",
       "k10 1.343136                                      \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__SELPLG\n",
       "k1         NA                                       \n",
       "k10 0.6540153                                       \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__SERPINA1\n",
       "k1        NA                                          \n",
       "k10 2.677473                                          \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__SPN\n",
       "k1       NA                                      \n",
       "k10 1.21784                                      \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__TGFB1\n",
       "k1         NA                                      \n",
       "k10 0.8279207                                      \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__TIMP1\n",
       "k1        NA                                       \n",
       "k10 2.885468                                       \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFRSF14\n",
       "k1         NA                                         \n",
       "k10 0.8118693                                         \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFRSF1B\n",
       "k1     NA                                             \n",
       "k10 2.711                                             \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFSF10\n",
       "k1        NA                                         \n",
       "k10 1.428577                                         \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFSF13B\n",
       "k1        NA                                          \n",
       "k10 1.190677                                          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_receptor_expression,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 458,
   "id": "482976fe-9fde-497b-aa0f-952b9ec58f4c",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Define Ligands and Receptors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 459,
   "id": "ecd17d9e-af13-4175-be76-156c2cf8eb13",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligands = unique(data_long_lig_rec$ident[data_long_lig_rec$interaction_type == 'Ligand'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 460,
   "id": "2115d5d7-db36-4946-a802-b311f6eef889",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "445"
      ],
      "text/latex": [
       "445"
      ],
      "text/markdown": [
       "445"
      ],
      "text/plain": [
       "[1] 445"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(ligands)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 461,
   "id": "b0203c53-2283-4a70-ab53-810c0d4e0187",
   "metadata": {},
   "outputs": [],
   "source": [
    "receptors = unique(data_long_lig_rec$ident[data_long_lig_rec$interaction_type == 'Receptor'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 462,
   "id": "14faf058-8aa4-461d-acb5-c12228f4c0e0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "367"
      ],
      "text/latex": [
       "367"
      ],
      "text/markdown": [
       "367"
      ],
      "text/plain": [
       "[1] 367"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(receptors)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 463,
   "id": "f96c84b8-ef54-4b0a-9963-c19fbd0b464a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "812"
      ],
      "text/latex": [
       "812"
      ],
      "text/markdown": [
       "812"
      ],
      "text/plain": [
       "[1] 812"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ncol(ligand_receptor_expression)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 464,
   "id": "fe2cfbc7-6b96-4331-a6d0-520d111dec39",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Calculate correlations between ligands and receptors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 465,
   "id": "5994a6d0-cf95-4995-ae88-cae800919f69",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_receptor_correlations = cor(ligand_receptor_expression[,colnames(ligand_receptor_expression) %in%ligands], ligand_receptor_expression[,colnames(ligand_receptor_expression) %in% receptors], method = 'spearman', use = 'pairwise.complete.obs')\n",
    "# currently calculated with spearman!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 466,
   "id": "e521389b-5f8a-4cb8-a185-bae9e6d5540f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Adjust format of calculated correlations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 467,
   "id": "1db15612-0c3d-473b-844d-d4bcf64acb5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_receptor_correlations = melt(ligand_receptor_correlations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 468,
   "id": "75f9fbbb-d239-4978-b82c-8b3fecec9a5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames(ligand_receptor_correlations) = c('ligand', 'receptor', 'correlation')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 469,
   "id": "9a343b45-c3e8-48d9-a3f1-08c04f807bf9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>correlation</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>cytokine_0_BCA1__CXCL13</td><td>neutrophil_0_ABCA7</td><td>0.09543501</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>cytokine_0_CTACK__CCL27</td><td>neutrophil_0_ABCA7</td><td>0.05916357</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & ligand & receptor & correlation\\\\\n",
       "  & <fct> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & cytokine\\_0\\_BCA1\\_\\_CXCL13 & neutrophil\\_0\\_ABCA7 & 0.09543501\\\\\n",
       "\t2 & cytokine\\_0\\_CTACK\\_\\_CCL27 & neutrophil\\_0\\_ABCA7 & 0.05916357\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 3\n",
       "\n",
       "| <!--/--> | ligand &lt;fct&gt; | receptor &lt;fct&gt; | correlation &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | cytokine_0_BCA1__CXCL13 | neutrophil_0_ABCA7 | 0.09543501 |\n",
       "| 2 | cytokine_0_CTACK__CCL27 | neutrophil_0_ABCA7 | 0.05916357 |\n",
       "\n"
      ],
      "text/plain": [
       "  ligand                  receptor           correlation\n",
       "1 cytokine_0_BCA1__CXCL13 neutrophil_0_ABCA7 0.09543501 \n",
       "2 cytokine_0_CTACK__CCL27 neutrophil_0_ABCA7 0.05916357 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_receptor_correlations,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 470,
   "id": "266f760d-f838-4d9d-b9f8-440ee073303e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAADhAAAAJYCAIAAAC2c9oYAAAABmJLR0QA/wD/AP+gvaeTAAAg\nAElEQVR4nOzdebiWc/4H8M9Z6rTTotCGKEspMrI0kkj2hiIihGHMIMaQtdGMbZhpjLHMZBnL\n/GRksoxC2qQSFYkhRKlQiea0nZbT8/uj0KlTPafOOc95nl6vP1xX9/093+dzPp/7flxX17v7\nzkokEgEAAAAAAAAAAABA5spOdQEAAAAAAAAAAAAAlC2BUQAAAAAAAAAAAIAMJzAKAAAAAAAA\nAAAAkOEERgEAAAAAAAAAAAAynMAoAAAAAAAAAAAAQIYTGAUAAAAAAAAAAADIcAKjAAAAAAAA\nAAAAABlOYBQAAAAAAAAAAAAgwwmMAgAAAAAAAAAAAGQ4gVEAAAAAAAAAAACADCcwCgAAAAAA\nAAAAAJDhBEYBAAAAAAAAAAAAMpzAKAAAAAAAAAAAAECGExgFAAAAAAAAAAAAyHACowAAAAAA\nAAAAAAAZTmAUAAAAAAAAAAAAIMMJjAIAAAAAAAAAAABkOIFRAAAAAAAAAAAAgAwnMAoAAAAA\nAAAAAACQ4QRGAQAAAAAAAAAAADKcwCgAAAAAAAAAAABAhhMYBQAAAAAAAAAAAMhwAqMAAAAA\nAAAAAAAAGU5gFAAAAAAAAAAAACDDCYwCAAAAAAAAAAAAZDiBUQAAAAAAAAAAAIAMJzAKAAAA\nAAAAAAAAkOEERgEAAAAAAAAAAAAynMAoAAAAAAAAAAAAQIYTGAUAAAAAAAAAAADIcAKjAAAA\nAAAAAAAAABlOYBQAAAAAAAAAAAAgwwmMAgAAAAAAAAAAAGQ4gVEAAAAAAAAAAACADCcwCgAA\nAAAAAAAAAJDhBEYBAAAAAAAAAAAAMpzAKAAAAAAAAAAAAECGExgFAAAAAAAAAAAAyHACowAA\nAAAV13NnV8nKytrxwpdTXQgRq+cOv+vCzm2a1q1eOTsrKyurxmbGUuzg0neaayuvd8lrqS6E\nzFGmt0M63Gtf/7l91iF3z0x1GWVoux8xAAAAQEUkMAoAAABQ5t68eresrKysepeM3tSKmXcf\nlJWVlZXVbdDq0vjA1U92zcrKytr5V5v8QEpoyfDL2h97zcPDp37x7bJViVRXQyq4rSoU48hs\n5gsAAABQJgRGAQAAAGBLvnvm7oEzE1Gt/Y0jZny7vDCRSCSWPNQl1VUBAAAAAEDSclNdAAAA\nAACb1PXJgsSTqS6CiPhg6tTCiOhwUd+j9qi+lXuYJpSPinuvJRa998z9f338hdGT/jtzXqya\nsE+9+/Zqse8BR57Y/fTTOreuXznV9aWNijtiAAAAgIpNYBQAAAAAtmTp0qURkVOr1tamRYHt\n3apPBp5+1KXPzVn9w5E1BQtnThs/c9r4oY/f9vzAxS9fWCOF5QEAAACwHfBKegAAAICK67mz\nq2RlZe144cvrHcufOqj/+V3aNqtfq0rlvFoNdtvnwJ/+7Fd3PfPWlyvWnh/cIyur0jnPR0TM\nu69j1o926ztp/b1Xzhl935Xdf7pPwzrVK+fVqNu4Zcczrx04fv7qKEZi0eRHfnPaIXvWr1ml\nSs16TVt37v27Zz9Ztq68epe89v26RQ8enZWVlbVn30mJRVOfuKnX0a2b7lSrSk7WjmuXJJbN\nffu5+68/74TDWjdvVLd65bwadRrtd8RpV/xl+MyCDT9y/a3e/UffM4/Yp2GdannV6jTar1Pv\nO1+ZtbbMwnnjHvh1t0ObN6xdNa9a3d0PPu26Z6YvK0mDt9yEr/96ZFZWVlaXh5ZEROHTP/uh\nnd0GleSDIoqfZkQk396Ire1hLP3omVvOO7pV47rV8/Kq12t2cNerHpr0XTE15r/7RN/uh+21\nfiUfl6ijmyig2OthrRVfjLj3qh4dWjauVzOvcrXaDfftcGbfRyYtXFPsjitmj77/mp5Ht9m9\n/g5V86rVbbJ32049rxv42qeLExutTGrbHytc8+3bD11zxk/33rVOtbyqOzZsefT5dwydufKH\nhaV8WyXVmWQk3ZBSKaz0B7qBZC/vLY1jk/dayfuQ/L2zhW/miE/vOefS5+asztq54zWPjZ3+\n/p2HR7vb3vvy43fGvvSPWy85cf8GeVnF9SS5EZfffGMbRlxa843UjBgAAAAgIyQAAAAAKGMT\nft00IqLuxaM2teLzu9pGRMRpT61a//CQnnkRscMFw74/8PWQc5tt4pUxHR5YkEgkEolnztjE\n3wM1vfbtHzae9/KVbYt9lF3tw28c+23R2lbPfPKMpjkbrazS8vLLjs+LiLoXD/9+6XcPdIqI\n2P3MX3dvvP5P1LzolUQikZjWb79N1Fazbd+Ri4p86vdbnfWbHht/eFaDU5/8YtnkP3bcaaN/\nDl37qAc+LkxuLEk14at7OxRf8mlPbWbrjQa3yYMlae/W9vDsG87ba+NXXefte+XIxesvL5z9\nr3OKubqq7HfZZccVU0kStnA9JBIrP33q/BZVivllsnY+9t6py4ruVvDR4+c0L25xRLMb3ll/\nZfLb/lDhVac12vif1tc85OZx+WsXlvJttcXOJCP5hpRWYaU50GJvh2Qv7y2No/h7bSv6kOS9\nk8w3c+Lz29tGRDT+xYjliUQi8dWAw6PdXZ+XxojLbb7bOuLSmm8qRgwAAACQIQRGAQAAAMpc\naQVGV428tH5ERO0jrvrH6P/OXbh0xfLv5n787vgXHux7RruzHv3mh59b9cQpERENflnsB675\n5N6ONSIiKu1+8q3PvTNncUHB/76Y/K9+XRrmRETUOeHROT8uLvzg9naVIyJqHnDRAyOnL1i2\nYvnCT9945PJDan+frNk4MBoRUaX5GXc9987sRSvWS29+8IfjD+9xzX1Dxkz+cOa8xQXLvp37\nyVtD7jz/oNoREfV7D12yXpHrbVXrgJ8/MPyjeUsLFn/14fABXXevFBFRb/82jbJymxx/y7/e\nmrWoYPl3Myc9/Zt2tSMiqp/yRNHcZLFK1IREYtgF1SMi54whW945kShBYLRE7d2GHtbY/7x7\nXnpvbn7B8m9nvjPkpk67ZEdE7N3vvR8bMv2Ph1eJiKje6ty/DP9w3rIVy775ZMzff9G21iYq\nScIWrodFI37ZLDsisht06PPwa//9avGKlcvmfzT8/vP2rxERWbtfsn5i65uXejfJioio2eqs\nOwe/OWPBkhXLv5szfcrIp+646Ogud07dum3XqzB71843Pz35i/8VFOTPnvLv/ic0yYmI2Pmc\n5xd+v7gUb6stdCYZSTekFAsrzYEWe4+U5PLe3DiK23xr+7DleyfJb+Zxl+0SEXHCo2u7sOXA\naHIjLr/5JrZ5xKU132I3L9MRAwAAAGQMgVEAAACAMrcuMJqEzQdG1z6eLevIvy7YwgduNmqz\n7D9n14mIqH38Q7OLhIFWffznI6tFROzyy1Er1x1bOuSMWhERTS98+X9Fdlk6tk+LtS9QLi4w\nutMZz8xLqjWJRCKxfOwVzSKi0gmPrvcRP2zV7LIx+UVWD7+4wdozNY968PMiv8CseztUiojc\nroOWb+kzS9SERJkFRkvW3s3YfA/3+OXIotvnv9izXkTEfv2mrTuy7IWzakdENDzn+aJPmM1/\n7RfNogSVrG+z10Phu31bRERUO+wPH60uemrhs93rRkT2oX/8bN2RVW/22T0iospBN01esuFO\nW7/tjxXmHfaH/xa591bPuOfI6hER+978wfdFlNpttZV3yvq7Jt2Q0iys9Aa6ySdEFqvYy7uE\ngdGt7EMS906y38wzbj0gImKXXi/8L5HYcmA0yRGX23wTZTfiEs+3+M3LcsQAAAAAGWPjFy0B\nAAAAUEFVqVIlImL16lXbsMnq4U//+9uI2PvyOy8o+hbu3L0uu+PnTSPiq0FPjV57aNUrzzyf\nH5F9+NW/PbZWkW2qtb/5+i55m/iMfS65uVv9pCuq0v60E+pHrHrzzSkbncs64oobj6hZZPVR\nx3aoFBFR/6zrLtytyC/Q5Nhj94mI1R9++OkWPrJETSg7W9neYmy+h31u6lh0+5rHn3JUpYiY\n/tFH6yp59V9DvovIanf170+uXXRpp/7XH1Mp+UqKU9z1kBj70CPTI2KvK+/7dYucoufqnPq7\nK9tErJnw7+e/joiINaMeeezziGhy6V/6HVh9cx9Vom1/tNPZt/TZp8grxXP2+OXvLtw1Iv47\n6OlpW/4Vt/KKKtmd8qPkG1ImhW3zQEtoc5d3krauD8ncO5HsN/Mep/c8pFLEV4+fdUj3mx55\nafJXBZtbneSIy22+UYYjLoX5RhmPGAAAACBjCIwCAAAAlJckXkm/eXsceWSTiMQbNxzTo/+T\noz78ZsXWVPHfSZOWRUTDY7u03OhcdrvjOteJiIWTJn2+dvGUKQUR0eKooxputLh2p04HFP8R\nOxx22L6b+PSCma/dd83ZxxzUfJfaNfJystY54i/zI2Lh11+v3nD9rgccsGFwKXvXXRtERLRq\n0yZnw9W77hoRkZ+fv4mP/16JmlB2tq69W9HDBhsey27ceNeIWJ2fvywiIj6cPHl5RLQ45pgm\nG31evWOO2UQlSSr2evhk3Lj5EbFTp2PaFPNXlM0PPLB6RLz77rtryxs//ruIqH9i10M3nPm2\nbPuDyh2O6bBRKDbnsKM7VouIjydN2tL1tLVX1GbulM1LviFlUdi2D3QzSnp5J2nr+pDMvRNJ\nfzPv2WfQkxe0rB5LPhz8+wtOPPEPk+Pdu07qeOrFNz/4n/8uWrPB4iRHXG7zjVIacRnNN8pm\nxFtfDQAAAEBFlbvlJQAAAABUDNmH3PyPa0edcueUD57ud87T/SKv7l5tDmnf8diuZ511Yqu6\nSf7L4Pnz50dENG7cuLizTZo0jvh27aLdIxYsWBAR0bDhxoHGiF0bNsyO2DDnFBF16tQp9qO/\nfP7iTj3+/tEmn6tXUFAQUaPIoXWP7isiJydnE6fWnVmzppiiiihRE8rOVrS3lHoYubm5ET+2\nal1DGjVqVMyejRo3zo63ttTTTSv2epg7d25ExIK/d8odGBGRSCQ2+G9ELP322xURefH1119H\nROy++xbHUaJtf1C/UaNiHqKa3ajRLhEzYv78BRG1Nj6/vq27ojZ1p2xR8g0pi8K2faCbshWX\nd5K2rg/J3DuR/DdzTtPTH3qnw8XPPPzov/4zYtw7nywomP/+6CHvjx7y999f07LXXwb97bz9\nfuhNkiMut/lGaYy47OYbZTPirSwFAAAAoALzhFEAAACANFKr4x0TP5n45O8u6XpY83qVVyz8\nZOJLj95x+Sn7737AhU9/XliSnbKyspI4uy4FtMnFiWKPZmcX95dOCwddeu7fPyqIHdv2/uO/\nRr/3+deLlq1YvSaRSCQSE39TbMSnzCXXhLJT4vaWdQ9LOOgkFXs9FBauvV7XFK61Zs2aNWvW\n/irrfdjKleu/5DuJiWzNtqU46hJeUcXfKaX1ccmvLGFhpTbQDZTDV0RZ3fLJfzPnNvjJmdff\nP2T89PduPyz2PnfAX27q/dOGlROL33/s/M6/emXxVpZU9vONbR9x+fwvINXf6gAAAAAVnMAo\nAAAAQHrJbXBwzxsfGDJu+oL8bz4Z99z9fU/dt0Ysfu/hc87+y2fJ/PxOO+0UEfHFF18Ud/aL\nL2avtyjq168fETFnzpxi1n45d24JcoSLhz710v8idjz9kZEPX9W9Q6vdGuxQtXJOVkTEmvnz\nFya/UWkoURPKTknbW3Y9XFfJ7Nmzizk5d86cbQqMFqtBgwYRET+5c1ZiM14+r0ZExC677BIR\n8dlnW7zES7TtD+bNnl1MyG3NnDlfRUTUr7/lC6Gcr6jkG1JuhW1d59dXpl8RZd+Hrflm3qFl\n18v6P/z6x+Oub1M54svHH3j+h9fcJzni8rzwtnHEZf2/gAryrQ4AAABQwQmMAgAAAKSrvLp7\nHnbKL25/dtyDXWtErBr/0qvfrTuTvfbV7EWe+/a9fQ86qFpEzH3llQ82Opd4+5Xh30ZE3YMO\n2m3t4gMPrBIR00eOnLvR4u9GjHinBNV+NXv26oho8ZOfbPhy78Jxw0ctK+5Hyk6JmlCGZZSw\nvWXXw33atq0aEdOHD984MfrNq6+WZNBJ2rd9+9oR8e4rr8zf8uK9DzusdkTM/89zE7bwHN0S\nbfuDlWOGj9koMVo4/rVRyyKi+UEHrW13qd1W264EDSmvwrau8+sr4eW9uXEUU145DmjT38yb\nUO2gXt32jYiVs2fP+/5YkiMuz99rG0dcpvONCvOtDgAAAFDBCYwCAAAApLta9erlRUSsWfN9\nsia7Vq0aEbHom282jhpV6tzj1DoR8dFfrvtH0UdYFs6477q/zYyIXXqceWTWusXdTq4VsWbc\n3b99Jb/INsve6H/byytKUOUOO+wQETF98uQlRY6v/OBP1z5U7CPhylCJmlCWZZSsvWXXw0qd\nT//ZjhGJiX+88cWi4bbFI/vd9trKbdm7eDkdL+rdLGLVyP6XP/3lmi0szu7Y+7w9IuKL+y+/\nZcpmg7El2vZHC5787Z8/XL3+kcLP77v5oS8jYp8zTm+1rorSuq22XfINKbfCtrLz6ynh5b25\ncWwsFbf8xt/Mm7Lwrbc/i4jcXXet//2hJEdcnr/XNo64TOcbFeZbHQAAAKCCExgFAAAASBv/\nuazVkWdd/aenhr817dO53y5buWLxgs/fHnJH90v+sTAip23nTnW+X9myVauIWPnq3+4aM+Pb\n5UXzNlWP73fLkdUjvn3x5x1Ou/2FqV8uWbly8Zx3Bvc/uWOfEUsj6pxw23VHVlq3uPrJN117\nUKWIWQ91P/LnD47++Jvlq1Z899m4R/t0OvnP00v0nvIGnbu0yY5YNOgXXW96dvLs7wpWLps/\nfcTAKzp0uHZCOT9fNErYhLJTwvaWYQ+rnnDTDYfmRcx5/MwO5/91xPQFy1cWfDtj7MBLj/rZ\n/TNK/4X0EZF70I1/+9VeuTHn6Z4Hdrj03ucnfjp/ycrVK/LnfzH93QnDHrvjstMOPer3U9Yt\nzjn4xr9e0CQrCib97shDzr7r3xM/X7hsZcH/vvzk3VGD/nBx5+P/8N5WbfujFeP6Hn1iv2em\nzFm8cuWSue8+9/uuHa8atTSifs9br9jv+0Wldlttu+QbUm6FbWXnf1TSy3tz49hYWfYhuW/m\nsde2a9/jqrufGjF5+hcLlqxKRERixTcfjx98+5kdLn0+P2KHrmefWP2HTZMccXleeNs24rKd\nb1SYb3UAAACACi4BAAAAQBmb8OumERF1Lx61qRWf39U2IiJOe2rV+oeH9MyLiB0uGLb+H4uT\ntXPne6atWO8n5/6jy4av/W167ds/nJ738pUHVi9uo9qH3Tj226K1rfrssW6NN/5nx1VaXv6r\nLpUjYpfLxny/9LsHOkVENFvvk9azZOy1rats/JG5u/e69qzGERHHPro4kcRWb69t5wnrrV5n\n+RMnREQ0uGLsphq9vpI0YdgF1SMi54whyWy80eA2c7Ak7U2UXg8Tb1/bLCLi2IE/Li+c9dRZ\nu+dstHfevr+67Li8iKh78fDkfvukClj7y88ccvmBG16oPyr6gwUfPdZzz+LvgGY3vLN1266r\ncPcefU5tuPEQahx8w+v/W3/jUrutttiZZCTfkNIqrDQHWtztUKLLe3PjKP5eK6U+bHzvJPfN\nPOriuuufyc0tcrPVPPDKYfM3/KgkR1xu801s24hLbb7FbV6arfhhxJvuAwAAAEC68oRRAAAA\ngLRx0l/fH/3Un67ueWy7VnvuWrtqbm6VHRrs+ZMu5//2yUkfDLu8ZeX1lu567tNjHru62+Et\ndtmhSu7G7+Ctf+yfJnw48q9XnHZ4i112rFqpUrXau+7b4Yzf/P2ND8f8rn3tomtzd+/1r3ff\nfOjXPzt4j3rVK1euXqdxq6PP6z/4vYl3HJi9MiJ23HHH5Mqv3v6OsW8+ce1phzSrV61SbpVa\n9Xdrc+x5v3166tRiE5NlryRNKDsla29Z9jC7SY8np7z1j2tObdesXvXKlavVbtTyqHNuHvTu\nW/ceVWfLP71Vcpt2vWfip5OeuOWik9o136V29cqVqu64y+77HND+pN433D9k4uib2q63OK9F\nryff++iVe/p0P2LfhnWqV65cY6em+7Q9uud1A4e/fG3rrd42IiK7ac9n3h3/4JWnHtp85x2q\nVsqrtfO+Hc+99cWpr//+p0USa6V4W5WC5BtSboWVuPNFlfDy3sI4NlZWfUjum/mIuya/OeSB\nfj//WceftNyzYe3KicLIqlxr570O6nLu9Q+N/fDNP3XZacONkxxxeV542zLisp5vVJRvdQAA\nAIAKLCuRKJM3SgEAAACQ4VaP/HnDTgPn5576f/nPnlk11dVkHO0tD4sePLr2L0ZEs2vf/vSO\ng1JdDNuVOXcf0nhwt8/fvHq3VFcCAAAAwHbEE0YBAAAA2AqFH9/3u/+bH5HbvnNHccZSp72Q\n2bKyknxoJgAAAACUHoFRAAAAADbv1d8ceXa/h1568/3P5i1esapg0ez3Xvlbn44/vWr00oiG\n517Tc+dUF5jetBe2QwKjAAAAAKRAbqoLAAAAAKCCW/nVm//855h/9t/oRN2f9h884LgaKSgp\nk2gvbIcERgEAAABIAU8YBQAAAGDzjv3j6//8/cWntG/dvHG9GpUrVam1024HHHP29Q+P/2DE\nTYfUTHV1aS/d2jv6V/WyknDkX79JdaXpR2+3I7tcOS7x5tW7pboMAAAAALYvnjAKAAAAwOZV\nanDwWTccfNYNqa4jQ2lvCu14yWuJS1JdBAAAAABA+chKJBKprgEAAAAAAAAAAACAMuSV9AAA\nAAAAAAAAAAAZTmAUAAAAAAAAAAAAIMMJjAIAAAAAAAAAAABkOIFRAAAAAAAAAAAAgAwnMAoA\nAAAAAAAAAACQ4QRGAQAAAAAAAAAAADKcwCgAAAAAAAAAAABAhhMYBQAAAAAAAAAAAMhwAqMA\nAAAAAAAAAAAAGU5gFAAAAAAAAAAAACDDCYwCAAAAAAAAAAAAZDiBUQAAAAAAAAAAAIAMJzAK\nAAAAAAAAAAAAkOEERgEAAAAAAAAAAAAynMAoAAAAAAAAAAAAQIYTGAUAAAAAAAAAAADIcAKj\nAAAAAAAAAAAAABlOYBQAAAAAAAAAAAAgwwmMAgAAAAAAAAAAAGQ4gVEAAAAAAAAAAACADCcw\nCgAAAAAAAAAAAJDhBEYBAAAAAAAAAAAAMpzAKAAAAAAAAAAAAECGExgFAAAAAAAAAAAAyHAC\nowAAAAAAAAAAAAAZTmAUAAAAAAAAAAAAIMMJjAIAAAAAAAAAAABkuNxUF1Byy2aNHvzPwUNf\nnzxt+qwFixYvW12pWs0d6jdt0bLtEcd369n9yKZVU10hAAAAAAAAAAAAQEWSlUgkUl1DCSwY\n1f/Mc24dMXflphZUbtjphicG3dyxXnlWBQAAAAAAAAAAAFCRpVNgdPW029r95IYpK6LGnl16\n9e52TLtWzRrWrVUld3VB/sK5M6ZNHD74kcde/nRp5LW9bdKb17VMw4enAgAAAAAAAAAAAJSB\nNAqMLnu6e/0eg5fu3PXhcYN675FX3JIVMx7p0f6C576u3m3Q/GfOqFbeFQIAAAAAAAAAAABU\nRNmpLiB5b48evTSiTZ+7N5EWjYi8Zr3vuqJ1xNIxYyaVa20AAAAAAAAAAAAAFVcaBUbz8/Mj\nolGjRptdtfb82rUAAAAAAAAAAAAApFVgtHHjxhHx9tixBZtZVPDGG5MiokmTJuVUFQAAAAAA\nAAAAAEBFl0aB0dbdT2+eFfMG9j79nvHzVhezYPW88fec3nvgvMhq0f20/cu9PgAAAAAAAAAA\nAICKKSuRSKS6hqQtGde3/bF3Tl0akVe/1eEdDm7ZrGG9mnk5hSsWfzN3xvtvjRk3bf6KiOpt\n+r7yxu2HV091tQAAAAAAAAAAAAAVQ1oFRiMi/71Hb+hz80Oj5hT7XvoqjTpe2P/Pt56/f63y\nrgsAAAAAAAAAAACgwkq3wGhERKxZMnPiqNcnTfv4i/mLliwvzKlaY8f6TZq3OqhDx3ZNa2Sn\nujoAAAAAAAAAAACAiiUtA6MAAAAAAAAAAAAAJM/zOAEAAAAAAAAAAAAyXG6qCyi5ZbNGD/7n\n4KGvT542fdaCRYuXra5UreYO9Zu2aNn2iOO79ex+ZNOqqa4QAAAAAAAAAAAAoCJJs1fSLxjV\n/8xzbh0xd+WmFlRu2OmGJwbd3LFeeVYFAAAAAAAAAAAAUJGlU2B09bTb2v3khikrosaeXXr1\n7nZMu1bNGtatVSV3dUH+wrkzpk0cPviRx17+dGnktb1t0pvXtUzDh6cCAAAAAAAAAAAAlIE0\nCowue7p7/R6Dl+7c9eFxg3rvkVfckhUzHunR/oLnvq7ebdD8Z86oVt4VAgAAAAAAAAAAAFRE\n2akuIHlvjx69NKJNn7s3kRaNiLxmve+6onXE0jFjJpVrbQAAAAAAAAAAAAAVVxoFRvPz8yOi\nUaNGm1219vzatQAAAAAAAAAAAABE5Ka6gOQ1btw4YsbbY8cW9Dy6yqYWFbzxxqSIaNKkSTlW\nBgAAAABsbwoLC4cOHVpQUJDqQrYjVapUOf7443NyclJdCAAAAACkpTQKjLbufnrzP9z+8cDe\np+8zaOAvD2uwUemr542/76LeA+dFVovup+2fihIBAAAAgO3EsGHDTj755FRXsd158cUXTzzx\nxFRXAQAAAABpKY0Co1ltr3/kmqHH3jn1xT6HN72t1eEdDm7ZrGG9mnk5hSsWfzN3xvtvjRk3\nbf6KiOpt+j58fdtUVwsAAAAAZLLly5dHRK9evVq3bp3qWrYLU6dOffzxx9e2HQAAAADYCmkU\nGI2ocfgdr49vcUOfmx8aNWfayGemjdxwQZVGHS/s/+dbz9+/eirKAwAAAICU8pL08jRhwoSI\naN26defOnVNdCwAAAADAlqVVYDQiau1//r0jz7195sRRr0+a9vEX8xctWV6YU7XGjvWbNG91\nUIeO7ZrWyE51iQAAAACQGl6SDgAAAADApqRbYDQiIrJr7HboSbsdelKq6+yJHuUAACAASURB\nVAAAAACAisRL0svTCy+8MGbMmFRXAQAAAACQrLQMjAIAAAAAm+Il6eVj6tSpqS4BAAAAAKAE\nvMAdAAAAAAAAAAAAIMOl4RNGl80aPfifg4e+Pnna9FkLFi1etrpStZo71G/aomXbI47v1rP7\nkU2rprpCAAAAAAAAAAAAgIokzQKjC0b1P/OcW0fMXVnk6NIl3y34auYHb77y9H23XN/phicG\n3dyxXooKBAAAAAAAAAAAAKhw0ikwunrabV2O6zdlRdTYs0uv3t2OadeqWcO6tarkri7IXzh3\nxrSJwwc/8tjLn47od1yXSpPevK5lOv1qAAAAAAAAAAAAAGUnjVKVy57tf9uUFbFz14fHDeq9\nR16Rc81atDr4qK4XXHXVIz3aX/Dc5FtvefaKZ86olqJCAQAAAAAAAAAAACqUNAqMvj169NKI\nNn3u3jAt+qO8Zr3vuuIvz103dcyYSXHGEUlsWlhYOHTo0IKCgs2sWbNmzfTp0/v27Vu5cuWt\nqBsAAAAAAAAAAAAgtdIoMJqfnx8RjRo12uyqRo0aRUxduzYJo0aNOvnkk5NZmZ2dfeONNya3\nKwAAAAAAAAAAAEAFkkaB0caNG0fMeHvs2IKeR1fZ1KKCN96YFBFNmjRJbtOOHTu+8MILm3/C\n6IQJEwYMGLDXXnuVrF4AAAAAAAAAAACAiiGNAqOtu5/e/A+3fzyw9+n7DBr4y8MabFT66nnj\n77uo98B5kdWi+2n7J7dpTk7OSSedtMVlAwYMyM7OLnHJAAAAAAAAAAAAABVAGgVGs9pe/8g1\nQ4+9c+qLfQ5velurwzsc3LJZw3o183IKVyz+Zu6M998aM27a/BUR1dv0ffj6tqmuFgAAAAAA\nAAAAAKCiSKPAaESNw+94fXyLG/rc/NCoOdNGPjNt5IYLqjTqeGH/P996/v7VU1EeAAAAAAAA\nAAAAQIWUVoHRiKi1//n3jjz39pkTR70+adrHX8xftGR5YU7VGjvWb9K81UEdOrZrWsOL4wEA\nAAAAAAAAAACKSLfAaEREZNfY7dCTdjv0pFTXAQAAAAAAAAAAAJAOPI8TAAAAAAAAAAAAIMOl\n5RNGiyj48q1hL439YE5+1Gy0b/vjjzukYdVUlwQAAAAAAAAAAABQkaRRYHTmiIde+zx2P/rC\nTrt9fyj/rT+e2e2GobNX/LCocuPjbnv26V//pGYqKgQAAAAAAAAAAACoiNLolfSTHrjooosu\nemDSDwe+fPKs464eOntFzk4H/OyCX/3qgp8dUD9n5exhVx/X6+n5KawTAAAAAAAAAAAAoGJJ\noyeMbiAx/u4bX/o2svb6+csTHjy6blZEJBa+dvEhnQd++ly/e94549YDUl0hAAAAAAAAAAAA\nQIWQRk8Y3cB/hw2bFVH9lH63r02LRkRW3aNv73dy9Yjpw4Z9ltrqAAAAAAAAAAAAACqM9A2M\nzpw5MyJaduhQZ/2jdTt02C8iPv3005QUBQAAAAAAAAAAAFDxpG9gNDc3NyLq1q1b9PBOO+0U\nEatWrUpFTQAAAAAAAAAAAAAVUG6qCyipxV9+9NFHERHVd2keMWXWrFkRe693/ssvv4yIxo0b\np6Y8AAAAAAAAAAAAgAon7QKjr16xzz7r/XH66NFf37j3zj/8edWHH86IqNa69Z7lXxoAAAAA\nAAAAAABAhZRGgdEG+3fqtGijo1lTR30ZZ+667k/Lnn/i3/lRs2fPk6qVb3EAAAAAAAAAAAAA\nFVYaBUZ/evNrr21pTX7DU+55tMuuhx8vLwoAAAAAAAAAAACwThoFRpOx86FnnndoqosAAAAA\nAAAAAAAAqFCyU10AAAAAAAAAAAAAAGUrHQOjq/K/nPHp5wuWJ4o9O3/aa6+99tq0+eVcFAAA\nAAAAAAAAAEBFlWaB0cVTHujZZuc6Dffca4/6dRofcenDU/I3XPL6Lcccc8wxt7yeivIAAAAA\nAAAAAAAAKqC0CozOfbzH0Zf+39RvCyOrSo1qq+aOfeDCww46+8kZq1NdGAAAAAAAAAAAAEAF\nlkaB0TVj/3Dj0O8ie7duD035dvHiJd9Nf+7Go+t+9s9zO575xMzCVFcHAAAAAAAAAAAAUFGl\nUWD0g5dfnh1R+8wBj1xwwI65kVWz+Sm/e/mtZy/eZ/7g8zv2enr2mlQXCAAAAAAAAAAAAFAh\npVFgdNasWRFxQIcONX88ltPwlAdHDr5ozzn/d3anC4d8lUhZcQAAAAAAAAAAAAAVVhoFRvPy\n8iIiO3vDkuuf+OCrj3Xf5dNHexz9y2ELUlEZAAAAAAAAAAAAQEWWm+oCkte0adOIabNmzYrY\nv+iZ7CZnPfnqwkXtL3/gtM7Vz6+bmvIAAAAAAAAAAAAAKqg0esJos0MP3Snis4kTvynmZOW9\nLxsyrF+7nHfvvn9EuVcGAAAAAAAAAAAAUJGlUWA0p+Opp9SOwlHPPvddseer/+S3Q5+/bL+8\nci4LAAAAAAAAAAAAoIJLo1fSR+5RfZ95osNXlfdcsakVdY6659Vh+zw+8X+xd6vyrAwAAAAA\nAAAAAACgAkunwGhUatbp7GabX5K1a8df9O1YPuUAAAAAAAAAAAAApIU0eiU9AAAAAAAAAAAA\nAFtDYBQAAAAAAAAAAAAgwwmMAgAAAAAAAAAAAGQ4gVEAAAAAAAAAAACADJeb6gIAAAAAAGAL\n1qxZExETJkxIdSHbkSpVqhx//PE5OTmpLgQAAACA0iEwCgAAAABARffRRx9FxIABAwYMGJDq\nWrYjL7744oknnpjqKgAAAAAoHQKjAAAAAABUdKtWrYqIXr16tW7dOtW1bBemTp36+OOPL1++\nPNWFAAAAAFBqBEYBAAAAAEgPrVu37ty5c6qrAAAAAIC0lJ3qAgAAAAAAAAAAAAAoWwKjAAAA\nAAAAAAAAABlOYBQAAAAAAAAAAAAgwwmMAgAAAAAAAAAAAGQ4gVEAAAAAAAAAAACADCcwCgAA\nAAAAAAAAAJDhBEYBAAAAAAAAAAAAMpzAKAAAAAAAAAAAAECGy011AQAAAABkrMLCwqFDhxYU\nFKS6kO3FhAkTUl0CAAAAAAAVlMAoAAAAAGVl2LBhJ598cqqrAAAAAAAABEYBAAAAKDPLly+P\niF69erVu3TrVtWwXXnjhhTFjxqS6CgAAAAAAKiKBUQAAAADKVuvWrTt37pzqKrYLU6dOTXUJ\nAAAAAABUUNmpLgAAAAAAAAAAAACAsiUwCgAAAAAAAAAAAJDhBEYBAAAAAAAAAAAAMpzAKAAA\nAAAAAAAAAECGExgFAAAAAAAAAAAAyHACowAAAAAAAAAAAAAZTmAUAAAAAAAAAAAAIMMJjAIA\nAAAAAAAAAABkOIFRAAAAAAAAAAAAgAwnMAoAAAAAAAAAAACQ4QRGAQAAAAAAAAAAADKcwCgA\nAAAAAAAAAABAhhMYBQAAAAAAAAAAAMhwuakuIMUKCwuHDh1aUFCwmTUTJkyIiDVr1pRXUQAA\nAAAAAAAAAAClaXsPjI4aNerkk09OZuUHH3xQ1sUAAAAAAAAAAAAAlIXtPTDasWPHF154YYtP\nGB0wYMB+++1XblUBAAAAAAAAAAAAlKLtPTCak5Nz0kknbXHZgAEDsrOzy6EeAAAAAAAAAAAA\ngFInBAkAAAAAAAAAAACQ4QRGAQAAAAAAAAAAADKcwCgAAAAAAAAAAABAhhMYBQAAAAAAAAAA\nAMhwAqMAAAAAAAAAAAAAGU5gFAAAAAAAAAAAACDDCYwCAAAAAAAAAAAAZDiBUQAAAAAAAAAA\nAIAMJzAKAAAAAAAAAAAAkOEERgEAAAAAAAAAAAAynMAoAAAAAAAAAAAAQIYTGAUAAAAAAAAA\nAADIcAKjAAAAAAAAAAAAABlOYBQAAAAAAAAAAAAgwwmMAgAAAAAAAAAAAGQ4gVEAAAAAAAAA\nAACADCcwCgAAAAAAAAAAAJDhBEYBAAAAAAAAAAAAMpzAKAAAAAAAAAAAAECGExgFAAAAAAAA\nAAAAyHACowAAAAAAAAAAAAAZTmAUAAAAAAAAAAAAIMMJjAIAAAAAAAAAAABkOIFRAAAAAAAA\nAAAAgAwnMAoAAAAAAAAAAACQ4QRGAQAAAAAAAAAAADKcwCgAAAAAAAAAAABAhhMYBQAAAAAA\nAAAAAMhwAqMAAAAAAAAAAAAAGU5gFAAAAAAAAAAAACDDCYwCAAAAAAAAAAAAZDiBUQAAAAAA\nAAAAAIAMJzAKAAAAAAAAAAAAkOEERgEAAAAAAAAAAAAyXG6qCyi5ZbNGD/7n4KGvT542fdaC\nRYuXra5UreYO9Zu2aNn2iOO79ex+ZNOqqa4QAAAAAAAAAAAAoCJJs8DoglH9zzzn1hFzVxY5\nunTJdwu+mvnBm688fd8t13e64YlBN3esl6ICAQAAAAAAAAAAACqcdAqMrp52W5fj+k1ZETX2\n7NKrd7dj2rVq1rBurSq5qwvyF86dMW3i8MGPPPbypyP6Hdel0qQ3r2uZTr8aAAAAAAAAAAAA\nQNlJo1Tlsmf73zZlRezc9eFxg3rvkVfkXLMWrQ4+qusFV131SI/2Fzw3+dZbnr3imTOqpahQ\nAAAAAAAAAAAAgAolO9UFJO/t0aOXRrTpc/eGadEf5TXrfdcVrSOWjhkzqVxrAwAAAAAAAAAA\nAKi40igwmp+fHxGNGjXa7Kq159euBQAAAAAAAAAAACDpwOgb91x27yufLinbWragcePGEfH2\n2LEFm1lU8MYbkyKiSZMm5VQVAAAAAAAAAAAAQEWXZGD067F/vbxLi0b7nXL1wNGzNhfYLEOt\nu5/ePCvmDex9+j3j560uZsHqeePvOb33wHmR1aL7afuXe30AAAAAAAAAAAAAFVNucsuOuPL+\nX6+6/9GXXvjjz1/48/VtTr3kyit/2ePQnSuXbXFFZbW9/pFrhh5759QX+xze9LZWh3c4uGWz\nhvVq5uUUrlj8zdwZ7781Zty0+Ssiqrfp+/D1bcuzMgAAAAAAAAAAAICKLMnAaP3Df3H387/4\n3RevD3rwvvsfHvLM7/+fvXsNr7K68wa8koAECGhFqUoCKAp1TNg6KkqxYBQjMJBqRcGiseBp\nbHVA27GKB0YsoNY22paeUPoK2uIA1sI0aJFDEIhosE0jChQ6oCIiVm1AEg4J7wccFVTYaPZ+\nspP7/sRe68+TXxbXlS/8sp4rpt/3/TOGfufGG//9opOPiPMhX1hWr3sWLe1226g7H1rweuX8\n6ZXz9x3IzM6/auwD44Z3b52kRAAAAAAAAAAAAAAN30F1PVt27D18fO/h//Xm89Mn/fznv3p8\nyh1Dp4zLOfvyG24cdfXAfzksztfbfyFtuw//6fwrJqxbtmBReeXqV996b2t1bUbLrMPad+ya\nd1qf/DM6ZSUjBQAAAAAAAAAAAEAK+TyXgx5yVI9hd/QYNvpHFX+Y9INb75ox6eaFk+46/rzL\nrrnu+mu+nnto4hub6Vmdew7q3HNQwr8QAAAAAAAAAAAAQCPwedudOzcvn/nrnzz4iz+urgkh\n44gTOlQt/NXN3+h+7OnffvLVunpNCAAAAAAAAAAAAMAXctCF0e1vPDf1vy4/s1POaUNGT16y\n5YQLb35o4f++tnrVq3+bc+vXvvTui7+4/DtT305E0g9tW79wyvjrh/bredKxx7T/UpusNoe3\nP+bY3J79hl4/fsrC9dUJ/doAAAAAAAAAAAAAqSf+V9JXry/93S8nTnzo9y++vSukH3pi4U3/\nMfKGonM6t/pgv1O/8Y/dPq/jTc+Xlr4YrihITNzNC8Zeevm4eRt27LX6/tZ3N29ct+K5px+f\neNfoc2+bOu3O/CMS8/UBAAAAAAAAAAAAUk+chdGld5w0cPzL79aFtLbd+o+8YeR/fOv841p/\nYionJyeE57dUVdVzyA/sqhzfr/+YF7eHrOP7FY0YfN4ZeV06tGub2WxXTdU/NqytXDZ3xuRH\nnlozb0z/fs3Ln7s1N/4uLAAAAAAAAAAAAEBjFmer8o1XXtlxXMG3bxg5cnj/rm3SPmvszBt/\n97vBIefM+kq3l20zx45/cXs46oKHl0wbcVyLvfa6dMvrcc4FV9500+ShZ1355PJxd80cOX1I\nq894EAAAAAAAAAAAAECTEmdhtNe4FRtOOPHQ9AOMZfcaOvQLR/osLyxc+H4IJ4+6f9+26Eda\ndBnxw5E/efLWitLS8jCkd8KiAAAAAAAAAAAAAKSQOAujR3fr+uZfF/75nWZHx87q9qWP1t9d\ntbhi467Dv/K17kdlJCbgR6qqqkII2dnZ+53Kzs4OoWLPLAAAAAAAcPDq6upCCGVlZVEHaUIy\nMzMHDBiQkZHw/24BAAAAmqw4C6Ph3ce/3WvY75tfOvPVs7p9bLn5SxO/MXjazsHTXps+5LBE\n5PuYnJycENa+8OyzNcP6Zn7WUM3ixeUhhI4dOyY4DQAAAAAANFYrV64MIRQXFxcXF0edpQmZ\nPXv2wIEDo04BAAAANFpxFkbfeuzXf9ga2l9/7QVZe61nXfjvw9pP+9kTv/7d20OuOyIB+T4m\ndvElXe+bsHrSiEtOnDbpO1/98iei79q0dOLVIyZtCmndLr6oe2LDAAAAAABAo7Vz584QQlFR\nUSwWizpLk1BRUTFlypTq6uqogwAAAACNWZyF0VdWrKgL4aS8vPR9NtJzc/8lhIUvv7wqhAQX\nRtNOHT355pLz762YPapXp/F5vfr0yO3S4Yg2LTJqt295e8Pal54vXVL51vYQWp98y8OjT01s\nFgAAAAAAaOxisVhBQUHUKQAAAACoH3EWRrds2bLf/XfffbcewhxIVq97Fi3tdtuoOx9a8Hrl\n/OmV8/cdyMzOv2rsA+OGd2+dhDQAAAAAAAAAAAAAqSHOwmh2dnYIa1956aW60HuvS0ZrV6xY\nGUI45phjEhDuU7TtPvyn86+YsG7ZgkXllatffeu9rdW1GS2zDmvfsWveaX3yz+iUte8dqAAA\nAAAAAAAAAABNXJyF0by+fdvfu/bNyT+Y9P2nr+2Q9n/LuzdM+sHkN0M4om/fWKISfor0rM49\nB3XuOSiJXxIAAAAAAAAAAAAgZcV5H2dG3+/eembL8P7cG3p//Y4pc19Y8cqKF+ZOuaOw93/M\nfT+0PGP0987LSGxOAAAAAAAAAAAAAD6nOG8YDeGEkdN/t/L8S3/18uwfXDH7Bx+ttzrp3387\nY1TXtM/+m/Vt2/qFMx6bUbJoeeWq9Zvf27JtV/NWbQ5t36lb7qm9BwwedvHZnVomLwsAAAAA\nAAAAAABAwxd3YTSkZX/9l8tXXvKbn0/549IVr72745DDO+Z+9d+Krhue37FFAgPubfOCsZde\nPm7ehh17rb6/9d3NG9eteO7pxyfeNfrc26ZOuzP/iKRFAgAAAAAAAAAAAGjg4i+MhhBCZsdz\nrrvnnOsSlOVAdlWO79d/zIvbQ9bx/YpGDD7vjLwuHdq1zWy2q6bqHxvWVi6bO2PyI0+tmTem\nf7/m5c/dmntw3xoAAAAAAAAAAABAY5VCrcptM8eOf3F7OOqCh5dMG3Hc3readumW1+OcC668\n6abJQ8+68snl4+6aOXL6kFYRBQUAAAAAAAAAAABoUA6qMFr9atlTf3ru5dfe3rq9dvfeW3mX\n3TMstz6DfdILCxe+H8LJo+7fty36kRZdRvxw5E+evLWitLQ8DOkdx0Nra2tLSkpqamr2M1NW\nVhZCqKurO/jMAAAAAAAAAAAAANGLuzC6ZdmECy68Y/7G2k/fvui0hBdGq6qqQgjZ2dn7ncrO\nzg6hYs9sHBYsWFBYWBjP5IoVK+J7JAAAAAAAAAAAAEDDEmdhdGfp6ItHz98YMo8dcM3V/btn\nH9Yibe+BTj3qP9s+cnJyQlj7wrPP1gzrm/lZQzWLF5eHEDp27BjfQ/Pz82fNmnXAG0aLi4tP\nOumkg8sLAAAAAAAAAAAA0DDEWRgtf+KJ10Joe+HDzz3xzfaJTfSZYhdf0vW+CasnjbjkxGmT\nvvPVL38i+q5NSydePWLSppDW7eKLusf30IyMjEGDBh1wrLi4OD09/aAjAwAAAA1MbW1tSUnJ\n/n93lHpUVlYWdQQAAAAAACCEuAujmzdvDiH0+Ld/i6otGkJIO3X05JtLzr+3YvaoXp3G5/Xq\n0yO3S4cj2rTIqN2+5e0Na196vnRJ5VvbQ2h98i0Pjz41upwAAABAwzVnzpzCwsKoUwAAAAAA\nACRbnIXR7OzsEP53x44diU1zAFm97lm0tNtto+58aMHrlfOnV87fdyAzO/+qsQ+MG969dRTx\nAAAAgAavuro6hFBUVBSLxaLO0iTMmjWrtLQ06hQAAAAAAEC8hdFThg478Uc/+Mv8+e9eN+RL\niU20f227D//p/CsmrFu2YFF55epX33pva3VtRsusw9p37Jp3Wp/8MzpleXE8AAAAcACxWKyg\noCDqFE1CRUVF1BEAAAAAAIAQ4i6Mpv3rbY9NWHTeLd+7YmKXR75zWqSd0RDSszr3HNS556Bo\nUwAAAAAAAAAAAACkhjgLo/O/f9bN83a0bPn67OtP//JtHU847siWe9/kee595feek4B8AAAA\nAAAAAAAAAHxBcRZG31m7fPnyD/6885+vvvznV/cZ6PxOfaY6GDVvPD/nj8+ueL0qtMn+l7MG\n9D+zQ8uoogAAAAAAAAAAAAA0RHEWRgc9tHHjz/Y3kJn4t9Svm/fQM/8bju171bmd/2+p6vkf\nXTr4tpLXtn84dEhO//EzH//u6W0SHgcAAAAAAAAAAAAgRcRZGG1x2FFHJTbIgZX/4uqrZ4aL\npn9YGH3j0W/2/17JOyHjyFMKC3t1CBuWzJ7159fmfK9/UfbLvx/SPtKwAAAAAAAAAAAAAA1G\nnIXRBmj30vtv/+M7Ie2Ea54q+2XfdmkhhN3/eObaMwsmrXlyzIN/HjLulKgTAgAAAAAAAAAA\nADQI6Qcz/P7q/3ngxssG5n+1x+mnfX9+CCG8uey/H3300XlrdyYm3f68PGfO+hBaf33MhD1t\n0RBCWru+E8YUtg5h1Zw5f09+IgAAAAAAAAAAAIAGKf4bRt/8wzX5Qyat3P7Bx47vhBBC6zW/\nu/byJ5sVHbrpkUGZicj32datWxdCyO3T5/CPr7br0+ek8Ifn16xZE8JxyQ0EAAAAAAAAAAAA\n0DDFe8Po5qnXXjZp5a7OQ365+Mf9P1puc+Flha1C1e9/V7L9s/9uYjRr1iyE0K5du72Xjzzy\nyBDCzp0R3HkKAAAAAAAAAAAA0CDFWRjd+NivZm0NbYf8+OFre3Vs/bGNVrm5x4WwZenSyoTE\n+6Qtb6zco/XRXUMI69ev33v/jTfeCCHk5OQkKQ8AAAAAAAAAAABAQxfnK+n/WlERQvjX3r1b\n77uTk5MTwksbN26s52Cf5U8jTzzxYx9XLVz45u1fOerDzztfeWVtCK1iseOTlAcAAAAAAAAA\nAACgoYuzMLpjx44QQlZWVgghLS0thN3/t7N169YQQvPmzRMS7+O+3P3cc9/7xGpaxYI3wqXH\nfPBp2x+mPlEV2gwbNqhVwvMAAAAAAAAAAAAApIY4C6NHH310COtff/31ELrstVH78surQwid\nO3eu92j7+tqdzzxzoJmqDl9/8Df9juk1QF8UAAAAAAAAAAAA4APp8Y2d3LdvuxAqfv/E33fv\nuWF0j91vTf3x1E0hdDr//G6JSnhQjup56be+9a2CEw6JOggAAAAAAAAAAABAgxFnYbTZ+f85\nukfL3cvGXHj1r5/bHEIIVX+b99u7L/rqNbOrQtu+d4zqmXagRwAAAAAAAAAAAAAQiThfSR/S\nut04c9raAd/8+cPX/jWEEMLc0X3nhhBC69io/370ypxE5Tt4a5/62Zw14fj+1/frEnUUAAAA\nAAAAAAAAgIYg3sJoCGnZhRPLVw2d8stH/mdx5fq3azIO63BSr0HDr7/y7OwG9QL4Pz90ww0z\nw0XTFUYBAAAAAAAAAAAAQggHUxgNIYRDOnztqru/dlWCsgAAAAAAAAAAAACQAAdXGI1Ubc3W\n6l0HHqupTXwUAAAAAAAAAAAAgBQSZ2H073N/+ae1+xvoUvDv5x1XH4E+2+8va3PxzMR+CQAA\nAAAAAAAAAIBGKM7C6Iu/uu66/ZY1L5qe8MIoAAAAAAAAAAAAAJ9HnIXR3EvvvvvkvVbqqjev\nXVYyY96atJOHjbroK6fkJiDc3o49tnMI606ZsPr57x27n7EnhjQf8kTCwwAAAAAAAAAAAACk\njDgLo1+56PbbP2X5nlt+OOD0m0uWfXvs2K/Ua6xP86/nnXfE/ZP+Mm/+e7dce8Rnj6WnJTwJ\nAAAAAAAAAAAAQCpJ/0J/u+WJ3x13Tfa7z3x39BPb6inQZ0vrXXBuZti9+E9zqxP+tQAAAAAA\nAAAAAAAajzhvGP1M6d27nxTC088883wYfHZ9BNqPzL5Fo75e80rbretCOPEzp067btKkfuHY\n0xIcBgAAAAAAAAAAACBVfNHCaPjnP/8ZQti6dWs9hDmQwwZMeHLAgYY6n3vVVUnIAgAAAAAA\nAAAAAJAqvtgr6evemXfvr54PIXTt2rV+8gAAAAAAAAAAAABQz+K8YfS5Hw2+v2zvpd07/rlh\nxQvP//2fu0Pb/jeOUBgFAAAAAAAAAAAAaJjiLIy+XjZz5sxP2zjkqB7D7pj4wPDs+gwFAAAA\nAAAAAAAAQP2JszB6zoSysu/tvZSWcUjrdjnHH3tkZlr9xwIAAAAAAAAAAACgvsRZGD38hDPP\nPCGxSQAAAAAAAAAAAABIiPSoAwAAAAAAAAAAAACQWHHeMLr2qZ/NWRPvM4/vf32/Lp83EAAA\nAAAAAAAAAAD1K87C6J8fuuGGmfE+86LpCqMAAAAAAAAAAAAADUachdG8yyZM6Fbx2APTXqpp\nFxv49d652W22vF5Z+uQf//pOZu7QUcNibT42/JW8hEQFAAAAAAAAAAAA4POIszDarX+/2nFj\nX6o9/dYlT48780tpe1brfrjk1vPOve8Pc3ffuvjW7ockLiUAAAAAmmD1EQAAIABJREFUAAAA\nAAAAn1t6fGPv/fa2u8urO1xXfPeHbdEQQvrhvcY9cO0x1S/cffu0fyYoIAAAAAAAAAAAAABf\nTJyF0eVLl24PITcWy9hno9kpp+SFUL1kyYv1Hg0AAAAAAAAAAACA+hBnYbSmpiaEsGnTpk/s\nvPnmmyGE6urqeo0FAAAAAAAAAAAAQH2JszCam5cXQvjrb36xpGav9W3PTvxNZQghLy+33qMB\nAAAAAAAAAAAAUB/iLIx2+tZNFx0W6lb9eFCv4ffPXFi+8m8ryxfO+OG3zip8cHVd+NJFN36r\nY2JzAgAAAAAAAAAAAPA5NYtzrv3QybP/9o9v/NfCF//ffw7+fx/bSD/y7LG/nzz0yARkAwAA\nAAAAAAAAAKAexFsYDaHtWXfMe6XwiYcnPzF/+ao33tvZ/LBjup16zjdGXPWN2OFx3lMKAAAA\nAAAAAAAAQPLFXxgNIaS3iw2++cHBNycqDAAAAAAAAAAAAAD1z92gAAAAAAAAAAAAAI3cQRVG\n31/9Pw/ceNnA/K/2OP20788PIYQ3l/33o48+Om/tzsSkAwAAAAAAAAAAAOALi/+V9G/+4Zr8\nIZNWbv/gY8d3Qgih9ZrfXXv5k82KDt30yKDMROQDAAAAAAAAAAAA4AuK94bRzVOvvWzSyl2d\nh/xy8Y/7f7Tc5sLLCluFqt//rmT7Z/9dAAAAAAAAAAAAACIUZ2F042O/mrU1tB3y44ev7dWx\n9cc2WuXmHhfClqVLKxMSDwAAAAAAAAAAAIAvKs7C6F8rKkII/9q7d+t9d3JyckIIGzdurN9c\nAAAAAAAAAAAAANSTOAujO3bsCCFkZWWFENLS0j62s3Xr1hBC8+bN6z8bAAAAAAAAAAAAAPUg\nzsLo0UcfHUJ4/fXX992offnl1SGEzp0712ssAAAAAAAAAAAAAOpLnIXRk/v2bRdCxe+f+Pvu\nj98wuvutqT+euimETuef3y1RCQEAAAAAAAAAAAD4QuIsjDY7/z9H92i5e9mYC6/+9XObQwih\n6m/zfnv3RV+9ZnZVaNv3jlE90w70CAAAAAAAAAAAAAAi0SzOubRuN86ctnbAN3/+8LV/DSGE\nMHd037khhNA6Nuq/H70yJ1H5AAAAAAAAAAAAAPiC4i2MhpCWXTixfNXQKb985H8WV65/uybj\nsA4n9Ro0/Porz84+JIEBAQAAAAAAAAAAAPhi4iyMlk/81k+X1Z12w5Qbrrr7a1clNhIAAAAA\nAAAAAAAA9Sk9vrE3lvx2ytSppesTGwYAAAAAAAAAAACA+hdnYTQ7OzuE8OabbyY0DAAAAAAA\nAAAAAAD1L87C6Cnf+MaxIVTMnftWYuMAAAAAAAAAAAAAUN/iLIymnflfj97Ro9ms7w4d+/Sr\nOxIbCQAAAAAAAAAAAID61Cy+sfnfP/vmeduzWm1eMKZf5wmH5xzX6ciWe3VNz72v/N5zEhEw\nwWpra0tKSmpqavYzU1ZWFkKoq6tLVigAAAAAAAAAAACA+hRnYfSdtcuXL//gz7tr3nn15Xde\n3Xug8zv1GitpFixYUFhYGM/kihUrEh0GAAAAAAAAAAAAIBHiLIwOemjjxp/tbyDzS/WRJvny\n8/NnzZp1wBtGi4uLTzrppKSlAgAAAAAAmo49bznb88YzkiMzM3PAgAEZGRlRBwEAAIDkibMw\n2uKwo45KbJCIZGRkDBo06IBjxcXF6enpScgDAAAAAAA0NStXrgwhFBcXFxcXR52lCZk9e/bA\ngQOjTgEAAADJE2dhFAAAAAAAgITYuXNnCKGoqCgWi0WdpUmoqKiYMmVKdXV11EEAAAAgqfZT\nGH3qhuOvnxP6/2zNT/t9sPLu38vXvhMO73LacSn6AnoAAAAAAICGKRaLFRQURJ0CAAAAaLT2\n85r1rRvXrl27duPWj1bm3Xz66aeffvO8xMcCAAAAAAAAAAAAoL7spzAKAAAAAAAAAAAAQGOg\nMAoAAAAAAAAAAADQyCmMAgAAAAAAAAAAADRyCqMAAAAAAAAAAAAAjVyzAw1s27xu3boP/rx5\n274rH2rdvvORreo1GQAAAAAAAAAAAAD14oCF0TnfPvbYA62EEC6avnvG4HpLBQAAAE1FbW1t\nSUlJTU1N1EGairKysqgjAAAAAAAARGA/hdFDWh966KHxPqf1IfWRBgAAAJqaOXPmFBYWRp0C\nAAAAAACARm4/hdHCR957L3lBAAAAoEmqrq4OIRQVFcVisaizNAmzZs0qLS2NOgUAAAAAAECy\nHfCV9AAAAEDCxWKxgoKCqFM0CRUVFVFHAAAAAAAAiEB61AEAAAAAAAAAAAAASCyFUQAAAAAA\nAAAAAIBGTmEUAAAAAAAAAAAAoJFTGAUAAAAAAAAAAABo5BRGAQAAAAAAAAAAABo5hVEAAAAA\nAAAAAACARk5hFAAAAAAAAAAAAKCRUxgFAAAAAAAAAAAAaOQURgEAAAAAAAAAAAAaOYVRAAAA\nAAAAAAAAgEZOYRQAAAAAAAAAAACgkVMYBQAAAAAAAAAAAGjkFEYBAAAAAAAAAAAAGjmFUQAA\nAAAAAAAAAIBGTmEUAAAAAAAAAAAAoJFTGAUAAAAAAAAAAABo5BRGAQAAAAAAAAAAABo5hVEA\nAAAAAAAAAACARk5hFAAAAAAAAAAAAKCRUxgFAAAAAAAAAAAAaOQURgEAAAAAAAAAAAAaOYVR\nAAAAAAAAAAAAgEZOYRQAAAAAAAAAAACgkVMYBQAAAAAAAAAAAGjkFEYBAAAAAAAAAAAAGjmF\nUQAAAAAAAAAAAIBGTmEUAAAAAAAAAAAAoJFTGAUAAAAAAAAAAABo5BRGAQAAAAAAAAAAABq5\nZlEHOHjb1i+c8diMkkXLK1et3/zelm27mrdqc2j7Tt1yT+09YPCwi8/u1DLqhAAAAAAAAAAA\nAAANSYoVRjcvGHvp5ePmbdix1+r7W9/dvHHdiueefnziXaPPvW3qtDvzj4goIAAAAAAAAAAA\nAECDk0qF0V2V4/v1H/Pi9pB1fL+iEYPPOyOvS4d2bTOb7aqp+seGtZXL5s6Y/MhTa+aN6d+v\neflzt+am0rcGAAAAAAAAAAAAkDgp1KrcNnPs+Be3h6MueHjJtBHHtdhrr0u3vB7nXHDlTTdN\nHnrWlU8uH3fXzJHTh7SKKCgAAAAAAAAAAABAg5IedYD4vbBw4fshnDzq/n3boh9p0WXED0fG\nQni/tLQ8qdkAAAAAAAAAAAAAGq4UKoxWVVWFELKzs/c7tWd/zywAAAAAAAAAAAAAKVUYzcnJ\nCSG88OyzNfsZqlm8uDyE0LFjxySlAgAAAAAAAAAAAGjoUqgwGrv4kq5pYdOkEZc8uHTTrk8Z\n2LVp6YOXjJi0KaR1u/ii7knPBwAAAAAAAAAAANAwNYs6QPzSTh09+eaS8++tmD2qV6fxeb36\n9Mjt0uGINi0yardveXvD2peeL11S+db2EFqffMvDo0+NOi0AAAAAAAAAAABAQ5FChdEQsnrd\ns2hpt9tG3fnQgtcr50+vnL/vQGZ2/lVjHxg3vHvrKOIBAAAAAAAAAAAANEgpVRgNIbTtPvyn\n86+YsG7ZgkXllatffeu9rdW1GS2zDmvfsWveaX3yz+iUlR51RAAAAAAAAAAAAICGJdUKoyGE\nENKzOvcc1LnnoKhzAAAAAAAAAAAAAKQC93ECAAAAAAAAAAAANHIpeMPotvULZzw2o2TR8spV\n6ze/t2Xbruat2hzavlO33FN7Dxg87OKzO7WMOiEAAAAAAAAAAABAQ5JihdHNC8Zeevm4eRt2\n7LX6/tZ3N29ct+K5px+feNfoc2+bOu3O/CMiCggAAAAAAAAAAADQ4KRSYXRX5fh+/ce8uD1k\nHd+vaMTg887I69KhXdvMZrtqqv6xYW3lsrkzJj/y1Jp5Y/r3a17+3K25qfStAQAAAAAAAAAA\nACROCrUqt80cO/7F7eGoCx5eMm3EcS322uvSLa/HORdcedNNk4eedeWTy8fdNXPk9CGtIgoK\nAAAAAAAAAAAA0KCkUGH0hYUL3w/h5FH379sW/UiLLiN+OPInT95aUVpaHob0Tmo8AAAAAAAA\nUkBdXV0IoaysLOogTUhmZuaAAQMyMjKiDgIAANCkpVBhtKqqKoSQnZ2936ns7OwQKvbMAgAA\nAAAAwN5WrlwZQiguLi4uLo46SxMye/bsgQMHRp0CAACgSUuhwmhOTk4Ia1949tmaYX0zP2uo\nZvHi8hBCx44dk5gMAAAAAACAVLFz584QQlFRUSwWizpLk1BRUTFlypTq6uqogwAAADR1KVQY\njV18Sdf7JqyeNOKSE6dN+s5Xv/yJ6Ls2LZ149YhJm0Jat4sv6h5FRAAAAAAAAFJCLBYrKCiI\nOgUAAAAkTwoVRtNOHT355pLz762YPapXp/F5vfr0yO3S4Yg2LTJqt295e8Pal54vXVL51vYQ\nWp98y8OjT406LQAAAAAAAAAAAEBDkUKF0RCyet2zaGm320bd+dCC1yvnT6+cv+9AZnb+VWMf\nGDe8e+so4gEAAAAAAAAAAAA0SClVGA0htO0+/Kfzr5iwbtmCReWVq199672t1bUZLbMOa9+x\na95pffLP6JSVflDPq62tLSkpqamp2c9MWVlZCKGuru4LJQcAAAAAAAAAAACISKoVRkMIIaRn\nde45qHPPQfXwqAULFhQWFsYzuWLFinr4egAAAAAAAAAAAABJl5KF0XqUn58/a9asA94wWlxc\nfNJJJyUtFQAAAAAAAAAAAEA9SsHC6Lb1C2c8NqNk0fLKVes3v7dl267mrdoc2r5Tt9xTew8Y\nPOziszu1PIiHZWRkDBp04KtKi4uL09MP7mX3AAAAKaq2trakpGT/v1lHPSorK4s6AgAAAAAA\nAI1fihVGNy8Ye+nl4+Zt2LHX6vtb3928cd2K555+fOJdo8+9beq0O/OPiCggAABAypszZ05h\nYWHUKQAAAAAAAID6lEqF0V2V4/v1H/Pi9pB1fL+iEYPPOyOvS4d2bTOb7aqp+seGtZXL5s6Y\n/MhTa+aN6d+veflzt+am0rcGAADQcFRXV4cQioqKYrFY1FmahFmzZpWWlkadAgAAAAAAgEYu\nhVqV22aOHf/i9nDUBQ8vmTbiuBZ77XXpltfjnAuuvOmmyUPPuvLJ5ePumjly+pBWEQUFAABo\nBGKxWEFBQdQpmoSKioqoIwAAAAAAAND4pUcdIH4vLFz4fggnj7p/37boR1p0GfHDkbEQ3i8t\nLU9qNgAAAAAAAAAAAICGK4UKo1VVVSGE7Ozs/U7t2d8zCwAAAAAAAAAAAEBKFUZzcnJCCC88\n+2zNfoZqFi8uDyF07NgxSakAAAAAAAAAAAAAGroUKozGLr6ka1rYNGnEJQ8u3bTrUwZ2bVr6\n4CUjJm0Kad0uvqh70vMBAAAAAAAAAAAANEzNog4Qv7RTR0++ueT8eytmj+rVaXxerz49crt0\nOKJNi4za7Vve3rD2pedLl1S+tT2E1iff8vDoU6NOCwAAAAAAAAAAANBQpFBhNISsXvcsWtrt\ntlF3PrTg9cr50yvn7zuQmZ1/1dgHxg3v3jqKeAAAAAAAAAAAAAANUkoVRkMIbbsP/+n8Kyas\nW7ZgUXnl6lffem9rdW1Gy6zD2nfsmndan/wzOmWlRx0RAAAAAAAAAAAAoGFJtcJoCCGE9KzO\nPQd17jko6hwAAAAAAAAAAAAAqcB9nAAAAAAAAAAAAACNXEreMLqXmjeen/PHZ1e8XhXaZP/L\nWQP6n9mhZdSRAAAAAAAAAAAAABqSFCqMrpv30DP/G47te9W5nf9vqer5H106+LaS17Z/OHRI\nTv/xMx//7ultokgIAAAAAAAAAAAA0BCl0Cvpy39x9dVXX/2L8g8X3nj0m/2/V/La9owjT7nw\nyuuvv/LCU9pn7Hhtzvf6Fz3+VoQ5AQAAAAAAAAAAABqWFLphdB+7l95/+x/fCWknXPNU2S/7\ntksLIez+xzPXnlkwac2TYx7885Bxp0SdEAAAAAAAAAAAAKBBSKEbRvfx8pw560No/fUxE/a0\nRUMIae36ThhT2DqEVXPm/D3adAAAAAAAAAAAAAANRuoWRtetWxdCyO3T5/CPr7br0+ekEMKa\nNWsiCQUAAAAAAAAAAADQ8KRuYbRZs2YhhHbt2u29fOSRR4YQdu7cGUUmAAAAAAAAAAAAgAao\nWdQBDtaWN1auXBlCCK2P7hrCi+vXrw/hKx/bf+ONN0IIOTk50cQDAAAAAAAAAAAAaHBSrjD6\np5Ennvixj6sWLnzz9q8c9eHnna+8sjaEVrHY8cmPBgAAAAAAAAAAANAgpVBh9Mvdzz33vU+s\nplUseCNceswHn7b9YeoTVaHNsGGDWiU3HAAAAAAAAAAAAECDlUKF0a/d+cwzB5qp6vD1B3/T\n75heA/RFAQAAAAAAAAAAAD6QQoXReBzV89Jv9Yw6BAAAAAAAAAAAAECDkh51AAAAAAAAAAAA\nAAASS2EUAAAAAAAAAAAAoJFrZK+kBwAAAAAAABqQurq6EEJZWVnUQZqQzMzMAQMGZGRkRB0E\nAABoWBRGAQAAAAAAgERZuXJlCKG4uLi4uDjqLE3I7NmzBw4cGHUKAACgYVEYBQAAAAAAABJl\n586dIYSioqJYLBZ1liahoqJiypQp1dXVUQcBAAAaHIVRAAAAAAAAILFisVhBQUHUKQAAAJq0\n9KgDAAAAAAAAAAAAAJBYCqMAAAAAAAAAAAAAjZzCKAAAAAAAAAAAAEAjpzAKAAAAAAAAAAAA\n0MgpjAIAAAAAAAAAAAA0cgqjAAAAAAAAAAAAAI1cs6gDAAAAHEBtbW1JSUlNTU3UQZqKsrKy\nqCMAAAAAAAAA9UxhFAAAaOjmzJlTWFgYdQoAAAAAAACAFKYwCgAANHTV1dUhhKKiolgsFnWW\nJmHWrFmlpaVRpwAAAAAAAADqk8IoAACQGmKxWEFBQdQpmoSKioqoIwAAAAAAAAD1LD3qAAAA\nAAAAAAAAAAAklsIoAAAAAAAAAAAAQCOnMAoAAAAAAAAAAADQyCmMAgAAAAAAAAAAADRyCqMA\nAAAAAAAAAAAAjZzCKAAAAAAAAAAAAEAj1yzqAAAAAAAAAADUj7q6uhBCWVlZ1EGakMzMzAED\nBmRkZEQdBAAADkBhFAAAAAAAAKCRWLlyZQihuLi4uLg46ixNyOzZswcOHBh1CgAAOACFUQAA\nAAAAAIBGYufOnSGEoqKiWCwWdZYmoaKiYsqUKdXV1VEHAQCAA1MYBQAAAAAAAGhUYrFYQUFB\n1CkAAICGJT3qAAAAAAAAAAAAAAAklhtGAQDgoNXW1paUlNTU1EQdpKkoKyuLOgIAAAAAAABA\nalMYBQCAgzZnzpzCwsKoUwAAAAAAAABAvBRGAQDgoFVXV4cQioqKYrFY1FmahFmzZpWWlkad\nAgAAAAAAACCFKYwCAMDnFIvFCgoKok7RJFRUVEQdAQAAAAAAACC1pUcdAAAAAAAAAAAAAIDE\ncsMoAAAAAAAAAHwedXV1IYSysrKogzQhmZmZAwYMyMjIiDoIAEDqURgFAAAAAAAAgM9j5cqV\nIYTi4uLi4uKoszQhs2fPHjhwYNQpAABSj8IoAAAAAAAAAHweO3fuDCEUFRXFYrGoszQJFRUV\nU6ZMqa6ujjoIAEBKUhgFAGgMamtrS0pKampqog7SVHjDFAAAAADwoVgsVlBQEHUKAAA4gBQs\njG5bv3DGYzNKFi2vXLV+83tbtu1q3qrNoe07dcs9tfeAwcMuPrtTy6gTAgAk3Zw5cwoLC6NO\nAQAAAAAAAAA0UClWGN28YOyll4+bt2HHXqvvb31388Z1K557+vGJd40+97ap0+7MPyKigAAA\n0djz/h2vPUqaWbNmlZaWRp0CAAAAAKBpqaurC94BlVyZmZkDBgzIyMiIOggAUA9SqTC6q3J8\nv/5jXtweso7vVzRi8Hln5HXp0K5tZrNdNVX/2LC2ctncGZMfeWrNvDH9+zUvf+7W3FT61gAA\n6oXXHiVNRUVF1BEAAAAAAJqclStXhhCKi4uLi4ujztKEzJ49e+DAgVGnAADqQQq1KrfNHDv+\nxe3hqAseXjJtxHEt9trr0i2vxzkXXHnTTZOHnnXlk8vH3TVz5PQhrSIKCgCEEGpra0tKSmpq\naqIO0lT4dWoAAAAAABq9nTt3Bq/bSqK//OUvU6dOnT9//p4XnZEErnQFIKFSqDD6wsKF74dw\n8qj7922LfqRFlxE/HPmTJ2+tKC0tD0N6x/HQeLose+oXe262J/n0jZKsrq7upZdeys3NTU9P\njzpLk+DAk8lpJ9ny5cvvvffeqFM0Oa69TJr169cHB55EDjzJHHiSOfAkc+BJ5sCTzIEnk9NO\nMgeeZA48yRx4kjnwJHPgSebAk8yBJ9meAydpNmzYEFzpmnTf//73Tz311KhTNH7++zjJHHjy\nKaDz6XanjFnfPCSEMPDRmv1OVU/9txBCi2Gz43vo3Llz4zyou+++ux6+CQ5e/P9GAAAAAAAA\nAAAAhBDmzp0bdfOLBieFbhjNyckJYe0Lzz5bM6xv5mcN1SxeXB5C6NixY3wPzc/PnzVr1v5v\nr6yrq1u1atXNN998cHmpJ/H8G1GP/EpHkjnwZHLaSebAk8yBJ5kDTzIHnmQOPMkceJI58CRz\n4EnmwJPJaSeZA08yB55kDjzJHHiSOfAkc+BJ5sCTzIEnmQNPMgeeTE47yRx48mVmZubn50ed\nggYnbffu3VFniNPu5aO/cvqE1Wk5g348bdJ3vvrlT3Rdd21aOvHqoTfNfm13t9EvrBzndm4A\nAAAAAAAAAACAEFKqMBrC1iW3nHX+vRXvh9CifV6vPj1yu3Q4ok2LjNrtW97esPal50uXVL61\nPYTWJ9/y9OIJvVpHnRYAAAAAAAAAAACgYUipwmgIoeqvv7lt1J0PLXj9U99Pnpmdf9XYB8YN\n79422bkAAAAAAAAAAAAAGqxUK4yGEEKo27pu2YJF5ZWrX33rva3VtRktsw5r37Fr3ml98s/o\nlJUedToAAAAAAAAAAACAhiUlC6MAAAAAAAAAAAAAxM99nAAAAAAAAAAAAACNnMIoAAAAAAAA\nAAAAQCOnMAoAAAAAAAAAAADQyCmMAgAAAAAAAAAAADRyCqMAAAAAAAAAAAAAjZzCKAAAAAAA\nAAAAAEAj1yzqAEDDsmPHjnvuuadbt27p6QrlAKmkrq6usrIyLy/PD3CAlONnOECK8gMcIHX5\nGQ6QuvwMB0hRdXV1q1atuuWWWw455JCos0CTpjAK7OW+++4bM2ZM1CkAAAAAAAAAAGhU0tPT\nb7/99qhTQJOmMArs5f+3d9/hVVRpA8DPDZAQEAhNuqCoa0NRQARsrAU7qIDYFRBkrdgVPuva\nd8W6goqIKNhR7AULCyKCisq6IiiuVEEICKFK7vdHiCQh9ZIQ7s3v94fPc2fOnPOeOXNfxnne\nzN1tt91CCAMHDuzQoUN5xwJACUyePHnIkCESOEA8ksMB4pQEDhC/5HCA+CWHA8SprASeVZQC\nlCMFo0AuWb/d0KFDhx49epR3LACUzJAhQyRwgDglhwPEKQkcIH7J4QDxSw4HiFNDhgzJKkoB\nypEvIQAAAAAAAAAAAECCUzAKAAAAAAAAAAAAkOAUjAIAAAAAAAAAAAAkOAWjAAAAAAAAAAAA\nAAlOwSgAAAAAAAAAAABAglMwCgAAAAAAAAAAAJDgFIwCAAAAAAAAAAAAJDgFowAAAAAAAAAA\nAAAJTsEoAAAAAAAAAAAAQIJTMArkkpqa+ud/AYgjEjhA/JLDAeKUBA4Qv+RwgPglhwPEKQkc\nthORaDRa3jEA25GNGzeOHz/+iCOOqFSpUnnHAkAJSOAA8UsOB4hTEjhA/JLDAeKXHA4QpyRw\n2E4oGAUAAAAAAAAAAABIcH6SHgAAAAAAAAAAACDBKRgFAAAAAAAAAAAASHAKRgEAAAAAAAAA\nAAASnIJRAAAAAAAAAAAAgASnYBQAAAAAAAAAAAAgwSkYBQAAAAAAAAAAAEhwCkYBAAAAAAAA\nAAAAEpyCUQAAAAAAAAAAAIAEp2AUAAAAAAAAAAAAIMEpGAUAAAAAAAAAAABIcApGAQAAAAAA\nAAAAABKcglEAAAAAAAAAAACABKdgFAAAAAAAAAAAACDBKRgFAAAAAAAAAAAASHAKRgEAAAAA\nAAAAAAASnIJRAAAAAAAAAAAAgASnYBQqqoz/jhl0WqfdG9SsWrXGjrt27HHdszNWFvPQ9fM+\nfODSUzrs1igtNTWt2V6dTr3q8U/mri3TaAH401Yk8M1WfdiveSQSiUQi3V8qgxgByF8MOXzD\n0v+8N+LW/l0Pbr1b47Sqyam1Gu/ZqfvAR/+98I9tEjFAxRbzvXep3LQDELPY8rB7b4DtgQfg\nAPFLFQrEiUg0Gi3vGIBt7vdJ1xx21L3T1+TamLL3FW//+5+daxd6ZPTXd6/scuqQrzPybO8y\nYuU75+1QymECkFfsCTynNZ9c2qrzI3MimZmZ4dQXoy91L+0wAchHTDl87TPdUs9+bcvtkTqH\n/WP821e0Ti31MAHYJOZ779K5aQcgVrHmYffeAOXPA3CA+KUKBeKHN4xCBbRuwqCz7p2+JqnZ\nCfe+/0P6mjXps8b/o1vzpHX/ue/saz9cU9iRG7+/r9vJQ77OiNTveNnwj79fnLFu7fJfvhh3\nf//Dm6dGtlX4ABVX7Ak8Vy9Tbu73yP/2u3TAIWUZKwC5xZjDI8n19j66943DXps4/Yf56atX\nL/vftJdvPXanStFln1xz3n0zt+EEACqYmO+9S+emHYBYxZ6H3XsDlDcPwAHilyoUiCfeMAoV\nz6rnutU//bW1Ow+c9N19Hatu2rhuyhX7HDRkdpUTRy4ed05aAUcuffK4Zn3eXpPc5qapE2/e\nt2oBrQAoI7En8Bw2TB90QLu7M6+a8lHLaxtcMN4fWANsG6WSw7Ot/2rQfgfc8X3o8MD8Ty9t\nXCbxAlR0MeftUk34AJRYaedh994A244H4ADxSxUKxBVvGIUMHNhoAAAY/ElEQVQKZ+NHb7+3\nNoRW513YMcc/tintB/Q+IIQN7731wR8FHTl/9PB31oTQsM89N/h3GmDbiz2B5+jj27v73vtd\ni4FP3NQmuewiBSCv0sjhmyXvf+hBO4QQ1q1bV8pxApAl5rxdugkfgJIq9Tzs3htgm/EAHCB+\nqUKB+KJgFCqcH7/9dk0ItQ48cPfc23dr3752COtmzJhdwIFrJ06YGg1hx26ndfb/WADlIOYE\n/qfMmUP6/v2LpgMeu6WD/+UC2Ka2Pofn8sMXX64KoUmnTi1KL0QAcog5b5dywgeghEo/D7v3\nBthWPAAHiF+qUCC+KBiFCmfx4sUhhCZNmuTdkbUpa3d+5s2ZsyGESNsD91/04d3nHNKybrWU\n1LQmex925uDRM1aWacgAhLAVCXyT6JyH+t30ef0+w+7sXK1sIgSgIFubw3PKnPOvi+7+JtTo\ncsvVHSOlFyIAOcSct0sz4QNQcqWch917A2xDHoADxC9VKBBfKpd3AMC2tmbNmhBCSkpK3h1V\nq1YNIaxevbqAA1etWhVCqJf65YD2F4yeuzGrswXfTRj93YSXX5rw6sShx9Qrs6AB2IoEnuWX\nx/oPnlDj3NfvPapGGQUIQIG2MofnsOTtvx132Qermp354tN9mpVmiADkEHPeLr2ED0AsSjUP\nu/cG2KY8AAeIX6pQIL54wygkqjfOqhrJqcVVn2XtSE1NDSGsW7cu7xFr164NIVSrVtBf3WUd\n+Nsrj7xc7dT7354xf8XqjCVzPnvuhsMbhHUzh/Ue9P76spoLQEWTfw6POYGHEMKCkRde+35q\nz4f+eULtsooagDK4Cc8lc8Fr/Q8/edgPDU9+4sMRp+xYmpEDkMtWPjzZ2oQPQKxKLQ+79wbY\n5jwAB4hfqlAgvigYhQpnxx13DCHMnz8/746sTfXr1y/gwAYNGoQQotF9Bo0dfdkxezeumVqt\nXov2p90+7rGz64SwcOzYz8oybABiTuAhLBl98ZVvJ530wIM96pZhgAAUaCtyeLYNP44685Du\nj81s1POpCS+cv2uVMogSgGwx5+1SSPgAbIXSycPuvQHKgwfgAPFLFQrEFwWjkKhOeGZtNKef\n/3FQ1o6WrVqlhrDi889/yH3ArClT0kNIadVqtwJ6TNt332YhhOoHtNuzUs7tNdq12yOEsGTR\nosxSnwVAxZR/Do85gYcw5a23lob0cWc0zPHWu9oXjA8hhJd7RCKRyK6Dp5flhAAqitK/Cd9k\n9df3n3Twuc/N3fncMRNGn71z5bIJH4BsMeftrU34AGydUsjD7r0ByokH4ADxSxUKxBcFo1Dh\nVOp87NFVQ/j2qaGfrt28dd3njz75ZQhVjj7uyAKfgLU9/vgGIWR8OfW/G3NuXjV16vchhAaN\nG8soAGUp9gQOQHnbqhy+bOLgIw4b+E76Xhe+MmFEj2aVCmkKQOmIOW+7aQcoX1ubh917A5Qf\n99IA8UsVCsQXXyyoeHY4eWDfFiHMeaBXz39+MGv5unXLf/zwvl49H5gdQpPzLu+elt3ut4cP\nj0QikSOH/rZpQ9Khl1/Zvkr49vaTz3zw3f8s/H3t6qX/+/z5QSf2G7UshJ169mxfPvMBqDBi\nTuBbvPEuGo1G0x8/IoQQTn0xGo1GZ/+9dTlMCKACiTWHZy54c8BhR9/+2Yb9r3jj40dPaBgp\np/gBKpqY772LeyAAZSP2hyfuvQHKmwfgAPFLFQrEFQWjUAGlHHb7qKv3S82c+/pVR+1eu2rV\n2rseceWr/8tM2WvgqLv/Wq2QAyN/ufzpB4/dcd3M5y87Zp/GtVKr12vRvtcdH/8aqu9/9ajb\nDq2yzWYAUEHFnMABKHcx5vCfnr5p6Iw1Iaz+6r6j6kfySLvwg204A4CKJeZ7bzftAOUr9jzs\n3hugvLmXBohfqlAgnigYhQqp5sH3TJz27PU9D2pZv3pycvV6uxzU/Zpnpn52X+faRRxYZfcL\nx03/cMiFx+zXrE61KlVSa++039F9bn9t+qR7Dq21TSIHqOBiTuAAlDs5HCC+xJy3JXyA8iUP\nA8QvORwgfqlCgfgRiUaj5R0DAAAAAAAAAAAAAGXIG0YBAAAAAAAAAAAAEpyCUQAAAAAAAAAA\nAIAEp2AUAAAAAAAAAAAAIMEpGAUAAAAAAAAAAABIcApGAQAAAAAAAAAAABKcglEAAAAAAAAA\nAACABKdgFAAAAAAAAAAAACDBKRgFAAAAAAAAAAAASHAKRgEAAAAAAAAAAAASnIJRAAAAAAAA\nAAAAgASnYBQAAAAAAAAAAAAgwSkYBQAAAAAAAAAAAEhwCkYBAAAAAAAAAAAAEpyCUQAAAAAA\nAAAAAIAEp2AUAAAAAAAAAAAAIMEpGAUAAAAAAAAAAABIcApGAQAAAAAAAAAAABKcglEAAAAA\nAAAAAACABKdgFAAAAAAAAAAAACDBKRgFAAAAAAAAAAAASHAKRgEAAAAAAAAAAAASnIJRAAAA\nAAAAAAAAgASnYBQAAAAAAAAAAAAgwSkYBQAAAAAAAAAAAEhwCkYBAAAAAAAAAAAAEpyCUQAA\nAAAAAAAAAIAEp2AUAAAAAAAAAAAAIMEpGAUAAAAAAAAAAABIcApGAQAAAAAAAAAAABKcglEA\nAAAAAAAAAACABKdgFAAAAAAAAAAAACDBKRgFAAAAIOH88Vy3SCSy63XTC9my3Zh9V9tIJHL4\nw4vKOxC2ldK9Greja/vjC+sdHJcXcqmfw+1oUQAAAABgMwWjAAAAAAAAAAAAAAlOwSgAAAAA\nFUDlXq9Go9HZd7Uu70AqllVPHROJRPYYPKO8A0kEBZ3M8r+2M3+b+sQV3Tvv17zr8KWTLt+t\n8V4dTuh/5/PTFq0vt4i2me13UQAAAAAgH5XLOwAAAAAAAOJU5sxHj+100XtLo9kbVi3872dv\n/vezNx+bMmrDq2d5AA0AAAAA2w9vGAUAAAAAICarxw6++r2l0TqHXD36szmv9qnb8R/f/fzN\npHFDr+vZplFKpLyjAwAAAAByUjAKAAAAQAXwx3PdIpHIrtdNz7V11YxnrzvlwJ3rVEupVnen\nNicNfPqblbPvahuJRA5/eFF2m8zfZ33w+KCzj2q7Z7M61ZJTazXeo1OPq4ZPWxbd3M2ihw+O\nRCJt7/p57eyX/u+0g1rWq55SNa3pvsdfOuKbVXniyPjuuRtObb9L3Rwjlmwif46V8d2z1/U4\naNcda6Yk7XDeG5sHmPXGXf2Oa7NL/Ropyam1Gu15SK9Bz3y9Ik8Qs968d8BJ7XdrWCs1ZYcd\nd9mv85k3j/ly6cY8bQrt588wVn49YuCJ++9Up1pKtXq7tO856PmZGZuazLv/4EiN898NIcy8\nvVUk25FDl2/av2HeRw9efNKBLevXSElOTWuy1+FnDB7zbe5Ii5ps4YqaZlEBFDJ6oYEVZwm2\nUMRlVtjJzP/aLvbsirhoNy789Ilrunfcs0WDGqnV6jbbo82R59w6ZtqSP7L3z5gyJSOEQ29+\n6Z7T27eoVTlEUmo3b9XxxP53Pj/t0R6VSndFCjnzpfG9KNGKlHxRSjK7ojMJAAAAAMQkCgAA\nAAAJZsOYriGEltd+VciW6OrJN7WtnudhWWq73r1ahRAOe2hhdrPJlzXJ56lalV37vbssu8nC\nhzqFEFqddfGhtfI0q3nEv37ePOKaz29rX6PoEQu1aazT+7avtrmPs1/btPONv+1VLWwhefcL\nxi3O7mD+q/33yKdN3Ys+yjFIkf1sCqNX73apeafc/rYv1kSj0ejcIZ3yOXFHPJoejUaj67++\n/8h6W75/MmW3fuMWFXOyhStqmsUIoJDRC95VnCXI72os4jIr7GTm01vxZ1fERbvsjd5N83nn\nQMq5r2e3+OL6nUNIOnb4ymg0Gv2of91OBV3IpbAihZz5rfxelHxFSrgoJZpdkZkEAAAAAGLk\nDaMAAAAAVESZX91x7q3TMiL1Ol35zORZi1euXDRz0shL9v3xyee+zd2wUu29Txj48GuTv/1p\n8aq1q9MX/veTUVcdXn/D7McuvueLXA2/febR2W2uHz159qKVq9PnTnmm/75Vw+/jb39w8qb9\n0a/vOvemKSsjdTpcPvLTWb+uXLlo5oTh/fectcWIxfDtmOE/tR44cuIPC1ZuiEZXP31SCCEs\nePKcXv/6LvMvp935/KezFqWvWZex5Oepr9x+YovMHx7vffUbq0II4ZehZ5w+7PvVyTsdddXT\nn8z4+beMjN/mfPPxszf1alOvcnbnxelnUxjPPTlrz78NnzBz0cqVv/4w8alL2tYOv0+56Zw7\nv46G0PTyidGVI7qEEP4y6Ns/n0d+cGFaCNEZd5155Qe/RWvs1+dfH343b8WqpT9OfWnwkY0r\nrZv12NkDXkgvxmQLV9Q0SxBAIaNvuav4py6vIi6zQk7mlkoyu8Iv2ozXHh05LzOlVd/hE3/4\n9fe161Yumj39o9G3nNm+QXJ2D/see2yTkPnurec98OHPqzK3xYoUsigxfy9KviIlWpQSzq6I\nTAIAAAAAsdum5akAAAAAsA0U4w2jn1zSOITQ6Pw3V+Y8MP3VXvVDKPJ9nyuePq5yCK1umZn1\nMeu9gGH3az5fk6PRHxMu3SmE0PG+X7I+T7i0SQih4bmv/56zq6UvnlqnGCP+adNYjfu+83ue\nPV8P3iOEKkcOXZBn+x+TBjYNofqZYzdEo9Ou2zWEkHbCiAWZBY5QnH6yw6jT66WlOdukjz2z\nXgih4cUTsz5vWU4XjUaj0X9f1iSEUPvkZ5fk3Lr+i2v3TAoh0vmR7FNR8GQLV+Q0ixVAIaMX\nsKtYpy6a7/ss85PnMivgZG7ZW0lmV/hFmzHi+KQQDsy+hPO3fNLth+1YKYQQKicnJ9U5oOcl\ntw4d982SP3K2KZ0ViRa8KFv3vYhxRaLFXZQSzq6ITAIAAAAAMfOGUQAAAAAqoIVffLEghLqn\nnHPcDjk3p3U996S8Pwa98dfJj19zeuf9WtSvUbVyUiQSiURqnfPWHyH88ssvOdvtdEL3dlVz\nfK607757hRB+++23EEIIi774Yn4IdU8974Rcv0pfp3ufbjVLHH6dE87okufH7Zf+e+L3IWwY\nf1GzypUrV65UqVKlSklJSUlJkcqdhswLIeOnnxaHxZMmzQ6hds9Lz2m05a9jl6CfbDW79c4q\nd82W1q3PKXVCWDRt2vxCos/aXa1b39Pr5dxc5YABfQ4MITp1au5Xt+Yz2cIVOc2SBFDI6Hl3\nleTUbaG4l1nRSjK7Ii7aaseedXLd8PmdvQb8c8wH039OX5/feLU63vDxj9+/O+ym/l1b101a\n9uULD9144Un7Nt/ztKFfr9nUpFRXJBS8KDF+L/JXPisSilwUAAAAAIhd5aKbAAAAAECiWbFi\nRQihSZMmeXc0adIkhBWbP897rueBZ76yML/f2l67dm3Oj/Xr18+9PyUlJYSwcePGEEIIy5cv\nz3/Epk2bhvBdycJv3rx53k2bysmi2ePltX79+pCenh5CaNS0aSF/Rl6cfrI1bdo07+6mTZuG\nsGz58uUhbDHTbFknv2mzZnmLB5s2axYJn61avnxjCJX+3JrPZAtX5DRLEkAho+fdVZJTl0fx\nL7OilWR2RVy0oUGvZz6v0fa2+0ffc/7Qq9aFKrVbtj38uJ79Bw7osnNKzsN22PXofjcf3e/w\nUG/QLu8P3XfyI4NvGPbCgBMb7T37/kOSS3lFQsGLEuP3Ij/ltSKh6EUBAAAAgJh5wygAAAAA\nFVCtWrVCCPPnb/EizDybPr3v2lcWZjb46+DnJ34397eVazdszIxGo3+MOyO1pCOmpaXlP+K8\nefNK2ldIStriuV5W/3UHjN+Y/08NTbuuRahdu3YIYeG8efmVwZWgn2zztww9azZZnRQk6+TP\nmzs3mvfguXOjIeyQllYp59Z8Jlu4IqdZkgAKGT3vrpKcutxK7zIr4eyKVnWX468b8f43i35P\nnzP94+fuOmf3eaMHHrP/iUNn5+19k0i1Rq2O+dvQN+/8ayTMfWXstBBCKa9IKHhRYvxe5GM7\nXhEAAAAAiJ2CUQAAAAAqoEZt2jQKYenYUW9n5Ny8YtzT43K8XjSs/v77X0Jods6dt/XstGfT\nujukVE6KhLD8vVc/XhNKqGGbNk1CWPryyDdX5dyc/vKTr/0e4yRyadCxY8sQlr78yJiFBZTy\nhbBjp067hpD+wkNPF9ymOP1kW/HaiFfSc294cuyyEBq2bZv1etEqVaqELd/ImLV79atPjMn1\nG9t/TB82YmoIkbZt2xQ1cOGKnGYZBVCSU5dL8S6z/E/mlspkdpHktBb7HXZKvxufev2Wg8OK\n9x9+Zmbh7aPRHMGW14qEWBel2F/8Yi1KWV/wAAAAAFBcCkYBAAAAqIgOPq/3bpGwYMS5x107\n5vOffstYtWT25GcGHtvnuSU5W6U2bFgrhHkv3H3/Rz8sWb1+zbK5M959uP+RZzy5IIYRe/fd\nIyksGnneMVc9O+XHJRmrFs+a9NTFXS54aWnpzKjdxdd0rhYWv9L30K6Dnv7kP78syVi3Jn3e\nD9MnvPzAwFPa9H9xXQihzYArD6sWlr/e/6Au14ya8N0vy1avXva/bz8ZfcsZXW6eWIJ+si0d\n07fLpU9NmrV41aolP346amCX3s8uCUl79+vTMWt/SsOGaSHMG//ix7+sXL/57ZKd+vTbOymk\nj73wqAuGffL9wpUZy+Z8Ofam40+89z8bQ82uF57ecCtPRZHTLKMASnLqcireZVbAydxS6c3u\nmzu6dD7/5uFvT5kxZ/Gq9eszlv405dk7Rn4VQsjM3BTBxP875Jj+d43+6JufFq3cEA0hunbx\n9x89eemJgz+KhtROnQ7IalVeKxJiXJRif/GLtyhlfcEDAAAAQHHl/0M8AAAAABC/NozpGkJo\nee1XhWyJZkwadEC1PA/LUtud22OvEMJRw5ZmNVo/9ca9k/M0ijTu0f+kuiGknPl6VqOFD3UK\nIbS5c07uKNaMyhpz6p8bPr+1/Q55R2zbu1erEMJhDy0s1tQKGCvLojcvbZV3gE26jlqT1Wb+\nq/32yOdntete9FFJ+skKY5/Tzm+bt6+a7W+btubPnlaPO6durt1HPJoejUaj6776Z+c6W/ae\n3LL32AXFnGzhippmMQIoZPSCdxVnCba8Got1mRV0MvO5tmOfXe6L9qtrW+Y3k0i9E0f8vOmA\njy7aHFNSpUqRzY3qH/3IzD9KdUUKOfNb+b2IcUWKvyhbNbstMgkAAAAAxMgbRgEAAAComKp1\n/PvHk5++umub5mlVk1NrN9v/hMuf+mz86bUWhlCpTp2aWY2qtL3544mPX3xs653rV09OTWuy\nz5F9731/2phejWIZsWq7/xs/Zcz1p7RrUbtqldS0pq2Pu2T4p+Ov3y9vYVrMGhz3wOczPnjw\n8u4H79kkLbVKcvV6O+1xQOeeAx8Y++XjPatmtWncddjUr169ve8x+7eoWz05uUaDlq3/etYt\nY96/5ZCS9RNCCCmtbxw/6fFLjtu3aVrVKlVrt2jX/foxU8YPbrO5SeqJ97/1YN8j92lSKyVH\nLWEIya2vePerD4YMOL5Ni7rVq1ROqdlwj0N7XT9qytTh3WI6tVsoapplFUBxT11uxbvMCjqZ\nWyqt2e036L2PR97W7/gO+7Sov0NKalqjnfc5tNf1w/89/ZXzmm9qcugdkz8ceVv/rge33r1J\nzSrRaCQlrcneh/S46tFPvnnrb7tX2txXea1IiGlRiv3FL+ailPkFDwAAAADFEYlGo+UdAwAA\nAABsH1a8c95ex45c0Pqu2QW8W5Esix4+uNElk9rcOWfadS3KOxa2Gx/0Tbu59fcTL/YT6wAA\nAACwXfKGUQAAAAAqqF8e79vzxqc++Gr2r6vWZiz5+Zv3Hr3w6LNHLgiVO5zdS7UolFhSkufN\nAAAAALAdq1zeAQAAAABA+chcMePF24a/eFvurbUPuXfYJc3zPwIoRCRSyK+yAwAAAADlzV98\nAwAAAFBBNe/35Iu39zmm3R7N6larkly93s4HHNvvnne+fP/yVlXKJ6Bp17WIFOqEp9aWT2Tb\nDadou9Z52FK/Rw8AAAAA2y9vGAUAAACggorU3Kv7DU90v6G844hHDS+eGL24vIMAAAAAAKAE\nItFotLxjAAAAAAAAAAAAAKAM+Ul6AAAAAAAAAAAAgASnYBQAAAAAAAAAAAAgwSkYBQAAAAAA\nAAAAAEhwCkYBAAAAAAAAAAAAEpyCUQAAAAAAAAAAAIAEp2AUAAAAAAAAAAAAIMEpGAUAAAAA\nAAAAAABIcApGAQAAAAAAAAAAABKcglEAAAAAAAAAAACABKdgFAAAAAAAAAAAACDBKRgFAAAA\nAAAAAAAASHAKRgEAAAAAAAAAAAASnIJRAAAAAAAAAAAAgASnYBQAAAAAAAAAAAAgwSkYBQAA\nAAAAAAAAAEhwCkYBAAAAAAAAAAAAEpyCUQAAAAAAAAAAAIAEp2AUAAAAAAAAAAAAIMEpGAUA\nAAAAAAAAAABIcApGAQAAAAAAAAAAABKcglEAAAAAAAAAAACABKdgFAAAAAAAAAAAACDBKRgF\nAAAAAAAAAAAASHD/DzaQxcWyWyl1AAAAAElFTkSuQmCC",
      "text/plain": [
       "Plot with title “Histogram of ligand_receptor_correlations$correlation”"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 300,
       "width": 1800
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "### Distribution of calculated correlations\n",
    "options(repr.plot.width=30, repr.plot.height=5)\n",
    "hist(ligand_receptor_correlations$correlation)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d88a1d4d-d554-4d8b-9317-18cc7181c3b4",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Combine nichenet interaction potential and expression of ligands/receptors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 471,
   "id": "fb94f6d7-7a75-4e65-99e0-a13f552cae1c",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Add nichenet interaction potential to ligand receptor expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 472,
   "id": "b5308b81-87f5-47e2-9ae3-c6d6107048fb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Ligand</th><th scope=col>Receptor</th><th scope=col>amount_sources</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>A2M</td><td>LRP1    </td><td>1</td></tr>\n",
       "\t<tr><td>A2M</td><td>TNFRSF14</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 2 × 3\n",
       "\\begin{tabular}{lll}\n",
       " Ligand & Receptor & amount\\_sources\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t A2M & LRP1     & 1\\\\\n",
       "\t A2M & TNFRSF14 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 2 × 3\n",
       "\n",
       "| Ligand &lt;chr&gt; | Receptor &lt;chr&gt; | amount_sources &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| A2M | LRP1     | 1 |\n",
       "| A2M | TNFRSF14 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand Receptor amount_sources\n",
       "1 A2M    LRP1     1             \n",
       "2 A2M    TNFRSF14 1             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_receptor_interactions,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 473,
   "id": "38a05c1f-6155-4441-a5b5-c37585f4e114",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>from</th><th scope=col>to</th><th scope=col>weight</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>A1BG</td><td>ABCC6</td><td>0.4216439</td></tr>\n",
       "\t<tr><td>A1BG</td><td>ACE2 </td><td>0.1007411</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 2 × 3\n",
       "\\begin{tabular}{lll}\n",
       " from & to & weight\\\\\n",
       " <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t A1BG & ABCC6 & 0.4216439\\\\\n",
       "\t A1BG & ACE2  & 0.1007411\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 2 × 3\n",
       "\n",
       "| from &lt;chr&gt; | to &lt;chr&gt; | weight &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| A1BG | ABCC6 | 0.4216439 |\n",
       "| A1BG | ACE2  | 0.1007411 |\n",
       "\n"
      ],
      "text/plain": [
       "  from to    weight   \n",
       "1 A1BG ABCC6 0.4216439\n",
       "2 A1BG ACE2  0.1007411"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(weighted_networks$lr_sig,2) # interaction potential from nichenet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 474,
   "id": "790487df-a457-4376-ad3a-7a45bc39e1d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_receptor_interactions =merge( ligand_receptor_interactions, weighted_networks$lr_sig, by.x = c('Ligand', 'Receptor'), by.y = c('from','to'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 475,
   "id": "52b72696-4a7c-45c4-bbf2-6c32ef3b82cb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "11639"
      ],
      "text/latex": [
       "11639"
      ],
      "text/markdown": [
       "11639"
      ],
      "text/plain": [
       "[1] 11639"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_receptor_interactions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 476,
   "id": "5a671e36-a121-4654-a3dc-90334cce4a6d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Ligand</th><th scope=col>Receptor</th><th scope=col>amount_sources</th><th scope=col>weight</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A2M</td><td>LRP1    </td><td>1</td><td>0.4678977</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A2M</td><td>TNFRSF14</td><td>1</td><td>0.6990671</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & Ligand & Receptor & amount\\_sources & weight\\\\\n",
       "  & <chr> & <chr> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & A2M & LRP1     & 1 & 0.4678977\\\\\n",
       "\t2 & A2M & TNFRSF14 & 1 & 0.6990671\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 4\n",
       "\n",
       "| <!--/--> | Ligand &lt;chr&gt; | Receptor &lt;chr&gt; | amount_sources &lt;int&gt; | weight &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | A2M | LRP1     | 1 | 0.4678977 |\n",
       "| 2 | A2M | TNFRSF14 | 1 | 0.6990671 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand Receptor amount_sources weight   \n",
       "1 A2M    LRP1     1              0.4678977\n",
       "2 A2M    TNFRSF14 1              0.6990671"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_receptor_interactions,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 477,
   "id": "b80b8cc8-5ee1-424a-bd13-3657b1da6e9e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAADhAAAAJYCAIAAAC2c9oYAAAABmJLR0QA/wD/AP+gvaeTAAAg\nAElEQVR4nOzdaXxU5dkH4HuSQNgRQVzYVBSqBVGhokKFxA2tWqrgUlzR1ta6UGsVVyot1FZb\na9W2b1XqWrFi3VqXUhIWQRBQMW64VFRQARFkTTAw74egEBLCJCyTjNf1jTnPPOc+93Oe8yG/\nP2cSyWQyAAAAAAAAAAAAAMhcWekuAAAAAAAAAAAAAIBtS2AUAAAAAAAAAAAAIMMJjAIAAAAA\nAAAAAABkOIFRAAAAAAAAAAAAgAwnMAoAAAAAAAAAAACQ4QRGAQAAAAAAAAAAADKcwCgAAAAA\nAAAAAABAhhMYBQAAAAAAAAAAAMhwAqMAAAAAAAAAAAAAGU5gFAAAAAAAAAAAACDDCYwCAAAA\nAAAAAAAAZDiBUQAAAAAAAAAAAIAMJzAKAAAAAAAAAAAAkOEERgEAAAAAAAAAAAAynMAoAAAA\nAAAAAAAAQIYTGAUAAAAAAAAAAADIcAKjAAAAAAAAAAAAABlOYBQAAAAAAAAAAAAgwwmMAgAA\nAAAAAAAAAGQ4gVEAAAAAAAAAAACADCcwCgAAAAAAAAAAAJDhBEYBAAAAAAAAAAAAMpzAKAAA\nAAAAAAAAAECGExgFAAAAAAAAAAAAyHACowAAAAAAAAAAAAAZTmAUAAAAAAAAAAAAIMMJjAIA\nAAAAAAAAAABkOIFRAAAAAAAAAAAAgAwnMAoAAAAAAAAAAACQ4QRGAQAAAAAAAAAAADKcwCgA\nAAAAAAAAAABAhhMYBQAAAAAAAAAAAMhwAqMAAAAAAAAAAAAAGU5gFAAAAAAAAAAAACDDCYwC\nAAAAAAAAAAAAZDiBUQAAAAAAAAAAAIAMJzAKAAAAAAAAAAAAkOEERgEAAAAAAAAAAAAynMAo\nAAAAwNfCY6c3SCQSO5z3TLoLIaJ03tgbzztq/w4tG9fPSiQSiSZVLEulC1d3V7Os8lY/+m+6\nC2G9uns7bQd1oTmf/KF34uCb5qS7jIit0K4lfzkikUgk9ho6Y2uWBQAAAMCXBEYBAAAA6pKp\nl+2eSCQSrX40flMj5tzUI5FIJBIDRpdujROW3t8/kUgkdrlwkyekmpaPvaj30ZffNXbWB5+t\n/CKZ7mpIhwzbVnX9cup6/V9n1g4AAACgegRGAQAAAGA7WvzwTXfMSUaj3teMe/ezVWuSyWRy\n+Z390l0VAAAAAACZLifdBQAAAACwPfS/vzh5f7qLICJemzVrTUT0+cHQ/D0b13AOq8lW5Haq\nQu1tTnLJKw//6bZ7nxg/4/U58+OL5/dpdfvenfc9oO9xA08+6ahureunpaja2y4AAAAAIkJg\nFAAAAAC2qxUrVkREdrNmNU2LAl93X7x9x8n5Fzw2t/SrT9YWL5pTNGVO0ZSn7h35+B3Lnjmv\nSRrLAwAAAKC28pP0AAAAAF8Lj53eIJFI7HDeMxt8tnTW6OHn9OvesXWzBvVzm+28+z4Hfvt7\nF9748AsflZQdH3NqIlHvjMcjIubfnpdYb/ehMzace/Xc8bf/dOC392mzY+P6uU1atuuSd9oV\nd0xZUBqVSC6ZOernJx28V+umDRo0bdWh21GDf/nI2yvXldfqR//9ctySvxyRSCQSew2dkVwy\n675rzzyiW4edmjXITuxQNiS5ct70x/501dnfObRbp7YtG9fPbbJj228edtIlfxw7p3jjU244\n1ct3Dz3tsH3a7Ngot9GObb95+ODfPPt+WZlr5k/+888GHNKpTYuGuY1a7nHQSVc+PHtldRq8\n+SZ8clvfRCKR6Hfn8ohY89D3vmrngNHVOVFE5asZEam3N6KmPYwVbz58/dlHdG3XsnFubuNW\nHQ/qf+mdMxZXUuPSl+8bOvDQvTes5K1qdXQTBVR6P5Qp+WDcrZee2qdLu1ZNc+s3atFm3z6n\nDR01Y9HaSmcs+XD8ny4fdMT+e7Ru3jC3Ucv23+h++KAr7/jvO8uS5YelOuf6Ctd+Nv3Oy0/5\n9jd227FRbsMd2nQ54pwbnpqz+quBW3lbpdSZqlV2O6W84qleToqd3Iob/8vzVr3Qm6t/k3ut\n+guU+t7ZzJM54p1bzrjgsbmliV3yLr9n0uxXf9Mreo585aO3Xpr077tH/Oi4/XbOTayf7c1f\ndEkkEom+t31S7iTP/6xtIpFIJBoOGF2udV88OahJIpFoeUFBuZ2Q4kbYKo+mdTbbrpTvPQAA\nAADWSwIAAABQdzz/sw4RES3PL9zUiPdu7B4RESc9+MWGHz86KDcimp/79JcffPLoWR038esz\nff68MJlMJpMPn7KJPyl1uGL6VxPPf+an3St9lV2LXtdM+qx8baVz7j+lQ3aFkQ26XHzRsbkR\n0fL8sV8OXfznwyMi9jjtZwPbbfiNpj94NplMJouGfXMTtTXtPrRgSbmzfjnV939+asWTJ3Y+\n8f4PVs78Xd5OFf5ndYv8P7+1JrVlSakJH9/ap/KST3qwiqkrLNwmP6xOe2vaw9OvPnvvij91\nnbvvTwuWbTh8zYf/OKOSu6vBNy+66JhKKknBZu6HZHL1Ow+e07lBJReT2OXoW2etLD9b8Zv3\nntGpssERHa9+6ath1ZrzqwovPaltxf+l3/Tg6yYvLRu4lbfVZjuzeZXdTimveEqXk3ont+LG\nT22hN1d/5XutBguU4t5J5cmcfO/X3SMi2v143KpkMpn8+OZe0fPG9za1vhMu2i0icr/3wIaN\nfvNX+6+bteUPxq3d4MBzQ3aNiNwT/7F+dDU2whY/mqrTrpTuPQAAAADK8YZRAAAAgK+j0sLh\n59/zbmm0OOzSu8e/Pm/RipJVi+e99fKUJ/4y9JSebRqUvZ5uwOhk8ov7vhsRsfNPCjf4m9Kc\nG3qUTZN857ZTB9w8c3nU2+OEEY+9NHdZcfHnH8z8x7B+bbJj8eRfffeMu+etP+fa1288ZfBD\n76+Jpgf84M8FsxeuLFm16J3nRl18cMNX/3jrUyUVaoyIeO/B3z28sOMpNz720odLStYkk8ml\nfz0qIiKrcYdep15++6MTZr4xZ/6y4pWfzXv7hUd/c06PFrFs5g2nXvr0ikqm+vuNoxfv98M/\nj31z/oriZR+/Mfbm/nvUi+T8fw457tDvXjapYb/r//HC+0uKVy2eM+Ohn/dsEbG44LKf//3z\nzfcy1SbscuH4ZDKZfPrcxhGRfcqjX7VzzKmpLltVqt3emvTw/hF3f/yNs2/59yvzlhav+mzO\nS49ee/iuWVHy+s0X3FS0viFv/eHUM+97tzQadz3rj2PfmL+yZOWnb0/464+713/t1lufrnyh\nU7Sp++Hzgp8ePehvs4uzdu4z5K7/vv7xspLVKxe8OfZPZ+/XJPnJsxf3v7Rw+fpJFj11wVFn\n3fdWcTTt+v3fjJn67sLlJasWz539YsGDN/zgiL2bffkX02rNuUGFv3/ko12Ouu6hmR98Xly8\n9MMX/zn8O+2zY9nU4Sf95InPIrb6ttpMZ7bQ5ld885dTk05ulY2f0kKnUH8FNVqgzXYyIlJ8\nMsdHH30UEbHfQQdVnoYt7+C8vg0jSsYXTF7/PtDPCgtfiWjWrFnEooKCWesHv1lY+HFEouw7\nEVHTjbCBGj35U2lXTdYOAAAA4GtvK4ROAQAAANhe1r1hNAVVv2G07EV9ib63LdzMCSuN43xp\n5b9O3zEiosWxd35Y7l2cX7z1h76NIiJ2/Unh6nWfrXj0lGYRER3Oe+bzcrOsmDSkc1kOquJ7\n5iJ2OuXh+Sm1JplMJldNuqRjRNT7zt82OMVXU3W8aMLScqPHnr9z2ZGm+X95r9wFvH9rn3oR\nkdN/9KrNnbNaTUhWFhitUqpvGK1ee6tQdQ/3/ElB+emXPjmoVUTEN4cVrftk5RPfbxER0eaM\nx8u/YXbpf3/cMapRyYaqvB/WvDy0c0REo0N/+2Zp+UOLHhnYMiKyDvnd/9Z98sXUIXtERDTo\nce3M5Zs+YbXmLFdh7qG/fb3c3it995a+jSMi9r3utS+L2GrbqoY7pZwq3jCa0opXfTnV6+TW\n2/gpLvTm6q+kOTVcoFQ6meKT+d0RB0RE7HrmE58nk5t9w2jyszuPyIqIPa6c+eUnxY8MbBjR\n6JTrLusUEYf87v0vD3x6e14iIrr84tV1H1RzI2zxo6l67drMvQcAAABABd4wCgAAAPB11KBB\ng4iI0tIvtmCS0rEP/fOziPjGxb85t/yvcOfsfdENP+wQER+PfnB82UdfPPvw40sjsnpd9ouj\nm5WbplHv667ql7uJc+zzo+sGtE65oga9T/pO64gvpk59scKxxGGXXHNY03Kj84/uUy8iovX3\nrzxv93IX0P7oo/eJiNI33nhnM6esVhO2nRq2txJV93DItXnlp2967Hfz60XE7DffXFfJf/7x\n6OKIRM/LfnVCi/JDDx9+1ZH1Uq+kMpXdD8lJd46aHRF7//T2n3Xe6Hevdzzxlz/dP2Lt8/98\n/JOIiFhbOOqe9yKi/QV/HHZg402ep1pzlrPT6dcP2afcT4pn7/mTX563W0S8Pvqhoopf2FgN\n76jq7ZRUpbLiVathJ7d446e40DVQswVKsZOpPZn3PHnQwfUiPr73+wcPvHbUv2d+XFxlxS3y\n8rpFxHsFBXPKPkhOLRy/KrJ65V/SL3/HiOkFBeteEloyvvD5ZETrvLxvrhtZ443wpZo9mrb8\nxgMAAACgUgKjAAAAAHVQy/MLN/UfhN+7sXsKE+zZt2/7iORzVx956vD7C9/4tEa/E/76jBkr\nI6LN0f26VDiW1fOYo3aMiEUzZrxXNvjFF4sjonN+fpsKg1scfvgBlZ+i+aGH7ruJsxfP+e/t\nl59+ZI9Ou7ZokpudWOewPy6IiEWffFK68fjdDjhg4wBa1m677RwR0XX//bM3Hr3bbhERS5cu\n3cTpv1StJmw7NWtvDXq488afZbVrt1tElC5dujIiIt6YOXNVRHQ+8sj2Fc7X6sgjN1FJiiq9\nH96ePHlBROx0+JH7V/LXzk4HHtg4Il5++eWy8qZMWRwRrY/rf8jGa17jOTdUv8+RfSqEYrMP\nPSKvUUS8NWPG5u6nmt5RVeyULZHKiletZp3c8o2f4kLXQM0WKMVOpvhk3mvI6PvP7dI4lr8x\n5lfnHnfcb2fGyzcen3fi+df95V+vL1lbYfie+fkdImJGQUHZ/fdqQcHCiAPy83c8NK9PbpRO\nLJhYGhGRnFIwvjiiYd+8nuu+WeON8JWaPZq2/MYDAAAAoFICowAAAABfR1kHX3f3FQc2jRWv\nPTTsjPx9d2reqtPBxw2+8tYnihZVTBttyoIFCyIi2rVrV9nR9u3bbTAoFi5cGBHRpk3F1FDE\nbm3aVP53qh133LHSzz96/PwD9jnywhsf+O/Mtz9ZsmL1xkUXF1d45d66V/eVk52dvYlD646s\nXbu5dlSrCdtODdq7lXoYOTk5Eetbte5a27ZtW0klbdu126I/SFZ6P8ybNy8iYuFfD8/JycnJ\nycnOzs7Ozs7KysrKykokso4dtSIiVnz2WUlExCeffBIRsccee1R5omrNuaHWbdtW8hLVrLZt\nd42IWLBg4WavsWZ31KZ2yhZKZcWrVrNObvnGT3Gha6BmC5RiJ1N9Mmd3OPnOl9594e8jftz/\nkE47NUhEyYJXxz/611/++Pgu7budffdrG92YPfLymkasmVQWDF1QWPh6RLv8/L2iYV7+wYlY\nVlAwIyKiqLDw04isQ/P6fPnizxpvhK/U7Mm/5TceAAAAAJUSGAUAAAD4emqWd8O0t6fd/8sf\n9T+0U6v6JYvenvbvv91w8Xf32+OA8x56b011ZkokEikcTSaTVQ5OVvppVlZlf79aNPqCs/76\nZnHs0H3w7/4x/pX3PlmysqR0bTKZTCan/bzSFNc2l1oTtp1qt3db97CaC52iSu+HNWvK7te1\na8qsXbt27dqyS9ngZKtXb/gj35tbkZrMmdLEKavmHVX5TqkFatbJrbbxt93W21ZbPvUnc87O\n3zrtqj89OmX2K78+NL5x1s1/vHbwt9vUTy579Z5zjrrw2WXlhh6W1zsnYnlhwQsRKwsKpkU0\nz8/vHhE75ed3iZhXUPBWxCcFBW9ERLf8/K/iujXeCF+p2ZMfAAAAgG2klv4ZEQAAAIBtL2fn\ngwZd8+dHJ89euPTTtyc/9qehJ+7bJJa9ctcZp//xf6l8f6eddoqI+OCDDyo7+sEHH24wKFq3\nbh0RMXfu3ErGfjRvXjViQ8ueevDfn0fscPKogrsuHdin6+47N29YPzsREbF2wYJFqU+0NVSr\nCdtOddu77Xq4rpIPP/ywkoPz5s7d+vmwnXfeOSLiW795P1mFZ85uEhGx6667RkT8739V3+LV\nmnND8z/8sJLs3Nq5cz+OiGjdevM3Qi25o7aWGneyomrdtCkudA1s+wWqyZO5eZf+Fw2/a+Jb\nk6/av37ER/f++fFyv9vetCwe+lFBwZtrnyuYuDrqHZb/7eyIiH3z8naOmFVQ8NmKgoIXImK3\nvLxOX31vy5dvKz75AQAAANhyAqMAAAAARG7LvQ797o9//cjkv/RvEvHFlH//Z/G6I1llP81e\n7nVyX9q3R49GETHv2Wdfq3AsOf3ZsZ9FRMsePXYvG3zggQ0iYnZBwbwKgxePG/dSNar9+MMP\nSyOi87e+1WyjI2smjy1cWdlXtp1qNWEbllHN9m67Hu7TvXvDiJg9dmzFxOin//lPdRY6Rfv2\n7t0iIl5+9tkFmx0b3zj00BYRseBfjz1f1Xt0qzXnhlZPGDuhQmJ0zZT/Fq6MiE49epS1e6tt\nq9qhysupaScrqtZNm+JCR0TV9Ve0PRdo00/mTWjU48wB+0bE6g8/nF/uQNu8vL0i4pWCgrGF\nhYsjDsrPb1x25OD8vEaxdkpBwbjCiV9ENMnL+9b6r2358m3FJ39lqrd2AAAAAAiMAgAAAPCV\nZq1a5UZErF37Zfomq1mzJhGx5NNPK6au6h116ok7RsSbf7zy7vLviVvz7u1X/t+ciNj11NP6\nJtYNHnBCs4i1k2/6xbNLy02z8rnhI58pqUaVzZs3j4iYPXPm8nKfr37t91fcWelb/7ahajVh\nW5ZRvfZuux7WO+rk7+0QkZz2u2ueLB9uW1YwbOR/V2/J3JXLzvvB4I4RXxQMv/ihj9ZuZnBW\n3uCz94yID/508fUvbjoYW605y1l4/y/+8Ebphp+see/26+78KCL2OeXkruuq2Frbqnao6nJq\n3skKqnXTprjQZWOrqL+idCxQxSfzpix6Yfr/IiJnt91alz9wQH5+i4jk8/cOe/qdiL3z89uu\nO1CvT16vrFg1/qaRBZ9HZPfO+3bO+m9t+fJtxSd/Zaq3dgAAAAAIjAIAAAB8Hf3roq59v3/Z\n7x8c+0LRO/M+W7m6ZNnC96Y/esPAH929KCK7+1GH7/jlyC5du0bE6v/8340T3v1sVflMTsNj\nh13ft3HEZ0/+sM9Jv35i1kfLV69eNvelMcNPyBsybkXEjt8ZeWXfeusGNz7h2it61It4/86B\nfX/4l/Fvfbrqi5LF/5v8tyGHn/CH2dV6O9zOR/XbPytiyegf97/2kZkfLi5evXLB7HF3XNKn\nzxXPb+f3i0Y1m7DtVLO927CHDb9z7dWH5EbMvfe0PufcNm72wlWriz97d9IdF+R/70/vbpPX\nAOb0uOb/Ltw7J+Y+NOjAPhfc+vi0dxYsX11asnTBB7Nffv7pe2646KRD8n/14rrB2Qddc9u5\n7RNRPOOXfQ8+/cZ/Tntv0crVxZ9/9PbLhaN/e/5Rx/72lerPWU7J5KFHHDfs4RfnLlu9evm8\nlx/7Vf+8SwtXRLQeNOKSb345aKttq9qhqsupeSc3Vr2bNsWF3mz9FW3LBUrtyTzpip69T730\npgfHzZz9wcLlXyQjIlny6VtTxvz6tD4XPL40onn/049rXH7mrF55h9WPKJ42bVZEq/z8rl8d\n2SE//4CIhdOm/S8iDszLa77h17Z8+bbek79S1Vs7AAAAAJIAAAAA1B3P/6xDRETL8ws3NeK9\nG7tHRMRJD36x4cePDsqNiObnPr3hPyuT2OWoW4pKNvjmvLv7bfwD0B2umP7V4fnP/PTAxpVN\n1OLQayZ9Vr62L/53z4B2Ff8Hc4MuF1/Yr35E7HrRhC+HLv7z4RERHTc40waWT7qiW4OKp8zZ\n48wrvt8uIuLovy1LpjDV9LJ2fmeD0eusuu87ERE7XzJpU43eUHWa8PS5jSMi+5RHU5m4wsJV\n8WF12pvcej1MTr+iY0TE0XesH77m/Qe/v0d2hblz973womNyI6Ll+WNTu/qUCii7+DmPXnzg\nxjfqeuW/WPzmPYP2qnwHdLz6pZrNua7CPU4dcmKbiovQ5KCrJ36+Yb1bbVtttjObV9ntVM0V\n38zlpN7JrbjxkykvdFX1V77XttICVexkak/mwvNbbngkJ6fcZmt64E+fXlBJ9z655dvrRjQc\n+FDxhkemXtZu3ZH2l79Q8YvV2Ahb/Giq5o2X3Oy9BwAAAEA53jAKAAAA8HV0/G2vjn/w95cN\nOrpn1712a9EwJ6dB8533+la/c35x/4zXnr64S/0Nhu521kMT7rlsQK/OuzZvkFPxZ5ZbH/37\n598ouO2Sk3p13nWHhvXqNWqx2759Tvn5X597Y8Ive7coPzZnjzP/8fLUO3/2vYP2bNW4fv3G\nO7bresTZw8e8Mu2GA7NWR8QOO+yQWvmNe98waep9V5x0cMdWjerlNGjWevf9jz77Fw/NmlVp\nLGnbq04Ttp3qtXdb9jCr/an3v/jC3Zef2LNjq8b16zdq0bZL/hnXjX75hVvzd9z8t2skp0P/\nW6a9M+O+639wfM9Ou7ZoXL9ewx123WOfA3ofP/jqPz06bfy13TcYnNv5zPtfefPZW4YMPGzf\nNjs2rl+/yU4d9ul+xKAr7xj7zBXdajbnl5feYdDDL0/5y09PPKTTLs0b1stttsu+eWeNeHLW\nxF99u1yqbStuq9pgM5dTk05Wpro3bYoLvdn6K9pWC5Tak/mwG2dOffTPw374vbxvddmrTYv6\nyTWRqN9sl7179DvrqjsnvTH19/12qmTunfPy9o2IiMSh+Xnlcqk98vPK7s5meXkHVvziFi/f\nVnvyV6baawcAAADwtZZIJrfJz0ABAAAAQKpKC37Y5vA7FuSc+Pelj5zWMN3VZBzt3R6W/OWI\nFj8eFx2vmP7ODT3SXQxfK3NvOrjdmAHvTb1s93RXUk0eTQAAAADbnTeMAgAAAJBea966/Zd/\nXxCR0/uoPJmhrU57IbMlEnXyxZoeTQAAAABpIDAKAAAAwHbzn5/3PX3Ynf+e+ur/5i8r+aJ4\nyYevPPt/Q/K+fen4FRFtzrp80C7pLrBu0174Gqr9gVGPJgAAAIDaIifdBQAAAADw9bH646kP\nPDDhgeEVDrT89vAxNx/TJA0lZRLtha+h2h8Y9WgCAAAAqC28YRQAAACA7ebo30184Ffnf7d3\nt07tWjWpX69Bs512P+DI06+6a8pr4649uGm6q6vz6lp7x1/YKpGCvrd9mu5K6x69/RrZ9aeT\nk1Mv2z3dZVShrj2aAAAAADJXIplMprsGAAAAAODrZ/yFrfJuX7TZYX1uXTj+wlbboZ5MorcA\nAAAAQAUCowAAAAAAAAAAAAAZzk/SAwAAAAAAAAAAAGQ4gVEAAAAAAAAAAACADCcwCgAAAAAA\nAAAAAJDhBEYBAAAAAAAAAAAAMpzAKAAAAAAAAAAAAECGExgFAAAAAAAAAAAAyHACowAAAAAA\nAAAAAAAZTmAUAAAAAAAAAAAAIMMJjAIAAAAAAAAAAABkOIFRAAAAAAAAAAAAgAwnMAoAAAAA\nAAAAAACQ4QRGAQAAAAAAAAAAADKcwCgAAAAAAAAAAABAhhMYBQAAAAAAAAAAAMhwAqMAAAAA\nAAAAAAAAGU5gFAAAAAAAAAAAACDDCYwCAAAAAAAAAAAAZDiBUQAAAAAAAAAAAIAMJzAKAAAA\nAAAAAAAAkOEERgEAAAAAAAAAAAAynMAoAAAAAAAAAAAAQIYTGAUAAAAAAAAAAADIcAKjAAAA\nAAAAAAAAABlOYBQAAAAAAAAAAAAgwwmMAgAAAAAAAAAAAGQ4gVEAAAAAAAAAAACADCcwCgAA\nAAAAAAAAAJDhBEYBAAAAAAAAAAAAMpzAKAAAAAAAAAAAAECGExgFAAAAAAAAAAAAyHACowAA\nAAAAAAAAAAAZTmAUAAAAAAAAAAAAIMMJjAIAAAAAAAAAAABkOIFRAAAAAAAAAAAAgAwnMAoA\nAAAAAAAAAACQ4QRGAQAAAAAAAAAAADKcwCgAAAAAAAAAAABAhhMYBQAAAAAAAAAAAMhwAqMA\nAAAAAAAAAAAAGU5gFAAAAAAAAAAAACDDCYwCAAAAAAAAAAAAZDiBUQAAAAAAAAAAAIAMJzAK\nAAAAAAAAAAAAkOEERgEAAAAAAAAAAAAynMAoAAAAAAAAAAAAQIYTGAUAAAAAAAAAAADIcDnp\nLgCAzLRmzZqnnnqquLg43YWwrTRo0ODYY4/Nzs5OdyEAAAAAAAAAAGyewCgA28TTTz99wgkn\npLsKtq0nn3zyuOOOS3cVAAAAAAAAAABsnsAokE5eQpnBnn/++Yg488wzu3Xrlu5a2PpmzZp1\n7733rlq1Kt2FAAAAAAAAAACQEoFRIJ28hDLjdevW7aijjkp3FQAAAAAAAAAA8HUnMAqkU9nr\nCb2EMiM98cQTEyZMSHcVAAAAAAAAAABAhMAoUBt4CWVGmjVrVrpLAAAAAAAAAAAA1slKdwEA\nAAAAAAAAAAAAbFsCowAAAAAAAAAAAAAZTmAUAAAAAAAAAAAAIMMJjAIAAAAAAAAAAABkOIFR\nAAAAAAAAAAAAgAwnMAoAAAAAAAAAAACQ4QRGAQAAAAAAAAAAADKcwCgAAAAAAAAAAABAhhMY\nBQAAAAAAAAAAAMhwAqMAAAAAAAAAAAAAGU5gFAAAAAAAAAAAACDDCYwCAAAAAAAAAAAAZDiB\nUQAAAAAAAAAAAIAMJzAKAAAAAAAAAAAAkOEERgEAAAAAAAAAAAAynMAoAO6QjOsAACAASURB\nVAAAAAAAAAAAQIYTGAUAAAAAAAAAAADIcAKjAAAAAAAAAAAAABkuJ90FVN/K98ePeWDMUxNn\nFs1+f+GSZStL6zVq2rx1h85duh927IBBA/t2aJjuCgEAAAAAAAAAAABqkzoWGF1YOPy0M0aM\nm7e63Kcrli9e+PGc16Y++9Dt1191+NX3jb4ur1WaCgQAAAAAAAAAAACodepSYLS0aGS/Y4a9\nWBJN9up35uABR/bs2rFNy2YNckqLly6a927RtLFjRt3zzDvjhh3Tr96MqVd2qUuXBgAAAAAA\nAAAAALDt1KFU5cpHho98sSR26X/X5NGD98wtd6xj564H5fc/99JLR53a+9zHZo64/pFLHj6l\nUZoKBQAAAAAAAAAAAKhVstJdQOqmjx+/ImL/ITdtnBZdL7fj4Bsv6RaxYsKEGdu1NgAAAAAA\nAAAAAIDaqw4FRpcuXRoRbdu2rXJU2fGysQAAAAAAAAAAAADUqcBou3btImL6pEnFVQwqfu65\nGRHRvn377VQVAAAAAAAAAAAAQG1XhwKj3Qae3CkR8+8YfPItU+aXVjKgdP6UW04efMf8SHQe\neNJ+270+AAAAAAAAAAAAgNopJ90FpC7R/apRlz919G9mPTmkV4eRXXv1OahLxzatmuZmrylZ\n9um8d199YcLkogUlEY33H3rXVd3TXS0AAAAAAAAAAABAbVGHAqMRTXrdMHFK56uHXHdn4dyi\ngoeLCjYe0KBt3nnD/zDinP0ap6M8AAAAAAAAAAAAgFqpTgVGI6LZfufcWnDWr+dMK5w4o+it\nDxYsWb5qTXbDJju0bt+pa48+eT07NMlKd4kAAAAAAAAAAAAAtUtdC4xGRERWk90POX73Q45P\ndx0AAAAAAAAAAAAAdYH3cQIAAAAAAAAAAABkuDr4htGV748f88CYpybOLJr9/sIly1aW1mvU\ntHnrDp27dD/s2AGDBvbt0DDdFQIAAAAAAAAAAADUJnUsMLqwcPhpZ4wYN291uU9XLF+88OM5\nr0199qHbr7/q8KvvG31dXqs0FQgAAAAAAAAAAABQ69SlwGhp0ch+xwx7sSSa7NXvzMEDjuzZ\ntWObls0a5JQWL100792iaWPHjLrnmXfGDTumX70ZU6/sUpcuDQAAAAAAAAAAAGDbqUOpypWP\nDB/5Ykns0v+uyaMH75lb7ljHzl0Pyu9/7qWXjjq197mPzRxx/SOXPHxKozQVCgAAAAAAAAAA\nAFCrZKW7gNRNHz9+RcT+Q27aOC26Xm7HwTde0i1ixYQJM7ZrbQAAAAAAAAAAAAC1Vx0KjC5d\nujQi2rZtW+WosuNlYwEAAAAAAAAAAACoU4HRdu3aRcT0SZOKqxhU/NxzMyKiffv226kqAAAA\nAAAAAAAAgNquDgVGuw08uVMi5t8x+ORbpswvrWRA6fwpt5w8+I75keg88KT9tnt9AAAAAAAA\nAAAAALVTTroLSF2i+1WjLn/q6N/MenJIrw4ju/bqc1CXjm1aNc3NXlOy7NN57776woTJRQtK\nIhrvP/Suq7qnu1oAAAAAAAAAAACA2qIOBUYjmvS6YeKUzlcPue7OwrlFBQ8XFWw8oEHbvPOG\n/2HEOfs1Tkd5AAAAAAAAAAAAALVSnQqMRkSz/c65teCsX8+ZVjhxRtFbHyxYsnzVmuyGTXZo\n3b5T1x598np2aJKV7hIBAAAAAAAAAAAAape6FhiNiIisJrsfcvzuhxy/FaZas2bNU089VVxc\nXMWYtWvXzp49e+jQofXr198KpwQAAAAAAAAAAADYvupkYHQrKiwsPOGEE1IZmZWVdc0112zr\negAAAAAAAAAAAAC2ujoYGF35/vgxD4x5auLMotnvL1yybGVpvUZNm7fu0LlL98OOHTBoYN8O\nDasxWV5e3hNPPFH1G0aff/75m2++ee+9997SygEAAAAAAAAAAADSoY4FRhcWDj/tjBHj5q0u\n9+mK5YsXfjzntanPPnT79VcdfvV9o6/La5XihNnZ2ccfv/nftr/55puzsrJqUDAAAAAAAAAA\nAABA2tWlwGhp0ch+xwx7sSSa7NXvzMEDjuzZtWObls0a5JQWL100792iaWPHjLrnmXfGDTum\nX70ZU6/sUpcuDQAAAAAAAAAAAGDbqUOpypWPDB/5Ykns0v+uyaMH75lb7ljHzl0Pyu9/7qWX\njjq197mPzRxx/SOXPHxKozQVCgAAAAAAAAAAAFCr1KGfWZ8+fvyKiP2H3LRxWnS93I6Db7yk\nW8SKCRNmbNfaAAAAAAAAAAAAAGqvOhQYXbp0aUS0bdu2ylFlx8vGAgAAAAAAAAAAAFCnAqPt\n2rWLiOmTJhVXMaj4uedmRET79u23U1UAAAAAAAAAAAAAtV0dCox2G3hyp0TMv2PwybdMmV9a\nyYDS+VNuOXnwHfMj0XngSftt9/oAAAAAAAAAAAAAaqecdBeQukT3q0Zd/tTRv5n15JBeHUZ2\n7dXnoC4d27Rqmpu9pmTZp/PeffWFCZOLFpRENN5/6F1XdU93tQAAAAAAAAAAAAC1RR0KjEY0\n6XXDxCmdrx5y3Z2Fc4sKHi4q2HhAg7Z55w3/w4hz9mucjvIAAAAAAAAAAAAAaqU6FRiNiGb7\nnXNrwVm/njOtcOKMorc+WLBk+ao12Q2b7NC6faeuPfrk9ezQJCvdJQIAAAAAAAAAAADULnUt\nMBoREVlNdj/k+N0POT7ddQAAAAAAAAAAAADUBRnzPs7P57z88ssvz/k83XUAAAAAAAAAAAAA\n1DYZExgde9kBBxxwwGVj010HAAAAAAAAAAAAQG2TMYFRAAAAAAAAAAAAACpXhwKjYwYkqjDw\nkYiIRwau++eAMekuFwAAAAAAAAAAAKCWqEOBUQAAAAAAAAAAAABqog4FRnfp0L5+ZO3Ue8jf\nX5m/uIJ7ToiIOOGedf+897vpLhcAAAAAAAAAAACglqhDgdHev3vtpft+svebf/z+oXk/ue+N\nL5rtsKFG9SIi6jUq908AAAAAAAAAAAAA6lJgNKLJvqf/8bk3ptzWP+vJi3vv0+vCe19dlu6S\nAAAAAAAAAAAAAGq9OhUYjYhItOr5k/tefP2Z63t9cudZB37zqOv+9V5JumsCAAAAAAAAAAAA\nqM3qXGA0IiLqtT36msdfnfXQhbu/MvL4Lvuf8rvn5q9Jd00AAAAAAAAAAAAAtVTdDIxGRETj\nzif/fsIbU/9yau6zlx22z4/HpbseAAAAAAAAAAAAgNqpDgdGIyISLXqcf9eM1wtG9m29Njc3\nN7d+drorAgAAAAAAAAAAAKhtctJdwFaQs1ve0H++OTTdZQAAAAAAAAAAAADUTnX8DaMAAAAA\nAAAAAAAAbI7AKAAAAAAAAAAAAECGExgFAAAAAAAAAAAAyHACowAAAAAAAAAAAAAZTmAUAAAA\nAAAAAAAAIMMJjAIAAAAAAAAAAABkOIFRAAAAAAAAAAAAgAwnMAoAAAAAAAAAAACQ4QRGAQAA\nAAAAAAAAADKcwCgAAAAAAAAAAABAhhMYBQAAAAAAAAAAAMhwAqMAAAAAAAAAAAAAGU5gFAAA\nAAAAAAAAACDDCYwCAAAAAAAAAAAAZDiBUQAAAAAAAAAAAIAMJzAKAAAAAAAAAAAAkOEERgEA\nAAAAAAAAAAAynMAoAAAAAAAAAAAAQIYTGAUAAAAAAAAAAADIcAKjAAAAAAAAAAAAABlOYBQA\nAAAAAAAAAAAgwwmMAgAAAAAAAAAAAGQ4gVEAAAAAAAAAAACADCcwCgAAAAAAAAAAAJDhBEYB\nAAAAAAAAAAAAMlxOuguovpXvjx/zwJinJs4smv3+wiXLVpbWa9S0eesOnbt0P+zYAYMG9u3Q\nMN0VAgAAAAAAAAAAANQmdSwwurBw+GlnjBg3b3W5T1csX7zw4zmvTX32oduvv+rwq+8bfV1e\nqzQVCAAAAAAAAAAAAFDr1KXAaGnRyH7HDHuxJJrs1e/MwQOO7Nm1Y5uWzRrklBYvXTTv3aJp\nY8eMuueZd8YNO6ZfvRlTr+xSly4NAAAAAAAAAAAAYNupQ6nKlY8MH/liSezS/67JowfvmVvu\nWMfOXQ/K73/upZeOOrX3uY/NHHH9I5c8fEqjNBUKAAAAAAAAAAAAUKtkpbuA1E0fP35FxP5D\nbto4LbpebsfBN17SLWLFhAkztmttAAAAAAAAAAAAALVXHQqMLl26NCLatm1b5aiy42VjAQAA\nAAAAAAAAAKhTgdF27dpFxPRJk4qrGFT83HMzIqJ9+/bbqSoAAAAAAAAAAACA2q4OBUa7DTy5\nUyLm3zH45FumzC+tZEDp/Cm3nDz4jvmR6DzwpP22e30AAAAAAAAAAAAAtVNOugtIXaL7VaMu\nf+ro38x6ckivDiO79upzUJeObVo1zc1eU7Ls03nvvvrChMlFC0oiGu8/9K6ruqe7WgAAAAAA\nAAAAAIDaog4FRiOa9Lph4pTOVw+57s7CuUUFDxcVbDygQdu884b/YcQ5+zVOR3kAAAAAAAAA\nAAAAtVKdCoxGRLP9zrm14Kxfz5lWOHFG0VsfLFiyfNWa7IZNdmjdvlPXHn3yenZokpXuEgEA\nAAAAAAAAAABql7oWGI2IiKwmux9y/O6HHJ/uOgAAAAAAAAAAAADqAu/jBAAAAAAAAAAAAMhw\ndfANoyvfHz/mgTFPTZxZNPv9hUuWrSyt16hp89YdOnfpftixAwYN7NuhYborBAAAAAAAAAAA\nAKhN6lhgdGHh8NPOGDFu3upyn65Yvnjhx3Nem/rsQ7dff9XhV983+rq8VmkqEAAAAAAAAAAA\nAKDWqUuB0dKikf2OGfZiSTTZq9+Zgwcc2bNrxzYtmzXIKS1eumjeu0XTxo4Zdc8z74wbdky/\nejOmXtmlLl0aAAAAAAAAAAAAwLZTh1KVKx8ZPvLFktil/12TRw/eM7fcsY6dux6U3//cSy8d\ndWrvcx+bOeL6Ry55+JRGaSoUAAAAAAAAAAAAoFbJSncBqZs+fvyKiP2H3LRxWnS93I6Db7yk\nW8SKCRNmbNfaAAAAAAAAAAAAAGqvOhQYXbp0aUS0bdu2ylFlx8vGAgAAAAAAAAAAAFCnAqPt\n2rWLiOmTJhVXMaj4uedmRET79u23U1UAAAAAAAAAAAAAtV0dCox2+3/27je+6rrw//9rG8iA\ngZZKGGNDh8xyYxiKESZOEYEPEMlfQyGGZVomUpFCSmKApp8mvyIthBI1MUANCjWEAQIDGdSc\nGJIYEHwQZkL82wbb+F2gbwohHHNnZ2fe75fg/X7t7LHrz9vrDBzULiHsnJY3aMrKnZUnOFC5\nc+WUQXnTdoaEzIH929d6HwAAAAAAAAAAAEDd1CDWAZFL6Dh2xpgF19xfPH9Ul/RJ2V26dsrK\naHVWs0ZJVRX73tm+6bVXlq4o2VURQtMOd0wf2zHWtQAAAAAAAAAAAAB1RRwNRkNI6XLfspWZ\n40bd/WjBtpLFs0sWH38gOTX3xgkPTRzRvmks8gAAAAAAAAAAAADqpAgHo8un3PqnC24bcU3b\nlOjmnFLz9iN+unj45M2rC5YVlWzcumvP/rKqpMYpZ7RIa5d9cdfcS9NTEj/U51VVVS1YsKC8\nvPwkZwoLC0MI1dXVH6kcAAAAAAAAAAAAIEYiHIy+/fLPvj3q53d9tveNo26/9YYr0pOjW3UK\niSltOvdp07lPDXxUQUFB3759Izm5fv36Gvh9AAAAAAAAAAAAALUuwsHo5bf//DuHf/6rP8z7\n36/Pe2hsh2u/cfvt3xzSueVp0Y2rBbm5ufPmzTvlDaP5+fkXXnhhrVUBAAAAAAAAAAAA1KAI\nB6Mtutz84O9uvnfrslmPTP359Gdn/2j47B9//9Ih37z99m/073BWhB9SQw5uWTLnyTkLlq0t\neWNL6Z59BysbNml2eov0zKyOl/caMHTgFemNP8SHJSUl9elz6qtK8/PzExM/3JfdAwAAAAAA\nAAAAANQRH2oE2Tjt8hGTnl7z962rn5gw/OKk4pl3Dbmo9Xm5X39g3ut7qqNVeIzSggnd2rXL\nHT5u6tMvrnp9847SPfsP7N9dumPz+lUvPj113PDcdud3m1DwTq20AAAAAAAAAAAAAMSH/+bW\nzNNadhp6169XbNn257mTB6SVLpk25ksXpmZ2/8YDz772z2juRitLJvXoOX7R9kMpbXvcMunR\nZxetfnXDm5s3b35zw6urFz376KRberRtemj7ovE9e0x+rTKKHQAAAAAAAAAAAABx5b/9mvXD\npWvn/vL/m/LwHzaWh5B01vmt9i75xZhr2597yS3PbY3SaPTg3AmT1lWElv2mF7/2/NQ7R/a7\nslN2ZkZ6enpGZnanK/uNvHPq868VT+/XMlSsnXjP3IPRiQAAAAAAAAAAAACIOx96MFrxf6se\n/+ENn09vffHgsTNW7Dv/y2MeXfK3v298Y+tfn7/zi5/Yve7hG775eHS+En7NkiUHQugw6sG8\n8xp9wJFGGXkP3JYTwoGlS4ui0gAAAAAAAAAAAAAQfyIfjJZtWTrjzkEdU9M7D7vniTUHz+s7\n+uFFb24rfub+kV1bJ4fQKL3HpCd/0CmE/UuXrotK6t69e0MIqampJz119P3RswAAAAAAAAAA\nAACE0CCyYyvvurD3pNd3V4eE5pk9b7v1tm9/9Zrzmv7HqdatW4fwyr4ojTVbt24dwqY1L79c\nPrRb8gcdKl++vCiEkJaWFpUGAAAAAAAAAAAAgPgT4Q2j//eXvxw6r/stU/6wYdtfFjz0zROt\nRUMI4fO3P/XUU0/d/vkaDHxPzsBB7RLCzml5g6as3Fl5ggOVO1dOGZQ3bWdIyBzYv31UGgAA\nAAAAAAAAAADiT4Q3jHaZuH77+Z85/VTz0tQuQ4Z85KQPktBx7IwxC665v3j+qC7pk7K7dO2U\nldHqrGaNkqoq9r2zfdNrryxdUbKrIoSmHe6YPrZj1DIAAAAAAAAAAAAA4kyEg9FzMtu9/eqS\nP73b4JycyzI/8d7z3W8sL95R+ckLvti+ZVJ0At8vpct9y1Zmjht196MF20oWzy5ZfPyB5NTc\nGyc8NHFE+xNfgAoAAAAAAAAAAADwcRThYDTsfvqWLkOfbXjd3K2XZb7vccPXpl47YNbhAbP+\nPnvwGdHoO17z9iN+unj45M2rC5YVlWzcumvP/rKqpMYpZ7RIa5d9cdfcS9NTTnULKgAAAAAA\nAAAAAMDHTISD0V1P/vJ3+0OLb93UL+WY5ylf/sbQFrN+9swvn3pn8M1nRaHvhBJT2nTu06Zz\nn9r6fQAAAAAAAAAAAADxLML7OP+yfn11CBdmZx9/PjEr67MhVL/++hs1ngYAAAAAAAAAAABA\nTYjwhtF9+/ad9P3u3btrICYi5TuKV/5pe0LL7Es+1/rodadVuwpnPjJ7xV/fSfzUZ3OH5A2+\nuIWvpQcAAAAAAAAAAAD4twgHo6mpqSFs+strr1WHy49ZY1atX78hhPDpT386CnH/4cArD1zb\nb+wfd1SGEBqcc+WP5v3++x3+/kjPzre89O6Royem/eTB6Q++9PvRHZJrowcAAAAAAAAAAAAg\nDkR4F2d2t24tQnh7xo+mbT/yvsdHtk/70Yy3QzirW7ecqOQdo2rNhOvG/HFHZdLp513U8fzm\npYvvHPLDPzz6rdteerdpzuA7H5w65QdDslOO/GPR966b9Keq6OcAAAAAAAAAAAAAxIcIB6NJ\n3b5z5+cbhwMLb738S3fNXLhm/V/Wr1k4866+l3974YHQ+NKx3706KbqdIYSDz/3kkbdCSP3K\nb994c13Rxo3PfjVt06Mj7l58KO1rc5fPmvSdW75971PLfj3kzFC94eGHF1mMAgAAAAAAAAAA\nABwV4WA0hPNvm/3UTZ9tfPit+T8a3r1T1mezOnUf/qPfv3W4yYXfeGrOqHYJ0Yw86s21a/eG\n0ParY679VEII4czeo4e3e7e0tCrjhlu6p/zrzBn9bv5KyxDeWb78jegHAQAAAAAAAAAAAMSF\nBhGfTEj90iNrNwz61c9n/mHl+r/vPnTaJ9OyvvA/w24ekZvWKIqB79m2bVsIoW3btv/vQdu2\nbUPY+P4nISRdcMH5Iby9devWED5bK1kAAAAAAAAAAAAAdVzkg9EQQkhOu/Lm+668OUotp9C8\nefMQyg4dOhRC0xBCCI0bNw4hhJSUlPedOv3000MIVVW+kh4AAAAAAAAAAADgqA83GI2pjIyM\nEHb+7W9/C+ETR5+cfWHXru+EC89+/6nt27eHENLS0mJQCAAAAAAAAAAAAFAXfajBaNnWwhf+\nuOr1v7+zv6LqyLGvsq+/b2hWTYb9p3N69mg/euWrixdvDp9rE0IIoes9S5Ycd6hy48a/hdAk\nJ6ft8T8OAAAAAAAAAAAA8DEV8WB03+rJ/b581+IdH/BV7/0vjvpgNFww+PrP/+/EDfPmvvXd\n75x34iPlv39i7p6Q8pXrejeOcgwAAAAAAAAAAABAvIhwMHp46diBYxfvCMnn9vr613q2Tz2j\nUcKxB9I71Xzbf2j3vcI93zvpid1nXj354ctade3VpBZyAAAAAAAAAAAAAOJChIPRomee+XsI\nzb88fdUzX2kR3aKP5JwvDv/GF2MdAQAAAAAAAAAAAFCnJEZ2rLS0NITQ6X/+py6vRQEAAAAA\nAAAAAAA4gQgHo6mpqSGEQ4cORTUGAAAAAAAAAAAAgJoX4WD0oiFDP5MY/rx48e7o5gAAAAAA\nAAAAAABQ0yIcjCZ8btyTky9vOPe7w6cW2YwCAAAAAAAAAAAAxJMGkR1b/P3Lxiw61Ljxtvnf\nuuRT49LOP+/sxsdOTa/6cdH9V0ahDwAAAAAAAAAAAICPKMLB6Lub1q5d+69/H/7n1tf/tPW4\nA23erckqAAAAAAAAAAAAAGpMhIPRPo/u2PGzkx1I/kRN1AAAAAAAAAAAAABQ4yIcjDY6o2XL\n6IYAAAAAAAAAAAAAEB2JsQ4AAAAAAAAAAAAAILo+1GD0wMbfP3T79b1zv9Dpkou/vziEEN5e\n/dsnnnhi0abD0akDAAAAAAAAAAAA4COL8CvpQwhv/+7ruYOnbaj413/T3g0hhKZvPnXTDc81\nGHb6zsf6JEejDwAAAAAAAAAAAICPKNIbRksfv+n6aRsq2wx+ZPlPer73uNmXr+/bJOx99qkF\nFR/8swAAAAAAAAAAAADEUISD0R1P/mLe/tB88E+m39Qlren7XjTJyjovhH0rV5ZEJQ8AAAAA\nAAAAAACAjyrCweirxcUhhM9dfnnT49+0bt06hLBjx46a7QIAAAAAAAAAAACghkQ4GD106FAI\nISUlJYSQkJDwvjf79+8PITRs2LDm2wAAAAAAAAAAAACoAREORs8555wQwrZt245/UfX66xtD\nCG3atKnRLAAAAAAAAAAAAABqSoSD0Q7dup0ZQvGzz7x15P03jB7Z9fhPHt8ZQvo112RGqxAA\nAAAAAAAAAACAjyTCwWiDa743tlPjI6vHf/lrv1xVGkIIe/+66Df39v/C1+fvDc273TWqc8Kp\nPgIAAAAAAAAAAACAmGgQ4bmEzNvnztrU6ys/n37TqyGEEBaO7bYwhBCa5oz67RMjW0erDwAA\nAAAAAAAAAICPKNLBaAgJqX2nFr0xZOYjj/1+ecmWd8qTzmh1YZc+I7418orU06IYCAAAAAAA\nAAAAAMBHE/lgNIQQTmv1xRvv/eKNUWoBAAAAAAAAAAAAIAoSYx0AAAAAAAAAAAAAQHRFeMPo\nWwsf+eOmkx3I6P6Nq8+riSAAAAAAAAAAAAAAalaEg9F1v7j55rknO9B/tsEoAAAAAAAAAAAA\nQJ0U4WA067p77+1wzJPqstJNqxfMWfRmQoeho/pfcFFWFOIAAAAAAAAAAAAA+OgiHIxe0P8H\nPzjB4/vueKDXJWMWrL5lwoQLajTrZA5uWTLnyTkLlq0teWNL6Z59BysbNml2eov0zKyOl/ca\nMHTgFemNay0FAAAAAAAAAAAAIA4kfqSfbvyZ70z8eurul74z9pmDNRR0cqUFE7q1a5c7fNzU\np19c9frmHaV79h/Yv7t0x+b1q158euq44bntzu82oeCdWmkBAAAAAAAAAAAAiA8R3jD6gRLb\nt78whBdfeumVMOCKmgg6icqSST16jl9XEVLa9hiWN+DqS7MzWp3ZPLlBZfnef2zfVLJ64ZwZ\nj73w5qLxPXs0LFp1Z9ZH/dMAAAAAAAAAAAAA6oePvKr85z//GULYv39/DcSc3MG5Eyatqwgt\n+01fMSvvvEbHvMvIzO50Zb+Ro0fPGHLZyOfWTrxn7m2zBzeJehIAAAAAAAAAAABAHPhoX0lf\n/e6i+3/xSgihXbt2NdNzEmuWLDkQQodRDx6/Fn1Po4y8B27LCeHA0qVFUe8BAAAAAAAAAAAA\niA8R3jC66n8HPFh47KMjh/65ff2aV97655HQvOftedEfjO7duzeEkJqaetJTqampIRQfPQsA\nAAAAAAAAAABAxIPRbYVz58490YvTWnYaetfUh0acfMVZI1q3bh3CpjUvv1w+tFvyBx0qX768\nKISQlpYW/SAAAAAAAAAAAACAuBDhYPTKyYWF3z32UULSaU3PbN323LOTE2o+60RyBg5q9+PJ\nG6flDfrMrGnf/MKn/iO9cufKqV/Lm7YzJGQO7N++dqIAAAAAAAAAAAAA6rwIB6OfPP/znz8/\nuiWnlNBx7IwxC665v3j+qC7pk7K7dO2UldHqrGaNkqoq9r2zfdNrryxdUbKrIoSmHe6YPrZj\njGMBAAAAAAAAAAAA6owIB6N1Q0qX+5atzBw36u5HC7aVLJ5dsvj4A8mpuTdOeGjiiPZNY5EH\nAAAAAAAAAAAAUCdFOBjd9MLPnn8z0s9s2/NbPTL+26BTaN5+xE8XD5+8eXXBsqKSjVt37dlf\nVpXUOOWMFmntsi/umntpekpilH4zAAAAAAAAAAAAQJyKcDD6p0dvvXVupJ/Zf3b0BqMhhBAS\nU9p07tOmc59o/g4AAAAAAAAAAACA+iLCwWj29ZMnZxY/+dCs18rPkF/seQAAIABJREFUzOn9\npcuzUpvt21ay9Lk/vPpuctaQUUNzmr3v8AXZUUkFAAAAAAAAAAAA4L8R4WA0s2ePqokTXqu6\n5M4VL078/CcSjj6tfmDFnVdf9ePfLTxy5/I7258WvcpjHNyyZM6TcxYsW1vyxpbSPfsOVjZs\n0uz0FumZWR0v7zVg6MAr0hvXUggAAAAAAAAAAABAXEiM7Nie34y7t6is1c359/57LRpCSPxk\nl4kP3fTpsjX3/mDWP6MUeKzSggnd2rXLHT5u6tMvrnp9847SPfsP7N9dumPz+lUvPj113PDc\ndud3m1DwTq20AAAAAAAAAAAAAMSHCG8YXbtyZUUIWTk5Scf//EUXZYfw4ooV68Kw3BqvO1Zl\nyaQePcevqwgpbXsMyxtw9aXZGa3ObJ7coLJ87z+2bypZvXDOjMdeeHPR+J49GhatujMrwj8N\nAAAAAAAAAAAAoJ6LcFVZXl4eQti5c2cIKce+efvtt0MIZWVlNV32Hw7OnTBpXUVo2W/6ill5\n5zU65l1GZnanK/uNHD16xpDLRj63duI9c2+bPbhJ1JMAAAAAAAAAAAAA4kCEX0mflZ0dQnj1\nVw+vKD/m+cGXp/6qJISQnZ1V42nHW7NkyYEQOox68Pi16HsaZeQ9cFtOCAeWLi2Keg8AAAAA\nAAAAAABAfIhwMJr+1dH9zwjVb/ykT5cRD85dUrThrxuKlsx54KuX9Z2ysTp8ov/tX02LbmcI\nYe/evSGE1NTUk546+v7oWQAAAAAAAAAAAAAi/kr60GLIjPl//ce1P1yy7tffG/Dr971IPPuK\nCc/OGHJ2FNqO07p16xA2rXn55fKh3ZI/6FD58uVFIYS0tOgPWAEAAAAAAAAAAADiQ4Q3jIYQ\nml9216K/rJt9/7evu6bL57IvzP5cl2uu+/b9s9dtWPSDLs2jWPhvOQMHtUsIO6flDZqycmfl\nCQ5U7lw5ZVDetJ0hIXNg//a1kQQAAAAAAAAAAAAQByK9YTSEEELimTkDxkwZMCZaMSeX0HHs\njDELrrm/eP6oLumTsrt07ZSV0eqsZo2Sqir2vbN902uvLF1RsqsihKYd7pg+tmNsGgEAAAAA\nAAAAAADqng81GI21lC73LVuZOW7U3Y8WbCtZPLtk8fEHklNzb5zw0MQR7ZvGIg8AAAAAAAAA\nAACgTvpQg9EDG38/7eFZL/35rV37D+XeX3T/leHt1b996a+Hzuk8+KqMhtFKPEbz9iN+unj4\n5M2rC5YVlWzcumvP/rKqpMYpZ7RIa5d9cdfcS9NTEj/U51VVVS1YsKC8vPwkZwoLC0MI1dXV\nH6kcAAAAAAAAAAAAIEYiH4y+/buv5w6etqHiX/9NezeEEJq++dRNNzzXYNjpOx/rkxyNvhNK\nTGnTuU+bzn1q4KMKCgr69u0bycn169fXwO8DAAAAAAAAAAAAqHWRDkZLH7/p+mkbKtsMfuSJ\nS3932ejn//W42Zev79vkuVnPPrXgl32ubRStyCjKzc2dN2/eKW8Yzc/Pv/DCC2utCgAAAAAA\nAAAAAKAGRTgY3fHkL+btD82/8pPpN3V54aX3vWiSlXVeCK+tXFkSrr04KoX/4eCWJXOenLNg\n2dqSN7aU7tl3sLJhk2ant0jPzOp4ea8BQwdekd74Q3xYUlJSnz6nvqo0Pz8/MfHDfdk9AAAA\nAAAAAAAAQB0R4WD01eLiEMLnLr+86fFvWrduHcJrO3bsqOGwEystmHDdDRMXbT90zNMD+3eX\n7ti8ftWLT0+9Z+xV4x6fdXfuWbWSAwAAAAAAAAAAABAHIhyMHjp0KISQkpISQkhISAjhyP97\ns3///hBCw4YNo5J3jMqSST16jl9XEVLa9hiWN+DqS7MzWp3ZPLlBZfnef2zfVLJ64ZwZj73w\n5qLxPXs0LFp1Z1aEfxoAAAAAAAAAAABAPRfhqvKcc84JYcu2bdtCyDjmRdXrr28MIbRp06bG\n0453cO6ESesqQst+01fMyjuv0THvMjKzO13Zb+To0TOGXDbyubUT75l72+zBTaKeBAAAAAAA\nAAAAABAHEiM71qFbtzNDKH72mbeOHL1h9Kgjux7/yeM7Q0i/5prMaBX+25olSw6E0GHUg8ev\nRd/TKCPvgdtyQjiwdGlR1HsAAAAAAAAAAAAA4kOEg9EG13xvbKfGR1aP//LXfrmqNIQQ9v51\n0W/u7f+Fr8/fG5p3u2tU54RTfcRHtnfv3hBCamrqSU8dfX/0LAAAAAAAAAAAAAARfyV9SMi8\nfe6sTb2+8vPpN70aQghh4dhuC0MIoWnOqN8+MbJ1tPrep3Xr1iFsWvPyy+VDuyV/0KHy5cuL\nQghpaWm1UAQAAAAAAAAAAAAQDyK8YTSEkJDad2rRG8um/WDkl67o1CGrfcfLeg77/s8LNryS\nf82nohj4npyBg9olhJ3T8gZNWbmz8gQHKneunDIob9rOkJA5sH/7WmkCAAAAAAAAAAAAqPsi\nvGG0aOpXf7q6+uJbZ956471fvDG6SR8koePYGWMWXHN/8fxRXdInZXfp2ikro9VZzRolVVXs\ne2f7ptdeWbqiZFdFCE073DF9bMfYNAIAAAAAAAAAAADUPREORv9vxW9mPnX4QN+Zt14S3Z6T\nSuly37KVmeNG3f1owbaSxbNLFh9/IDk198YJD00c0b5pLPIAAAAAAAAAAAAA6qQIB6Opqakh\n/O3tt98OoWV0g06hefsRP108fPLm1QXLiko2bt21Z39ZVVLjlDNapLXLvrhr7qXpKYkx7QMA\nAAAAAAAAAACocyIcjF507bXnPvC/xQsX7vrWDS2iWxSBxJQ2nfu06dwn1h0AAAAAAAAAAAAA\n8SDC+zgTPv/DJ+7q1GDed4ZMeHHroegmAQAAAAAAAAAAAFCTIrxhdPH3rxizqCKlSWnB+B5t\nJn+y9XnpZzc+Zmt61Y+L7r8yGoEncHjv/23dVdG8VZuzGyf859tdJS+9ujN8qn237NhfhQoA\nAAAAAAAAAABQB0Q4GH1309q1a//17yPl7259/d2txx5o826NZn2Qfese/kbeD54ufrcqhORW\nXxxxz0P3jfxc82OOLLvn6oFzQ//ZR+YMqJUkAAAAAAAAAAAAgDouwsFon0d37PjZyQ4kf6Im\nak5h+8wh3W5ZsDuEkJCc0vjw9pcfvvELLxU8+vyvr8+I8O8AAAAAAAAAAAAA+PiJcGjZ6IyW\nLaMbcmrVL//4Bwt2h8Q2A375zLThF51etnHej795y+Qnh+dWHFk264Y2SbEOBAAAAAAAAAAA\nAKiTEmMdELn1L7zw9xA+cV3+jJEXndEgJDRr96V7X3hl7k2f2TVnRO6wp/9eHetAAAAAAAAA\nAAAAgDrpJIPRF25t27Zt21tfeO/J7reKioqK3tod/awT2bJlSwjhoq5dm733LKnVlx5ZPOdr\nbbf95vqrbnx2x5HYlAEAAAAAAAAAAADUZScZjO7fsWnTpk079r/3ZNGYSy655JIxi6KfdSKN\nGjUKISQmHp/covcjf3xs4Dlv/mpIt28+XxqLMgAAAAAAAAAAAIC6rEGsAyKXnp4eQsmWLVtC\naH/sm8S0rzzxx3/suezbD/fv3nTEmbHJAwAAAAAAAAAAAKijTnLDaF2T0bnz2SG8tXr1Oyd4\nedoFtz77/PhLk/784M9jdAEqAAAAAAAAAAAAQB0VR4PRpNxrv/SJUFUw97ndJ3zf9JIfLvjd\nrRc2quUsAAAAAAAAAAAAgDoujr6SPjS48o7Zj3fdcVrbig868ckrp/zx+c/MXP3PcEF2bZYB\nAAAAAAAAAAAA1GGnHIweLN28efO//l168Pgn/9a0RZuzm9Ro2Qk0zLjq+oyTH0n4dO7Nd+RG\nOwQAAAAAAAAAAAAgjpxyMPr8Leeee6onIYT+s4/MGVBjVQAAAAAAAAAAAADUmJMMRk9revrp\np0f6OU1Pq4kaAAAAAAAAAAAAAGrcSQajfR/bs6f2QgAAAAAAAAAAAACIjsRYBwAAAAAAAAAA\nAAAQXQajAAAAAAAAAAAAAPWcwSgAAAAAAAAAAABAPWcwCgAAAAAAAAAAAFDPGYwCAAAAAAAA\nAAAA1HMGowAAAAAAAAAAAAD1nMEoAAAAAAAAAAAAQD1nMAoAAAAAAAAAAABQzxmMAgAAAAAA\nAAAAANRzBqMAAAAAAAAAAAAA9ZzBKAAAAAAAAAAAAEA9ZzAKAAAAAAAAAAAAUM8ZjAIAAAAA\nAAAAAADUcwajAAAAAAAAAAAAAPWcwSgAAAAAAAAAAABAPWcwCgAAAAAAAAAAAFDPGYwCAAAA\nAAAAAAAA1HMGowAAAAAAAAAAAAD1nMEoAAAAAAAAAAAAQD1nMAoAAAAAAAAAAABQzxmMAgAA\nAAAAAAAAANRzBqMAAAAAAAAAAAAA9VyDWAd8eAe3LJnz5JwFy9aWvLGldM++g5UNmzQ7vUV6\nZlbHy3sNGDrwivTGsS4EAAAAAAAAAAAAqEvibDBaWjDhuhsmLtp+6JinB/bvLt2xef2qF5+e\nes/Yq8Y9Puvu3LNiFAgAAAAAAAAAAABQ58TTYLSyZFKPnuPXVYSUtj2G5Q24+tLsjFZnNk9u\nUFm+9x/bN5WsXjhnxmMvvLlofM8eDYtW3ZkVT38aAAAAAAAAAAAAQPTE0ary4NwJk9ZVhJb9\npq+YlXdeo2PeZWRmd7qy38jRo2cMuWzkc2sn3jP3ttmDm8QoFAAAAAAAAAAAAKBOSYx1QOTW\nLFlyIIQOox48fi36nkYZeQ/clhPCgaVLi2q1DQAAAAAAAAAAAKDuiqPB6N69e0MIqampJz11\n9P3RswAAAAAAAAAAAADE1WC0devWIYQ1L79cfpJD5cuXF4UQ0tLSaqkKAAAAAAAAAAAAoK6L\no8FozsBB7RLCzml5g6as3Fl5ggOVO1dOGZQ3bWdIyBzYv32t9wEAAAAAAAAAAADUTQ1iHRC5\nhI5jZ4xZcM39xfNHdUmflN2la6esjFZnNWuUVFWx753tm157ZemKkl0VITTtcMf0sR1jXQsA\nAAAAAAAAAABQV8TRYDSElC73LVuZOW7U3Y8WbCtZPLtk8fEHklNzb5zw0MQR7ZvGIg8AAAAA\nAAAAAACgToqrwWgIoXn7ET9dPHzy5tUFy4pKNm7dtWd/WVVS45QzWqS1y764a+6l6SmJsU4E\nAAAAAAAAAAAAqFvibTAaQgghMaVN5z5tOveJdQcAAAAAAAAAAABAPHAfJwAAAAAAAAAAAEA9\nF4c3jB7csmTOk3MWLFtb8saW0j37DlY2bNLs9BbpmVkdL+81YOjAK9Ibx7oQAAAAAAAAAAAA\noC6Js8FoacGE626YuGj7oWOeHti/u3TH5vWrXnx66j1jrxr3+Ky7c8+KUSAAAAAAAAAAAABA\nnRNPg9HKkkk9eo5fVxFS2vYYljfg6kuzM1qd2Ty5QWX53n9s31SyeuGcGY+98Oai8T17NCxa\ndWdWPP1pAAAAAAAAAAAAANETR6vKg3MnTFpXEVr2m75iVt55jY55l5GZ3enKfiNHj54x5LKR\nz62deM/c22YPbhKjUAAAAAAAAAAAAIA6JTHWAZFbs2TJgRA6jHrw+LXoexpl5D1wW04IB5Yu\nLarVNgAAAAAAAAAAAIC6K44Go3v37g0hpKamnvTU0fdHzwIAAAAAAAAAAAAQV4PR1q1bhxDW\nvPxy+UkOlS9fXhRCSEtLq6UqAAAAAAAAAAAAgLoujgajOQMHtUsIO6flDZqycmflCQ5U7lw5\nZVDetJ0hIXNg//a13gcAAAAAAAAAAABQNzWIdUDkEjqOnTFmwTX3F88f1SV9UnaXrp2yMlqd\n1axRUlXFvne2b3rtlaUrSnZVhNC0wx3Tx3aMdS0AAAAAAAAAAABAXRFHg9EQUrrct2xl5rhR\ndz9asK1k8eySxccfSE7NvXHCQxNHtG8aizwAAAAAAAAAAACAOimuBqMhhObtR/x08fDJm1cX\nLCsq2bh11579ZVVJjVPOaJHWLvvirrmXpqckfqjPq6qqWrBgQXl5+UnOFBYWhhCqq6s/UjkA\nAAAAAAAAAABAjMTbYDSEEEJiSpvOfdp07lMDH1VQUNC3b99ITq5fv74Gfh8AAAAAAAAAAABA\nrYvLwWgNys3NnTdv3ilvGM3Pz7/wwgtrrQoAAAAAAAAAAACgBsXhYPTgliVznpyzYNnakje2\nlO7Zd7CyYZNmp7dIz8zqeHmvAUMHXpHe+EN8WFJSUp8+p76qND8/PzHxw33ZPQAAAAAAAAAA\nAEAdEWeD0dKCCdfdMHHR9kPHPD2wf3fpjs3rV7349NR7xl417vFZd+eeFaNAAAAAAAAAAAAA\ngDonngajlSWTevQcv64ipLTtMSxvwNWXZme0OrN5coPK8r3/2L6pZPXCOTMee+HNReN79mhY\ntOrOrHj60wAAAAAAAAAAAACiJ45WlQfnTpi0riK07Dd9xay88xod8y4jM7vTlf1Gjh49Y8hl\nI59bO/GeubfNHtwkRqEAAAAAAAAAAAAAdUpirAMit2bJkgMhdBj14PFr0fc0ysh74LacEA4s\nXVpUq20AAAAAAAAAAAAAdVccDUb37t0bQkhNTT3pqaPvj54FAAAAAAAAAAAAIK4Go61btw4h\nrHn55fKTHCpfvrwohJCWllZLVQAAAAAAAAAAAAB1XRwNRnMGDmqXEHZOyxs0ZeXOyhMcqNy5\ncsqgvGk7Q0LmwP7ta70PAAAAAAAAAAAAoG5qEOuAyCV0HDtjzIJr7i+eP6pL+qTsLl07ZWW0\nOqtZo6Sqin3vbN/02itLV5TsqgihaYc7po/tGOtaAAAAAAAAAAAAgLoijgajIaR0uW/Zysxx\no+5+tGBbyeLZJYuPP5CcmnvjhIcmjmjfNBZ5AAAAAAAAAAAAAHVSXA1GQwjN24/46eLhkzev\nLlhWVLJx6649+8uqkhqnnNEirV32xV1zL01PSYx1IgAAAAAAAAAAAEDdEm+D0RBCCIkpbTr3\nadO5T6w7AAAAAAAAAAAAAOKB+zgBAAAAAAAAAAAA6rn6Nxjd9MLPfvazn72wKdYdAAAAAAAA\nAAAAAHVE/RuM/unRW2+99dZH/xTrDgAAAAAAAAAAAIA6ov4NRgEAAAAAAAAAAAA4RoNYB0Su\nqnx/WeWpj5VXRT8FAAAAAAAAAAAAII7E0WD02eubDZwb6wgAAAAAAAAAAACAuOMr6QEAAAAA\nAAAAAADquTi6YfTcc9uEsPmiyRtf+e65Jzn2zOCGg5+pnSJqRVVV1YIFC8rLy2MdQlQUFhbG\nOgEAAAAAAAAAAKD+i6PB6OeuvvqsB6f9edHiPXfcdNYHH0tMqL0kasPzzz/ft2/fWFcAAAAA\nAAAAAABAHIujwWjC5d2vSp729PI/Liy76brGsa6h1pSVlYUQhg0blpOTE+sWat68efOWLl0a\n6woAAAAAAAAAAIB6Lo4GoyG527BRXyr/S/P9m0P4zAeeuvjmadN6hHMvrr0uakNOTk737t1j\nXUHNKy4ujnUCAAAAAAAAAABA/RdPg9FwRq/Jz/U61aE2V914Y23EAAAAAAAAAAAAAMSJxFgH\nAAAAAAAAAAAAABBdcXXDKABQN1RXV4cQCgsLYx1CtCQnJ/fq1SspKSnWIQAAAAAAAABAzTAY\nBQA+tA0bNoQQ8vPz8/PzY91CtMyfP793796xrgAAAAAAAAAAaobBKADwoR0+fDiEMGzYsJyc\nnFi3UPOKi4tnzpxZVlYW6xAAAAAAAAAAoMYYjAIA/6WcnJzu3bvHugIAAAAAAAAAgFNLjHUA\nAAAAAAAAAAAAANFlMAoAAAAAAAAAAABQzxmMAgAAAAAAAAAAANRzBqMAAAAAAAAAAAAA9ZzB\nKAAAAAAAAAAAAEA9ZzAKAAAAAAAAAAAAUM8ZjAIAAAAAAAAAAADUcwajAAAAAAAAAAAAAPWc\nwSgAAAAAAAAAAABAPWcwCgAAAAAAAAAAAFDPGYwCAAAAAAAAAAAA1HMGowAAAAAAAAAAAAD1\nnMEoAAAAAAAAAAAAQD1nMAoAAAAAAAAAAABQzxmMAgAAAAAAAAAAANRzBqMAAAAAAAAAAAAA\n9ZzBKAAAAAAAAAAAAEA9ZzAKAAAAAAAAAAAAUM8ZjAIAAAAAAAAAAADUcw1iHQAAQN1SXV0d\nQigsLIx1CNGSnJzcq1evpKSkWIcAAAAAAAAAUHsMRgEAOMaGDRtCCPn5+fn5+bFuIVrmz5/f\nu3fvWFcAAAAAAAAAUHsMRgEAOMbhw4dDCMOGDcvJyYl1CzWvuLh45syZZWVlsQ4BAAAAAAAA\noFYZjAIAcAI5OTndu3ePdQUAAAAAAAAAUDMSYx0AAAAAAAAAAAAAQHQZjAIAAAAAAAAAAADU\ncwajAAAAAAAAAAAAAPWcwSgAAAAAAAAAAABAPdcg1gEAAEDtqa6uDiEUFhbGOoSoSE5O7tWr\nV1JSUqxDAAAAAAAAgDrHYBQAAD5GNmzYEELIz8/Pz8+PdQtRMX/+/N69e8e6AgAAAAAAAKhz\nDEYBAOBj5PDhwyGEYcOG5eTkxLqFGlZcXDxz5syysrJYhwAAAAAAAAB1kcEoAAB87OTk5HTv\n3j3WFdSw6urqEEJhYWGsQ4iW5OTkXr16JSUlxToEAAAAAACAuGQwCgAAUB9s2LAhhJCfn5+f\nnx/rFqJl/vz5vXv3jnUFAAAAAAAAcclgFAAAoD44fPhwCGHYsGE5OTmxbqHmFRcXz5w5s6ys\nLNYhAAAAAAAAxCuDUQAAgPojJyene/fusa4AAAAAAAAA6pzEWAcAAAAAAAAAAAAAEF1uGAUA\nAIC6rrq6OoRQWFgY6xCiJTk5uVevXklJSbEOAQAAAAAA6i2DUQAAAKjrNmzYEELIz8/Pz8+P\ndQvRMn/+/N69e8e6AgAAAAAAqLcMRgEAAKCuO3z4cAhh2LBhOTk5sW6h5hUXF8+cObOsrCzW\nIQAAAAAAQH1mMAoAAADxIScnp3v37rGuoOZVV1eHEAoLC2MdQrQkJyf36tUrKSkp1iEAAAAA\nAHysGYwCAAAAxNKGDRtCCPn5+fn5+bFuIVrmz5/fu3fvWFcAAAAAAPCxFoeD0YNblsx5cs6C\nZWtL3thSumffwcqGTZqd3iI9M6vj5b0GDB14RXrjWBcCAAAAROzw4cMhhGHDhuXk5MS6hZpX\nXFw8c+bMsrKyWIcAAAAAAPBxF2eD0dKCCdfdMHHR9kPHPD2wf3fpjs3rV7349NR7xl417vFZ\nd+eeFaNAAAAAgP9GTk5O9+7dY11Bzauurg4hFBYWxjqEqEhOTu7Vq1dSUlKsQwAAAAAATi2e\nBqOVJZN69By/riKktO0xLG/A1ZdmZ7Q6s3lyg8ryvf/Yvqlk9cI5Mx574c1F43v2aFi06s6s\nePrTAAAAAKiXNmzYEELIz8/Pz8+PdQtR8f3vf79jx46xriAqDIIBAAAAqGfiaFV5cO6ESesq\nQst+01fMyjuv0THvMjKzO13Zb+To0TOGXDbyubUT75l72+zBTWIUCgAAAABHHT58OIQwbNiw\nnJycWLdQw/7whz/8/+3dZ3hWRdoA4HkTSAhFuqKAYK8oCjbAgkIQRUVXEHXtrOiuuuK3i1jW\nuirqrmV1V1gLdrC3RSWAAgJKsVBEKYqFJl1qKMn7/XgBSUKSN5CYcLzvH15mzpw5z8yck2uY\n68k5H3zwwb333lvegVCGJARHmIRgAAAA4DdoB0oYHT98+KoQml/7j/zZor9I3+vS+//8rzdv\nmDhixIRwznFJNJqTk/Puu+9mZ2cXUSfxybDE58MoLxMnTizvECgT33//fTC/EWVyo838Rpv5\njTbzG2EmN9rMb7SZ32hLzC+RlJOTE0I49NBDd9lll/KOhdI3Y8aMWbNmSQiONgnBUZWbmztl\nypSDDz44JSWlvGOh9JnfCDO50WZ+o838Rpv5jTbzG23+VJKti+8w3j4vLYTQ6fnsImutee7U\nEEL6+e8k1+iQIUOSHKg777yzFDpByb3zzjtl+AAAAAAAAAAAAABEzpAhQ8o784sKZwd6w2jj\nxo1D+Gb8Rx9ln9+uSmGVskeNmhBC2H333ZNrtG3btm+//XbRbxjNzc2dNm1ar169ShYvpaRj\nx47FzhE7Ln+tEmEmN9rMb7SZ32gzvxFmcqPN/Eab+Y028xthJjfazG+0md9oM7/RZn4jzORG\nm/mNNvMbbeY32sxvtFWpUqVt27blHQUVTiwej5d3DEmKf3rj/kfcMz3W+LQHBj7+p1a7FMh1\n3fDTmH//odt17/wY3+/G8V/f5TsyAAAAAAAAAAAAACHsUAmjIawc3btNh3snrgohfedmrY8/\n8uC9GtarkZ6as3bFojnfTBk3YvTkBWtDqNa89+BR97SuVt7RAgAAAAAAAAAAAFQMO1TCaAhh\n+aT+N117yxMfzt7q98mrNGrb/Y6H7rrkkJ1+7bgAAAAAAAAAAAAAKqwdLWE0hBBC7srvxn44\ncsLk6T8sWLZyTU5qRvVaO+++b7OWx7c9qkn1lPKODgAAAAAAAAAAAKBi2SETRgEAAAAAAAAA\nAABInvdxAgAAAAAAAAAAAESchFEAAAAAAAAAAACAiJMwCgAAAAAAAAAAABBxEkYBAAAAAAAA\nAAAAIk7CKAAAAAAAAAAAAEDESRgFAAAAAAAAAAAAiLhK5R0AFGPdunV9+vTZb7/9UlLkNwMA\nbK/c3NzJkyc3a9bM4goAYDtZWQEAlCKLKwCAUpSbmztt2rTevXunpaWVdyxULBJGqejuu+++\nW2+9tbyjAAAAAAAAAAAA2GGkpKTcfPPN5R0FFYuEUSq6ffbZJ4TQs2fPY445prxjAQDY4X38\n8ccPPvigxRUAwPazsgIAKEUWVwAApSixuEqkXcGWJIxS0SW+OnHMMcd06dKlvGMBAIiCBx98\n0OIKAKBUWFkBAJQiiysAgFL04IMPJtKuYEvuCQAAAAAAAAAAAICIkzAKAAAAAAAAAAAAEHES\nRgEAAAAAAAAAAAAiTsIoAAAAAAAAAAAAQMRJGAUAAAAAAADHH8oeAAAcgUlEQVQAAACIOAmj\nAAAAAAAAAAAAABEnYRQAAAAAAAAAAAAg4iSMAgAAAAAAAAAAAESchFEAAAAAAAAAAACAiJMw\nSkWXkZGx+b8AAGwniysAgNJiZQUAUIosrgAASpHFFYWJxePx8o4BipKTkzNs2LCTTjopNTW1\nvGMBANjhWVwBAJQWKysAgFJkcQUAUIosriiMhFEAAAAAAAAAAACAiPNJegAAAAAAAAAAAICI\nkzAKAAAAAAAAAAAAEHESRgEAAAAAAAAAAAAiTsIoAAAAAAAAAAAAQMRJGAUAAAAAAAAAAACI\nOAmjAAAAAAAAAAAAABEnYRQAAAAAAAAAAAAg4iSMAgAAAAAAAAAAAESchFEAAAAAAAAAAACA\niJMwCgAAAAAAAAAAABBxEkYBAAAAAAAAAAAAIk7CKAAAAAAAAAAAAEDESRgFAAAAAAAAAAAA\niDgJowAAAAAAAAAAAAARJ2EUAAAAAAAAAAAAIOIkjAIAAAAAAAAAAABEnIRRytWqrwbcdE7r\nfXfZqUqVGjvv3apL7xemrPg1zgUAiKRtWSAt69sutlUnP7Hs14gZAKBC2rBs5qjX/nNLj85H\n71GzUiwWi5389MqkT7ZtBQCQ17YuruxcAQDktX7xl1n97+hxRpvm++xWq0paRs3dDmh9ds/H\nPpq3IdkW7Fz95sXi8Xh5x8Bv1fLRvY5vf/8Xa/IUph903Xsf/bNt7TI8FwAgkrZxgbSsb7va\nVw7byoEOjy99v3ut0o0RAGAHseH5zpUveGvLkg79V7x/cfUkTrVtBQCQ3zYvruxcAQDkkf18\n54y866qEWJ3j/zHsveuaZxRzvp0rvGGU8rN25E2/v/+LNSmNO90/ZPrSNWuWzhj2j85NUtZ+\n+cAF13+wpuzOBQCIpO1cIB1051fxfOy5AwC/YbHKtfdqfeYVtz32+pjpfTuV4ETbVgAABW3z\n4irBzhUAwEaxtHoHZV56S7+3Rn0xfc7S1auXfD/htTs67p4aXzKi18UPTCvmbDtXhOANo5Sb\nlQM71z/3rew9eo6e+kCrKhsL14697uCjH5xZ+bRnFrx9YeH/zNuecwEAImnbF0iJ9zQcdOdX\nU27e/9eKFgBgB7JhYOfK576V5EuwbFsBABStRIsrO1cAAElY9/lNhx5+99fhmIfnjLlmt8Lr\n2bkihOANo5SXnA/fy8oOodnFV2z+BRRCSD/qyksPD2F91rtDN5TNuQAAkWSBBABQEViVAQAA\nAPArSzvsuKOrhxDWrl1bVDU7VyRIGKV8fDN58poQah555L55y/c56qjaIaydMmVm2ZwLABBJ\n271A+vHZ3x+8a430tKq1Gx107Dl/7Td6Xk6ZBQsAEFm2rQAAyoCdKwCAIk3/9LOVITRs3bpp\nUbXsXJEgYZTysWDBghBCw4YN8x9IFCUOl8G5AACRtN0LpOUzPv1y/sp169csmzN11Mv/uOLY\nQ068/9PVZRIrAEB02bYCACgDdq4AAAqXO+s/f7p3UqjR4fa/tooVVdHOFQkSRikfa9asCSGk\np6fnP1ClSpUQwurVRfwjb3vOBQCIpO1YIMWq7X3Kdf9+c/SU75esXrXoh6kjnr3hlD2rxBeN\n7PW7Xh9ll2HMAADRY9sKAKBU2bkCACjawvf+eMqfh65sfP7Tz17WuOiqdq5IkDBK+cjIyAgh\nrF27Nv+B7OzsEELVqlXL5lwAgEjajgVSzQv6DvrnH89oddDutTOq1m18wHEX3D1o7PNd6ofw\nff9+7xdoEACAwtm2AgAoVXauAAAKlzv3rR4nnNlveoMzn/ig/1k7F1fdzhUJEkYpHzvvvHMI\nYc6cOfkPJIrq169fNucCAERSKS+Q6p116em1Qlg9ZcqsUgoQAOA3wbYVAECZs3MFABBCWP/N\nc+cfe/Z/p+3a9emRL1+yd+Xiz7BzRYKEUcrHXs2aZYTw87hx0/OWzxg7dmkI6c2a7VM25wIA\nRJIFEgBARWBVBgAAAECZWz3xodPbXDTwxz0uGjDyxQv2qJTUSXauSJAwSvlIbdsxs0oIk5/u\nOyb7l9K14x576rMQKmee0q6IX2Xbcy4AQCSV8gJpyVvPvLMshKoHHti0dOMEAIg221YAAGXO\nzhUA8Bu3ZNTNJx3f8/2lB17x+sj+XRqnJnuenSsSJIxSTqqf2bN70xBmPdyt6z+Hzli2du2y\nbz54oFvXh2eG0PDia8+utaneokdPiMVisXZ9F5X8XACA34xtXVytGnj50efe+MS7n3w566cV\na9csnT1t9Iu3dj7m/IELQmh84eUdq5RTfwAAdggFdq5sWwEAbDs7VwAAxcidO+jK4zPv+mT9\nYdf9b/hjnRrEiqhr54qti8Xj8fKOgd+q5aN6HZd5/8Q1eQrTD+z53qgH2tbe9POiR0+of/WI\ncNJjC4deUa+E5wIA/JZs0+Jq5dOdalwyqGBjsbrH3fPeoOuPqF7WUQMAVEzLnji59h8Gb/XQ\n8Y/MG35VgxDC1neubFsBABSwbYsrO1cAAPnM7NNynxs+LeRgzR5DlvVtt/lHO1dsnTeMUn52\nanPfqAkv3ND16L3qV0tLq1Zvz6PP7vX8+E+S+gW0PecCAETSNi2Qqp/35ITXHvzz744/dO/d\naqZXSq9Rb4/m7X5/45Mff/mBPXcAgG1h2woAoJTYuQIAKGV2rvCGUQAAAAAAAAAAAIDI84ZR\nAAAAAAAAAAAAgIiTMAoAAAAAAAAAAAAQcRJGAQAAAAAAAAAAACJOwigAAAAAAAAAAABAxEkY\nBQAAAAAAAAAAAIg4CaMAAAAAAAAAAAAAESdhFAAAAAAAAAAAACDiJIwCAAAAAAAAAAAARJyE\nUQAAAAAAAAAAAICIkzAKAAAAAAAAAAAAEHESRgEAAAAAAAAAAAAiTsIoAAAAAAAAAAAAQMRJ\nGAUAAAAAAAAAAACIOAmjAAAAAAAAAAAAABEnYRQAAAAAAAAAAAAg4iSMAgAAAAAAAAAAAESc\nhFEAAAAAAAAAAACAiJMwCgAAAAAAAAAAABBxEkYBAAAAAAAAAAAAIk7CKAAAAAAAAAAAAEDE\nSRgFAAAAAAAAAAAAiDgJowAAAAAAAAAAAAARJ2EUAAAAAAAAAAAAIOIkjAIAAAAAAAAAAABE\nnIRRAAAAAAAAAAAAgIiTMAoAAAAAAAAAAAAQcRJGAQAAAAAAAAAAACJOwigAAAAAAAAAAABA\nxEkYBQAAAAAAAAAAAIg4CaMAAAAAAAAAAAAAESdhFAAAAAAAAAAAACDiJIwCAAAAQF4bBnaO\nxWJ79/6iiJIKY2aflrFY7IRH55d3IFFRgee6TFSg/g6/ol6bX/VG3ua+z3+0TSwWa9nnuzII\nCgAAAADKjoRRAAAAAAAAAAAAgIiTMAoAAAAAxanU7c14PD6zT/PyDuS3ZeXTJ8disf1vnvJr\nXnSb57pcok1eYeGV/72du2j8E9ed3fbQJmc8uXj0tfvsduAxnXrc89KE+evK/Mpl3fcKfksA\nAAAA8NtTqbwDAAAAAADgtyl32mMdW/8pa3F8U8HKeV99MuirTwb9d+xz69/8vf1rAAAAAChF\n3jAKAAAAAEB5WP3GzX/NWhyvc+xfX/xk1puX1W31j6nfTRr9dt/eXVvsmh4r7+gAAAAAIGIk\njAIAAABAcTYM7ByLxfbu/cUvRSunvND7rCP3qFM1vWrd3Vuc3vPZSStm9mkZi8VOeHT+pjq5\ny2cMffymC9q3PKBxnappGTV32791l788OWFJPE/j8x9tE4vFWvb5Lnvmq3875+i96lVLr1Kr\n0SGnXtN/0sp8cayaOvDG3x21Z90tLlqyjmy+1qqpL/TucvTeO++UnlL94v/9coEZ/+tz+Skt\n9qxfIz0to+auBxzb7abnJ/6cL4gZg+6/8vSj9mlQMyO9+s57Htr2/NsGfLY4J5Sgkc1hrJjY\nv+dph+1ep2p61Xp7HtX1ppemrdpYZfZDbWI1LhkcQph2V7PYJu36Ltt4fP3sD/911elH7lW/\nRnpaRq2GB55w3s0DJueNtLjObl3BuU5igoqLtiRjkj/a5O6iIqemqPC2cm8nMbzJ3rQ588Y8\n0evsVgc03aVGRtW6jfdv0e7COwZMWLhh0/EpY8euCuG4216979yjmtasFGLptZs0a3Vaj3te\nmvBYl9REnbVvnFMtFmt4zZhfms1+qUvVWCwWO/CWqb8ULu3XPjUWO/jvX285IEUO+9b7nsyj\nvTmQEt8SBWcNAAAAAH41PukDAAAAACW15pPb2ra7fcLG5MY1P372zkMXDR196Rnr8lYbd0vb\n9g/P+eWsedPGvDptzFtvjfvfuH6ZtfPWXfflPzv0eXTkpmS2OZPffeTSUVOzJw29ssnGouzx\nfz/ppL+NXVHMRZOwbtJdJ935xNjViZ8ycnMT/zN/0J9O6vqfqas3V5v/9aiX7h71xqsjXx31\n39PqhxBCmPvWFSed1+/rzXWyZ00aPmvS8MGLdl306AnJNrLp0MQ7T7rjqfFrNnZo1rhX7u42\neNiMD4fffHiVojuwftLDp5zUc+iiTTmT6+Z+NWLAXSNef3n4Kx/1O22XZDpbYsVPUKFKMiYF\no032LipianqXrKtJD29xY7J00OVHnv7U7M1DPnvaktnTPhv28rc1sp/uFEIIoVKlSiGkVKtW\ntYhw0k/MPK7Sy+8PHjw5tGoWQggh96OsYWtCCOGrwYN/vOPAxomBGZo1Kjc0bt9+/43nlWDY\nf5Hko51U9wEAAACgovGGUQAAAAAomdzP777ojgmrYvVa/9/zH89YsGLF/Gmjn7n6kG+eGjg5\nb8XU2gd16vnoWx9P/nbByuzVS+d9NeK5v5xQf/3M/15136f5G538/GMzW9zw4scz569YvfTH\nsc/3OKRKWD7srn99vPF4fGKfi24duyJW55hrnxkz46cVK+ZPG/lkjwNmFLhoEiYPePLb5j2f\nGTV97or18fjqZ08PIYS5T13Y7T9Tc/c7556XxsyYv3TN2lULvxv/+l2nNc2d/vilf/3fyhBC\n+KHveef2+3p12u7t//LsiCnfLVq1atGsScNfuLVbi3qJv0xPppFfwhj41IwD/vjkyGnzV6z4\nafqop69uWTssH3vrhfdMjIfQ6NpR8RX9O4QQ9rtpcnyToVfUCiE+pc/5/zd0UbzGoZf954Op\ns39eufib8a/e3G631LUz/nvBlS8vTaKz26DICSoi2hKOScFok7uLipyaosIroATDW8xNu+qt\nx56ZnZverPuTo6b/tDx77Yr5M7/48MXbzz9ql7RNLRzSsWPDkDv4josf/uC7lYXl8tbMzDwy\nhOlZWd9vLBifNWRpOPLkk+uECVlZSxJlOSOzPsgOtdtnHpH4uUTDvknSj3bx3S9szAvpIwAA\nAAD8KuIAAAAAwJbWDzgjhLDX9Z8XUjLi6t1CCLteMmjFlmctfbNb/RBCOP6ReUU1/vOzp1QK\nodnt0zaXzHukdQgh7Ntr3Jot6m0Yec3uIYRWD/yQ+HnkNQ1DCA0uemf5lq0tfuV3dZK4aP5r\n7db9/eX5jky8ef8QKrfrOzdf+YbRPRuFUO38N9bH4xN67x1CqNWp/9zcrTefTCNbhFGn26uL\nt6y29I3z64UQGlw1KvFzwXy7eDwej3/054YhhNpnvrBwy9J1n15/QEoIsbb/3jQUhXe2KAVn\nP7kJKizaEo5JctEWuIuKnZrCwivY36SGN6kxWdX/1JQQjtw8Qlu1bPRdx++cGkIIldLSUuoc\n3vXqO/q+PWnhhjyVpty2fwip7fsticfj8fjUOw4JYb9bx7/WJSNkdHkpOx6Px+NjrmsSQvrZ\nA1clzkhq2Av0PdlHe/tuCQAAAAAoL94wCgAAAAAlMu/TT+eGUPesC0+pvmVxrTMuOr1mvqo5\nP338eK9z2x7atH6NKpVSYrFYLFbzwnc3hPDDDz/kq7p7p7OP2PJD7KmHHHJgCGHRokUhhBDm\nf/rpnBDq/u7iTjW2PKvO2Zd13qnEPajT6bwONfIWLf5o1NchrB/2p8aVKlWqlJqampqakpKS\nkhKr1PrB2SGs+vbbBWHB6NEzQ6jd9ZoLd41ttd1kGtnCTp0vTaS7blKr82Vn1Qlh/oQJc0Lh\nEoerdu5+br0tiysffuVlR4YQHz8+79tbt9LZbVLMBBWmhGOytWiTuIuKnZrklWR4ixmTqh1/\nf2bdMO6eblf+c8DQL75burXPuoearW4c/s3Xg/vd2uOM5nVTlnz28iO3XHH6IU0OOKfvxDWb\nKx2Umdkw5Gz8EP1PWVmTw26ZmS1Pyjwudc2wrI9yQwjTsrK+DymtM9slPm5fwmHfqASPdhLd\nBwAAAIAKR8IoAAAAAJTIzz//HEJo2LBh/gP5i2YP7HpYm8vvHzh80veLVq7NiW9xKDs7O9/J\n9evXz1uQnp4eQsjJyQkhhLBs2bKtX7RRo0Yl7kGTJk3yF21Mcovn5OTk5OTk5ubmJt5Uuen4\nunXrwtKlS0MIuzZqVNimYjKNFB16oijR2cIkxr9R48b5UyMTRSuXLcspprPbpJgJKkwJx6Rg\ntEndRcVOTfJKMrzFjcku3Z4f9797Tq06+r5L2h+2R53qdfZuddY1Dw2etTZf09X3zrz8tr6v\n3nt6zVb3fjbpvX/3aFF5xstXnnbDR5vH5sgOmbVD9rDBI3PCiqysT+K1MjscFWp2yDwyLMka\nPD6EeVlZU0I4PDOzbqJ+CYc9T9+TeLST6z4AAAAAVDQSRgEAAACgRGrWrBlCmDOnwFsw8xWN\neeD61+fl7nLizS+NmvrjohXZ63Ny4/H4hrfPy9iGi9aqVWvrF509e3aJG0tJKbAtmGi/7pXD\ncrb+paIJvZuG2rVrhxDmzZ6dW2SQRTeyhTkFQ0/0JtFOYRLjP/vHH+P5DiSKqteqlVpMZ39N\nJRyTAtEmdxcVOzXJK+HwFqPKnqf27j9k0vzlS2d9MXxgnwv3nf1iz5MPO63vzPytbxSrumuz\nk//Yd9A9J8bCj6+/MWFTeeoJmSemhWVZWePWfZA1YkPaiZknpIbQODNzv/BjVtZXy7OyxoWw\nT/v2m9JtSzjsefpe/KMNAAAAADsoCaMAAAAAUCK7tmixawiL33juvVVbFv/89rNv/7zFz6u/\n/vqHEBpfeM+dXVsf0Khu9fRKKbEQlmW9OXxNKLkGLVo0DGHxa88MWrll8dLXnnpr+bZ0Ir9d\nWrXaK4TFr/17wLxCUvlC2Ll1671DWPryI88WUieZRrbw81v9X1+at+CpN5aE0KBly8T7HCtX\nrhwKvo01cXj1m08MyPPl7w1f9Os/PoRYy5Ytkrh2Gdh6tCUck/ySvIuKnZrCwiuoTIY3llar\n6aHHn3X5LU+/c3ub8POQR5+fVnT9eDxfsFXbZ7ZOCT9kDX4qa+iqWOsO7auFEMLBmZm7hUlZ\ng17IGr4h1M/MPGxT9W0b9iQf7aQlO+YAAAAA8CuRMAoAAAAAJdPm4kv3iYW5/S865foB475d\ntGrlwpkfP9+z42UDF25ZK6NBg5ohzH753oc+nL5w9bo1S36cMvjRHu3Oe2rutl300u77p4T5\nz1x88l9eGPvNwlUrF8wY/fRVHf7w6uJS6VM44qpebauGBa93P+6Mm54d8eUPC1etXbN09vQv\nRr72cM+zWvR4ZW0IocWV/3d81bDsnR5Hd+j13MipPyxZvXrJ95NHvHj7eR1uG5VsI1tYPKB7\nh2ueHj1jwcqVC78Z81zPDpe+sDCkHHT5Za0Sx9MbNKgVwuxhrwz/YcW6X96d2fqyyw9KCUvf\nuKL9H/qN+HreilVLZn32xq2nnnb/lzlhpzOuOLdB6YxISRUSbQnHJJ9k76Jip6aw8AoqveGd\ndHeHtpfc9uR7Y6fMWrBy3bpVi78d+8Ldz3weQsjN3RjBqL8de3KPPi9+OOnb+SvWx0OIZy/4\n+sOnrjnt5g/jIaN168N/aaxuh8zDQpjS97ZX54fDMzPrJUqPymxfM3zyj9vfXxmqtctsFdtc\nfduGPblHO2lJjzkAAAAA/Eq2/kUeAAAAAPjNWj/gjBDCXtd/XmjJqtE3HV4130ZbxhEXdTkw\nhNC+3+JEpXXjbzkoLV+l2G5depxeN4T089/Z3Pi8R1qHEFrcMytvFGueS1xz/OaCcXccVT3/\nRVte2q1ZCOH4R+Yl1bVCrpUwf9A1zfJfYKMznluTqDPnzcv3zyh4vO6fPixBIxvDOPicS1rm\nb2uno+6csGZzRKvfvrBunsMnPbY0Ho/H137+z7Z1Cl4gba9L35ibZGcLVXD2k52gwqItyZgU\nuEqyd1HxU1NIeAX7m8zwJjUmn1+/19b6HKt3Wv/vNp7w4Z9+iSklNfWXhM9Y/cx/T9uQp/FP\nb9gncWzfGz/bXLjyhTMTw1P51P7L8wZT/LBvpe9JPdrbc0vEAQAAAKD8eMMoAAAAAJRU1VZ/\nH/7xs389o0WTWlXSMmo3PqzTtU9/MuzcmvNCSK1TZ6dEpcotbxs+6vGrOjbfo361tIxaDQ9u\n1/3+IRMGdNt1Gy9a5Yi/DRs74Iazjmhau0rljFqNmp9y9ZNjht1waP50wm22yykPj5sy9F/X\nnt3mgIa1MiqnVau3+/6Ht+3a8+E3Pnu8a5VEnd3O6Df+8zfv6n7yYU3rVktLq7HLXs1P/P3t\nA4bcfmwJGtkovfktw0Y/fvUphzSqVaVyldpNjzj7hgFjh93c4pdaGac99O6/urc7uGHN9C1y\nCUNIa37d4M+HPnjlqS2a1q1WuVL6Tg32P67bDc+NHf9k520d3VJQWLQlGZMCkr+LipuaQsMr\noLSG99CbsoY/c+flpx5zcNP61dMzau26x8HHdbvhyY++eP3iJhurHHf3xx88c2ePM9o037fh\nTpXj8Vh6rYYHHdvlL4+NmPTuH/dNzdPcYR067BxCCDtnZjbfXFitfYfEe0WPzmxfI+/lt2nY\nk3q0k5b8mAMAAADAryEWj8fLOwYAAAAA2PH9/P7FB3Z8Zm7zPjMLebEim81/tM2uV49ucc+s\nCb2blncsVBhDu9e6rfnXo65K/qP3vw6PNgAAAABR4Q2jAAAAAFBiPzzevestTw/9fOZPK7NX\nLfxuUtZjV2Re8MzcUOmYC7pJKYNtkZJSAbarPdoAAAAARFil8g4AAAAAAHY8uT9PeeXOJ1+5\nM29p7WPv73d1k62fARQtFqsAn233aAMAAAAQYRXgT7YBAAAAYEfT5PKnXrnrspOP2L9x3aqV\n06rV2+Pwjpff9/5nQ65tVrncYprQu2msSJ2ezi634CoGQ1Shte23uPy/R18RH20AAAAAKCXe\nMAoAAAAAJRbb6cCzb3zi7BvLO44dVIOrRsWvKu8gYCs82gAAAABEWCwej5d3DAAAAAAAAAAA\nAACUIZ+kBwAAAAAAAAAAAIg4CaMAAAAAAAAAAAAAESdhFAAAAAAAAAAAACDiJIwCAAAAAAAA\nAAAARJyEUQAAAAAAAAAAAICIkzAKAAAAAAAAAAAAEHESRgEAAAAAAAAAAAAiTsIoAAAAAAAA\nAAAAQMRJGAUAAAAAAAAAAACIOAmjAAAAAAAAAAAAABEnYRQAAAAAAAAAAAAg4iSMAgAAAAAA\nAAAAAESchFEAAAAAAAAAAACAiJMwCgAAAAAAAAAAABBxEkYBAAAAAAAAAAAAIk7CKAAAAAAA\nAAAAAEDESRgFAAAAAAAAAAAAiDgJowAAAAAAAAAAAAARJ2EUAAAAAAAAAAAAIOIkjAIAAAAA\nAAAAAABEnIRRAAAAAAAAAAAAgIiTMAoAAAAAAAAAAAAQcRJGAQAAAAAAAAAAACLu/wE2zaja\nYrXHowAAAABJRU5ErkJggg==",
      "text/plain": [
       "Plot with title “Histogram of ligand_receptor_interactions$weight”"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 300,
       "width": 1800
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "hist(ligand_receptor_interactions$weight)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 478,
   "id": "36033e6e-1ca2-4b40-92cc-f4149450897b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Add expression data of ligands and receptors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 479,
   "id": "c077fa5b-ae17-4fd7-a3b6-abc50cb8c559",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'proteomics'</li><li>'single_cell'</li><li>'neutrophil'</li><li>'cytokine'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'proteomics'\n",
       "\\item 'single\\_cell'\n",
       "\\item 'neutrophil'\n",
       "\\item 'cytokine'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'proteomics'\n",
       "2. 'single_cell'\n",
       "3. 'neutrophil'\n",
       "4. 'cytokine'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"proteomics\"  \"single_cell\" \"neutrophil\"  \"cytokine\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(analyze_expr_cluster$type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 480,
   "id": "e361930a-1928-48a0-8edb-ca9bd0429f1e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>interaction_type</th><th scope=col>gene</th><th scope=col>type</th><th scope=col>cluster</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th><th scope=col>ident</th><th scope=col>mean_expr</th><th scope=col>var_expression</th><th scope=col>cell_type</th><th scope=col>cluster_vis</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Ligand</td><td>A2M   </td><td>proteomics </td><td>proteomics         </td><td>69.10381</td><td>  NA</td><td>proteomics_0_A2M_P01023                  </td><td>8.7445302</td><td>0.32468578</td><td>proteomics      </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>Ligand</td><td>ADAM17</td><td>single_cell</td><td>X4_Monocytes...CD14</td><td>49.68889</td><td>7267</td><td>single_cell_0_X4_Monocytes...CD14__ADAM17</td><td>0.8383966</td><td>0.03136488</td><td>Monocytes...CD14</td><td> 3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & interaction\\_type & gene & type & cluster & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene & ident & mean\\_expr & var\\_expression & cell\\_type & cluster\\_vis\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <dbl> & <int> & <chr> & <dbl> & <dbl> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & Ligand & A2M    & proteomics  & proteomics          & 69.10381 &   NA & proteomics\\_0\\_A2M\\_P01023                   & 8.7445302 & 0.32468578 & proteomics       & NA\\\\\n",
       "\t2 & Ligand & ADAM17 & single\\_cell & X4\\_Monocytes...CD14 & 49.68889 & 7267 & single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_ADAM17 & 0.8383966 & 0.03136488 & Monocytes...CD14 &  3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 11\n",
       "\n",
       "| <!--/--> | interaction_type &lt;chr&gt; | gene &lt;chr&gt; | type &lt;chr&gt; | cluster &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; | ident &lt;chr&gt; | mean_expr &lt;dbl&gt; | var_expression &lt;dbl&gt; | cell_type &lt;chr&gt; | cluster_vis &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | Ligand | A2M    | proteomics  | proteomics          | 69.10381 |   NA | proteomics_0_A2M_P01023                   | 8.7445302 | 0.32468578 | proteomics       | NA |\n",
       "| 2 | Ligand | ADAM17 | single_cell | X4_Monocytes...CD14 | 49.68889 | 7267 | single_cell_0_X4_Monocytes...CD14__ADAM17 | 0.8383966 | 0.03136488 | Monocytes...CD14 |  3 |\n",
       "\n"
      ],
      "text/plain": [
       "  interaction_type gene   type        cluster            \n",
       "1 Ligand           A2M    proteomics  proteomics         \n",
       "2 Ligand           ADAM17 single_cell X4_Monocytes...CD14\n",
       "  perc_cells_expressing_gene total_amount_cells_expressing_gene\n",
       "1 69.10381                     NA                              \n",
       "2 49.68889                   7267                              \n",
       "  ident                                     mean_expr var_expression\n",
       "1 proteomics_0_A2M_P01023                   8.7445302 0.32468578    \n",
       "2 single_cell_0_X4_Monocytes...CD14__ADAM17 0.8383966 0.03136488    \n",
       "  cell_type        cluster_vis\n",
       "1 proteomics       NA         \n",
       "2 Monocytes...CD14  3         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(analyze_expr_cluster,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 481,
   "id": "a443e8bf-7bf6-45fe-a294-b4339e14e55b",
   "metadata": {},
   "outputs": [],
   "source": [
    "analyze_expr_cluster$variable = paste0(analyze_expr_cluster$cluster, '_0_', analyze_expr_cluster$gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 482,
   "id": "1ba9b4ea-be0c-4e5e-9d3b-4c2818fdaa58",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "11639"
      ],
      "text/latex": [
       "11639"
      ],
      "text/markdown": [
       "11639"
      ],
      "text/plain": [
       "[1] 11639"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_receptor_interactions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 483,
   "id": "471e600a-eefb-402d-8c8b-c6592f6c4c99",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Add data for ligands\n",
    "data  = merge(ligand_receptor_interactions, analyze_expr_cluster[analyze_expr_cluster$interaction_type == 'Ligand',], by.x = c('Ligand'), by.y = c('gene'),  suffixes = c(\"\",\"_L\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 484,
   "id": "85ca594a-a361-4c85-bf34-5abd5daddd89",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Add data for Receptors\n",
    "data  = merge(data, analyze_expr_cluster[analyze_expr_cluster$interaction_type == 'Receptor',], by.x = c('Receptor'), by.y = c('gene'),  suffixes = c(\"\",\"_R\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 485,
   "id": "47c9edaf-8194-400b-a3ca-e0a9e8fb4a4a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 26</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Receptor</th><th scope=col>Ligand</th><th scope=col>amount_sources</th><th scope=col>weight</th><th scope=col>interaction_type</th><th scope=col>type</th><th scope=col>cluster</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th><th scope=col>ident</th><th scope=col>⋯</th><th scope=col>type_R</th><th scope=col>cluster_R</th><th scope=col>perc_cells_expressing_gene_R</th><th scope=col>total_amount_cells_expressing_gene_R</th><th scope=col>ident_R</th><th scope=col>mean_expr_R</th><th scope=col>var_expression_R</th><th scope=col>cell_type_R</th><th scope=col>cluster_vis_R</th><th scope=col>variable_R</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ABCA7  </td><td>APOA1 </td><td>1</td><td>0.4572367</td><td>Ligand</td><td>proteomics</td><td>proteomics</td><td>69.10381</td><td>NA</td><td>proteomics_0_APOA1_P02647 </td><td>⋯</td><td>neutrophil</td><td>neutrophil</td><td>69.10381</td><td>NA</td><td>neutrophil_0_ABCA7  </td><td>1.387586</td><td>0.8848451</td><td>neutrophil</td><td>NA</td><td>neutrophil_0_ABCA7  </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ADIPOR1</td><td>ADIPOQ</td><td>1</td><td>0.7307022</td><td>Ligand</td><td>proteomics</td><td>proteomics</td><td>69.10381</td><td>NA</td><td>proteomics_0_ADIPOQ_Q15848</td><td>⋯</td><td>neutrophil</td><td>neutrophil</td><td>69.10381</td><td>NA</td><td>neutrophil_0_ADIPOR1</td><td>2.269589</td><td>0.8178776</td><td>neutrophil</td><td>NA</td><td>neutrophil_0_ADIPOR1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 26\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & Receptor & Ligand & amount\\_sources & weight & interaction\\_type & type & cluster & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene & ident & ⋯ & type\\_R & cluster\\_R & perc\\_cells\\_expressing\\_gene\\_R & total\\_amount\\_cells\\_expressing\\_gene\\_R & ident\\_R & mean\\_expr\\_R & var\\_expression\\_R & cell\\_type\\_R & cluster\\_vis\\_R & variable\\_R\\\\\n",
       "  & <chr> & <chr> & <int> & <dbl> & <chr> & <chr> & <chr> & <dbl> & <int> & <chr> & ⋯ & <chr> & <chr> & <dbl> & <int> & <chr> & <dbl> & <dbl> & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & ABCA7   & APOA1  & 1 & 0.4572367 & Ligand & proteomics & proteomics & 69.10381 & NA & proteomics\\_0\\_APOA1\\_P02647  & ⋯ & neutrophil & neutrophil & 69.10381 & NA & neutrophil\\_0\\_ABCA7   & 1.387586 & 0.8848451 & neutrophil & NA & neutrophil\\_0\\_ABCA7  \\\\\n",
       "\t2 & ADIPOR1 & ADIPOQ & 1 & 0.7307022 & Ligand & proteomics & proteomics & 69.10381 & NA & proteomics\\_0\\_ADIPOQ\\_Q15848 & ⋯ & neutrophil & neutrophil & 69.10381 & NA & neutrophil\\_0\\_ADIPOR1 & 2.269589 & 0.8178776 & neutrophil & NA & neutrophil\\_0\\_ADIPOR1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 26\n",
       "\n",
       "| <!--/--> | Receptor &lt;chr&gt; | Ligand &lt;chr&gt; | amount_sources &lt;int&gt; | weight &lt;dbl&gt; | interaction_type &lt;chr&gt; | type &lt;chr&gt; | cluster &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; | ident &lt;chr&gt; | ⋯ ⋯ | type_R &lt;chr&gt; | cluster_R &lt;chr&gt; | perc_cells_expressing_gene_R &lt;dbl&gt; | total_amount_cells_expressing_gene_R &lt;int&gt; | ident_R &lt;chr&gt; | mean_expr_R &lt;dbl&gt; | var_expression_R &lt;dbl&gt; | cell_type_R &lt;chr&gt; | cluster_vis_R &lt;int&gt; | variable_R &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | ABCA7   | APOA1  | 1 | 0.4572367 | Ligand | proteomics | proteomics | 69.10381 | NA | proteomics_0_APOA1_P02647  | ⋯ | neutrophil | neutrophil | 69.10381 | NA | neutrophil_0_ABCA7   | 1.387586 | 0.8848451 | neutrophil | NA | neutrophil_0_ABCA7   |\n",
       "| 2 | ADIPOR1 | ADIPOQ | 1 | 0.7307022 | Ligand | proteomics | proteomics | 69.10381 | NA | proteomics_0_ADIPOQ_Q15848 | ⋯ | neutrophil | neutrophil | 69.10381 | NA | neutrophil_0_ADIPOR1 | 2.269589 | 0.8178776 | neutrophil | NA | neutrophil_0_ADIPOR1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Receptor Ligand amount_sources weight    interaction_type type      \n",
       "1 ABCA7    APOA1  1              0.4572367 Ligand           proteomics\n",
       "2 ADIPOR1  ADIPOQ 1              0.7307022 Ligand           proteomics\n",
       "  cluster    perc_cells_expressing_gene total_amount_cells_expressing_gene\n",
       "1 proteomics 69.10381                   NA                                \n",
       "2 proteomics 69.10381                   NA                                \n",
       "  ident                      ⋯ type_R     cluster_R \n",
       "1 proteomics_0_APOA1_P02647  ⋯ neutrophil neutrophil\n",
       "2 proteomics_0_ADIPOQ_Q15848 ⋯ neutrophil neutrophil\n",
       "  perc_cells_expressing_gene_R total_amount_cells_expressing_gene_R\n",
       "1 69.10381                     NA                                  \n",
       "2 69.10381                     NA                                  \n",
       "  ident_R              mean_expr_R var_expression_R cell_type_R cluster_vis_R\n",
       "1 neutrophil_0_ABCA7   1.387586    0.8848451        neutrophil  NA           \n",
       "2 neutrophil_0_ADIPOR1 2.269589    0.8178776        neutrophil  NA           \n",
       "  variable_R          \n",
       "1 neutrophil_0_ABCA7  \n",
       "2 neutrophil_0_ADIPOR1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 486,
   "id": "515c5f77-f33d-448b-92cc-49f5192cfe72",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Receptor'</li><li>'Ligand'</li><li>'amount_sources'</li><li>'weight'</li><li>'interaction_type'</li><li>'type'</li><li>'cluster'</li><li>'perc_cells_expressing_gene'</li><li>'total_amount_cells_expressing_gene'</li><li>'ident'</li><li>'mean_expr'</li><li>'var_expression'</li><li>'cell_type'</li><li>'cluster_vis'</li><li>'variable'</li><li>'interaction_type_R'</li><li>'type_R'</li><li>'cluster_R'</li><li>'perc_cells_expressing_gene_R'</li><li>'total_amount_cells_expressing_gene_R'</li><li>'ident_R'</li><li>'mean_expr_R'</li><li>'var_expression_R'</li><li>'cell_type_R'</li><li>'cluster_vis_R'</li><li>'variable_R'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Receptor'\n",
       "\\item 'Ligand'\n",
       "\\item 'amount\\_sources'\n",
       "\\item 'weight'\n",
       "\\item 'interaction\\_type'\n",
       "\\item 'type'\n",
       "\\item 'cluster'\n",
       "\\item 'perc\\_cells\\_expressing\\_gene'\n",
       "\\item 'total\\_amount\\_cells\\_expressing\\_gene'\n",
       "\\item 'ident'\n",
       "\\item 'mean\\_expr'\n",
       "\\item 'var\\_expression'\n",
       "\\item 'cell\\_type'\n",
       "\\item 'cluster\\_vis'\n",
       "\\item 'variable'\n",
       "\\item 'interaction\\_type\\_R'\n",
       "\\item 'type\\_R'\n",
       "\\item 'cluster\\_R'\n",
       "\\item 'perc\\_cells\\_expressing\\_gene\\_R'\n",
       "\\item 'total\\_amount\\_cells\\_expressing\\_gene\\_R'\n",
       "\\item 'ident\\_R'\n",
       "\\item 'mean\\_expr\\_R'\n",
       "\\item 'var\\_expression\\_R'\n",
       "\\item 'cell\\_type\\_R'\n",
       "\\item 'cluster\\_vis\\_R'\n",
       "\\item 'variable\\_R'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Receptor'\n",
       "2. 'Ligand'\n",
       "3. 'amount_sources'\n",
       "4. 'weight'\n",
       "5. 'interaction_type'\n",
       "6. 'type'\n",
       "7. 'cluster'\n",
       "8. 'perc_cells_expressing_gene'\n",
       "9. 'total_amount_cells_expressing_gene'\n",
       "10. 'ident'\n",
       "11. 'mean_expr'\n",
       "12. 'var_expression'\n",
       "13. 'cell_type'\n",
       "14. 'cluster_vis'\n",
       "15. 'variable'\n",
       "16. 'interaction_type_R'\n",
       "17. 'type_R'\n",
       "18. 'cluster_R'\n",
       "19. 'perc_cells_expressing_gene_R'\n",
       "20. 'total_amount_cells_expressing_gene_R'\n",
       "21. 'ident_R'\n",
       "22. 'mean_expr_R'\n",
       "23. 'var_expression_R'\n",
       "24. 'cell_type_R'\n",
       "25. 'cluster_vis_R'\n",
       "26. 'variable_R'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Receptor\"                            \n",
       " [2] \"Ligand\"                              \n",
       " [3] \"amount_sources\"                      \n",
       " [4] \"weight\"                              \n",
       " [5] \"interaction_type\"                    \n",
       " [6] \"type\"                                \n",
       " [7] \"cluster\"                             \n",
       " [8] \"perc_cells_expressing_gene\"          \n",
       " [9] \"total_amount_cells_expressing_gene\"  \n",
       "[10] \"ident\"                               \n",
       "[11] \"mean_expr\"                           \n",
       "[12] \"var_expression\"                      \n",
       "[13] \"cell_type\"                           \n",
       "[14] \"cluster_vis\"                         \n",
       "[15] \"variable\"                            \n",
       "[16] \"interaction_type_R\"                  \n",
       "[17] \"type_R\"                              \n",
       "[18] \"cluster_R\"                           \n",
       "[19] \"perc_cells_expressing_gene_R\"        \n",
       "[20] \"total_amount_cells_expressing_gene_R\"\n",
       "[21] \"ident_R\"                             \n",
       "[22] \"mean_expr_R\"                         \n",
       "[23] \"var_expression_R\"                    \n",
       "[24] \"cell_type_R\"                         \n",
       "[25] \"cluster_vis_R\"                       \n",
       "[26] \"variable_R\"                          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 487,
   "id": "cf52413f-7332-4e0b-89f1-8e7ca86bd238",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "5835"
      ],
      "text/latex": [
       "5835"
      ],
      "text/markdown": [
       "5835"
      ],
      "text/plain": [
       "[1] 5835"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 488,
   "id": "29b387e7-62ba-4524-a95d-cd6abeed4572",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'proteomics'</li><li>'single_cell'</li><li>'neutrophil'</li><li>'cytokine'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'proteomics'\n",
       "\\item 'single\\_cell'\n",
       "\\item 'neutrophil'\n",
       "\\item 'cytokine'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'proteomics'\n",
       "2. 'single_cell'\n",
       "3. 'neutrophil'\n",
       "4. 'cytokine'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"proteomics\"  \"single_cell\" \"neutrophil\"  \"cytokine\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(data$type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 489,
   "id": "3044cb11-4e12-4354-8496-e28d8e6e8bb9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Remove same type links (ligands to receptors of same type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 490,
   "id": "fdcbeaeb-214f-4f7c-b707-e22d27c3981f",
   "metadata": {},
   "outputs": [],
   "source": [
    "data = data[data$cell_type != data$cell_type_R,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 491,
   "id": "f4fd0a82-e0ef-49d5-96a5-4f06e59f83fa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "4930"
      ],
      "text/latex": [
       "4930"
      ],
      "text/markdown": [
       "4930"
      ],
      "text/plain": [
       "[1] 4930"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 492,
   "id": "f17e0a8b-9e45-48e5-b72e-0302f9f3f313",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'proteomics'</li><li>'single_cell'</li><li>'neutrophil'</li><li>'cytokine'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'proteomics'\n",
       "\\item 'single\\_cell'\n",
       "\\item 'neutrophil'\n",
       "\\item 'cytokine'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'proteomics'\n",
       "2. 'single_cell'\n",
       "3. 'neutrophil'\n",
       "4. 'cytokine'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"proteomics\"  \"single_cell\" \"neutrophil\"  \"cytokine\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(data$type)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "344dde5b-e2fa-478c-9adc-f3ded4c0943a",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Combine expression data and interaction potential with correlations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 493,
   "id": "c5537ee5-4e56-4884-8b88-2233063a353c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ligand</th><th scope=col>receptor</th><th scope=col>correlation</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>cytokine_0_BCA1__CXCL13</td><td>neutrophil_0_ABCA7</td><td>0.09543501</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>cytokine_0_CTACK__CCL27</td><td>neutrophil_0_ABCA7</td><td>0.05916357</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & ligand & receptor & correlation\\\\\n",
       "  & <fct> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & cytokine\\_0\\_BCA1\\_\\_CXCL13 & neutrophil\\_0\\_ABCA7 & 0.09543501\\\\\n",
       "\t2 & cytokine\\_0\\_CTACK\\_\\_CCL27 & neutrophil\\_0\\_ABCA7 & 0.05916357\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 3\n",
       "\n",
       "| <!--/--> | ligand &lt;fct&gt; | receptor &lt;fct&gt; | correlation &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | cytokine_0_BCA1__CXCL13 | neutrophil_0_ABCA7 | 0.09543501 |\n",
       "| 2 | cytokine_0_CTACK__CCL27 | neutrophil_0_ABCA7 | 0.05916357 |\n",
       "\n"
      ],
      "text/plain": [
       "  ligand                  receptor           correlation\n",
       "1 cytokine_0_BCA1__CXCL13 neutrophil_0_ABCA7 0.09543501 \n",
       "2 cytokine_0_CTACK__CCL27 neutrophil_0_ABCA7 0.05916357 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_receptor_correlations,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 494,
   "id": "f12703f0-81c2-4639-b183-7cdd32171749",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 26</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Receptor</th><th scope=col>Ligand</th><th scope=col>amount_sources</th><th scope=col>weight</th><th scope=col>interaction_type</th><th scope=col>type</th><th scope=col>cluster</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th><th scope=col>ident</th><th scope=col>⋯</th><th scope=col>type_R</th><th scope=col>cluster_R</th><th scope=col>perc_cells_expressing_gene_R</th><th scope=col>total_amount_cells_expressing_gene_R</th><th scope=col>ident_R</th><th scope=col>mean_expr_R</th><th scope=col>var_expression_R</th><th scope=col>cell_type_R</th><th scope=col>cluster_vis_R</th><th scope=col>variable_R</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ABCA7  </td><td>APOA1 </td><td>1</td><td>0.4572367</td><td>Ligand</td><td>proteomics</td><td>proteomics</td><td>69.10381</td><td>NA</td><td>proteomics_0_APOA1_P02647 </td><td>⋯</td><td>neutrophil</td><td>neutrophil</td><td>69.10381</td><td>NA</td><td>neutrophil_0_ABCA7  </td><td>1.387586</td><td>0.8848451</td><td>neutrophil</td><td>NA</td><td>neutrophil_0_ABCA7  </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ADIPOR1</td><td>ADIPOQ</td><td>1</td><td>0.7307022</td><td>Ligand</td><td>proteomics</td><td>proteomics</td><td>69.10381</td><td>NA</td><td>proteomics_0_ADIPOQ_Q15848</td><td>⋯</td><td>neutrophil</td><td>neutrophil</td><td>69.10381</td><td>NA</td><td>neutrophil_0_ADIPOR1</td><td>2.269589</td><td>0.8178776</td><td>neutrophil</td><td>NA</td><td>neutrophil_0_ADIPOR1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 26\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & Receptor & Ligand & amount\\_sources & weight & interaction\\_type & type & cluster & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene & ident & ⋯ & type\\_R & cluster\\_R & perc\\_cells\\_expressing\\_gene\\_R & total\\_amount\\_cells\\_expressing\\_gene\\_R & ident\\_R & mean\\_expr\\_R & var\\_expression\\_R & cell\\_type\\_R & cluster\\_vis\\_R & variable\\_R\\\\\n",
       "  & <chr> & <chr> & <int> & <dbl> & <chr> & <chr> & <chr> & <dbl> & <int> & <chr> & ⋯ & <chr> & <chr> & <dbl> & <int> & <chr> & <dbl> & <dbl> & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & ABCA7   & APOA1  & 1 & 0.4572367 & Ligand & proteomics & proteomics & 69.10381 & NA & proteomics\\_0\\_APOA1\\_P02647  & ⋯ & neutrophil & neutrophil & 69.10381 & NA & neutrophil\\_0\\_ABCA7   & 1.387586 & 0.8848451 & neutrophil & NA & neutrophil\\_0\\_ABCA7  \\\\\n",
       "\t2 & ADIPOR1 & ADIPOQ & 1 & 0.7307022 & Ligand & proteomics & proteomics & 69.10381 & NA & proteomics\\_0\\_ADIPOQ\\_Q15848 & ⋯ & neutrophil & neutrophil & 69.10381 & NA & neutrophil\\_0\\_ADIPOR1 & 2.269589 & 0.8178776 & neutrophil & NA & neutrophil\\_0\\_ADIPOR1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 26\n",
       "\n",
       "| <!--/--> | Receptor &lt;chr&gt; | Ligand &lt;chr&gt; | amount_sources &lt;int&gt; | weight &lt;dbl&gt; | interaction_type &lt;chr&gt; | type &lt;chr&gt; | cluster &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; | ident &lt;chr&gt; | ⋯ ⋯ | type_R &lt;chr&gt; | cluster_R &lt;chr&gt; | perc_cells_expressing_gene_R &lt;dbl&gt; | total_amount_cells_expressing_gene_R &lt;int&gt; | ident_R &lt;chr&gt; | mean_expr_R &lt;dbl&gt; | var_expression_R &lt;dbl&gt; | cell_type_R &lt;chr&gt; | cluster_vis_R &lt;int&gt; | variable_R &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | ABCA7   | APOA1  | 1 | 0.4572367 | Ligand | proteomics | proteomics | 69.10381 | NA | proteomics_0_APOA1_P02647  | ⋯ | neutrophil | neutrophil | 69.10381 | NA | neutrophil_0_ABCA7   | 1.387586 | 0.8848451 | neutrophil | NA | neutrophil_0_ABCA7   |\n",
       "| 2 | ADIPOR1 | ADIPOQ | 1 | 0.7307022 | Ligand | proteomics | proteomics | 69.10381 | NA | proteomics_0_ADIPOQ_Q15848 | ⋯ | neutrophil | neutrophil | 69.10381 | NA | neutrophil_0_ADIPOR1 | 2.269589 | 0.8178776 | neutrophil | NA | neutrophil_0_ADIPOR1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Receptor Ligand amount_sources weight    interaction_type type      \n",
       "1 ABCA7    APOA1  1              0.4572367 Ligand           proteomics\n",
       "2 ADIPOR1  ADIPOQ 1              0.7307022 Ligand           proteomics\n",
       "  cluster    perc_cells_expressing_gene total_amount_cells_expressing_gene\n",
       "1 proteomics 69.10381                   NA                                \n",
       "2 proteomics 69.10381                   NA                                \n",
       "  ident                      ⋯ type_R     cluster_R \n",
       "1 proteomics_0_APOA1_P02647  ⋯ neutrophil neutrophil\n",
       "2 proteomics_0_ADIPOQ_Q15848 ⋯ neutrophil neutrophil\n",
       "  perc_cells_expressing_gene_R total_amount_cells_expressing_gene_R\n",
       "1 69.10381                     NA                                  \n",
       "2 69.10381                     NA                                  \n",
       "  ident_R              mean_expr_R var_expression_R cell_type_R cluster_vis_R\n",
       "1 neutrophil_0_ABCA7   1.387586    0.8848451        neutrophil  NA           \n",
       "2 neutrophil_0_ADIPOR1 2.269589    0.8178776        neutrophil  NA           \n",
       "  variable_R          \n",
       "1 neutrophil_0_ABCA7  \n",
       "2 neutrophil_0_ADIPOR1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 495,
   "id": "6635545d-bc10-44df-94d6-5c224fd231c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "4930"
      ],
      "text/latex": [
       "4930"
      ],
      "text/markdown": [
       "4930"
      ],
      "text/plain": [
       "[1] 4930"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 496,
   "id": "69497364-7065-492d-ae1e-9d5f47386ce3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "163315"
      ],
      "text/latex": [
       "163315"
      ],
      "text/markdown": [
       "163315"
      ],
      "text/plain": [
       "[1] 163315"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_receptor_correlations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 497,
   "id": "d3ee5c37-08e4-4fad-9066-d4463b444a10",
   "metadata": {},
   "outputs": [],
   "source": [
    "data = merge(data, ligand_receptor_correlations, by.x = c('ident', 'ident_R'), by.y = c('ligand', 'receptor'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 498,
   "id": "d280e13f-bfed-4f8d-8575-3751c666d9c9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "4930"
      ],
      "text/latex": [
       "4930"
      ],
      "text/markdown": [
       "4930"
      ],
      "text/plain": [
       "[1] 4930"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 499,
   "id": "1e14e57c-1ab0-47c6-8dd9-1cc06746a4ce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'all'"
      ],
      "text/latex": [
       "'all'"
      ],
      "text/markdown": [
       "'all'"
      ],
      "text/plain": [
       "[1] \"all\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "filter_samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 500,
   "id": "78e10b39-0a5b-4ba6-a68c-2444567db07b",
   "metadata": {},
   "outputs": [],
   "source": [
    "## save the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 501,
   "id": "8c4a960e-142b-4881-997f-392f7056f953",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(data,gzfile(paste0(result_path, '/F-Analysis/' ,'F4_Ligand_Receptor_Data', name, filter_samples,'.gz')))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "05d5e645-60c4-42da-a581-9578200fcd0e",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Define expressed ligands and potential ligand - target combinations based on expressed receptors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 502,
   "id": "14202f35-ea65-4cd2-8c31-f7a2594f4bff",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Define minimum amount of expression of receptor in receiving cell to be able to react to ligand"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 503,
   "id": "61e16fac-4041-46d6-9942-b1946de77de0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Ligand</th><th scope=col>Receptor</th><th scope=col>amount_sources</th><th scope=col>weight</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A2M</td><td>LRP1    </td><td>1</td><td>0.4678977</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A2M</td><td>TNFRSF14</td><td>1</td><td>0.6990671</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & Ligand & Receptor & amount\\_sources & weight\\\\\n",
       "  & <chr> & <chr> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & A2M & LRP1     & 1 & 0.4678977\\\\\n",
       "\t2 & A2M & TNFRSF14 & 1 & 0.6990671\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 4\n",
       "\n",
       "| <!--/--> | Ligand &lt;chr&gt; | Receptor &lt;chr&gt; | amount_sources &lt;int&gt; | weight &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | A2M | LRP1     | 1 | 0.4678977 |\n",
       "| 2 | A2M | TNFRSF14 | 1 | 0.6990671 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand Receptor amount_sources weight   \n",
       "1 A2M    LRP1     1              0.4678977\n",
       "2 A2M    TNFRSF14 1              0.6990671"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_receptor_interactions,2) # all potential ligand receptor interactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 504,
   "id": "0f8f3c06-73da-492b-aa33-c83e36465e45",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th><th scope=col>gene</th><th scope=col>cluster</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AL627309.1</td><td>0.24327612</td><td>18</td><td>AL627309.1</td><td>X8_B.cell</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AL627309.4</td><td>0.05406136</td><td> 4</td><td>AL627309.4</td><td>X8_B.cell</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & X & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene & gene & cluster\\\\\n",
       "  & <chr> & <dbl> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & AL627309.1 & 0.24327612 & 18 & AL627309.1 & X8\\_B.cell\\\\\n",
       "\t2 & AL627309.4 & 0.05406136 &  4 & AL627309.4 & X8\\_B.cell\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 5\n",
       "\n",
       "| <!--/--> | X &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; | gene &lt;chr&gt; | cluster &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | AL627309.1 | 0.24327612 | 18 | AL627309.1 | X8_B.cell |\n",
       "| 2 | AL627309.4 | 0.05406136 |  4 | AL627309.4 | X8_B.cell |\n",
       "\n"
      ],
      "text/plain": [
       "  X          perc_cells_expressing_gene total_amount_cells_expressing_gene\n",
       "1 AL627309.1 0.24327612                 18                                \n",
       "2 AL627309.4 0.05406136                  4                                \n",
       "  gene       cluster  \n",
       "1 AL627309.1 X8_B.cell\n",
       "2 AL627309.4 X8_B.cell"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cell_perc_cluster,2)   # expression of the genes in cells"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 505,
   "id": "d60b6d7f-93e4-4337-88b1-0305c5fb5cce",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Set threshold to filter out receptors that are to low expressed\n",
    "cell_perc_cluster_filtered =  cell_perc_cluster[((cell_perc_cluster$perc_cells > 30) & (cell_perc_cluster$total_amount_cells_expressing_gene > 600)) | ((cell_perc_cluster$perc_cells > 10) & (cell_perc_cluster$total_amount_cells_expressing_gene > 1200)),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 506,
   "id": "bd54a0d2-ac92-49bf-900b-4f38e529d313",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Define all potential ligand receptor interactions based on expression\n",
    "potential_ligand_target_types_comb = merge(ligand_receptor_interactions, cell_perc_cluster_filtered, by.x = 'Receptor', by.y = 'gene')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 507,
   "id": "8892c6ea-aa4f-4514-9237-6e8648a94242",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Receptor</th><th scope=col>Ligand</th><th scope=col>amount_sources</th><th scope=col>weight</th><th scope=col>X</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th><th scope=col>cluster</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ABCA1</td><td>APOA1</td><td>1</td><td>0.8194814</td><td>ABCA19</td><td>15.74701</td><td>2303</td><td>X4_Monocytes...CD14</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ABCA1</td><td>APOA1</td><td>1</td><td>0.8194814</td><td>ABCA17</td><td>16.46135</td><td>1844</td><td>X6_Monocytes...CD14</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Receptor & Ligand & amount\\_sources & weight & X & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene & cluster\\\\\n",
       "  & <chr> & <chr> & <int> & <dbl> & <chr> & <dbl> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & ABCA1 & APOA1 & 1 & 0.8194814 & ABCA19 & 15.74701 & 2303 & X4\\_Monocytes...CD14\\\\\n",
       "\t2 & ABCA1 & APOA1 & 1 & 0.8194814 & ABCA17 & 16.46135 & 1844 & X6\\_Monocytes...CD14\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 8\n",
       "\n",
       "| <!--/--> | Receptor &lt;chr&gt; | Ligand &lt;chr&gt; | amount_sources &lt;int&gt; | weight &lt;dbl&gt; | X &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; | cluster &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | ABCA1 | APOA1 | 1 | 0.8194814 | ABCA19 | 15.74701 | 2303 | X4_Monocytes...CD14 |\n",
       "| 2 | ABCA1 | APOA1 | 1 | 0.8194814 | ABCA17 | 16.46135 | 1844 | X6_Monocytes...CD14 |\n",
       "\n"
      ],
      "text/plain": [
       "  Receptor Ligand amount_sources weight    X      perc_cells_expressing_gene\n",
       "1 ABCA1    APOA1  1              0.8194814 ABCA19 15.74701                  \n",
       "2 ABCA1    APOA1  1              0.8194814 ABCA17 16.46135                  \n",
       "  total_amount_cells_expressing_gene cluster            \n",
       "1 2303                               X4_Monocytes...CD14\n",
       "2 1844                               X6_Monocytes...CD14"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(potential_ligand_target_types_comb,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 508,
   "id": "742e810d-442f-4f78-a259-90b8b73e5f42",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Define potential ligand interactions with target cell-types (minimum one receptor that can interact with the ligand on the cell-type)\n",
    "potential_ligand_target_types_comb = unique(potential_ligand_target_types_comb[,c('Ligand', 'cluster')] )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 509,
   "id": "524ede29-4dc4-48f9-aa42-2fe7c1991182",
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames(potential_ligand_target_types_comb) = c('Ligand', 'cluster_R')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 510,
   "id": "f8338a36-14f7-4430-b9eb-db2e060ffa83",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Ligand</th><th scope=col>cluster_R</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>APOA1</td><td>X4_Monocytes...CD14</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>APOA1</td><td>X6_Monocytes...CD14</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Ligand & cluster\\_R\\\\\n",
       "  & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & APOA1 & X4\\_Monocytes...CD14\\\\\n",
       "\t2 & APOA1 & X6\\_Monocytes...CD14\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 2\n",
       "\n",
       "| <!--/--> | Ligand &lt;chr&gt; | cluster_R &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| 1 | APOA1 | X4_Monocytes...CD14 |\n",
       "| 2 | APOA1 | X6_Monocytes...CD14 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand cluster_R          \n",
       "1 APOA1  X4_Monocytes...CD14\n",
       "2 APOA1  X6_Monocytes...CD14"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(potential_ligand_target_types_comb,2) # ligands and the cluster-cell types they can affect"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 511,
   "id": "4c061112-0e85-451d-9a8d-75d56732928d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'X4_Monocytes...CD14'</li><li>'X6_Monocytes...CD14'</li><li>'X2_T.cell.CD4'</li><li>'X1_T.cell.CD8'</li><li>'X7_Monocytes...CD14'</li><li>'X0_T.cell.CD4'</li><li>'X12_Monocytes...CD16_FCGR3A'</li><li>'X3_NK'</li><li>'X9_Monocytes...CD16_FCGR3A'</li><li>'X8_B.cell'</li><li>'X11_T.cell.CD4'</li><li>'X10_B.cell'</li><li>'X5_T.cell.CD4'</li><li>'X13_Dendritic'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'X4\\_Monocytes...CD14'\n",
       "\\item 'X6\\_Monocytes...CD14'\n",
       "\\item 'X2\\_T.cell.CD4'\n",
       "\\item 'X1\\_T.cell.CD8'\n",
       "\\item 'X7\\_Monocytes...CD14'\n",
       "\\item 'X0\\_T.cell.CD4'\n",
       "\\item 'X12\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'X3\\_NK'\n",
       "\\item 'X9\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'X8\\_B.cell'\n",
       "\\item 'X11\\_T.cell.CD4'\n",
       "\\item 'X10\\_B.cell'\n",
       "\\item 'X5\\_T.cell.CD4'\n",
       "\\item 'X13\\_Dendritic'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'X4_Monocytes...CD14'\n",
       "2. 'X6_Monocytes...CD14'\n",
       "3. 'X2_T.cell.CD4'\n",
       "4. 'X1_T.cell.CD8'\n",
       "5. 'X7_Monocytes...CD14'\n",
       "6. 'X0_T.cell.CD4'\n",
       "7. 'X12_Monocytes...CD16_FCGR3A'\n",
       "8. 'X3_NK'\n",
       "9. 'X9_Monocytes...CD16_FCGR3A'\n",
       "10. 'X8_B.cell'\n",
       "11. 'X11_T.cell.CD4'\n",
       "12. 'X10_B.cell'\n",
       "13. 'X5_T.cell.CD4'\n",
       "14. 'X13_Dendritic'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"X4_Monocytes...CD14\"         \"X6_Monocytes...CD14\"        \n",
       " [3] \"X2_T.cell.CD4\"               \"X1_T.cell.CD8\"              \n",
       " [5] \"X7_Monocytes...CD14\"         \"X0_T.cell.CD4\"              \n",
       " [7] \"X12_Monocytes...CD16_FCGR3A\" \"X3_NK\"                      \n",
       " [9] \"X9_Monocytes...CD16_FCGR3A\"  \"X8_B.cell\"                  \n",
       "[11] \"X11_T.cell.CD4\"              \"X10_B.cell\"                 \n",
       "[13] \"X5_T.cell.CD4\"               \"X13_Dendritic\"              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(potential_ligand_target_types_comb$cluster_R)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 512,
   "id": "c89f95e5-12fc-404a-96a1-3f0b9121aa5c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 19 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Ligand</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ARTN   </td><td>1</td></tr>\n",
       "\t<tr><td>CNTN2  </td><td>1</td></tr>\n",
       "\t<tr><td>DEFB4A </td><td>1</td></tr>\n",
       "\t<tr><td>DEFB4B </td><td>1</td></tr>\n",
       "\t<tr><td>DHH    </td><td>1</td></tr>\n",
       "\t<tr><td>FLT3LG </td><td>1</td></tr>\n",
       "\t<tr><td>GPHA2  </td><td>1</td></tr>\n",
       "\t<tr><td>IHH    </td><td>1</td></tr>\n",
       "\t<tr><td>IL18   </td><td>1</td></tr>\n",
       "\t<tr><td>IL23A  </td><td>1</td></tr>\n",
       "\t<tr><td>LIPH   </td><td>1</td></tr>\n",
       "\t<tr><td>NRG1   </td><td>1</td></tr>\n",
       "\t<tr><td>NRTN   </td><td>1</td></tr>\n",
       "\t<tr><td>PSPN   </td><td>1</td></tr>\n",
       "\t<tr><td>SEMA3A </td><td>1</td></tr>\n",
       "\t<tr><td>SEMA3F </td><td>1</td></tr>\n",
       "\t<tr><td>SEMA6D </td><td>1</td></tr>\n",
       "\t<tr><td>TNFSF18</td><td>1</td></tr>\n",
       "\t<tr><td>TNFSF4 </td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 19 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Ligand & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t ARTN    & 1\\\\\n",
       "\t CNTN2   & 1\\\\\n",
       "\t DEFB4A  & 1\\\\\n",
       "\t DEFB4B  & 1\\\\\n",
       "\t DHH     & 1\\\\\n",
       "\t FLT3LG  & 1\\\\\n",
       "\t GPHA2   & 1\\\\\n",
       "\t IHH     & 1\\\\\n",
       "\t IL18    & 1\\\\\n",
       "\t IL23A   & 1\\\\\n",
       "\t LIPH    & 1\\\\\n",
       "\t NRG1    & 1\\\\\n",
       "\t NRTN    & 1\\\\\n",
       "\t PSPN    & 1\\\\\n",
       "\t SEMA3A  & 1\\\\\n",
       "\t SEMA3F  & 1\\\\\n",
       "\t SEMA6D  & 1\\\\\n",
       "\t TNFSF18 & 1\\\\\n",
       "\t TNFSF4  & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 19 × 2\n",
       "\n",
       "| Ligand &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| ARTN    | 1 |\n",
       "| CNTN2   | 1 |\n",
       "| DEFB4A  | 1 |\n",
       "| DEFB4B  | 1 |\n",
       "| DHH     | 1 |\n",
       "| FLT3LG  | 1 |\n",
       "| GPHA2   | 1 |\n",
       "| IHH     | 1 |\n",
       "| IL18    | 1 |\n",
       "| IL23A   | 1 |\n",
       "| LIPH    | 1 |\n",
       "| NRG1    | 1 |\n",
       "| NRTN    | 1 |\n",
       "| PSPN    | 1 |\n",
       "| SEMA3A  | 1 |\n",
       "| SEMA3F  | 1 |\n",
       "| SEMA6D  | 1 |\n",
       "| TNFSF18 | 1 |\n",
       "| TNFSF4  | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "   Ligand  n\n",
       "1  ARTN    1\n",
       "2  CNTN2   1\n",
       "3  DEFB4A  1\n",
       "4  DEFB4B  1\n",
       "5  DHH     1\n",
       "6  FLT3LG  1\n",
       "7  GPHA2   1\n",
       "8  IHH     1\n",
       "9  IL18    1\n",
       "10 IL23A   1\n",
       "11 LIPH    1\n",
       "12 NRG1    1\n",
       "13 NRTN    1\n",
       "14 PSPN    1\n",
       "15 SEMA3A  1\n",
       "16 SEMA3F  1\n",
       "17 SEMA6D  1\n",
       "18 TNFSF18 1\n",
       "19 TNFSF4  1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Below: ligands where receptors are only expressed in very low amount of targets  (removed ligands)\n",
    "unique(potential_ligand_target_types_comb[,c('Ligand', 'cluster_R')]) %>% group_by(Ligand) %>% count() %>% filter(n == 1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 513,
   "id": "48c2e0f9-0c42-4c9d-9d72-5454f1a5a42b",
   "metadata": {},
   "outputs": [],
   "source": [
    "## get all expressed ligands per cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 514,
   "id": "49003b10-bc5a-45a8-9c06-0909157616ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "expressed_ligand_types = unique(data[,c('Ligand', 'cluster', 'mean_expr', 'type', 'variable', 'ident')])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8de04252-820d-4d43-893f-aa6f8f5ddda2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "dc74a9fe-4037-4e1b-b030-80964257adfb",
   "metadata": {},
   "source": [
    "# Ligand Target Data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "408f8793-3437-4325-b517-d589dc3cf3de",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Calculate ligand - target correlations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 515,
   "id": "f0bf8c31-a78e-4ccd-b2a8-d185d659016b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Ligand</th><th scope=col>cluster</th><th scope=col>mean_expr</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>ident</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>CXCL13</td><td>cytokine</td><td>5.429313</td><td>cytokine</td><td>cytokine_0_CXCL13</td><td>cytokine_0_BCA1__CXCL13</td></tr>\n",
       "\t<tr><th scope=row>30</th><td>CCL27 </td><td>cytokine</td><td>9.723839</td><td>cytokine</td><td>cytokine_0_CCL27 </td><td>cytokine_0_CTACK__CCL27</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & Ligand & cluster & mean\\_expr & type & variable & ident\\\\\n",
       "  & <chr> & <chr> & <dbl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & CXCL13 & cytokine & 5.429313 & cytokine & cytokine\\_0\\_CXCL13 & cytokine\\_0\\_BCA1\\_\\_CXCL13\\\\\n",
       "\t30 & CCL27  & cytokine & 9.723839 & cytokine & cytokine\\_0\\_CCL27  & cytokine\\_0\\_CTACK\\_\\_CCL27\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 6\n",
       "\n",
       "| <!--/--> | Ligand &lt;chr&gt; | cluster &lt;chr&gt; | mean_expr &lt;dbl&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | ident &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | CXCL13 | cytokine | 5.429313 | cytokine | cytokine_0_CXCL13 | cytokine_0_BCA1__CXCL13 |\n",
       "| 30 | CCL27  | cytokine | 9.723839 | cytokine | cytokine_0_CCL27  | cytokine_0_CTACK__CCL27 |\n",
       "\n"
      ],
      "text/plain": [
       "   Ligand cluster  mean_expr type     variable          ident                  \n",
       "1  CXCL13 cytokine 5.429313  cytokine cytokine_0_CXCL13 cytokine_0_BCA1__CXCL13\n",
       "30 CCL27  cytokine 9.723839  cytokine cytokine_0_CCL27  cytokine_0_CTACK__CCL27"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### expressed ligands per cluster\n",
    "head(expressed_ligand_types,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 516,
   "id": "ae1073d9-f2e1-41f2-9cc3-68b353fb24d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Determine expressed ligands in single-cell data\n",
    "expressed_ligand_types$variable[expressed_ligand_types$type == 'single_cell'] = paste0(expressed_ligand_types$type[expressed_ligand_types$type == 'single_cell'], '_0_',expressed_ligand_types$cluster[expressed_ligand_types$type == 'single_cell'], '__', expressed_ligand_types$Ligand[expressed_ligand_types$type == 'single_cell'] )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 517,
   "id": "0ad0a10a-42bd-40c2-b6d0-e5d30c7cf3d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "expressed_ligands = unique(expressed_ligand_types$ident)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 518,
   "id": "6a71d50b-d417-4283-a751-5f8aa4fd2af4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "344"
      ],
      "text/latex": [
       "344"
      ],
      "text/markdown": [
       "344"
      ],
      "text/plain": [
       "[1] 344"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(expressed_ligands)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 519,
   "id": "938631c6-3a2f-4844-a2c3-05a59b0a6479",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Transform data_long to matrix format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 520,
   "id": "dbf9a48b-08c1-471e-9abb-c2d5c923608d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>type</th><th scope=col>variable</th><th scope=col>X</th><th scope=col>sample_id</th><th scope=col>value</th><th scope=col>config</th><th scope=col>expression</th><th scope=col>ident</th><th scope=col>gene</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>clinical_data</td><td>CK</td><td>1557210</td><td>k1 </td><td>5.459432</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>clinical_data_0_CK</td><td>CK</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>clinical_data</td><td>CK</td><td>1557211</td><td>k10</td><td>      NA</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>clinical_data_0_CK</td><td>CK</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & type & variable & X & sample\\_id & value & config & expression & ident & gene\\\\\n",
       "  & <chr> & <chr> & <int> & <chr> & <dbl> & <chr> & <lgl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & clinical\\_data & CK & 1557210 & k1  & 5.459432 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & clinical\\_data\\_0\\_CK & CK\\\\\n",
       "\t2 & clinical\\_data & CK & 1557211 & k10 &       NA & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & clinical\\_data\\_0\\_CK & CK\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 9\n",
       "\n",
       "| <!--/--> | type &lt;chr&gt; | variable &lt;chr&gt; | X &lt;int&gt; | sample_id &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; | expression &lt;lgl&gt; | ident &lt;chr&gt; | gene &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | clinical_data | CK | 1557210 | k1  | 5.459432 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | clinical_data_0_CK | CK |\n",
       "| 2 | clinical_data | CK | 1557211 | k10 |       NA | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | clinical_data_0_CK | CK |\n",
       "\n"
      ],
      "text/plain": [
       "  type          variable X       sample_id value   \n",
       "1 clinical_data CK       1557210 k1        5.459432\n",
       "2 clinical_data CK       1557211 k10             NA\n",
       "  config                           expression ident              gene\n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       clinical_data_0_CK CK  \n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       clinical_data_0_CK CK  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 521,
   "id": "e5b42efe-5ab0-43c8-a3a8-d1526dfefa15",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_data = data_long %>% dcast(sample_id ~ ident, value.var = 'value')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 522,
   "id": "099dab7a-bf4a-4dba-9f18-0068c7642efe",
   "metadata": {},
   "outputs": [],
   "source": [
    "rownames(expression_data) = expression_data$sample_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 523,
   "id": "62386c1c-2e61-4eef-ace9-ad42e732e1f9",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_data$sample_id = NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 524,
   "id": "6bcaf00a-2aa3-47b5-8250-6d51291e71bc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 13283</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>clinical_data_0_CK</th><th scope=col>clinical_data_0_CK_MB</th><th scope=col>clinical_data_0_CRP</th><th scope=col>clinical_data_0_delta_ef_value</th><th scope=col>clinical_data_0_Troponin</th><th scope=col>cytokine_0_BCA1__CXCL13</th><th scope=col>cytokine_0_CTACK__CCL27</th><th scope=col>cytokine_0_EGF__EGF</th><th scope=col>cytokine_0_ENA78__CXCL5</th><th scope=col>cytokine_0_Eotaxin__CCL11</th><th scope=col>⋯</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>k1</th><td>5.459432</td><td>NA</td><td>0.4854268</td><td>NA</td><td>0.01863417</td><td>5.696550</td><td>9.821662</td><td>5.045268</td><td>10.562366</td><td>7.42341</td><td>⋯</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>       NA</td><td>      NA</td><td>       NA</td></tr>\n",
       "\t<tr><th scope=row>k10</th><td>      NA</td><td>NA</td><td>0.2630344</td><td>NA</td><td>        NA</td><td>4.697663</td><td>9.701220</td><td>1.922198</td><td> 7.724173</td><td>9.19842</td><td>⋯</td><td>2.461054</td><td>2.157169</td><td>1.975326</td><td>3.139595</td><td>2.078417</td><td>1.843355</td><td>1.343136</td><td>0.6540153</td><td>1.190677</td><td>0.8572279</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 13283\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & clinical\\_data\\_0\\_CK & clinical\\_data\\_0\\_CK\\_MB & clinical\\_data\\_0\\_CRP & clinical\\_data\\_0\\_delta\\_ef\\_value & clinical\\_data\\_0\\_Troponin & cytokine\\_0\\_BCA1\\_\\_CXCL13 & cytokine\\_0\\_CTACK\\_\\_CCL27 & cytokine\\_0\\_EGF\\_\\_EGF & cytokine\\_0\\_ENA78\\_\\_CXCL5 & cytokine\\_0\\_Eotaxin\\_\\_CCL11 & ⋯ & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZEB2 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFAND5 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFAS1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36L1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36L2 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF106 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF207 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF706 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZYX\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tk1 & 5.459432 & NA & 0.4854268 & NA & 0.01863417 & 5.696550 & 9.821662 & 5.045268 & 10.562366 & 7.42341 & ⋯ &       NA &       NA &       NA &       NA &       NA &       NA &       NA &        NA &       NA &        NA\\\\\n",
       "\tk10 &       NA & NA & 0.2630344 & NA &         NA & 4.697663 & 9.701220 & 1.922198 &  7.724173 & 9.19842 & ⋯ & 2.461054 & 2.157169 & 1.975326 & 3.139595 & 2.078417 & 1.843355 & 1.343136 & 0.6540153 & 1.190677 & 0.8572279\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 13283\n",
       "\n",
       "| <!--/--> | clinical_data_0_CK &lt;dbl&gt; | clinical_data_0_CK_MB &lt;dbl&gt; | clinical_data_0_CRP &lt;dbl&gt; | clinical_data_0_delta_ef_value &lt;dbl&gt; | clinical_data_0_Troponin &lt;dbl&gt; | cytokine_0_BCA1__CXCL13 &lt;dbl&gt; | cytokine_0_CTACK__CCL27 &lt;dbl&gt; | cytokine_0_EGF__EGF &lt;dbl&gt; | cytokine_0_ENA78__CXCL5 &lt;dbl&gt; | cytokine_0_Eotaxin__CCL11 &lt;dbl&gt; | ⋯ ⋯ | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| k1 | 5.459432 | NA | 0.4854268 | NA | 0.01863417 | 5.696550 | 9.821662 | 5.045268 | 10.562366 | 7.42341 | ⋯ |       NA |       NA |       NA |       NA |       NA |       NA |       NA |        NA |       NA |        NA |\n",
       "| k10 |       NA | NA | 0.2630344 | NA |         NA | 4.697663 | 9.701220 | 1.922198 |  7.724173 | 9.19842 | ⋯ | 2.461054 | 2.157169 | 1.975326 | 3.139595 | 2.078417 | 1.843355 | 1.343136 | 0.6540153 | 1.190677 | 0.8572279 |\n",
       "\n"
      ],
      "text/plain": [
       "    clinical_data_0_CK clinical_data_0_CK_MB clinical_data_0_CRP\n",
       "k1  5.459432           NA                    0.4854268          \n",
       "k10       NA           NA                    0.2630344          \n",
       "    clinical_data_0_delta_ef_value clinical_data_0_Troponin\n",
       "k1  NA                             0.01863417              \n",
       "k10 NA                                     NA              \n",
       "    cytokine_0_BCA1__CXCL13 cytokine_0_CTACK__CCL27 cytokine_0_EGF__EGF\n",
       "k1  5.696550                9.821662                5.045268           \n",
       "k10 4.697663                9.701220                1.922198           \n",
       "    cytokine_0_ENA78__CXCL5 cytokine_0_Eotaxin__CCL11 ⋯\n",
       "k1  10.562366               7.42341                   ⋯\n",
       "k10  7.724173               9.19842                   ⋯\n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2\n",
       "k1        NA                                      \n",
       "k10 2.461054                                      \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5\n",
       "k1        NA                                        \n",
       "k10 2.157169                                        \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1\n",
       "k1        NA                                       \n",
       "k10 1.975326                                       \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36\n",
       "k1        NA                                       \n",
       "k10 3.139595                                       \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1\n",
       "k1        NA                                         \n",
       "k10 2.078417                                         \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2\n",
       "k1        NA                                         \n",
       "k10 1.843355                                         \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106\n",
       "k1        NA                                        \n",
       "k10 1.343136                                        \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207\n",
       "k1         NA                                       \n",
       "k10 0.6540153                                       \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706\n",
       "k1        NA                                        \n",
       "k10 1.190677                                        \n",
       "    single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX\n",
       "k1         NA                                    \n",
       "k10 0.8572279                                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(expression_data,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 525,
   "id": "dc74d19d-d14c-40cb-8913-8d718d4b6334",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Calculate all correlations between ligands and targets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 526,
   "id": "5494afdd-6628-4711-9beb-739f43ad13a1",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n"
     ]
    }
   ],
   "source": [
    "ligand_target_correlations = cor(expression_data[,colnames(expression_data) %in% expressed_ligands], expression_data[,!colnames(expression_data) %in% expressed_ligands], method = 'spearman', use = 'pairwise.complete.obs') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 527,
   "id": "1aeb596e-d579-45c9-836a-929524e508b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 2 × 12939 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>clinical_data_0_CK</th><th scope=col>clinical_data_0_CK_MB</th><th scope=col>clinical_data_0_CRP</th><th scope=col>clinical_data_0_delta_ef_value</th><th scope=col>clinical_data_0_Troponin</th><th scope=col>cytokine_0_Eotaxin__CCL11</th><th scope=col>cytokine_0_Eotaxin2__CCL24</th><th scope=col>cytokine_0_FLT3L__FLT3LG</th><th scope=col>cytokine_0_GROalpha__CXCL1</th><th scope=col>cytokine_0_IL12p40__IL12B</th><th scope=col>⋯</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>cytokine_0_BCA1__CXCL13</th><td> 0.04568048</td><td> 0.1841583</td><td>0.03474096</td><td>-0.263390374</td><td> 0.08664079</td><td>-0.4042097</td><td>0.07479910</td><td>-0.12506712</td><td>-0.20342629</td><td>-0.06320019</td><td>⋯</td><td> 0.14780538</td><td> 0.06469496</td><td>-0.033172594</td><td> 0.02807027</td><td> 0.006225096</td><td> 0.08782851</td><td>-0.06057393</td><td> 0.1023231</td><td> 0.20825309</td><td>-0.02657856</td></tr>\n",
       "\t<tr><th scope=row>cytokine_0_CTACK__CCL27</th><td>-0.19761446</td><td>-0.3175698</td><td>0.01813627</td><td>-0.008483006</td><td>-0.09251747</td><td> 0.1002265</td><td>0.07672164</td><td>-0.02487356</td><td>-0.03717696</td><td> 0.30506995</td><td>⋯</td><td>-0.02130075</td><td>-0.15482846</td><td> 0.004267636</td><td>-0.11350619</td><td>-0.029636124</td><td>-0.04159127</td><td> 0.08755046</td><td>-0.1470658</td><td>-0.01133356</td><td>-0.05600466</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 2 × 12939 of type dbl\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & clinical\\_data\\_0\\_CK & clinical\\_data\\_0\\_CK\\_MB & clinical\\_data\\_0\\_CRP & clinical\\_data\\_0\\_delta\\_ef\\_value & clinical\\_data\\_0\\_Troponin & cytokine\\_0\\_Eotaxin\\_\\_CCL11 & cytokine\\_0\\_Eotaxin2\\_\\_CCL24 & cytokine\\_0\\_FLT3L\\_\\_FLT3LG & cytokine\\_0\\_GROalpha\\_\\_CXCL1 & cytokine\\_0\\_IL12p40\\_\\_IL12B & ⋯ & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZEB2 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFAND5 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFAS1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36L1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36L2 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF106 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF207 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF706 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZYX\\\\\n",
       "\\hline\n",
       "\tcytokine\\_0\\_BCA1\\_\\_CXCL13 &  0.04568048 &  0.1841583 & 0.03474096 & -0.263390374 &  0.08664079 & -0.4042097 & 0.07479910 & -0.12506712 & -0.20342629 & -0.06320019 & ⋯ &  0.14780538 &  0.06469496 & -0.033172594 &  0.02807027 &  0.006225096 &  0.08782851 & -0.06057393 &  0.1023231 &  0.20825309 & -0.02657856\\\\\n",
       "\tcytokine\\_0\\_CTACK\\_\\_CCL27 & -0.19761446 & -0.3175698 & 0.01813627 & -0.008483006 & -0.09251747 &  0.1002265 & 0.07672164 & -0.02487356 & -0.03717696 &  0.30506995 & ⋯ & -0.02130075 & -0.15482846 &  0.004267636 & -0.11350619 & -0.029636124 & -0.04159127 &  0.08755046 & -0.1470658 & -0.01133356 & -0.05600466\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 2 × 12939 of type dbl\n",
       "\n",
       "| <!--/--> | clinical_data_0_CK | clinical_data_0_CK_MB | clinical_data_0_CRP | clinical_data_0_delta_ef_value | clinical_data_0_Troponin | cytokine_0_Eotaxin__CCL11 | cytokine_0_Eotaxin2__CCL24 | cytokine_0_FLT3L__FLT3LG | cytokine_0_GROalpha__CXCL1 | cytokine_0_IL12p40__IL12B | ⋯ | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cytokine_0_BCA1__CXCL13 |  0.04568048 |  0.1841583 | 0.03474096 | -0.263390374 |  0.08664079 | -0.4042097 | 0.07479910 | -0.12506712 | -0.20342629 | -0.06320019 | ⋯ |  0.14780538 |  0.06469496 | -0.033172594 |  0.02807027 |  0.006225096 |  0.08782851 | -0.06057393 |  0.1023231 |  0.20825309 | -0.02657856 |\n",
       "| cytokine_0_CTACK__CCL27 | -0.19761446 | -0.3175698 | 0.01813627 | -0.008483006 | -0.09251747 |  0.1002265 | 0.07672164 | -0.02487356 | -0.03717696 |  0.30506995 | ⋯ | -0.02130075 | -0.15482846 |  0.004267636 | -0.11350619 | -0.029636124 | -0.04159127 |  0.08755046 | -0.1470658 | -0.01133356 | -0.05600466 |\n",
       "\n"
      ],
      "text/plain": [
       "                        clinical_data_0_CK clinical_data_0_CK_MB\n",
       "cytokine_0_BCA1__CXCL13  0.04568048         0.1841583           \n",
       "cytokine_0_CTACK__CCL27 -0.19761446        -0.3175698           \n",
       "                        clinical_data_0_CRP clinical_data_0_delta_ef_value\n",
       "cytokine_0_BCA1__CXCL13 0.03474096          -0.263390374                  \n",
       "cytokine_0_CTACK__CCL27 0.01813627          -0.008483006                  \n",
       "                        clinical_data_0_Troponin cytokine_0_Eotaxin__CCL11\n",
       "cytokine_0_BCA1__CXCL13  0.08664079              -0.4042097               \n",
       "cytokine_0_CTACK__CCL27 -0.09251747               0.1002265               \n",
       "                        cytokine_0_Eotaxin2__CCL24 cytokine_0_FLT3L__FLT3LG\n",
       "cytokine_0_BCA1__CXCL13 0.07479910                 -0.12506712             \n",
       "cytokine_0_CTACK__CCL27 0.07672164                 -0.02487356             \n",
       "                        cytokine_0_GROalpha__CXCL1 cytokine_0_IL12p40__IL12B ⋯\n",
       "cytokine_0_BCA1__CXCL13 -0.20342629                -0.06320019               ⋯\n",
       "cytokine_0_CTACK__CCL27 -0.03717696                 0.30506995               ⋯\n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2\n",
       "cytokine_0_BCA1__CXCL13  0.14780538                                   \n",
       "cytokine_0_CTACK__CCL27 -0.02130075                                   \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5\n",
       "cytokine_0_BCA1__CXCL13  0.06469496                                     \n",
       "cytokine_0_CTACK__CCL27 -0.15482846                                     \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1\n",
       "cytokine_0_BCA1__CXCL13 -0.033172594                                   \n",
       "cytokine_0_CTACK__CCL27  0.004267636                                   \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36\n",
       "cytokine_0_BCA1__CXCL13  0.02807027                                    \n",
       "cytokine_0_CTACK__CCL27 -0.11350619                                    \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1\n",
       "cytokine_0_BCA1__CXCL13  0.006225096                                     \n",
       "cytokine_0_CTACK__CCL27 -0.029636124                                     \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2\n",
       "cytokine_0_BCA1__CXCL13  0.08782851                                      \n",
       "cytokine_0_CTACK__CCL27 -0.04159127                                      \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106\n",
       "cytokine_0_BCA1__CXCL13 -0.06057393                                     \n",
       "cytokine_0_CTACK__CCL27  0.08755046                                     \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207\n",
       "cytokine_0_BCA1__CXCL13  0.1023231                                      \n",
       "cytokine_0_CTACK__CCL27 -0.1470658                                      \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706\n",
       "cytokine_0_BCA1__CXCL13  0.20825309                                     \n",
       "cytokine_0_CTACK__CCL27 -0.01133356                                     \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX\n",
       "cytokine_0_BCA1__CXCL13 -0.02657856                                  \n",
       "cytokine_0_CTACK__CCL27 -0.05600466                                  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_target_correlations,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 528,
   "id": "88ae81b8-96d5-4fed-9a7d-a7f55c51df6b",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Transform to long format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 529,
   "id": "f7fd06cc-1689-4038-83aa-420d6e2f7119",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations = melt(ligand_target_correlations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 530,
   "id": "d64d14aa-8de6-435d-aeba-7a246ce5c438",
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames(ligand_target_correlations) = c('Ligand', 'Target', 'correlation')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 531,
   "id": "dbb19653-a67b-4c21-b10d-3bcc9667532d",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Adjust format and extract certain columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 532,
   "id": "d99fd456-355e-40ad-9c1f-fcc15088ec1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Target_Gene = str_replace(ligand_target_correlations$Target, '.*_0_', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 533,
   "id": "345aa6c1-90b5-447c-a0e8-2cf6f6c1701b",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Target_Gene = str_replace(ligand_target_correlations$Target_Gene, '.*__', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 534,
   "id": "12bd9b23-45c6-47de-bf05-d845c1959994",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Ligand_Gene = str_replace(ligand_target_correlations$Ligand, '.*_0_', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 535,
   "id": "429f3bf9-d4cb-497d-ab13-380473e818e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Ligand_Gene = str_replace(ligand_target_correlations$Ligand_Gene, '.*__', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 536,
   "id": "5dc6497b-6389-4f82-bfdc-bafe74421b29",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Target_Type = str_replace(ligand_target_correlations$Target, '_0_.*', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 537,
   "id": "e28db9c1-7f65-443b-9eae-35c4a5b2325b",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Target_Type[ligand_target_correlations$Target_Type == 'single_cell']  = str_replace(ligand_target_correlations$Target[ligand_target_correlations$Target_Type == 'single_cell'] , '__.*', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 538,
   "id": "49e1e65e-9f94-4255-8a65-75e55e27120c",
   "metadata": {},
   "outputs": [],
   "source": [
    "###  Adapt target gene format for proteomics\n",
    "ligand_target_correlations$Target_Gene[ligand_target_correlations$Target_Type == 'proteomics'] = str_replace(ligand_target_correlations$Target_Gene[ligand_target_correlations$Target_Type == 'proteomics'], '_.*', '')\n",
    "ligand_target_correlations$Ligand_Gene[ligand_target_correlations$Ligand_Type == 'proteomics'] = str_replace(ligand_target_correlations$Ligand_Gene[ligand_target_correlations$Ligand_Type == 'proteomics'], '_.*', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 539,
   "id": "f93e38f2-6b42-4968-8986-b41ae12195b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Ligand_Type = str_replace(ligand_target_correlations$Ligand, '_0_.*', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 540,
   "id": "3c4c5166-477f-451a-a631-cc8703201ee3",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Ligand_Type[ligand_target_correlations$Ligand_Type == 'single_cell']  = str_replace(ligand_target_correlations$Ligand[ligand_target_correlations$Ligand_Type == 'single_cell'] , '__.*', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 541,
   "id": "62fee583-2df2-4bbf-83d9-a1d86b1d02c1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>cytokine_0_BCA1__CXCL13</li><li>cytokine_0_CTACK__CCL27</li><li>cytokine_0_EGF__EGF</li><li>cytokine_0_ENA78__CXCL5</li><li>cytokine_0_Eotaxin3__CCL26</li><li>cytokine_0_FGF2__FGF2</li></ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'cytokine_0_BCA1__CXCL13'</li><li>'cytokine_0_CTACK__CCL27'</li><li>'cytokine_0_EGF__EGF'</li><li>'cytokine_0_ENA78__CXCL5'</li><li>'cytokine_0_Eotaxin3__CCL26'</li><li>'cytokine_0_FGF2__FGF2'</li><li>'cytokine_0_Fractalkine__CX3CL1'</li><li>'cytokine_0_GCSF__CSF3'</li><li>'cytokine_0_GMCSF__CSF2'</li><li>'cytokine_0_I309__CCL1'</li><li>'cytokine_0_IFNalpha2__IFNA2'</li><li>'cytokine_0_IFNgamma__IFNG'</li><li>'cytokine_0_IL10__IL10'</li><li>'cytokine_0_IL12p70__IL12A'</li><li>'cytokine_0_IL13__IL13'</li><li>'cytokine_0_IL15__IL15'</li><li>'cytokine_0_IL17A__IL17A'</li><li>'cytokine_0_IL17E_IL25__IL25'</li><li>'cytokine_0_IL17F__IL17F'</li><li>'cytokine_0_IL1alpha__IL1A'</li><li>'cytokine_0_IL1beta__IL1B'</li><li>'cytokine_0_IL1RA__IL1RN'</li><li>'cytokine_0_IL2__IL2'</li><li>'cytokine_0_IL22__IL22'</li><li>'cytokine_0_IL27__IL27'</li><li>'cytokine_0_IL4__IL4'</li><li>'cytokine_0_IL5__IL5'</li><li>'cytokine_0_IL6__IL6'</li><li>'cytokine_0_IL7__IL7'</li><li>'cytokine_0_IL9__IL9'</li><li>'cytokine_0_IP10__CXCL10'</li><li>'cytokine_0_LIF__LIF'</li><li>'cytokine_0_MCP1__CCL2'</li><li>'cytokine_0_MCP2__CCL8'</li><li>'cytokine_0_MCP3__CCL7'</li><li>'cytokine_0_MCP4__CCL13'</li><li>'cytokine_0_MCSF__CSF1'</li><li>'cytokine_0_MDC__CCL22'</li><li>'cytokine_0_MIG_CXCL9__CXCL9'</li><li>'cytokine_0_MIP1alpha__CCL3'</li><li>'cytokine_0_MIP1beta__CCL4'</li><li>'cytokine_0_MIP1delta__CCL15'</li><li>'cytokine_0_RANTES__CCL5'</li><li>'cytokine_0_SCF__KITLG'</li><li>'cytokine_0_SDF1alpha_beta__CXCL12'</li><li>'cytokine_0_TARC__CCL17'</li><li>'cytokine_0_TNFalpha__TNF'</li><li>'cytokine_0_TNFbeta__LTA'</li><li>'cytokine_0_VEGFA__VEGFA'</li><li>'cytokine_0_X6CKine__CCL21'</li><li>'neutrophil_0_ADM'</li><li>'neutrophil_0_FBRS'</li><li>'neutrophil_0_GRN'</li><li>'neutrophil_0_HMGB1'</li><li>'neutrophil_0_HMGB2'</li><li>'neutrophil_0_IL16'</li><li>'neutrophil_0_MMP9'</li><li>'neutrophil_0_PGLYRP1'</li><li>'neutrophil_0_RTN4'</li><li>'neutrophil_0_SECTM1'</li><li>'neutrophil_0_TGFB1'</li><li>'neutrophil_0_TIMP1'</li><li>'neutrophil_0_TNFSF13B'</li><li>'proteomics_0_A2M_P01023'</li><li>'proteomics_0_ADIPOQ_Q15848'</li><li>'proteomics_0_AGT_P01019'</li><li>'proteomics_0_APOA1_P02647'</li><li>'proteomics_0_APOB_P04114'</li><li>'proteomics_0_APOE_P02649'</li><li>'proteomics_0_C3_M0R1Q1'</li><li>'proteomics_0_C3_P01024'</li><li>'proteomics_0_C5_P01031'</li><li>'proteomics_0_CAMP_J3KNB4'</li><li>'proteomics_0_CDH5_P33151'</li><li>'proteomics_0_CEACAM1_P13688'</li><li>'proteomics_0_CFH_P08603'</li><li>'proteomics_0_COL18A1_P39060'</li><li>'proteomics_0_COMP_G3XAP6'</li><li>'proteomics_0_DSG2_Q14126'</li><li>'proteomics_0_F2_P00734'</li><li>'proteomics_0_F9_P00740'</li><li>'proteomics_0_FGA_P02671'</li><li>'proteomics_0_FN1_H0Y4K8'</li><li>'proteomics_0_FN1_P02751'</li><li>'proteomics_0_GSTP1_P09211'</li><li>'proteomics_0_HSP90B1_P14625'</li><li>'proteomics_0_HSPG2_P98160'</li><li>'proteomics_0_ICAM1_P05362'</li><li>'proteomics_0_ICAM2_J3QQR8'</li><li>'proteomics_0_IGF2_P01344'</li><li>'proteomics_0_KNG1_P01042'</li><li>'proteomics_0_KNG1_P01042.2'</li><li>'proteomics_0_LTF_E7EQB2'</li><li>'proteomics_0_MMP9_P14780'</li><li>'proteomics_0_NID1_P14543'</li><li>'proteomics_0_PF4_P02776'</li><li>'proteomics_0_PLG_P00747'</li><li>'proteomics_0_PPBP_P02775'</li><li>'proteomics_0_SAA1_P0DJI8'</li><li>'proteomics_0_SELL_P14151'</li><li>'proteomics_0_SERPINA1_A0A024R6I7'</li><li>'proteomics_0_SERPINA1_P01009'</li><li>'proteomics_0_SERPINC1_P01008'</li><li>'proteomics_0_SERPING1_P05155'</li><li>'proteomics_0_TF_P02787'</li><li>'proteomics_0_THBS1_P07996'</li><li>'proteomics_0_TIMP1_P01033'</li><li>'proteomics_0_TTR_P02766'</li><li>'proteomics_0_VTN_P04004'</li><li>'single_cell_0_X0_T.cell.CD4__ANXA1'</li><li>'single_cell_0_X0_T.cell.CD4__ARF1'</li><li>'single_cell_0_X0_T.cell.CD4__CALM1'</li><li>'single_cell_0_X0_T.cell.CD4__CALR'</li><li>'single_cell_0_X0_T.cell.CD4__CD2'</li><li>'single_cell_0_X0_T.cell.CD4__HMGB1'</li><li>'single_cell_0_X0_T.cell.CD4__HMGB2'</li><li>'single_cell_0_X0_T.cell.CD4__HSP90B1'</li><li>'single_cell_0_X0_T.cell.CD4__ICAM3'</li><li>'single_cell_0_X0_T.cell.CD4__LTB'</li><li>'single_cell_0_X0_T.cell.CD4__MIF'</li><li>'single_cell_0_X0_T.cell.CD4__PTPRC'</li><li>'single_cell_0_X0_T.cell.CD4__SELL'</li><li>'single_cell_0_X1_T.cell.CD8__ANXA1'</li><li>'single_cell_0_X1_T.cell.CD8__ARF1'</li><li>'single_cell_0_X1_T.cell.CD8__CALM1'</li><li>'single_cell_0_X1_T.cell.CD8__CALR'</li><li>'single_cell_0_X1_T.cell.CD8__CCL4'</li><li>'single_cell_0_X1_T.cell.CD8__CCL5'</li><li>'single_cell_0_X1_T.cell.CD8__CD2'</li><li>'single_cell_0_X1_T.cell.CD8__GSTP1'</li><li>'single_cell_0_X1_T.cell.CD8__GZMB'</li><li>'single_cell_0_X1_T.cell.CD8__HMGB1'</li><li>'single_cell_0_X1_T.cell.CD8__HMGB2'</li><li>'single_cell_0_X1_T.cell.CD8__HSP90B1'</li><li>'single_cell_0_X1_T.cell.CD8__ICAM3'</li><li>'single_cell_0_X1_T.cell.CD8__LTB'</li><li>'single_cell_0_X1_T.cell.CD8__PTPRC'</li><li>'single_cell_0_X1_T.cell.CD8__TGFB1'</li><li>'single_cell_0_X10_B.cell__ANXA1'</li><li>'single_cell_0_X10_B.cell__ARF1'</li><li>'single_cell_0_X10_B.cell__CALM1'</li><li>'single_cell_0_X10_B.cell__CALR'</li><li>'single_cell_0_X10_B.cell__CD2'</li><li>'single_cell_0_X10_B.cell__GSTP1'</li><li>'single_cell_0_X10_B.cell__HMGB1'</li><li>'single_cell_0_X10_B.cell__HMGB2'</li><li>'single_cell_0_X10_B.cell__HSP90B1'</li><li>'single_cell_0_X10_B.cell__ICAM3'</li><li>'single_cell_0_X10_B.cell__LTB'</li><li>'single_cell_0_X10_B.cell__PTPRC'</li><li>'single_cell_0_X10_B.cell__SELL'</li><li>'single_cell_0_X10_B.cell__TGFB1'</li><li>'single_cell_0_X11_T.cell.CD4__ANXA1'</li><li>'single_cell_0_X11_T.cell.CD4__ARF1'</li><li>'single_cell_0_X11_T.cell.CD4__CALM1'</li><li>'single_cell_0_X11_T.cell.CD4__CALR'</li><li>'single_cell_0_X11_T.cell.CD4__CD2'</li><li>'single_cell_0_X11_T.cell.CD4__HMGB1'</li><li>'single_cell_0_X11_T.cell.CD4__HMGB2'</li><li>'single_cell_0_X11_T.cell.CD4__HSP90B1'</li><li>'single_cell_0_X11_T.cell.CD4__ICAM3'</li><li>'single_cell_0_X11_T.cell.CD4__LTB'</li><li>'single_cell_0_X11_T.cell.CD4__PTPRC'</li><li>'single_cell_0_X11_T.cell.CD4__SELL'</li><li>'single_cell_0_X11_T.cell.CD4__TGFB1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__ANXA1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__ARF1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__CALM1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__CALR'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__GSTP1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__HMGB1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__HMGB2'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__HSP90B1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__ICAM3'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__IL1B'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__LGALS3'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__LTB'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__NAMPT'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__PECAM1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__PSAP'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__PTPRC'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__RTN4'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__SERPINA1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__TGFB1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__TIMP1'</li><li>'single_cell_0_X13_Dendritic__ANXA1'</li><li>'single_cell_0_X13_Dendritic__CALM1'</li><li>'single_cell_0_X13_Dendritic__CALR'</li><li>'single_cell_0_X13_Dendritic__GRN'</li><li>'single_cell_0_X13_Dendritic__GSTP1'</li><li>'single_cell_0_X13_Dendritic__HMGB1'</li><li>'single_cell_0_X13_Dendritic__HSP90B1'</li><li>'single_cell_0_X13_Dendritic__NAMPT'</li><li>'single_cell_0_X13_Dendritic__PSAP'</li><li>'single_cell_0_X13_Dendritic__PTPRC'</li><li>'single_cell_0_X13_Dendritic__RTN4'</li><li>'single_cell_0_X13_Dendritic__TIMP1'</li><li>'single_cell_0_X2_T.cell.CD4__ANXA1'</li><li>'single_cell_0_X2_T.cell.CD4__ARF1'</li><li>'single_cell_0_X2_T.cell.CD4__CALM1'</li><li>'single_cell_0_X2_T.cell.CD4__CALR'</li><li>'single_cell_0_X2_T.cell.CD4__CD2'</li><li>'single_cell_0_X2_T.cell.CD4__HMGB1'</li><li>'single_cell_0_X2_T.cell.CD4__HMGB2'</li><li>'single_cell_0_X2_T.cell.CD4__HSP90B1'</li><li>'single_cell_0_X2_T.cell.CD4__LTB'</li><li>'single_cell_0_X2_T.cell.CD4__PTPRC'</li><li>'single_cell_0_X2_T.cell.CD4__SELL'</li><li>'single_cell_0_X3_NK__ANXA1'</li><li>'single_cell_0_X3_NK__ARF1'</li><li>'single_cell_0_X3_NK__CALM1'</li><li>'single_cell_0_X3_NK__CALR'</li><li>'single_cell_0_X3_NK__CCL4'</li><li>'single_cell_0_X3_NK__CCL5'</li><li>'single_cell_0_X3_NK__GSTP1'</li><li>'single_cell_0_X3_NK__GZMB'</li><li>'single_cell_0_X3_NK__HMGB1'</li><li>'single_cell_0_X3_NK__HMGB2'</li><li>'single_cell_0_X3_NK__HSP90B1'</li><li>'single_cell_0_X3_NK__PTPRC'</li><li>'single_cell_0_X3_NK__TGFB1'</li><li>'single_cell_0_X4_Monocytes...CD14__ADAM17'</li><li>'single_cell_0_X4_Monocytes...CD14__AIMP1'</li><li>'single_cell_0_X4_Monocytes...CD14__ALOX5AP'</li><li>'single_cell_0_X4_Monocytes...CD14__ANXA1'</li><li>'single_cell_0_X4_Monocytes...CD14__ARF1'</li><li>'single_cell_0_X4_Monocytes...CD14__CALM1'</li><li>'single_cell_0_X4_Monocytes...CD14__CALR'</li><li>'single_cell_0_X4_Monocytes...CD14__CCL5'</li><li>'single_cell_0_X4_Monocytes...CD14__COPA'</li><li>'single_cell_0_X4_Monocytes...CD14__CXCL2'</li><li>'single_cell_0_X4_Monocytes...CD14__GRN'</li><li>'single_cell_0_X4_Monocytes...CD14__GSTP1'</li><li>'single_cell_0_X4_Monocytes...CD14__HDGF'</li><li>'single_cell_0_X4_Monocytes...CD14__HMGB1'</li><li>'single_cell_0_X4_Monocytes...CD14__HMGB2'</li><li>'single_cell_0_X4_Monocytes...CD14__HSP90B1'</li><li>'single_cell_0_X4_Monocytes...CD14__ICAM1'</li><li>'single_cell_0_X4_Monocytes...CD14__ICAM3'</li><li>'single_cell_0_X4_Monocytes...CD14__IL1B'</li><li>'single_cell_0_X4_Monocytes...CD14__LGALS3'</li><li>'single_cell_0_X4_Monocytes...CD14__LRPAP1'</li><li>'single_cell_0_X4_Monocytes...CD14__LTB'</li><li>'single_cell_0_X4_Monocytes...CD14__MIF'</li><li>'single_cell_0_X4_Monocytes...CD14__PECAM1'</li><li>'single_cell_0_X4_Monocytes...CD14__PSAP'</li><li>'single_cell_0_X4_Monocytes...CD14__PSEN1'</li><li>'single_cell_0_X4_Monocytes...CD14__PTPRC'</li><li>'single_cell_0_X4_Monocytes...CD14__RTN4'</li><li>'single_cell_0_X4_Monocytes...CD14__SERPINA1'</li><li>'single_cell_0_X4_Monocytes...CD14__TGFB1'</li><li>'single_cell_0_X4_Monocytes...CD14__TIMP1'</li><li>'single_cell_0_X4_Monocytes...CD14__TNFSF10'</li><li>'single_cell_0_X4_Monocytes...CD14__TNFSF13B'</li><li>'single_cell_0_X5_T.cell.CD4__ANXA1'</li><li>'single_cell_0_X5_T.cell.CD4__CALM1'</li><li>'single_cell_0_X5_T.cell.CD4__CD2'</li><li>'single_cell_0_X5_T.cell.CD4__HMGB1'</li><li>'single_cell_0_X5_T.cell.CD4__HMGB2'</li><li>'single_cell_0_X5_T.cell.CD4__HSP90B1'</li><li>'single_cell_0_X5_T.cell.CD4__ICAM3'</li><li>'single_cell_0_X5_T.cell.CD4__LTB'</li><li>'single_cell_0_X5_T.cell.CD4__PTPRC'</li><li>'single_cell_0_X5_T.cell.CD4__SELL'</li><li>'single_cell_0_X6_Monocytes...CD14__ADAM17'</li><li>'single_cell_0_X6_Monocytes...CD14__ANXA1'</li><li>'single_cell_0_X6_Monocytes...CD14__ARF1'</li><li>'single_cell_0_X6_Monocytes...CD14__CALM1'</li><li>'single_cell_0_X6_Monocytes...CD14__CALR'</li><li>'single_cell_0_X6_Monocytes...CD14__CXCL2'</li><li>'single_cell_0_X6_Monocytes...CD14__GRN'</li><li>'single_cell_0_X6_Monocytes...CD14__GSTP1'</li><li>'single_cell_0_X6_Monocytes...CD14__HMGB1'</li><li>'single_cell_0_X6_Monocytes...CD14__HMGB2'</li><li>'single_cell_0_X6_Monocytes...CD14__HSP90B1'</li><li>'single_cell_0_X6_Monocytes...CD14__ICAM1'</li><li>'single_cell_0_X6_Monocytes...CD14__ICAM3'</li><li>'single_cell_0_X6_Monocytes...CD14__IL1B'</li><li>'single_cell_0_X6_Monocytes...CD14__LGALS3'</li><li>'single_cell_0_X6_Monocytes...CD14__LRPAP1'</li><li>'single_cell_0_X6_Monocytes...CD14__PECAM1'</li><li>'single_cell_0_X6_Monocytes...CD14__PSAP'</li><li>'single_cell_0_X6_Monocytes...CD14__PSEN1'</li><li>'single_cell_0_X6_Monocytes...CD14__PTPRC'</li><li>'single_cell_0_X6_Monocytes...CD14__RTN4'</li><li>'single_cell_0_X6_Monocytes...CD14__SERPINA1'</li><li>'single_cell_0_X6_Monocytes...CD14__TGFB1'</li><li>'single_cell_0_X6_Monocytes...CD14__THBS1'</li><li>'single_cell_0_X6_Monocytes...CD14__TIMP1'</li><li>'single_cell_0_X6_Monocytes...CD14__TNFSF10'</li><li>'single_cell_0_X6_Monocytes...CD14__TNFSF13B'</li><li>'single_cell_0_X7_Monocytes...CD14__ANXA1'</li><li>'single_cell_0_X7_Monocytes...CD14__ARF1'</li><li>'single_cell_0_X7_Monocytes...CD14__CALM1'</li><li>'single_cell_0_X7_Monocytes...CD14__CALR'</li><li>'single_cell_0_X7_Monocytes...CD14__GRN'</li><li>'single_cell_0_X7_Monocytes...CD14__GSTP1'</li><li>'single_cell_0_X7_Monocytes...CD14__HMGB1'</li><li>'single_cell_0_X7_Monocytes...CD14__HMGB2'</li><li>'single_cell_0_X7_Monocytes...CD14__HSP90B1'</li><li>'single_cell_0_X7_Monocytes...CD14__ICAM3'</li><li>'single_cell_0_X7_Monocytes...CD14__IL1B'</li><li>'single_cell_0_X7_Monocytes...CD14__LGALS3'</li><li>'single_cell_0_X7_Monocytes...CD14__LRPAP1'</li><li>'single_cell_0_X7_Monocytes...CD14__PECAM1'</li><li>'single_cell_0_X7_Monocytes...CD14__PSAP'</li><li>'single_cell_0_X7_Monocytes...CD14__PTPRC'</li><li>'single_cell_0_X7_Monocytes...CD14__RTN4'</li><li>'single_cell_0_X7_Monocytes...CD14__SERPINA1'</li><li>'single_cell_0_X7_Monocytes...CD14__TGFB1'</li><li>'single_cell_0_X7_Monocytes...CD14__TIMP1'</li><li>'single_cell_0_X7_Monocytes...CD14__TNFSF10'</li><li>'single_cell_0_X7_Monocytes...CD14__TNFSF13B'</li><li>'single_cell_0_X8_B.cell__CALM1'</li><li>'single_cell_0_X8_B.cell__CALR'</li><li>'single_cell_0_X8_B.cell__HMGB1'</li><li>'single_cell_0_X8_B.cell__HMGB2'</li><li>'single_cell_0_X8_B.cell__HSP90B1'</li><li>'single_cell_0_X8_B.cell__LTB'</li><li>'single_cell_0_X8_B.cell__PTPRC'</li><li>'single_cell_0_X8_B.cell__SELL'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__ANXA1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__ARF1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__CALM1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__CALR'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__CXCL16'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__GRN'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__GSTP1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__HMGB1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__HMGB2'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__HSP90B1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__ICAM3'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__IL1B'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__LGALS3'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__NAMPT'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__PECAM1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__PSAP'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__PTPRC'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__RTN4'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__SERPINA1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__TGFB1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__TIMP1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFSF10'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFSF13B'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item cytokine\\_0\\_BCA1\\_\\_CXCL13\n",
       "\\item cytokine\\_0\\_CTACK\\_\\_CCL27\n",
       "\\item cytokine\\_0\\_EGF\\_\\_EGF\n",
       "\\item cytokine\\_0\\_ENA78\\_\\_CXCL5\n",
       "\\item cytokine\\_0\\_Eotaxin3\\_\\_CCL26\n",
       "\\item cytokine\\_0\\_FGF2\\_\\_FGF2\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'cytokine\\_0\\_BCA1\\_\\_CXCL13'\n",
       "\\item 'cytokine\\_0\\_CTACK\\_\\_CCL27'\n",
       "\\item 'cytokine\\_0\\_EGF\\_\\_EGF'\n",
       "\\item 'cytokine\\_0\\_ENA78\\_\\_CXCL5'\n",
       "\\item 'cytokine\\_0\\_Eotaxin3\\_\\_CCL26'\n",
       "\\item 'cytokine\\_0\\_FGF2\\_\\_FGF2'\n",
       "\\item 'cytokine\\_0\\_Fractalkine\\_\\_CX3CL1'\n",
       "\\item 'cytokine\\_0\\_GCSF\\_\\_CSF3'\n",
       "\\item 'cytokine\\_0\\_GMCSF\\_\\_CSF2'\n",
       "\\item 'cytokine\\_0\\_I309\\_\\_CCL1'\n",
       "\\item 'cytokine\\_0\\_IFNalpha2\\_\\_IFNA2'\n",
       "\\item 'cytokine\\_0\\_IFNgamma\\_\\_IFNG'\n",
       "\\item 'cytokine\\_0\\_IL10\\_\\_IL10'\n",
       "\\item 'cytokine\\_0\\_IL12p70\\_\\_IL12A'\n",
       "\\item 'cytokine\\_0\\_IL13\\_\\_IL13'\n",
       "\\item 'cytokine\\_0\\_IL15\\_\\_IL15'\n",
       "\\item 'cytokine\\_0\\_IL17A\\_\\_IL17A'\n",
       "\\item 'cytokine\\_0\\_IL17E\\_IL25\\_\\_IL25'\n",
       "\\item 'cytokine\\_0\\_IL17F\\_\\_IL17F'\n",
       "\\item 'cytokine\\_0\\_IL1alpha\\_\\_IL1A'\n",
       "\\item 'cytokine\\_0\\_IL1beta\\_\\_IL1B'\n",
       "\\item 'cytokine\\_0\\_IL1RA\\_\\_IL1RN'\n",
       "\\item 'cytokine\\_0\\_IL2\\_\\_IL2'\n",
       "\\item 'cytokine\\_0\\_IL22\\_\\_IL22'\n",
       "\\item 'cytokine\\_0\\_IL27\\_\\_IL27'\n",
       "\\item 'cytokine\\_0\\_IL4\\_\\_IL4'\n",
       "\\item 'cytokine\\_0\\_IL5\\_\\_IL5'\n",
       "\\item 'cytokine\\_0\\_IL6\\_\\_IL6'\n",
       "\\item 'cytokine\\_0\\_IL7\\_\\_IL7'\n",
       "\\item 'cytokine\\_0\\_IL9\\_\\_IL9'\n",
       "\\item 'cytokine\\_0\\_IP10\\_\\_CXCL10'\n",
       "\\item 'cytokine\\_0\\_LIF\\_\\_LIF'\n",
       "\\item 'cytokine\\_0\\_MCP1\\_\\_CCL2'\n",
       "\\item 'cytokine\\_0\\_MCP2\\_\\_CCL8'\n",
       "\\item 'cytokine\\_0\\_MCP3\\_\\_CCL7'\n",
       "\\item 'cytokine\\_0\\_MCP4\\_\\_CCL13'\n",
       "\\item 'cytokine\\_0\\_MCSF\\_\\_CSF1'\n",
       "\\item 'cytokine\\_0\\_MDC\\_\\_CCL22'\n",
       "\\item 'cytokine\\_0\\_MIG\\_CXCL9\\_\\_CXCL9'\n",
       "\\item 'cytokine\\_0\\_MIP1alpha\\_\\_CCL3'\n",
       "\\item 'cytokine\\_0\\_MIP1beta\\_\\_CCL4'\n",
       "\\item 'cytokine\\_0\\_MIP1delta\\_\\_CCL15'\n",
       "\\item 'cytokine\\_0\\_RANTES\\_\\_CCL5'\n",
       "\\item 'cytokine\\_0\\_SCF\\_\\_KITLG'\n",
       "\\item 'cytokine\\_0\\_SDF1alpha\\_beta\\_\\_CXCL12'\n",
       "\\item 'cytokine\\_0\\_TARC\\_\\_CCL17'\n",
       "\\item 'cytokine\\_0\\_TNFalpha\\_\\_TNF'\n",
       "\\item 'cytokine\\_0\\_TNFbeta\\_\\_LTA'\n",
       "\\item 'cytokine\\_0\\_VEGFA\\_\\_VEGFA'\n",
       "\\item 'cytokine\\_0\\_X6CKine\\_\\_CCL21'\n",
       "\\item 'neutrophil\\_0\\_ADM'\n",
       "\\item 'neutrophil\\_0\\_FBRS'\n",
       "\\item 'neutrophil\\_0\\_GRN'\n",
       "\\item 'neutrophil\\_0\\_HMGB1'\n",
       "\\item 'neutrophil\\_0\\_HMGB2'\n",
       "\\item 'neutrophil\\_0\\_IL16'\n",
       "\\item 'neutrophil\\_0\\_MMP9'\n",
       "\\item 'neutrophil\\_0\\_PGLYRP1'\n",
       "\\item 'neutrophil\\_0\\_RTN4'\n",
       "\\item 'neutrophil\\_0\\_SECTM1'\n",
       "\\item 'neutrophil\\_0\\_TGFB1'\n",
       "\\item 'neutrophil\\_0\\_TIMP1'\n",
       "\\item 'neutrophil\\_0\\_TNFSF13B'\n",
       "\\item 'proteomics\\_0\\_A2M\\_P01023'\n",
       "\\item 'proteomics\\_0\\_ADIPOQ\\_Q15848'\n",
       "\\item 'proteomics\\_0\\_AGT\\_P01019'\n",
       "\\item 'proteomics\\_0\\_APOA1\\_P02647'\n",
       "\\item 'proteomics\\_0\\_APOB\\_P04114'\n",
       "\\item 'proteomics\\_0\\_APOE\\_P02649'\n",
       "\\item 'proteomics\\_0\\_C3\\_M0R1Q1'\n",
       "\\item 'proteomics\\_0\\_C3\\_P01024'\n",
       "\\item 'proteomics\\_0\\_C5\\_P01031'\n",
       "\\item 'proteomics\\_0\\_CAMP\\_J3KNB4'\n",
       "\\item 'proteomics\\_0\\_CDH5\\_P33151'\n",
       "\\item 'proteomics\\_0\\_CEACAM1\\_P13688'\n",
       "\\item 'proteomics\\_0\\_CFH\\_P08603'\n",
       "\\item 'proteomics\\_0\\_COL18A1\\_P39060'\n",
       "\\item 'proteomics\\_0\\_COMP\\_G3XAP6'\n",
       "\\item 'proteomics\\_0\\_DSG2\\_Q14126'\n",
       "\\item 'proteomics\\_0\\_F2\\_P00734'\n",
       "\\item 'proteomics\\_0\\_F9\\_P00740'\n",
       "\\item 'proteomics\\_0\\_FGA\\_P02671'\n",
       "\\item 'proteomics\\_0\\_FN1\\_H0Y4K8'\n",
       "\\item 'proteomics\\_0\\_FN1\\_P02751'\n",
       "\\item 'proteomics\\_0\\_GSTP1\\_P09211'\n",
       "\\item 'proteomics\\_0\\_HSP90B1\\_P14625'\n",
       "\\item 'proteomics\\_0\\_HSPG2\\_P98160'\n",
       "\\item 'proteomics\\_0\\_ICAM1\\_P05362'\n",
       "\\item 'proteomics\\_0\\_ICAM2\\_J3QQR8'\n",
       "\\item 'proteomics\\_0\\_IGF2\\_P01344'\n",
       "\\item 'proteomics\\_0\\_KNG1\\_P01042'\n",
       "\\item 'proteomics\\_0\\_KNG1\\_P01042.2'\n",
       "\\item 'proteomics\\_0\\_LTF\\_E7EQB2'\n",
       "\\item 'proteomics\\_0\\_MMP9\\_P14780'\n",
       "\\item 'proteomics\\_0\\_NID1\\_P14543'\n",
       "\\item 'proteomics\\_0\\_PF4\\_P02776'\n",
       "\\item 'proteomics\\_0\\_PLG\\_P00747'\n",
       "\\item 'proteomics\\_0\\_PPBP\\_P02775'\n",
       "\\item 'proteomics\\_0\\_SAA1\\_P0DJI8'\n",
       "\\item 'proteomics\\_0\\_SELL\\_P14151'\n",
       "\\item 'proteomics\\_0\\_SERPINA1\\_A0A024R6I7'\n",
       "\\item 'proteomics\\_0\\_SERPINA1\\_P01009'\n",
       "\\item 'proteomics\\_0\\_SERPINC1\\_P01008'\n",
       "\\item 'proteomics\\_0\\_SERPING1\\_P05155'\n",
       "\\item 'proteomics\\_0\\_TF\\_P02787'\n",
       "\\item 'proteomics\\_0\\_THBS1\\_P07996'\n",
       "\\item 'proteomics\\_0\\_TIMP1\\_P01033'\n",
       "\\item 'proteomics\\_0\\_TTR\\_P02766'\n",
       "\\item 'proteomics\\_0\\_VTN\\_P04004'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_CD2'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_MIF'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_SELL'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_CCL4'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_CCL5'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_CD2'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_GZMB'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_CD2'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_SELL'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_CD2'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_SELL'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_IL1B'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_LGALS3'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_NAMPT'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_PECAM1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_PSAP'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_RTN4'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_SERPINA1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_TIMP1'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_GRN'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_NAMPT'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_PSAP'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_RTN4'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_TIMP1'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_CD2'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_SELL'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_CCL4'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_CCL5'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_GZMB'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_ADAM17'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_AIMP1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_ALOX5AP'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_CCL5'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_COPA'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_CXCL2'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_GRN'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_HDGF'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_ICAM1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_IL1B'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_LGALS3'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_LRPAP1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_MIF'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_PECAM1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_PSAP'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_PSEN1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_RTN4'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_SERPINA1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_TIMP1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_TNFSF10'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_TNFSF13B'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_CD2'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_SELL'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_ADAM17'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_CXCL2'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_GRN'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_ICAM1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_IL1B'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_LGALS3'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_LRPAP1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_PECAM1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_PSAP'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_PSEN1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_RTN4'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_SERPINA1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_THBS1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_TIMP1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_TNFSF10'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_TNFSF13B'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_GRN'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_IL1B'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_LGALS3'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_LRPAP1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_PECAM1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_PSAP'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_RTN4'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_SERPINA1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_TIMP1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_TNFSF10'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_TNFSF13B'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell\\_\\_SELL'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_CXCL16'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_GRN'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_IL1B'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_LGALS3'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_NAMPT'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_PECAM1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_PSAP'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_RTN4'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_SERPINA1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_TIMP1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_TNFSF10'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_TNFSF13B'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. cytokine_0_BCA1__CXCL13\n",
       "2. cytokine_0_CTACK__CCL27\n",
       "3. cytokine_0_EGF__EGF\n",
       "4. cytokine_0_ENA78__CXCL5\n",
       "5. cytokine_0_Eotaxin3__CCL26\n",
       "6. cytokine_0_FGF2__FGF2\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'cytokine_0_BCA1__CXCL13'\n",
       "2. 'cytokine_0_CTACK__CCL27'\n",
       "3. 'cytokine_0_EGF__EGF'\n",
       "4. 'cytokine_0_ENA78__CXCL5'\n",
       "5. 'cytokine_0_Eotaxin3__CCL26'\n",
       "6. 'cytokine_0_FGF2__FGF2'\n",
       "7. 'cytokine_0_Fractalkine__CX3CL1'\n",
       "8. 'cytokine_0_GCSF__CSF3'\n",
       "9. 'cytokine_0_GMCSF__CSF2'\n",
       "10. 'cytokine_0_I309__CCL1'\n",
       "11. 'cytokine_0_IFNalpha2__IFNA2'\n",
       "12. 'cytokine_0_IFNgamma__IFNG'\n",
       "13. 'cytokine_0_IL10__IL10'\n",
       "14. 'cytokine_0_IL12p70__IL12A'\n",
       "15. 'cytokine_0_IL13__IL13'\n",
       "16. 'cytokine_0_IL15__IL15'\n",
       "17. 'cytokine_0_IL17A__IL17A'\n",
       "18. 'cytokine_0_IL17E_IL25__IL25'\n",
       "19. 'cytokine_0_IL17F__IL17F'\n",
       "20. 'cytokine_0_IL1alpha__IL1A'\n",
       "21. 'cytokine_0_IL1beta__IL1B'\n",
       "22. 'cytokine_0_IL1RA__IL1RN'\n",
       "23. 'cytokine_0_IL2__IL2'\n",
       "24. 'cytokine_0_IL22__IL22'\n",
       "25. 'cytokine_0_IL27__IL27'\n",
       "26. 'cytokine_0_IL4__IL4'\n",
       "27. 'cytokine_0_IL5__IL5'\n",
       "28. 'cytokine_0_IL6__IL6'\n",
       "29. 'cytokine_0_IL7__IL7'\n",
       "30. 'cytokine_0_IL9__IL9'\n",
       "31. 'cytokine_0_IP10__CXCL10'\n",
       "32. 'cytokine_0_LIF__LIF'\n",
       "33. 'cytokine_0_MCP1__CCL2'\n",
       "34. 'cytokine_0_MCP2__CCL8'\n",
       "35. 'cytokine_0_MCP3__CCL7'\n",
       "36. 'cytokine_0_MCP4__CCL13'\n",
       "37. 'cytokine_0_MCSF__CSF1'\n",
       "38. 'cytokine_0_MDC__CCL22'\n",
       "39. 'cytokine_0_MIG_CXCL9__CXCL9'\n",
       "40. 'cytokine_0_MIP1alpha__CCL3'\n",
       "41. 'cytokine_0_MIP1beta__CCL4'\n",
       "42. 'cytokine_0_MIP1delta__CCL15'\n",
       "43. 'cytokine_0_RANTES__CCL5'\n",
       "44. 'cytokine_0_SCF__KITLG'\n",
       "45. 'cytokine_0_SDF1alpha_beta__CXCL12'\n",
       "46. 'cytokine_0_TARC__CCL17'\n",
       "47. 'cytokine_0_TNFalpha__TNF'\n",
       "48. 'cytokine_0_TNFbeta__LTA'\n",
       "49. 'cytokine_0_VEGFA__VEGFA'\n",
       "50. 'cytokine_0_X6CKine__CCL21'\n",
       "51. 'neutrophil_0_ADM'\n",
       "52. 'neutrophil_0_FBRS'\n",
       "53. 'neutrophil_0_GRN'\n",
       "54. 'neutrophil_0_HMGB1'\n",
       "55. 'neutrophil_0_HMGB2'\n",
       "56. 'neutrophil_0_IL16'\n",
       "57. 'neutrophil_0_MMP9'\n",
       "58. 'neutrophil_0_PGLYRP1'\n",
       "59. 'neutrophil_0_RTN4'\n",
       "60. 'neutrophil_0_SECTM1'\n",
       "61. 'neutrophil_0_TGFB1'\n",
       "62. 'neutrophil_0_TIMP1'\n",
       "63. 'neutrophil_0_TNFSF13B'\n",
       "64. 'proteomics_0_A2M_P01023'\n",
       "65. 'proteomics_0_ADIPOQ_Q15848'\n",
       "66. 'proteomics_0_AGT_P01019'\n",
       "67. 'proteomics_0_APOA1_P02647'\n",
       "68. 'proteomics_0_APOB_P04114'\n",
       "69. 'proteomics_0_APOE_P02649'\n",
       "70. 'proteomics_0_C3_M0R1Q1'\n",
       "71. 'proteomics_0_C3_P01024'\n",
       "72. 'proteomics_0_C5_P01031'\n",
       "73. 'proteomics_0_CAMP_J3KNB4'\n",
       "74. 'proteomics_0_CDH5_P33151'\n",
       "75. 'proteomics_0_CEACAM1_P13688'\n",
       "76. 'proteomics_0_CFH_P08603'\n",
       "77. 'proteomics_0_COL18A1_P39060'\n",
       "78. 'proteomics_0_COMP_G3XAP6'\n",
       "79. 'proteomics_0_DSG2_Q14126'\n",
       "80. 'proteomics_0_F2_P00734'\n",
       "81. 'proteomics_0_F9_P00740'\n",
       "82. 'proteomics_0_FGA_P02671'\n",
       "83. 'proteomics_0_FN1_H0Y4K8'\n",
       "84. 'proteomics_0_FN1_P02751'\n",
       "85. 'proteomics_0_GSTP1_P09211'\n",
       "86. 'proteomics_0_HSP90B1_P14625'\n",
       "87. 'proteomics_0_HSPG2_P98160'\n",
       "88. 'proteomics_0_ICAM1_P05362'\n",
       "89. 'proteomics_0_ICAM2_J3QQR8'\n",
       "90. 'proteomics_0_IGF2_P01344'\n",
       "91. 'proteomics_0_KNG1_P01042'\n",
       "92. 'proteomics_0_KNG1_P01042.2'\n",
       "93. 'proteomics_0_LTF_E7EQB2'\n",
       "94. 'proteomics_0_MMP9_P14780'\n",
       "95. 'proteomics_0_NID1_P14543'\n",
       "96. 'proteomics_0_PF4_P02776'\n",
       "97. 'proteomics_0_PLG_P00747'\n",
       "98. 'proteomics_0_PPBP_P02775'\n",
       "99. 'proteomics_0_SAA1_P0DJI8'\n",
       "100. 'proteomics_0_SELL_P14151'\n",
       "101. 'proteomics_0_SERPINA1_A0A024R6I7'\n",
       "102. 'proteomics_0_SERPINA1_P01009'\n",
       "103. 'proteomics_0_SERPINC1_P01008'\n",
       "104. 'proteomics_0_SERPING1_P05155'\n",
       "105. 'proteomics_0_TF_P02787'\n",
       "106. 'proteomics_0_THBS1_P07996'\n",
       "107. 'proteomics_0_TIMP1_P01033'\n",
       "108. 'proteomics_0_TTR_P02766'\n",
       "109. 'proteomics_0_VTN_P04004'\n",
       "110. 'single_cell_0_X0_T.cell.CD4__ANXA1'\n",
       "111. 'single_cell_0_X0_T.cell.CD4__ARF1'\n",
       "112. 'single_cell_0_X0_T.cell.CD4__CALM1'\n",
       "113. 'single_cell_0_X0_T.cell.CD4__CALR'\n",
       "114. 'single_cell_0_X0_T.cell.CD4__CD2'\n",
       "115. 'single_cell_0_X0_T.cell.CD4__HMGB1'\n",
       "116. 'single_cell_0_X0_T.cell.CD4__HMGB2'\n",
       "117. 'single_cell_0_X0_T.cell.CD4__HSP90B1'\n",
       "118. 'single_cell_0_X0_T.cell.CD4__ICAM3'\n",
       "119. 'single_cell_0_X0_T.cell.CD4__LTB'\n",
       "120. 'single_cell_0_X0_T.cell.CD4__MIF'\n",
       "121. 'single_cell_0_X0_T.cell.CD4__PTPRC'\n",
       "122. 'single_cell_0_X0_T.cell.CD4__SELL'\n",
       "123. 'single_cell_0_X1_T.cell.CD8__ANXA1'\n",
       "124. 'single_cell_0_X1_T.cell.CD8__ARF1'\n",
       "125. 'single_cell_0_X1_T.cell.CD8__CALM1'\n",
       "126. 'single_cell_0_X1_T.cell.CD8__CALR'\n",
       "127. 'single_cell_0_X1_T.cell.CD8__CCL4'\n",
       "128. 'single_cell_0_X1_T.cell.CD8__CCL5'\n",
       "129. 'single_cell_0_X1_T.cell.CD8__CD2'\n",
       "130. 'single_cell_0_X1_T.cell.CD8__GSTP1'\n",
       "131. 'single_cell_0_X1_T.cell.CD8__GZMB'\n",
       "132. 'single_cell_0_X1_T.cell.CD8__HMGB1'\n",
       "133. 'single_cell_0_X1_T.cell.CD8__HMGB2'\n",
       "134. 'single_cell_0_X1_T.cell.CD8__HSP90B1'\n",
       "135. 'single_cell_0_X1_T.cell.CD8__ICAM3'\n",
       "136. 'single_cell_0_X1_T.cell.CD8__LTB'\n",
       "137. 'single_cell_0_X1_T.cell.CD8__PTPRC'\n",
       "138. 'single_cell_0_X1_T.cell.CD8__TGFB1'\n",
       "139. 'single_cell_0_X10_B.cell__ANXA1'\n",
       "140. 'single_cell_0_X10_B.cell__ARF1'\n",
       "141. 'single_cell_0_X10_B.cell__CALM1'\n",
       "142. 'single_cell_0_X10_B.cell__CALR'\n",
       "143. 'single_cell_0_X10_B.cell__CD2'\n",
       "144. 'single_cell_0_X10_B.cell__GSTP1'\n",
       "145. 'single_cell_0_X10_B.cell__HMGB1'\n",
       "146. 'single_cell_0_X10_B.cell__HMGB2'\n",
       "147. 'single_cell_0_X10_B.cell__HSP90B1'\n",
       "148. 'single_cell_0_X10_B.cell__ICAM3'\n",
       "149. 'single_cell_0_X10_B.cell__LTB'\n",
       "150. 'single_cell_0_X10_B.cell__PTPRC'\n",
       "151. 'single_cell_0_X10_B.cell__SELL'\n",
       "152. 'single_cell_0_X10_B.cell__TGFB1'\n",
       "153. 'single_cell_0_X11_T.cell.CD4__ANXA1'\n",
       "154. 'single_cell_0_X11_T.cell.CD4__ARF1'\n",
       "155. 'single_cell_0_X11_T.cell.CD4__CALM1'\n",
       "156. 'single_cell_0_X11_T.cell.CD4__CALR'\n",
       "157. 'single_cell_0_X11_T.cell.CD4__CD2'\n",
       "158. 'single_cell_0_X11_T.cell.CD4__HMGB1'\n",
       "159. 'single_cell_0_X11_T.cell.CD4__HMGB2'\n",
       "160. 'single_cell_0_X11_T.cell.CD4__HSP90B1'\n",
       "161. 'single_cell_0_X11_T.cell.CD4__ICAM3'\n",
       "162. 'single_cell_0_X11_T.cell.CD4__LTB'\n",
       "163. 'single_cell_0_X11_T.cell.CD4__PTPRC'\n",
       "164. 'single_cell_0_X11_T.cell.CD4__SELL'\n",
       "165. 'single_cell_0_X11_T.cell.CD4__TGFB1'\n",
       "166. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__ANXA1'\n",
       "167. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__ARF1'\n",
       "168. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__CALM1'\n",
       "169. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__CALR'\n",
       "170. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__GSTP1'\n",
       "171. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__HMGB1'\n",
       "172. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__HMGB2'\n",
       "173. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__HSP90B1'\n",
       "174. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__ICAM3'\n",
       "175. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__IL1B'\n",
       "176. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__LGALS3'\n",
       "177. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__LTB'\n",
       "178. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__NAMPT'\n",
       "179. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__PECAM1'\n",
       "180. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__PSAP'\n",
       "181. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__PTPRC'\n",
       "182. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__RTN4'\n",
       "183. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__SERPINA1'\n",
       "184. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__TGFB1'\n",
       "185. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__TIMP1'\n",
       "186. 'single_cell_0_X13_Dendritic__ANXA1'\n",
       "187. 'single_cell_0_X13_Dendritic__CALM1'\n",
       "188. 'single_cell_0_X13_Dendritic__CALR'\n",
       "189. 'single_cell_0_X13_Dendritic__GRN'\n",
       "190. 'single_cell_0_X13_Dendritic__GSTP1'\n",
       "191. 'single_cell_0_X13_Dendritic__HMGB1'\n",
       "192. 'single_cell_0_X13_Dendritic__HSP90B1'\n",
       "193. 'single_cell_0_X13_Dendritic__NAMPT'\n",
       "194. 'single_cell_0_X13_Dendritic__PSAP'\n",
       "195. 'single_cell_0_X13_Dendritic__PTPRC'\n",
       "196. 'single_cell_0_X13_Dendritic__RTN4'\n",
       "197. 'single_cell_0_X13_Dendritic__TIMP1'\n",
       "198. 'single_cell_0_X2_T.cell.CD4__ANXA1'\n",
       "199. 'single_cell_0_X2_T.cell.CD4__ARF1'\n",
       "200. 'single_cell_0_X2_T.cell.CD4__CALM1'\n",
       "201. 'single_cell_0_X2_T.cell.CD4__CALR'\n",
       "202. 'single_cell_0_X2_T.cell.CD4__CD2'\n",
       "203. 'single_cell_0_X2_T.cell.CD4__HMGB1'\n",
       "204. 'single_cell_0_X2_T.cell.CD4__HMGB2'\n",
       "205. 'single_cell_0_X2_T.cell.CD4__HSP90B1'\n",
       "206. 'single_cell_0_X2_T.cell.CD4__LTB'\n",
       "207. 'single_cell_0_X2_T.cell.CD4__PTPRC'\n",
       "208. 'single_cell_0_X2_T.cell.CD4__SELL'\n",
       "209. 'single_cell_0_X3_NK__ANXA1'\n",
       "210. 'single_cell_0_X3_NK__ARF1'\n",
       "211. 'single_cell_0_X3_NK__CALM1'\n",
       "212. 'single_cell_0_X3_NK__CALR'\n",
       "213. 'single_cell_0_X3_NK__CCL4'\n",
       "214. 'single_cell_0_X3_NK__CCL5'\n",
       "215. 'single_cell_0_X3_NK__GSTP1'\n",
       "216. 'single_cell_0_X3_NK__GZMB'\n",
       "217. 'single_cell_0_X3_NK__HMGB1'\n",
       "218. 'single_cell_0_X3_NK__HMGB2'\n",
       "219. 'single_cell_0_X3_NK__HSP90B1'\n",
       "220. 'single_cell_0_X3_NK__PTPRC'\n",
       "221. 'single_cell_0_X3_NK__TGFB1'\n",
       "222. 'single_cell_0_X4_Monocytes...CD14__ADAM17'\n",
       "223. 'single_cell_0_X4_Monocytes...CD14__AIMP1'\n",
       "224. 'single_cell_0_X4_Monocytes...CD14__ALOX5AP'\n",
       "225. 'single_cell_0_X4_Monocytes...CD14__ANXA1'\n",
       "226. 'single_cell_0_X4_Monocytes...CD14__ARF1'\n",
       "227. 'single_cell_0_X4_Monocytes...CD14__CALM1'\n",
       "228. 'single_cell_0_X4_Monocytes...CD14__CALR'\n",
       "229. 'single_cell_0_X4_Monocytes...CD14__CCL5'\n",
       "230. 'single_cell_0_X4_Monocytes...CD14__COPA'\n",
       "231. 'single_cell_0_X4_Monocytes...CD14__CXCL2'\n",
       "232. 'single_cell_0_X4_Monocytes...CD14__GRN'\n",
       "233. 'single_cell_0_X4_Monocytes...CD14__GSTP1'\n",
       "234. 'single_cell_0_X4_Monocytes...CD14__HDGF'\n",
       "235. 'single_cell_0_X4_Monocytes...CD14__HMGB1'\n",
       "236. 'single_cell_0_X4_Monocytes...CD14__HMGB2'\n",
       "237. 'single_cell_0_X4_Monocytes...CD14__HSP90B1'\n",
       "238. 'single_cell_0_X4_Monocytes...CD14__ICAM1'\n",
       "239. 'single_cell_0_X4_Monocytes...CD14__ICAM3'\n",
       "240. 'single_cell_0_X4_Monocytes...CD14__IL1B'\n",
       "241. 'single_cell_0_X4_Monocytes...CD14__LGALS3'\n",
       "242. 'single_cell_0_X4_Monocytes...CD14__LRPAP1'\n",
       "243. 'single_cell_0_X4_Monocytes...CD14__LTB'\n",
       "244. 'single_cell_0_X4_Monocytes...CD14__MIF'\n",
       "245. 'single_cell_0_X4_Monocytes...CD14__PECAM1'\n",
       "246. 'single_cell_0_X4_Monocytes...CD14__PSAP'\n",
       "247. 'single_cell_0_X4_Monocytes...CD14__PSEN1'\n",
       "248. 'single_cell_0_X4_Monocytes...CD14__PTPRC'\n",
       "249. 'single_cell_0_X4_Monocytes...CD14__RTN4'\n",
       "250. 'single_cell_0_X4_Monocytes...CD14__SERPINA1'\n",
       "251. 'single_cell_0_X4_Monocytes...CD14__TGFB1'\n",
       "252. 'single_cell_0_X4_Monocytes...CD14__TIMP1'\n",
       "253. 'single_cell_0_X4_Monocytes...CD14__TNFSF10'\n",
       "254. 'single_cell_0_X4_Monocytes...CD14__TNFSF13B'\n",
       "255. 'single_cell_0_X5_T.cell.CD4__ANXA1'\n",
       "256. 'single_cell_0_X5_T.cell.CD4__CALM1'\n",
       "257. 'single_cell_0_X5_T.cell.CD4__CD2'\n",
       "258. 'single_cell_0_X5_T.cell.CD4__HMGB1'\n",
       "259. 'single_cell_0_X5_T.cell.CD4__HMGB2'\n",
       "260. 'single_cell_0_X5_T.cell.CD4__HSP90B1'\n",
       "261. 'single_cell_0_X5_T.cell.CD4__ICAM3'\n",
       "262. 'single_cell_0_X5_T.cell.CD4__LTB'\n",
       "263. 'single_cell_0_X5_T.cell.CD4__PTPRC'\n",
       "264. 'single_cell_0_X5_T.cell.CD4__SELL'\n",
       "265. 'single_cell_0_X6_Monocytes...CD14__ADAM17'\n",
       "266. 'single_cell_0_X6_Monocytes...CD14__ANXA1'\n",
       "267. 'single_cell_0_X6_Monocytes...CD14__ARF1'\n",
       "268. 'single_cell_0_X6_Monocytes...CD14__CALM1'\n",
       "269. 'single_cell_0_X6_Monocytes...CD14__CALR'\n",
       "270. 'single_cell_0_X6_Monocytes...CD14__CXCL2'\n",
       "271. 'single_cell_0_X6_Monocytes...CD14__GRN'\n",
       "272. 'single_cell_0_X6_Monocytes...CD14__GSTP1'\n",
       "273. 'single_cell_0_X6_Monocytes...CD14__HMGB1'\n",
       "274. 'single_cell_0_X6_Monocytes...CD14__HMGB2'\n",
       "275. 'single_cell_0_X6_Monocytes...CD14__HSP90B1'\n",
       "276. 'single_cell_0_X6_Monocytes...CD14__ICAM1'\n",
       "277. 'single_cell_0_X6_Monocytes...CD14__ICAM3'\n",
       "278. 'single_cell_0_X6_Monocytes...CD14__IL1B'\n",
       "279. 'single_cell_0_X6_Monocytes...CD14__LGALS3'\n",
       "280. 'single_cell_0_X6_Monocytes...CD14__LRPAP1'\n",
       "281. 'single_cell_0_X6_Monocytes...CD14__PECAM1'\n",
       "282. 'single_cell_0_X6_Monocytes...CD14__PSAP'\n",
       "283. 'single_cell_0_X6_Monocytes...CD14__PSEN1'\n",
       "284. 'single_cell_0_X6_Monocytes...CD14__PTPRC'\n",
       "285. 'single_cell_0_X6_Monocytes...CD14__RTN4'\n",
       "286. 'single_cell_0_X6_Monocytes...CD14__SERPINA1'\n",
       "287. 'single_cell_0_X6_Monocytes...CD14__TGFB1'\n",
       "288. 'single_cell_0_X6_Monocytes...CD14__THBS1'\n",
       "289. 'single_cell_0_X6_Monocytes...CD14__TIMP1'\n",
       "290. 'single_cell_0_X6_Monocytes...CD14__TNFSF10'\n",
       "291. 'single_cell_0_X6_Monocytes...CD14__TNFSF13B'\n",
       "292. 'single_cell_0_X7_Monocytes...CD14__ANXA1'\n",
       "293. 'single_cell_0_X7_Monocytes...CD14__ARF1'\n",
       "294. 'single_cell_0_X7_Monocytes...CD14__CALM1'\n",
       "295. 'single_cell_0_X7_Monocytes...CD14__CALR'\n",
       "296. 'single_cell_0_X7_Monocytes...CD14__GRN'\n",
       "297. 'single_cell_0_X7_Monocytes...CD14__GSTP1'\n",
       "298. 'single_cell_0_X7_Monocytes...CD14__HMGB1'\n",
       "299. 'single_cell_0_X7_Monocytes...CD14__HMGB2'\n",
       "300. 'single_cell_0_X7_Monocytes...CD14__HSP90B1'\n",
       "301. 'single_cell_0_X7_Monocytes...CD14__ICAM3'\n",
       "302. 'single_cell_0_X7_Monocytes...CD14__IL1B'\n",
       "303. 'single_cell_0_X7_Monocytes...CD14__LGALS3'\n",
       "304. 'single_cell_0_X7_Monocytes...CD14__LRPAP1'\n",
       "305. 'single_cell_0_X7_Monocytes...CD14__PECAM1'\n",
       "306. 'single_cell_0_X7_Monocytes...CD14__PSAP'\n",
       "307. 'single_cell_0_X7_Monocytes...CD14__PTPRC'\n",
       "308. 'single_cell_0_X7_Monocytes...CD14__RTN4'\n",
       "309. 'single_cell_0_X7_Monocytes...CD14__SERPINA1'\n",
       "310. 'single_cell_0_X7_Monocytes...CD14__TGFB1'\n",
       "311. 'single_cell_0_X7_Monocytes...CD14__TIMP1'\n",
       "312. 'single_cell_0_X7_Monocytes...CD14__TNFSF10'\n",
       "313. 'single_cell_0_X7_Monocytes...CD14__TNFSF13B'\n",
       "314. 'single_cell_0_X8_B.cell__CALM1'\n",
       "315. 'single_cell_0_X8_B.cell__CALR'\n",
       "316. 'single_cell_0_X8_B.cell__HMGB1'\n",
       "317. 'single_cell_0_X8_B.cell__HMGB2'\n",
       "318. 'single_cell_0_X8_B.cell__HSP90B1'\n",
       "319. 'single_cell_0_X8_B.cell__LTB'\n",
       "320. 'single_cell_0_X8_B.cell__PTPRC'\n",
       "321. 'single_cell_0_X8_B.cell__SELL'\n",
       "322. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__ANXA1'\n",
       "323. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__ARF1'\n",
       "324. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__CALM1'\n",
       "325. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__CALR'\n",
       "326. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__CXCL16'\n",
       "327. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__GRN'\n",
       "328. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__GSTP1'\n",
       "329. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__HMGB1'\n",
       "330. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__HMGB2'\n",
       "331. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__HSP90B1'\n",
       "332. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__ICAM3'\n",
       "333. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__IL1B'\n",
       "334. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__LGALS3'\n",
       "335. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__NAMPT'\n",
       "336. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__PECAM1'\n",
       "337. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__PSAP'\n",
       "338. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__PTPRC'\n",
       "339. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__RTN4'\n",
       "340. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__SERPINA1'\n",
       "341. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__TGFB1'\n",
       "342. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__TIMP1'\n",
       "343. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFSF10'\n",
       "344. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFSF13B'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] cytokine_0_BCA1__CXCL13    cytokine_0_CTACK__CCL27   \n",
       "[3] cytokine_0_EGF__EGF        cytokine_0_ENA78__CXCL5   \n",
       "[5] cytokine_0_Eotaxin3__CCL26 cytokine_0_FGF2__FGF2     \n",
       "344 Levels: cytokine_0_BCA1__CXCL13 ... single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFSF13B"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(sort(unique(ligand_target_correlations$Ligand)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 542,
   "id": "6782e731-c592-4e50-874a-25682a0037c1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'clinical_data'</li><li>'cytokine'</li><li>'neutrophil'</li><li>'proteomics'</li><li>'single_cell_0_X0_T.cell.CD4'</li><li>'single_cell_0_X1_T.cell.CD8'</li><li>'single_cell_0_X10_B.cell'</li><li>'single_cell_0_X11_T.cell.CD4'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A'</li><li>'single_cell_0_X13_Dendritic'</li><li>'single_cell_0_X2_T.cell.CD4'</li><li>'single_cell_0_X3_NK'</li><li>'single_cell_0_X4_Monocytes...CD14'</li><li>'single_cell_0_X5_T.cell.CD4'</li><li>'single_cell_0_X6_Monocytes...CD14'</li><li>'single_cell_0_X7_Monocytes...CD14'</li><li>'single_cell_0_X8_B.cell'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'clinical\\_data'\n",
       "\\item 'cytokine'\n",
       "\\item 'neutrophil'\n",
       "\\item 'proteomics'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'clinical_data'\n",
       "2. 'cytokine'\n",
       "3. 'neutrophil'\n",
       "4. 'proteomics'\n",
       "5. 'single_cell_0_X0_T.cell.CD4'\n",
       "6. 'single_cell_0_X1_T.cell.CD8'\n",
       "7. 'single_cell_0_X10_B.cell'\n",
       "8. 'single_cell_0_X11_T.cell.CD4'\n",
       "9. 'single_cell_0_X12_Monocytes...CD16_FCGR3A'\n",
       "10. 'single_cell_0_X13_Dendritic'\n",
       "11. 'single_cell_0_X2_T.cell.CD4'\n",
       "12. 'single_cell_0_X3_NK'\n",
       "13. 'single_cell_0_X4_Monocytes...CD14'\n",
       "14. 'single_cell_0_X5_T.cell.CD4'\n",
       "15. 'single_cell_0_X6_Monocytes...CD14'\n",
       "16. 'single_cell_0_X7_Monocytes...CD14'\n",
       "17. 'single_cell_0_X8_B.cell'\n",
       "18. 'single_cell_0_X9_Monocytes...CD16_FCGR3A'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"clinical_data\"                            \n",
       " [2] \"cytokine\"                                 \n",
       " [3] \"neutrophil\"                               \n",
       " [4] \"proteomics\"                               \n",
       " [5] \"single_cell_0_X0_T.cell.CD4\"              \n",
       " [6] \"single_cell_0_X1_T.cell.CD8\"              \n",
       " [7] \"single_cell_0_X10_B.cell\"                 \n",
       " [8] \"single_cell_0_X11_T.cell.CD4\"             \n",
       " [9] \"single_cell_0_X12_Monocytes...CD16_FCGR3A\"\n",
       "[10] \"single_cell_0_X13_Dendritic\"              \n",
       "[11] \"single_cell_0_X2_T.cell.CD4\"              \n",
       "[12] \"single_cell_0_X3_NK\"                      \n",
       "[13] \"single_cell_0_X4_Monocytes...CD14\"        \n",
       "[14] \"single_cell_0_X5_T.cell.CD4\"              \n",
       "[15] \"single_cell_0_X6_Monocytes...CD14\"        \n",
       "[16] \"single_cell_0_X7_Monocytes...CD14\"        \n",
       "[17] \"single_cell_0_X8_B.cell\"                  \n",
       "[18] \"single_cell_0_X9_Monocytes...CD16_FCGR3A\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(ligand_target_correlations$Target_Type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 543,
   "id": "fcd0583b-7e85-4d78-ab3c-fce11cf7ba5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$cluster = str_replace(ligand_target_correlations$Target_Type, 'single_cell_0_', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 544,
   "id": "3cf25a82-61f2-4a07-8d3c-a547fbd16226",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'clinical_data'</li><li>'cytokine'</li><li>'neutrophil'</li><li>'proteomics'</li><li>'X0_T.cell.CD4'</li><li>'X1_T.cell.CD8'</li><li>'X10_B.cell'</li><li>'X11_T.cell.CD4'</li><li>'X12_Monocytes...CD16_FCGR3A'</li><li>'X13_Dendritic'</li><li>'X2_T.cell.CD4'</li><li>'X3_NK'</li><li>'X4_Monocytes...CD14'</li><li>'X5_T.cell.CD4'</li><li>'X6_Monocytes...CD14'</li><li>'X7_Monocytes...CD14'</li><li>'X8_B.cell'</li><li>'X9_Monocytes...CD16_FCGR3A'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'clinical\\_data'\n",
       "\\item 'cytokine'\n",
       "\\item 'neutrophil'\n",
       "\\item 'proteomics'\n",
       "\\item 'X0\\_T.cell.CD4'\n",
       "\\item 'X1\\_T.cell.CD8'\n",
       "\\item 'X10\\_B.cell'\n",
       "\\item 'X11\\_T.cell.CD4'\n",
       "\\item 'X12\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'X13\\_Dendritic'\n",
       "\\item 'X2\\_T.cell.CD4'\n",
       "\\item 'X3\\_NK'\n",
       "\\item 'X4\\_Monocytes...CD14'\n",
       "\\item 'X5\\_T.cell.CD4'\n",
       "\\item 'X6\\_Monocytes...CD14'\n",
       "\\item 'X7\\_Monocytes...CD14'\n",
       "\\item 'X8\\_B.cell'\n",
       "\\item 'X9\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'clinical_data'\n",
       "2. 'cytokine'\n",
       "3. 'neutrophil'\n",
       "4. 'proteomics'\n",
       "5. 'X0_T.cell.CD4'\n",
       "6. 'X1_T.cell.CD8'\n",
       "7. 'X10_B.cell'\n",
       "8. 'X11_T.cell.CD4'\n",
       "9. 'X12_Monocytes...CD16_FCGR3A'\n",
       "10. 'X13_Dendritic'\n",
       "11. 'X2_T.cell.CD4'\n",
       "12. 'X3_NK'\n",
       "13. 'X4_Monocytes...CD14'\n",
       "14. 'X5_T.cell.CD4'\n",
       "15. 'X6_Monocytes...CD14'\n",
       "16. 'X7_Monocytes...CD14'\n",
       "17. 'X8_B.cell'\n",
       "18. 'X9_Monocytes...CD16_FCGR3A'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"clinical_data\"               \"cytokine\"                   \n",
       " [3] \"neutrophil\"                  \"proteomics\"                 \n",
       " [5] \"X0_T.cell.CD4\"               \"X1_T.cell.CD8\"              \n",
       " [7] \"X10_B.cell\"                  \"X11_T.cell.CD4\"             \n",
       " [9] \"X12_Monocytes...CD16_FCGR3A\" \"X13_Dendritic\"              \n",
       "[11] \"X2_T.cell.CD4\"               \"X3_NK\"                      \n",
       "[13] \"X4_Monocytes...CD14\"         \"X5_T.cell.CD4\"              \n",
       "[15] \"X6_Monocytes...CD14\"         \"X7_Monocytes...CD14\"        \n",
       "[17] \"X8_B.cell\"                   \"X9_Monocytes...CD16_FCGR3A\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(ligand_target_correlations$cluster)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 545,
   "id": "4e5c2d62-0ac1-4b8c-b360-dc03df6aea14",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Ligand</th><th scope=col>Target</th><th scope=col>correlation</th><th scope=col>Target_Gene</th><th scope=col>Ligand_Gene</th><th scope=col>Target_Type</th><th scope=col>Ligand_Type</th><th scope=col>cluster</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>cytokine_0_BCA1__CXCL13</td><td>clinical_data_0_CK</td><td> 0.04568048</td><td>CK</td><td>CXCL13</td><td>clinical_data</td><td>cytokine</td><td>clinical_data</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>cytokine_0_CTACK__CCL27</td><td>clinical_data_0_CK</td><td>-0.19761446</td><td>CK</td><td>CCL27 </td><td>clinical_data</td><td>cytokine</td><td>clinical_data</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Ligand & Target & correlation & Target\\_Gene & Ligand\\_Gene & Target\\_Type & Ligand\\_Type & cluster\\\\\n",
       "  & <fct> & <fct> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & cytokine\\_0\\_BCA1\\_\\_CXCL13 & clinical\\_data\\_0\\_CK &  0.04568048 & CK & CXCL13 & clinical\\_data & cytokine & clinical\\_data\\\\\n",
       "\t2 & cytokine\\_0\\_CTACK\\_\\_CCL27 & clinical\\_data\\_0\\_CK & -0.19761446 & CK & CCL27  & clinical\\_data & cytokine & clinical\\_data\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 8\n",
       "\n",
       "| <!--/--> | Ligand &lt;fct&gt; | Target &lt;fct&gt; | correlation &lt;dbl&gt; | Target_Gene &lt;chr&gt; | Ligand_Gene &lt;chr&gt; | Target_Type &lt;chr&gt; | Ligand_Type &lt;chr&gt; | cluster &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | cytokine_0_BCA1__CXCL13 | clinical_data_0_CK |  0.04568048 | CK | CXCL13 | clinical_data | cytokine | clinical_data |\n",
       "| 2 | cytokine_0_CTACK__CCL27 | clinical_data_0_CK | -0.19761446 | CK | CCL27  | clinical_data | cytokine | clinical_data |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand                  Target             correlation Target_Gene\n",
       "1 cytokine_0_BCA1__CXCL13 clinical_data_0_CK  0.04568048 CK         \n",
       "2 cytokine_0_CTACK__CCL27 clinical_data_0_CK -0.19761446 CK         \n",
       "  Ligand_Gene Target_Type   Ligand_Type cluster      \n",
       "1 CXCL13      clinical_data cytokine    clinical_data\n",
       "2 CCL27       clinical_data cytokine    clinical_data"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_target_correlations,2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "34fd691d-2cd4-40ec-94ae-f69f4ac37db7",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Filter on ligands with matching receptor in target"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 546,
   "id": "df075c80-9948-425e-8729-52d9441124dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Ligand</th><th scope=col>cluster_R</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>APOA1</td><td>X4_Monocytes...CD14</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>APOA1</td><td>X6_Monocytes...CD14</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Ligand & cluster\\_R\\\\\n",
       "  & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & APOA1 & X4\\_Monocytes...CD14\\\\\n",
       "\t2 & APOA1 & X6\\_Monocytes...CD14\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 2\n",
       "\n",
       "| <!--/--> | Ligand &lt;chr&gt; | cluster_R &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| 1 | APOA1 | X4_Monocytes...CD14 |\n",
       "| 2 | APOA1 | X6_Monocytes...CD14 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand cluster_R          \n",
       "1 APOA1  X4_Monocytes...CD14\n",
       "2 APOA1  X6_Monocytes...CD14"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(potential_ligand_target_types_comb,2) # dataset with available receptor for ligands in target types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 547,
   "id": "7a76f257-44b3-48cf-b136-139563f302be",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "4451016"
      ],
      "text/latex": [
       "4451016"
      ],
      "text/markdown": [
       "4451016"
      ],
      "text/plain": [
       "[1] 4451016"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_target_correlations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 548,
   "id": "9ad3e57a-784f-4e49-baea-3c9c058ea6e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "filter_matched_receptor = TRUE  # filter out ligands for which there is a receptor in the target type if true"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 549,
   "id": "d457c105-e8cd-4f72-b452-0c01a13febb4",
   "metadata": {},
   "outputs": [],
   "source": [
    "if(filter_matched_receptor == TRUE){\n",
    "    ### seperately add clinical, proteomics as target (as here no filtering based on expression is applied)\n",
    "    no_sc = ligand_target_correlations[ligand_target_correlations$Target_Type %in%  c('clinical_data', 'proteomics', 'neutrophil'),]\n",
    "    ### filter the ligands\n",
    "    ligand_target_correlations = merge(ligand_target_correlations, potential_ligand_target_types_comb[,c('Ligand', 'cluster_R')], by.x = c('Ligand_Gene', 'cluster'), by.y = c('Ligand', 'cluster_R'))\n",
    "    \n",
    "    ### add other omics data\n",
    "    ligand_target_correlations = rbind(ligand_target_correlations, no_sc)\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 550,
   "id": "6d24c6e9-3487-441c-b8c3-ccd5b55ccba8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "3522537"
      ],
      "text/latex": [
       "3522537"
      ],
      "text/markdown": [
       "3522537"
      ],
      "text/plain": [
       "[1] 3522537"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_target_correlations)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c3d3f22-e7e4-4719-bb86-f67014b2c53c",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Add regulatory potential"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 551,
   "id": "97fc1c00-8c1c-4b73-ae0e-c898a0ed3a65",
   "metadata": {},
   "outputs": [],
   "source": [
    "## add ligand target regulatory potential score from nichenet to the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 552,
   "id": "58e7451a-647d-47f7-aa51-a234494a6cdb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>regulatory_potential</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A1BG    </td><td>CXCL1</td><td>0.0003534343</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A1BG-AS1</td><td>CXCL1</td><td>0.0001650894</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & target & ligand & regulatory\\_potential\\\\\n",
       "  & <fct> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & A1BG     & CXCL1 & 0.0003534343\\\\\n",
       "\t2 & A1BG-AS1 & CXCL1 & 0.0001650894\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 3\n",
       "\n",
       "| <!--/--> | target &lt;fct&gt; | ligand &lt;fct&gt; | regulatory_potential &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | A1BG     | CXCL1 | 0.0003534343 |\n",
       "| 2 | A1BG-AS1 | CXCL1 | 0.0001650894 |\n",
       "\n"
      ],
      "text/plain": [
       "  target   ligand regulatory_potential\n",
       "1 A1BG     CXCL1  0.0003534343        \n",
       "2 A1BG-AS1 CXCL1  0.0001650894        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_target_matrix_long,2) # regulatory potential scores from nichenet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 553,
   "id": "319efddf-541e-4d57-9a96-163e018913f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Adjust target gene names for clinical data (to be included later on)\n",
    "ligand_target_correlations$Target_Gene[ligand_target_correlations$Target == 'clinical_data_0_CK'] = 'CKM'\n",
    "ligand_target_correlations$Target_Gene[ligand_target_correlations$Target == 'clinical_data_0_CK_MB'] = 'CKB'\n",
    "ligand_target_correlations$Target_Gene[ligand_target_correlations$Target == 'clinical_data_0_Troponin'] = 'TNNT2'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 554,
   "id": "7d5ebdbd-7480-4f48-9bac-f6a343d27d47",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Merge regulatory potential scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 555,
   "id": "42a3fc61-d2ed-42e2-bbc8-199264d61a23",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations = merge(ligand_target_correlations, ligand_target_matrix_long, by.x = c('Ligand_Gene', 'Target_Gene'), by.y = c('ligand', 'target'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 556,
   "id": "15ec68da-5550-4e2f-b0cc-062eeb2bf4b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "3314320"
      ],
      "text/latex": [
       "3314320"
      ],
      "text/markdown": [
       "3314320"
      ],
      "text/plain": [
       "[1] 3314320"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_target_correlations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 557,
   "id": "0c050d14-2bc5-484b-b014-5294bf3e65de",
   "metadata": {},
   "outputs": [],
   "source": [
    "## scale the regulatory potential score based on max value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 558,
   "id": "bc40ffa8-96b3-4af3-ba5e-6e22132ef518",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$regulatory_potential_scaled = ligand_target_correlations$regulatory_potential/ max(ligand_target_correlations$regulatory_potential)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6bbcfe81-c047-41ce-890c-c0752bc26ac0",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Adjustments to ligand-target correlations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 559,
   "id": "e9658dd3-77e7-41fd-8395-9906316f760d",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Remove cytokines that have been calculated as targets (because not included in ligand list above)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 560,
   "id": "66250797-eb55-4a8d-8a04-f17d9d16e1a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "3314320"
      ],
      "text/latex": [
       "3314320"
      ],
      "text/markdown": [
       "3314320"
      ],
      "text/plain": [
       "[1] 3314320"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_target_correlations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 561,
   "id": "7fd55f4e-4e5a-4c9d-a1ed-420c1452b3b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations = ligand_target_correlations[ligand_target_correlations$Target_Type != 'cytokine',]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 562,
   "id": "328490cd-82c4-482f-9bcb-8de1381d8c18",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "3314320"
      ],
      "text/latex": [
       "3314320"
      ],
      "text/markdown": [
       "3314320"
      ],
      "text/plain": [
       "[1] 3314320"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_target_correlations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 563,
   "id": "98cb0ba0-d953-4732-97d2-ccaef93a6dfb",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Only look at between cell-type communications --> remove same cell-type Links"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 564,
   "id": "4f41e8de-a4d9-4169-84dd-ee046a17b098",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Target_Cell_Type = str_replace(ligand_target_correlations$Target_Type, '.*_X\\\\d_|.*_X\\\\d\\\\d_','')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 565,
   "id": "347e8379-2463-48dc-be1c-a3f777e112a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Ligand_Cell_Type = str_replace(ligand_target_correlations$Ligand_Type, '.*_X\\\\d_|.*_X\\\\d\\\\d_','')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 566,
   "id": "ae650eda-ac47-4b64-a8c0-3c47ecc11c03",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations = ligand_target_correlations[ligand_target_correlations$Target_Cell_Type != ligand_target_correlations$Ligand_Cell_Type,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 567,
   "id": "8dfa3e4f-1719-4cf9-a02d-919a8c3435a8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "2721225"
      ],
      "text/latex": [
       "2721225"
      ],
      "text/markdown": [
       "2721225"
      ],
      "text/plain": [
       "[1] 2721225"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_target_correlations)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1db9326e-d1b9-48a9-bcfc-8c4dac9e3a8b",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Save the ligand target data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 568,
   "id": "3e1a1849-eeb6-4df8-ae9b-b4af8dfef178",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'V_FINAL'"
      ],
      "text/latex": [
       "'V\\_FINAL'"
      ],
      "text/markdown": [
       "'V_FINAL'"
      ],
      "text/plain": [
       "[1] \"V_FINAL\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 569,
   "id": "0e56f093-2e64-4d0b-ad60-345341f11a61",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(ligand_target_correlations,gzfile(paste0(result_path, '/F-Analysis/' ,'F4_Ligand_Target_Data', name, filter_samples,'.gz')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39c8bbf2-36d5-4ba6-b0f2-492a9d6b750d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.1"
  },
  "toc-autonumbering": true
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
