{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a5af7905-9b65-4d41-b747-0c795477d895",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "### Execute ligand target analysis for lagged ligands "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7ad3b8fc-2638-401d-8299-974e2612bdfd",
   "metadata": {
    "tags": []
   },
   "source": [
    "#############################################\n",
    "# Prerequisites - Load Libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8bc0aa68-a56c-4954-b7f1-c940eed22535",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in read.table(file = file, header = header, sep = sep, quote = quote, :\n",
      "“incomplete final line found by readTableHeader on '../conda_environment/Environment_Configs.csv'”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/home/icb/corinna.losert/miniconda3/envs/stark_stemi_R_Env_4_1//lib/R/library\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘igraph’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    decompose, spectrum\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    union\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:igraph’:\n",
      "\n",
      "    normalize, path, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘sp’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    %over%\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘SeuratObject’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SummarizedExperiment’:\n",
      "\n",
      "    Assays\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    intersect\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomeInfoDb’:\n",
      "\n",
      "    intersect\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    intersect\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    intersect\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    intersect\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    intersect\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Seurat’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SummarizedExperiment’:\n",
      "\n",
      "    Assays\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:igraph’:\n",
      "\n",
      "    components\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Matrix’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘DelayedArray’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:igraph’:\n",
      "\n",
      "    simplify\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    aperm, apply, rowsum, scale, sweep\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘HDF5Array’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:rhdf5’:\n",
      "\n",
      "    h5ls\n",
      "\n",
      "\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Warning message in if (is.na(desc)) {:\n",
      "“the condition has length > 1 and only the first element will be used”\n",
      "Registered S3 method overwritten by 'SeuratDisk':\n",
      "  method            from  \n",
      "  as.sparse.H5Group Seurat\n",
      "\n",
      "\n",
      "Attaching package: ‘gridExtra’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    combine\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    combine\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:gridExtra’:\n",
      "\n",
      "    combine\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    combine\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomeInfoDb’:\n",
      "\n",
      "    intersect\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:IRanges’:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:matrixStats’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:igraph’:\n",
      "\n",
      "    as_data_frame, groups, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘plotly’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:ggplot2’:\n",
      "\n",
      "    last_plot\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    rename\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:igraph’:\n",
      "\n",
      "    groups\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    filter\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:graphics’:\n",
      "\n",
      "    layout\n",
      "\n",
      "\n",
      "Warning message in checkMatrixPackageVersion():\n",
      "“Package version inconsistency detected.\n",
      "TMB was built with Matrix version 1.5.3\n",
      "Current Matrix version is 1.6.4\n",
      "Please re-install 'TMB' from source using install.packages('TMB', type = 'source') or ask CRAN for a binary version of 'TMB' matching CRAN's 'Matrix' package”\n",
      "\n",
      "Attaching package: ‘limma’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:scater’:\n",
      "\n",
      "    plotMDS\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    plotMA\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘DESeq2’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:scater’:\n",
      "\n",
      "    fpkm\n",
      "\n",
      "\n",
      "Loading required package: AnnotationDbi\n",
      "\n",
      "\n",
      "Attaching package: ‘AnnotationDbi’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:plotly’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Registered S3 method overwritten by 'ggtree':\n",
      "  method      from \n",
      "  identify.gg ggfun\n",
      "\n",
      "clusterProfiler v4.0.5  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/\n",
      "\n",
      "If you use clusterProfiler in published research, please cite:\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, 2(3):100141. doi: 10.1016/j.xinn.2021.100141\n",
      "\n",
      "\n",
      "Attaching package: ‘clusterProfiler’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:AnnotationDbi’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:DelayedArray’:\n",
      "\n",
      "    simplify\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    rename\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:igraph’:\n",
      "\n",
      "    simplify\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    filter\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SummarizedExperiment’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘MOFA2’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    predict\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘ggraph’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:sp’:\n",
      "\n",
      "    geometry\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Hmisc’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MOFA2’:\n",
      "\n",
      "    impute\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:AnnotationDbi’:\n",
      "\n",
      "    contents\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:plotly’:\n",
      "\n",
      "    subplot\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    src, summarize\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Seurat’:\n",
      "\n",
      "    Key\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SeuratObject’:\n",
      "\n",
      "    Key\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    contents\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    format.pval, units\n",
      "\n",
      "\n",
      "Registered S3 method overwritten by 'GGally':\n",
      "  method from   \n",
      "  +.gg   ggplot2\n",
      "\n",
      "\n",
      "Attaching package: ‘rstatix’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:AnnotationDbi’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    desc\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    filter\n",
      "\n",
      "\n",
      "corrplot 0.92 loaded\n",
      "\n",
      "\n",
      "‘network’ 1.18.1 (2023-01-24), part of the Statnet Project\n",
      "* ‘news(package=\"network\")’ for changes since last version\n",
      "* ‘citation(\"network\")’ for citation information\n",
      "* ‘https://statnet.org’ for help, support, and other information\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘network’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Hmisc’:\n",
      "\n",
      "    is.discrete\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:igraph’:\n",
      "\n",
      "    %c%, %s%, add.edges, add.vertices, delete.edges, delete.vertices,\n",
      "    get.edge.attribute, get.edges, get.vertex.attribute, is.bipartite,\n",
      "    is.directed, list.edge.attributes, list.vertex.attributes,\n",
      "    set.edge.attribute, set.vertex.attribute\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘enrichplot’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GGally’:\n",
      "\n",
      "    ggtable\n",
      "\n",
      "\n",
      "DOSE v3.18.3  For help: https://guangchuangyu.github.io/software/DOSE\n",
      "\n",
      "If you use DOSE in published research, please cite:\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘UpSetR’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:ComplexUpset’:\n",
      "\n",
      "    upset\n",
      "\n",
      "\n",
      "Registered S3 method overwritten by 'gdata':\n",
      "  method         from  \n",
      "  reorder.factor gplots\n",
      "\n",
      "gdata: read.xls support for 'XLS' (Excel 97-2004) files ENABLED.\n",
      "\n",
      "\n",
      "\n",
      "gdata: read.xls support for 'XLSX' (Excel 2007+) files ENABLED.\n",
      "\n",
      "\n",
      "Attaching package: ‘gdata’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:data.table’:\n",
      "\n",
      "    first, last\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    combine, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:gridExtra’:\n",
      "\n",
      "    combine\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SummarizedExperiment’:\n",
      "\n",
      "    trim\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    combine\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    trim\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:IRanges’:\n",
      "\n",
      "    startsWith, trim\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, first<-\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    combine\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:stats4’:\n",
      "\n",
      "    nobs\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    nobs\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    object.size\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    startsWith\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘reshape2’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:data.table’:\n",
      "\n",
      "    dcast, melt\n",
      "\n",
      "\n",
      "ReactomePA v1.36.0  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/reactomepa.html\n",
      "\n",
      "If you use ReactomePA in published research, please cite:\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization. Molecular BioSystems 2016, 12(2):477-479\n",
      "\n",
      "Loading required package: mgcv\n",
      "\n",
      "Loading required package: nlme\n",
      "\n",
      "\n",
      "Attaching package: ‘nlme’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    collapse\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    collapse\n",
      "\n",
      "\n",
      "This is mgcv 1.8-41. For overview type 'help(\"mgcv-package\")'.\n",
      "\n",
      "Loading required package: genefilter\n",
      "\n",
      "\n",
      "Attaching package: ‘genefilter’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:rstatix’:\n",
      "\n",
      "    Anova\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowSds, rowVars\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    rowSds, rowVars\n",
      "\n",
      "\n",
      "Loading required package: BiocParallel\n",
      "\n",
      "\n",
      "Attaching package: ‘patchwork’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:genefilter’:\n",
      "\n",
      "    area\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:cowplot’:\n",
      "\n",
      "    align_plots\n",
      "\n",
      "\n",
      "Type 'citation(\"pROC\")' for a citation.\n",
      "\n",
      "\n",
      "Attaching package: ‘pROC’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:colorspace’:\n",
      "\n",
      "    coords\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:IRanges’:\n",
      "\n",
      "    cov, var\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    cov, var\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    var\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    cov, smooth, var\n",
      "\n",
      "\n",
      "Loaded glmnet 4.1-6\n",
      "\n",
      "\n",
      "Attaching package: ‘nnet’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:mgcv’:\n",
      "\n",
      "    multinom\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘dfidx’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:AnnotationDbi’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    filter\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘cvAUC’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:genefilter’:\n",
      "\n",
      "    AUC\n",
      "\n",
      "\n",
      "Loading required package: maps\n",
      "\n",
      "Loading required package: shapefiles\n",
      "\n",
      "Loading required package: foreign\n",
      "\n",
      "\n",
      "Attaching package: ‘shapefiles’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:foreign’:\n",
      "\n",
      "    read.dbf, write.dbf\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘plotROC’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:pROC’:\n",
      "\n",
      "    ggroc\n",
      "\n",
      "\n",
      "Welcome! Want to learn more? See two factoextra-related books at https://goo.gl/ve3WBa\n",
      "\n",
      "Loading required package: grid\n",
      "\n",
      "========================================\n",
      "ComplexHeatmap version 2.8.0\n",
      "Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/\n",
      "Github page: https://github.com/jokergoo/ComplexHeatmap\n",
      "Documentation: http://jokergoo.github.io/ComplexHeatmap-reference\n",
      "\n",
      "If you use it in published research, please cite:\n",
      "Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional \n",
      "  genomic data. Bioinformatics 2016.\n",
      "\n",
      "The new InteractiveComplexHeatmap package can directly export static \n",
      "complex heatmaps into an interactive Shiny app with zero effort. Have a try!\n",
      "\n",
      "This message can be suppressed by:\n",
      "  suppressPackageStartupMessages(library(ComplexHeatmap))\n",
      "========================================\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘ComplexHeatmap’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:genefilter’:\n",
      "\n",
      "    dist2\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:network’:\n",
      "\n",
      "    %v%\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:plotly’:\n",
      "\n",
      "    add_heatmap\n",
      "\n",
      "\n",
      "Welcome to compositions, a package for compositional data analysis.\n",
      "Find an intro with \"? compositions\"\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘compositions’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:glmnet’:\n",
      "\n",
      "    rmult\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:pROC’:\n",
      "\n",
      "    cov, var\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:DelayedArray’:\n",
      "\n",
      "    scale\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Matrix’:\n",
      "\n",
      "    norm\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:scater’:\n",
      "\n",
      "    normalize\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:IRanges’:\n",
      "\n",
      "    cor, cov, var\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    cor, cov, var\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    normalize, var\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:igraph’:\n",
      "\n",
      "    normalize\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    anova, cor, cov, dist, var\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    %*%, norm, scale, scale.default\n",
      "\n",
      "\n",
      "========================================\n",
      "circlize version 0.4.15\n",
      "CRAN page: https://cran.r-project.org/package=circlize\n",
      "Github page: https://github.com/jokergoo/circlize\n",
      "Documentation: https://jokergoo.github.io/circlize_book/book/\n",
      "\n",
      "If you use it in published research, please cite:\n",
      "Gu, Z. circlize implements and enhances circular visualization\n",
      "  in R. Bioinformatics 2014.\n",
      "\n",
      "This message can be suppressed by:\n",
      "  suppressPackageStartupMessages(library(circlize))\n",
      "========================================\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘circlize’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:igraph’:\n",
      "\n",
      "    degree\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘ggpubr’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:enrichplot’:\n",
      "\n",
      "    color_palette\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:cowplot’:\n",
      "\n",
      "    get_legend\n",
      "\n",
      "\n",
      "Loading required package: survival\n",
      "\n",
      "Loading required package: TH.data\n",
      "\n",
      "Loading required package: MASS\n",
      "\n",
      "\n",
      "Attaching package: ‘MASS’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:dfidx’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:patchwork’:\n",
      "\n",
      "    area\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:genefilter’:\n",
      "\n",
      "    area\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:rstatix’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:clusterProfiler’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:AnnotationDbi’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:plotly’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘TH.data’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MASS’:\n",
      "\n",
      "    geyser\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘purrr’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:maps’:\n",
      "\n",
      "    map\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:gdata’:\n",
      "\n",
      "    keep\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:data.table’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:clusterProfiler’:\n",
      "\n",
      "    simplify\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:DelayedArray’:\n",
      "\n",
      "    simplify\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    reduce\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    reduce\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:igraph’:\n",
      "\n",
      "    compose, simplify\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "source('MS0_Libraries.r')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a1fe13f0-3991-4bcf-8a72-1e57f5de3639",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“\u001b[1m\u001b[22mThe `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.\n",
      "\u001b[36mℹ\u001b[39m Please use the `linewidth` argument instead.”\n"
     ]
    }
   ],
   "source": [
    "source('MS4_Plot_Config.r')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c0d8f19c-38da-490a-904c-ead6cbd8fada",
   "metadata": {
    "tags": []
   },
   "source": [
    "###############################################\n",
    "# Preqrequisites Configurations & Parameters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "32a08c89-f24e-4022-bcf8-70120b317e3b",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_path ='../data/current'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "41b26d19-d26f-4917-bf53-88d39efe056c",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_path = '../results/current'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "61a059d8-eb1d-4224-84dd-a61292c1b6c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'../data/current'"
      ],
      "text/latex": [
       "'../data/current'"
      ],
      "text/markdown": [
       "'../data/current'"
      ],
      "text/plain": [
       "[1] \"../data/current\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "data_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "434926b9-50d3-42e5-8d6c-2e98586db31b",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Version of the pre-processed data to load"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "55ae9b4e-d653-4f50-82ce-c45e566e4bf3",
   "metadata": {},
   "outputs": [],
   "source": [
    "name = 'V_FINAL'  \n",
    "# of dataset to load, prepared and combined data of E1 - skip E2 step "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "98f9d1fb-733a-41a7-8432-96ca94d08820",
   "metadata": {},
   "outputs": [],
   "source": [
    "filter_samples = 'acs'\n",
    "#filter_samples = 'acs_w_o_infection_ccs'\n",
    "#filter_samples = 'no_ccs_ccs'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4d78b773-c356-43bd-a1b7-988fd621712b",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Load Data "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "88d6ff01-805b-4e59-bfbc-122177418d3c",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Sample Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a2c784bb-e348-4eba-877b-419b2dd9c918",
   "metadata": {},
   "outputs": [],
   "source": [
    "## sample meta and clinical data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b5d4992b-aefd-4b17-8114-bc765b0700f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_data = read.csv(paste0(result_path, '/00_Data_Overview/Merged_Sample_Meta_Data.csv'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "248e225a-e54f-4535-b981-a3e7955c616c",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Prepared combined data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f9d4e05a-0aa9-444b-887e-484ba2d665af",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Select and load pre-processed and combined data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "ae3feb85-f2b2-4b90-a67c-151d3cf87723",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"../results/current/E-Analysis/Combined_Data_V_FINAL.csv\"\n",
      "[1] \"2024-01-03 14:22:52 CET\"\n"
     ]
    }
   ],
   "source": [
    "path = paste0(result_path, '/E-Analysis/Combined_Data_',name,'.csv')\n",
    "print(path)\n",
    "data_long = read.csv(path)\n",
    "print(file.info(path)$mtime)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "126a3bfb-1256-4c6b-b778-aeb0c6f1a343",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>sample_id</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>value</th><th scope=col>config</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>k10</td><td>single_cell</td><td>X0_T.cell.CD4__SSU72</td><td>0.5896835</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>2</td><td>k10</td><td>single_cell</td><td>X0_T.cell.CD4__PARK7</td><td>0.8587511</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & X & sample\\_id & type & variable & value & config\\\\\n",
       "  & <int> & <chr> & <chr> & <chr> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & k10 & single\\_cell & X0\\_T.cell.CD4\\_\\_SSU72 & 0.5896835 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\\\\\n",
       "\t2 & 2 & k10 & single\\_cell & X0\\_T.cell.CD4\\_\\_PARK7 & 0.8587511 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 6\n",
       "\n",
       "| <!--/--> | X &lt;int&gt; | sample_id &lt;chr&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | k10 | single_cell | X0_T.cell.CD4__SSU72 | 0.5896835 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- |\n",
       "| 2 | 2 | k10 | single_cell | X0_T.cell.CD4__PARK7 | 0.8587511 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- |\n",
       "\n"
      ],
      "text/plain": [
       "  X sample_id type        variable             value    \n",
       "1 1 k10       single_cell X0_T.cell.CD4__SSU72 0.5896835\n",
       "2 2 k10       single_cell X0_T.cell.CD4__PARK7 0.8587511\n",
       "  config                          \n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE-"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "5fc8233d-b29a-41cc-b9e7-06ffddb55a22",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1557939"
      ],
      "text/latex": [
       "1557939"
      ],
      "text/markdown": [
       "1557939"
      ],
      "text/plain": [
       "[1] 1557939"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(data_long)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "db789d09-2390-4ab9-aba9-d0e2730752dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "13289"
      ],
      "text/latex": [
       "13289"
      ],
      "text/markdown": [
       "13289"
      ],
      "text/plain": [
       "[1] 13289"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique(data_long$variable))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0a077569-b065-469d-9d21-df4d73ecf422",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Ligand Receptor Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "7c7fb68c-e0da-4d4a-b15a-cf9c6c97e7b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Load ligand receptor data from nichenet database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "9699012a-a3f6-4e2b-86a1-0c98b182a8e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "lr_network = readRDS(url(\"https://zenodo.org/record/3260758/files/lr_network.rds\")) # ligand receptor interactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "0439aa90-4c14-4714-81de-86753c5ba3cf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 2 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>from</th><th scope=col>to</th><th scope=col>source</th><th scope=col>database</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>CXCL1</td><td>CXCR2</td><td>kegg_cytokines</td><td>kegg</td></tr>\n",
       "\t<tr><td>CXCL2</td><td>CXCR2</td><td>kegg_cytokines</td><td>kegg</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 2 × 4\n",
       "\\begin{tabular}{llll}\n",
       " from & to & source & database\\\\\n",
       " <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t CXCL1 & CXCR2 & kegg\\_cytokines & kegg\\\\\n",
       "\t CXCL2 & CXCR2 & kegg\\_cytokines & kegg\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 2 × 4\n",
       "\n",
       "| from &lt;chr&gt; | to &lt;chr&gt; | source &lt;chr&gt; | database &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| CXCL1 | CXCR2 | kegg_cytokines | kegg |\n",
       "| CXCL2 | CXCR2 | kegg_cytokines | kegg |\n",
       "\n"
      ],
      "text/plain": [
       "  from  to    source         database\n",
       "1 CXCL1 CXCR2 kegg_cytokines kegg    \n",
       "2 CXCL2 CXCR2 kegg_cytokines kegg    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(lr_network,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "76be70e6-58f7-4229-9096-1b6e36acccf7",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Summarise to have only ligand or receptor assignment not double"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "9372b74d-318f-4838-867d-328b7941248a",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_receptor_interactions = lr_network %>% group_by(from,to) %>% count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "33767995-6fba-4c90-9e4f-4bb8b237c489",
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames(ligand_receptor_interactions) = c('Ligand', 'Receptor', 'amount_sources')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "f6174978-a76f-40d8-9c71-2698e688f7b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Ligand</th><th scope=col>Receptor</th><th scope=col>amount_sources</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>A2M</td><td>LRP1    </td><td>1</td></tr>\n",
       "\t<tr><td>A2M</td><td>TNFRSF14</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 2 × 3\n",
       "\\begin{tabular}{lll}\n",
       " Ligand & Receptor & amount\\_sources\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t A2M & LRP1     & 1\\\\\n",
       "\t A2M & TNFRSF14 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 2 × 3\n",
       "\n",
       "| Ligand &lt;chr&gt; | Receptor &lt;chr&gt; | amount_sources &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| A2M | LRP1     | 1 |\n",
       "| A2M | TNFRSF14 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand Receptor amount_sources\n",
       "1 A2M    LRP1     1             \n",
       "2 A2M    TNFRSF14 1             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_receptor_interactions,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "9eb4c6a2-e588-408f-b6fc-0f7c86abac85",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Aggregate to contain only unique ligand-receptor interactions across different data sources"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "e3ffd5fa-677c-476d-9082-babe75735e35",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_amounts = ligand_receptor_interactions %>% group_by(Ligand) %>% summarise(amount_l = sum(amount_sources))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "a7aa8dff-aaae-4410-bef7-2aaae9366e09",
   "metadata": {},
   "outputs": [],
   "source": [
    "receptor_amounts = ligand_receptor_interactions %>% group_by(Receptor) %>% summarise(amount_r = sum(amount_sources))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "4ea1be21-be31-4f73-b456-37169478be6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "l_r_merged = merge(ligand_amounts, receptor_amounts, by.x = 'Ligand', by.y = 'Receptor')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "8a79dc18-b5dd-4e67-aa55-b6136397c473",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand = l_r_merged$Ligand[l_r_merged$amount_l >= l_r_merged$amount_r]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "2e77c390-00da-4e46-bbcd-7250f7f686b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "receptor = l_r_merged$Ligand[l_r_merged$amount_l < l_r_merged$amount_r]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "379fd38a-a70e-44db-a833-73b96112fb9f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "115"
      ],
      "text/latex": [
       "115"
      ],
      "text/markdown": [
       "115"
      ],
      "text/plain": [
       "[1] 115"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(l_r_merged)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "07eeea0f-f1d5-4888-9262-7e8328f6db06",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Ligand</th><th scope=col>amount_l</th><th scope=col>amount_r</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ALCAM </td><td>3</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AMIGO1</td><td>2</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & Ligand & amount\\_l & amount\\_r\\\\\n",
       "  & <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & ALCAM  & 3 & 2\\\\\n",
       "\t2 & AMIGO1 & 2 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 3\n",
       "\n",
       "| <!--/--> | Ligand &lt;chr&gt; | amount_l &lt;int&gt; | amount_r &lt;int&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | ALCAM  | 3 | 2 |\n",
       "| 2 | AMIGO1 | 2 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand amount_l amount_r\n",
       "1 ALCAM  3        2       \n",
       "2 AMIGO1 2        1       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(l_r_merged,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "677019a5-f8a9-4b83-9e27-2aeab665e012",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "84"
      ],
      "text/latex": [
       "84"
      ],
      "text/markdown": [
       "84"
      ],
      "text/plain": [
       "[1] 84"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(ligand)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "7c33378a-42d9-4dd0-b737-406b98afc231",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'ALCAM'</li><li>'AMIGO1'</li><li>'CADM1'</li><li>'CADM3'</li><li>'CADM4'</li><li>'CALM1'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'ALCAM'\n",
       "\\item 'AMIGO1'\n",
       "\\item 'CADM1'\n",
       "\\item 'CADM3'\n",
       "\\item 'CADM4'\n",
       "\\item 'CALM1'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'ALCAM'\n",
       "2. 'AMIGO1'\n",
       "3. 'CADM1'\n",
       "4. 'CADM3'\n",
       "5. 'CADM4'\n",
       "6. 'CALM1'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"ALCAM\"  \"AMIGO1\" \"CADM1\"  \"CADM3\"  \"CADM4\"  \"CALM1\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "bb368b72-cce0-458a-a4e4-dd2e6224193b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "31"
      ],
      "text/latex": [
       "31"
      ],
      "text/markdown": [
       "31"
      ],
      "text/plain": [
       "[1] 31"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(receptor)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "40ff416e-fea0-4356-a36a-b4c7eb8a5dd3",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Filter out duplicate assignments and decide whether it is a receptor or ligand"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "cbdb333d-637b-42fe-b26d-5add4a7b28cb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Ligand</th><th scope=col>Receptor</th><th scope=col>amount_sources</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>A2M</td><td>LRP1    </td><td>1</td></tr>\n",
       "\t<tr><td>A2M</td><td>TNFRSF14</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 2 × 3\n",
       "\\begin{tabular}{lll}\n",
       " Ligand & Receptor & amount\\_sources\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t A2M & LRP1     & 1\\\\\n",
       "\t A2M & TNFRSF14 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 2 × 3\n",
       "\n",
       "| Ligand &lt;chr&gt; | Receptor &lt;chr&gt; | amount_sources &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| A2M | LRP1     | 1 |\n",
       "| A2M | TNFRSF14 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand Receptor amount_sources\n",
       "1 A2M    LRP1     1             \n",
       "2 A2M    TNFRSF14 1             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_receptor_interactions,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "3b097b9b-9cbe-4b26-85db-7cf5c6fb1c1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_receptor_interactions = ligand_receptor_interactions[!ligand_receptor_interactions$Receptor %in% ligand,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "8c6a6b28-81aa-4e80-85cf-9352224bc7d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_receptor_interactions = ligand_receptor_interactions[!ligand_receptor_interactions$Ligand %in% receptor,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "40e083b4-3b56-4061-970b-4d1588665a55",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "11639"
      ],
      "text/latex": [
       "11639"
      ],
      "text/markdown": [
       "11639"
      ],
      "text/plain": [
       "[1] 11639"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_receptor_interactions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "bbe47f2a-c4dc-4b6d-9b0a-695feae2a384",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Generate dataset specifying ligand receptor variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "8c36e108-4550-47f3-a239-bac3a03f6e34",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "ligand_receptor_vars = rbind(data.frame(gene = ligand_receptor_interactions$Ligand, interaction_type = 'Ligand'), data.frame(gene = ligand_receptor_interactions$Receptor, interaction_type = 'Receptor'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "56fdf875-b311-4d7e-97c9-dc1f7ebd52d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_receptor_vars = unique(ligand_receptor_vars)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "33fd1c26-aa15-4f7f-b93c-40e17576d4b8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>interaction_type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A2M  </td><td>Ligand</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>AANAT</td><td>Ligand</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & gene & interaction\\_type\\\\\n",
       "  & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & A2M   & Ligand\\\\\n",
       "\t4 & AANAT & Ligand\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 2\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | interaction_type &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| 1 | A2M   | Ligand |\n",
       "| 4 | AANAT | Ligand |\n",
       "\n"
      ],
      "text/plain": [
       "  gene  interaction_type\n",
       "1 A2M   Ligand          \n",
       "4 AANAT Ligand          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_receptor_vars,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "c9129614-bfe8-447b-8aff-fd43415ac148",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1414"
      ],
      "text/latex": [
       "1414"
      ],
      "text/markdown": [
       "1414"
      ],
      "text/plain": [
       "[1] 1414"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_receptor_vars)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "d91a51dd-0332-4e24-b8d6-d60b4d1bf653",
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove empty spaces\n",
    "ligand_receptor_vars$gene = str_replace(ligand_receptor_vars$gene, ' ', '')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f977280-9d12-4b5b-b3c9-4c715443bbac",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Available ligand receptor potential + transcription factor datasets from  nichenet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "e489e0dc-7a6b-48c0-a6d4-b87dd5456fdb",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Weighted networks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "b5b53376-5c56-43fd-85b9-694972d38e02",
   "metadata": {},
   "outputs": [],
   "source": [
    "weighted_networks = readRDS(url(\"https://zenodo.org/record/3260758/files/weighted_networks.rds\")) # integrated weighted ligand-signaling and gene regulatory network"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "98be6e73-1cc4-443f-834b-46408c899c8c",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Prepared influential matrices"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "218857a4-9abf-4155-b1d0-b220a9a84f6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_matrix = readRDS(url(\"https://zenodo.org/record/3260758/files/ligand_target_matrix.rds\"))  # gene-gene regulatory scores\n",
    "# gives the regulatory potential of a target gene based on the ligand-tf & tf- target-gene links and weights of the signaling and gr network (multiply PPR of each ligand -tf  pair with the weights of the gr network)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "fb475f82-0f33-43cb-a614-ea5428142561",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Generate data in long format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "fb089af0-96e9-469c-9e64-debf11e69b6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_matrix_long = melt(ligand_target_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "a27dddb7-4c9d-4f91-b8a9-522d1a2a9aac",
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames(ligand_target_matrix_long) = c('target', 'ligand', 'regulatory_potential')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "80e5fe18-967f-403e-a487-ea20378bc381",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>regulatory_potential</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A1BG    </td><td>CXCL1</td><td>0.0003534343</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A1BG-AS1</td><td>CXCL1</td><td>0.0001650894</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & target & ligand & regulatory\\_potential\\\\\n",
       "  & <fct> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & A1BG     & CXCL1 & 0.0003534343\\\\\n",
       "\t2 & A1BG-AS1 & CXCL1 & 0.0001650894\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 3\n",
       "\n",
       "| <!--/--> | target &lt;fct&gt; | ligand &lt;fct&gt; | regulatory_potential &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | A1BG     | CXCL1 | 0.0003534343 |\n",
       "| 2 | A1BG-AS1 | CXCL1 | 0.0001650894 |\n",
       "\n"
      ],
      "text/plain": [
       "  target   ligand regulatory_potential\n",
       "1 A1BG     CXCL1  0.0003534343        \n",
       "2 A1BG-AS1 CXCL1  0.0001650894        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_target_matrix_long,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "d871bfeb-4931-4970-b0c6-bf0808c4965a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "688"
      ],
      "text/latex": [
       "688"
      ],
      "text/markdown": [
       "688"
      ],
      "text/plain": [
       "[1] 688"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique(ligand_target_matrix_long$ligand))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea0860ee-f5b2-4984-8e29-613b0b604f53",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "e6ec2a5a-ee0f-498c-b2a3-217cbef10b92",
   "metadata": {},
   "source": [
    "## Cell Gene Expression Data (percentage of cells expressing gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "91689551-d337-4635-b045-965a43f62c24",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Load information about how many cells express a gene generated in C0 Script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "66aff71b-4ad6-4196-a424-8d8ccf873e23",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"2024-01-03 14:10:30 CET\"\n"
     ]
    }
   ],
   "source": [
    "path = paste0(result_path, '/C-Analysis/C0_Filter_Genes_Input_Correlations_Perc_Values.csv')\n",
    "cell_perc_cluster = read.csv( path) # cluster alternative\n",
    "print(file.info(path)$mtime)\n",
    "#cell_perc_cluster = read.csv( paste0(result_path, '/C-Analysis/C0_Filter_Genes_Input_Correlations_Perc_Values_cell_type_Scanorama.csv')) # cell-type alternative"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "637602d5-8dfa-4a03-a4ab-04b53cb37679",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Filter on relevant cell types (clusters with low amount of cells per sample; similar to MOFA analysis)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "e248ad84-cbca-4ca3-bf2f-8e4b2f894b1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "cell_perc_cluster = cell_perc_cluster[!(cell_perc_cluster$cluster %in% c('16_Plasma Blast', '14_Other', '17_Progenitor', '15_Plasma Blast', '18_Megakaryocytes' ,'Plasma Blast', 'Other', 'Progenitor', 'Plasma Blast', 'Megakaryocytes' )),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "6cb18288-4fb6-4865-8158-81ab077ad6a5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'8_B-cell'</li><li>'9_Monocytes - CD16_FCGR3A'</li><li>'5_T-cell-CD4'</li><li>'0_T-cell-CD4'</li><li>'1_T-cell-CD8'</li><li>'6_Monocytes - CD14'</li><li>'2_T-cell-CD4'</li><li>'4_Monocytes - CD14'</li><li>'11_T-cell-CD4'</li><li>'3_NK'</li><li>'7_Monocytes - CD14'</li><li>'13_Dendritic'</li><li>'10_B-cell'</li><li>'12_Monocytes - CD16_FCGR3A'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '8\\_B-cell'\n",
       "\\item '9\\_Monocytes - CD16\\_FCGR3A'\n",
       "\\item '5\\_T-cell-CD4'\n",
       "\\item '0\\_T-cell-CD4'\n",
       "\\item '1\\_T-cell-CD8'\n",
       "\\item '6\\_Monocytes - CD14'\n",
       "\\item '2\\_T-cell-CD4'\n",
       "\\item '4\\_Monocytes - CD14'\n",
       "\\item '11\\_T-cell-CD4'\n",
       "\\item '3\\_NK'\n",
       "\\item '7\\_Monocytes - CD14'\n",
       "\\item '13\\_Dendritic'\n",
       "\\item '10\\_B-cell'\n",
       "\\item '12\\_Monocytes - CD16\\_FCGR3A'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '8_B-cell'\n",
       "2. '9_Monocytes - CD16_FCGR3A'\n",
       "3. '5_T-cell-CD4'\n",
       "4. '0_T-cell-CD4'\n",
       "5. '1_T-cell-CD8'\n",
       "6. '6_Monocytes - CD14'\n",
       "7. '2_T-cell-CD4'\n",
       "8. '4_Monocytes - CD14'\n",
       "9. '11_T-cell-CD4'\n",
       "10. '3_NK'\n",
       "11. '7_Monocytes - CD14'\n",
       "12. '13_Dendritic'\n",
       "13. '10_B-cell'\n",
       "14. '12_Monocytes - CD16_FCGR3A'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"8_B-cell\"                   \"9_Monocytes - CD16_FCGR3A\" \n",
       " [3] \"5_T-cell-CD4\"               \"0_T-cell-CD4\"              \n",
       " [5] \"1_T-cell-CD8\"               \"6_Monocytes - CD14\"        \n",
       " [7] \"2_T-cell-CD4\"               \"4_Monocytes - CD14\"        \n",
       " [9] \"11_T-cell-CD4\"              \"3_NK\"                      \n",
       "[11] \"7_Monocytes - CD14\"         \"13_Dendritic\"              \n",
       "[13] \"10_B-cell\"                  \"12_Monocytes - CD16_FCGR3A\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(cell_perc_cluster$cluster)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "ad0a4241-f062-4733-9cb8-6fbe9878aa51",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th><th scope=col>gene</th><th scope=col>cluster</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AL627309.1</td><td>0.24327612</td><td>18</td><td>AL627309.1</td><td>8_B-cell</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AL627309.4</td><td>0.05406136</td><td> 4</td><td>AL627309.4</td><td>8_B-cell</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & X & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene & gene & cluster\\\\\n",
       "  & <chr> & <dbl> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & AL627309.1 & 0.24327612 & 18 & AL627309.1 & 8\\_B-cell\\\\\n",
       "\t2 & AL627309.4 & 0.05406136 &  4 & AL627309.4 & 8\\_B-cell\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 5\n",
       "\n",
       "| <!--/--> | X &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; | gene &lt;chr&gt; | cluster &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | AL627309.1 | 0.24327612 | 18 | AL627309.1 | 8_B-cell |\n",
       "| 2 | AL627309.4 | 0.05406136 |  4 | AL627309.4 | 8_B-cell |\n",
       "\n"
      ],
      "text/plain": [
       "  X          perc_cells_expressing_gene total_amount_cells_expressing_gene\n",
       "1 AL627309.1 0.24327612                 18                                \n",
       "2 AL627309.4 0.05406136                  4                                \n",
       "  gene       cluster \n",
       "1 AL627309.1 8_B-cell\n",
       "2 AL627309.4 8_B-cell"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cell_perc_cluster,2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "663e9dbc-c818-4c80-bc91-5757eb1da480",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Potential Pre-Processing / Data Adjustments/ Data Preparations"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f795ec1-31e8-4b45-815b-81fe4e0c0853",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Prepare list of cytokines in data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "057902db-f598-4038-b934-a1c665ac62d8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>sample_id</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>value</th><th scope=col>config</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>k10</td><td>single_cell</td><td>X0_T.cell.CD4__SSU72</td><td>0.5896835</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>2</td><td>k10</td><td>single_cell</td><td>X0_T.cell.CD4__PARK7</td><td>0.8587511</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & X & sample\\_id & type & variable & value & config\\\\\n",
       "  & <int> & <chr> & <chr> & <chr> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & k10 & single\\_cell & X0\\_T.cell.CD4\\_\\_SSU72 & 0.5896835 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\\\\\n",
       "\t2 & 2 & k10 & single\\_cell & X0\\_T.cell.CD4\\_\\_PARK7 & 0.8587511 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 6\n",
       "\n",
       "| <!--/--> | X &lt;int&gt; | sample_id &lt;chr&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | k10 | single_cell | X0_T.cell.CD4__SSU72 | 0.5896835 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- |\n",
       "| 2 | 2 | k10 | single_cell | X0_T.cell.CD4__PARK7 | 0.8587511 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- |\n",
       "\n"
      ],
      "text/plain": [
       "  X sample_id type        variable             value    \n",
       "1 1 k10       single_cell X0_T.cell.CD4__SSU72 0.5896835\n",
       "2 2 k10       single_cell X0_T.cell.CD4__PARK7 0.8587511\n",
       "  config                          \n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE-"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "15f87b86-6183-4aa3-94dd-dfc3cd2eef43",
   "metadata": {},
   "outputs": [],
   "source": [
    "cytokines = unique(data_long$variable[data_long$type == 'cytokine'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "9a0e272a-a40f-440a-ba1a-a2dd078806e4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'BCA1__CXCL13'</li><li>'CTACK__CCL27'</li><li>'EGF__EGF'</li><li>'ENA78__CXCL5'</li><li>'Eotaxin__CCL11'</li><li>'Eotaxin2__CCL24'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'BCA1\\_\\_CXCL13'\n",
       "\\item 'CTACK\\_\\_CCL27'\n",
       "\\item 'EGF\\_\\_EGF'\n",
       "\\item 'ENA78\\_\\_CXCL5'\n",
       "\\item 'Eotaxin\\_\\_CCL11'\n",
       "\\item 'Eotaxin2\\_\\_CCL24'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'BCA1__CXCL13'\n",
       "2. 'CTACK__CCL27'\n",
       "3. 'EGF__EGF'\n",
       "4. 'ENA78__CXCL5'\n",
       "5. 'Eotaxin__CCL11'\n",
       "6. 'Eotaxin2__CCL24'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"BCA1__CXCL13\"    \"CTACK__CCL27\"    \"EGF__EGF\"        \"ENA78__CXCL5\"   \n",
       "[5] \"Eotaxin__CCL11\"  \"Eotaxin2__CCL24\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cytokines)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "796d030c-fed8-49eb-ab76-8ffaeeeb1d94",
   "metadata": {},
   "outputs": [],
   "source": [
    "cytokines = str_replace(cytokines, '.*__', '') # extract gene code from cytokines"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "33a0a3b2-876e-4bd5-89a3-6161b21d414e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'CXCL13'</li><li>'CCL27'</li><li>'EGF'</li><li>'CXCL5'</li><li>'CCL11'</li><li>'CCL24'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'CXCL13'\n",
       "\\item 'CCL27'\n",
       "\\item 'EGF'\n",
       "\\item 'CXCL5'\n",
       "\\item 'CCL11'\n",
       "\\item 'CCL24'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'CXCL13'\n",
       "2. 'CCL27'\n",
       "3. 'EGF'\n",
       "4. 'CXCL5'\n",
       "5. 'CCL11'\n",
       "6. 'CCL24'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"CXCL13\" \"CCL27\"  \"EGF\"    \"CXCL5\"  \"CCL11\"  \"CCL24\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cytokines)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "63d66c8c-9338-4670-98b9-5c8e3da23694",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1236"
      ],
      "text/latex": [
       "1236"
      ],
      "text/markdown": [
       "1236"
      ],
      "text/plain": [
       "[1] 1236"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_receptor_interactions[ligand_receptor_interactions$Ligand %in% cytokines,])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "45e5085a-e03e-4220-b5ac-e24bb59ce434",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Adjustments to data_long"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a89f9812-3b9f-49a4-8377-967f37c2d635",
   "metadata": {},
   "source": [
    "### Filter samples (optional)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "a2385779-1afd-49d9-b45b-fce030341f5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "## filter out some samples if specified in the beginning"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "c02dca7f-62b6-450f-948c-bbe80f51499e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>sample_id</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>value</th><th scope=col>config</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>k10</td><td>single_cell</td><td>X0_T.cell.CD4__SSU72</td><td>0.5896835</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>2</td><td>k10</td><td>single_cell</td><td>X0_T.cell.CD4__PARK7</td><td>0.8587511</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & X & sample\\_id & type & variable & value & config\\\\\n",
       "  & <int> & <chr> & <chr> & <chr> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & k10 & single\\_cell & X0\\_T.cell.CD4\\_\\_SSU72 & 0.5896835 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\\\\\n",
       "\t2 & 2 & k10 & single\\_cell & X0\\_T.cell.CD4\\_\\_PARK7 & 0.8587511 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 6\n",
       "\n",
       "| <!--/--> | X &lt;int&gt; | sample_id &lt;chr&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | k10 | single_cell | X0_T.cell.CD4__SSU72 | 0.5896835 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- |\n",
       "| 2 | 2 | k10 | single_cell | X0_T.cell.CD4__PARK7 | 0.8587511 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- |\n",
       "\n"
      ],
      "text/plain": [
       "  X sample_id type        variable             value    \n",
       "1 1 k10       single_cell X0_T.cell.CD4__SSU72 0.5896835\n",
       "2 2 k10       single_cell X0_T.cell.CD4__PARK7 0.8587511\n",
       "  config                          \n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE-\n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE-"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "cde8fd1c-8200-49d2-bf72-e01d33bd77e9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'vollstaendiger_ausschluss'</li><li>'ccs'</li><li>'koronarsklerose'</li><li>'acs_subacute'</li><li>'acs_w_infection'</li><li>'acs_w_o_infection'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'vollstaendiger\\_ausschluss'\n",
       "\\item 'ccs'\n",
       "\\item 'koronarsklerose'\n",
       "\\item 'acs\\_subacute'\n",
       "\\item 'acs\\_w\\_infection'\n",
       "\\item 'acs\\_w\\_o\\_infection'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'vollstaendiger_ausschluss'\n",
       "2. 'ccs'\n",
       "3. 'koronarsklerose'\n",
       "4. 'acs_subacute'\n",
       "5. 'acs_w_infection'\n",
       "6. 'acs_w_o_infection'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"vollstaendiger_ausschluss\" \"ccs\"                      \n",
       "[3] \"koronarsklerose\"           \"acs_subacute\"             \n",
       "[5] \"acs_w_infection\"           \"acs_w_o_infection\"        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(sample_data$classification)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "7c5c9e60-e52a-46ad-8e93-5ad018306922",
   "metadata": {},
   "outputs": [],
   "source": [
    "if(filter_samples == 'all'){\n",
    "    filter = unique(sample_data$sample_id)\n",
    "    }\n",
    "\n",
    "if(filter_samples == 'acs'){\n",
    "    filter = unique(sample_data$sample_id[sample_data$classification %in% c('acs_w_infection', 'acs_w_o_infection', 'acs_subacute')])\n",
    "    }\n",
    "\n",
    "\n",
    "if(filter_samples == 'acs_w_o_infection_ccs'){\n",
    "    filter =unique(sample_data$sample_id[sample_data$classification %in% c('ccs', 'acs_w_o_infection')])\n",
    "    \n",
    "    }\n",
    "\n",
    "if(filter_samples == 'no_ccs_ccs'){\n",
    "    filter =unique(sample_data$sample_id[sample_data$classification %in% c('vollstaendiger_ausschluss','ccs','koronarsklerose')])\n",
    "    \n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "6715853e-8f7b-4b25-a8fb-b9458ee0bc21",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "112"
      ],
      "text/latex": [
       "112"
      ],
      "text/markdown": [
       "112"
      ],
      "text/plain": [
       "[1] 112"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(filter)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "a2e20c9f-7c75-4119-a9f3-b430b6a2d72c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "9242c888-c0b6-4a94-ab4d-dd9415b698d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1557939"
      ],
      "text/latex": [
       "1557939"
      ],
      "text/markdown": [
       "1557939"
      ],
      "text/plain": [
       "[1] 1557939"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(data_long)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "a9378624-840d-4fc5-b03b-c73cc06873c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long = data_long[data_long$sample_id %in% filter,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "cb94911c-51a9-481b-9f4d-fc62e6cd83e4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1128736"
      ],
      "text/latex": [
       "1128736"
      ],
      "text/markdown": [
       "1128736"
      ],
      "text/plain": [
       "[1] 1128736"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(data_long)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "424dcc05-eeea-4d92-8af3-dba8a3516c1c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'m1.1'</li><li>'m10.2'</li><li>'m10.3'</li><li>'m11.2'</li><li>'m11.3'</li><li>'m11.4'</li><li>'m12.1'</li><li>'m12.2'</li><li>'m13.1'</li><li>'m13.2'</li><li>'m13.3'</li><li>'m13.4'</li><li>'m14.1'</li><li>'m14.2'</li><li>'m15.1'</li><li>'m15.2'</li><li>'m15.3'</li><li>'m15.4'</li><li>'m16.1'</li><li>'m16.2'</li><li>'m17.1'</li><li>'m18.1'</li><li>'m18.2'</li><li>'m18.3'</li><li>'m19.1'</li><li>'m19.2'</li><li>'m19.3'</li><li>'m2.1'</li><li>'m2.2'</li><li>'m2.3'</li><li>'m2.4'</li><li>'m20.3'</li><li>'m20.4'</li><li>'m21.1'</li><li>'m21.2'</li><li>'m21.3'</li><li>'m21.4'</li><li>'m22.1'</li><li>'m22.2'</li><li>'m22.3'</li><li>'m22.4'</li><li>'m23.1'</li><li>'m23.2'</li><li>'m23.3'</li><li>'m23.4'</li><li>'m24.2'</li><li>'m24.3'</li><li>'m24.4'</li><li>'m25.1'</li><li>'m25.2'</li><li>'m25.3'</li><li>'m26.1'</li><li>'m26.2'</li><li>'m26.3'</li><li>'m26.4'</li><li>'m27.1'</li><li>'m27.2'</li><li>'m27.3'</li><li>'m28.2'</li><li>'m28.3'</li><li>'m28.4'</li><li>'m3.2'</li><li>'m3.3'</li><li>'m3.4'</li><li>'m4.1'</li><li>'m4.2'</li><li>'m4.3'</li><li>'m4.4'</li><li>'m5.1'</li><li>'m5.3'</li><li>'m5.4'</li><li>'m6.1'</li><li>'m6.2'</li><li>'m6.3'</li><li>'m6.4'</li><li>'m7.1'</li><li>'m7.2'</li><li>'m7.3'</li><li>'m7.4'</li><li>'m8.1'</li><li>'m8.2'</li><li>'m8.3'</li><li>'m8.4'</li><li>'m9.1'</li><li>'m9.2'</li><li>'m9.3'</li><li>'m11.1'</li><li>'m10.1'</li><li>'m20.1'</li><li>'m24.1'</li><li>'m5.2'</li><li>'m20.2'</li><li>'m14.3'</li><li>'m14.4'</li><li>'m1.2'</li><li>'m1.3'</li><li>'m1.4'</li><li>'m10.4'</li><li>'m12.3'</li><li>'m12.4'</li><li>'m16.3'</li><li>'m16.4'</li><li>'m17.2'</li><li>'m17.3'</li><li>'m17.4'</li><li>'m18.4'</li><li>'m19.4'</li><li>'m25.4'</li><li>'m27.4'</li><li>'m28.1'</li><li>'m3.1'</li><li>'m9.4'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'm1.1'\n",
       "\\item 'm10.2'\n",
       "\\item 'm10.3'\n",
       "\\item 'm11.2'\n",
       "\\item 'm11.3'\n",
       "\\item 'm11.4'\n",
       "\\item 'm12.1'\n",
       "\\item 'm12.2'\n",
       "\\item 'm13.1'\n",
       "\\item 'm13.2'\n",
       "\\item 'm13.3'\n",
       "\\item 'm13.4'\n",
       "\\item 'm14.1'\n",
       "\\item 'm14.2'\n",
       "\\item 'm15.1'\n",
       "\\item 'm15.2'\n",
       "\\item 'm15.3'\n",
       "\\item 'm15.4'\n",
       "\\item 'm16.1'\n",
       "\\item 'm16.2'\n",
       "\\item 'm17.1'\n",
       "\\item 'm18.1'\n",
       "\\item 'm18.2'\n",
       "\\item 'm18.3'\n",
       "\\item 'm19.1'\n",
       "\\item 'm19.2'\n",
       "\\item 'm19.3'\n",
       "\\item 'm2.1'\n",
       "\\item 'm2.2'\n",
       "\\item 'm2.3'\n",
       "\\item 'm2.4'\n",
       "\\item 'm20.3'\n",
       "\\item 'm20.4'\n",
       "\\item 'm21.1'\n",
       "\\item 'm21.2'\n",
       "\\item 'm21.3'\n",
       "\\item 'm21.4'\n",
       "\\item 'm22.1'\n",
       "\\item 'm22.2'\n",
       "\\item 'm22.3'\n",
       "\\item 'm22.4'\n",
       "\\item 'm23.1'\n",
       "\\item 'm23.2'\n",
       "\\item 'm23.3'\n",
       "\\item 'm23.4'\n",
       "\\item 'm24.2'\n",
       "\\item 'm24.3'\n",
       "\\item 'm24.4'\n",
       "\\item 'm25.1'\n",
       "\\item 'm25.2'\n",
       "\\item 'm25.3'\n",
       "\\item 'm26.1'\n",
       "\\item 'm26.2'\n",
       "\\item 'm26.3'\n",
       "\\item 'm26.4'\n",
       "\\item 'm27.1'\n",
       "\\item 'm27.2'\n",
       "\\item 'm27.3'\n",
       "\\item 'm28.2'\n",
       "\\item 'm28.3'\n",
       "\\item 'm28.4'\n",
       "\\item 'm3.2'\n",
       "\\item 'm3.3'\n",
       "\\item 'm3.4'\n",
       "\\item 'm4.1'\n",
       "\\item 'm4.2'\n",
       "\\item 'm4.3'\n",
       "\\item 'm4.4'\n",
       "\\item 'm5.1'\n",
       "\\item 'm5.3'\n",
       "\\item 'm5.4'\n",
       "\\item 'm6.1'\n",
       "\\item 'm6.2'\n",
       "\\item 'm6.3'\n",
       "\\item 'm6.4'\n",
       "\\item 'm7.1'\n",
       "\\item 'm7.2'\n",
       "\\item 'm7.3'\n",
       "\\item 'm7.4'\n",
       "\\item 'm8.1'\n",
       "\\item 'm8.2'\n",
       "\\item 'm8.3'\n",
       "\\item 'm8.4'\n",
       "\\item 'm9.1'\n",
       "\\item 'm9.2'\n",
       "\\item 'm9.3'\n",
       "\\item 'm11.1'\n",
       "\\item 'm10.1'\n",
       "\\item 'm20.1'\n",
       "\\item 'm24.1'\n",
       "\\item 'm5.2'\n",
       "\\item 'm20.2'\n",
       "\\item 'm14.3'\n",
       "\\item 'm14.4'\n",
       "\\item 'm1.2'\n",
       "\\item 'm1.3'\n",
       "\\item 'm1.4'\n",
       "\\item 'm10.4'\n",
       "\\item 'm12.3'\n",
       "\\item 'm12.4'\n",
       "\\item 'm16.3'\n",
       "\\item 'm16.4'\n",
       "\\item 'm17.2'\n",
       "\\item 'm17.3'\n",
       "\\item 'm17.4'\n",
       "\\item 'm18.4'\n",
       "\\item 'm19.4'\n",
       "\\item 'm25.4'\n",
       "\\item 'm27.4'\n",
       "\\item 'm28.1'\n",
       "\\item 'm3.1'\n",
       "\\item 'm9.4'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'm1.1'\n",
       "2. 'm10.2'\n",
       "3. 'm10.3'\n",
       "4. 'm11.2'\n",
       "5. 'm11.3'\n",
       "6. 'm11.4'\n",
       "7. 'm12.1'\n",
       "8. 'm12.2'\n",
       "9. 'm13.1'\n",
       "10. 'm13.2'\n",
       "11. 'm13.3'\n",
       "12. 'm13.4'\n",
       "13. 'm14.1'\n",
       "14. 'm14.2'\n",
       "15. 'm15.1'\n",
       "16. 'm15.2'\n",
       "17. 'm15.3'\n",
       "18. 'm15.4'\n",
       "19. 'm16.1'\n",
       "20. 'm16.2'\n",
       "21. 'm17.1'\n",
       "22. 'm18.1'\n",
       "23. 'm18.2'\n",
       "24. 'm18.3'\n",
       "25. 'm19.1'\n",
       "26. 'm19.2'\n",
       "27. 'm19.3'\n",
       "28. 'm2.1'\n",
       "29. 'm2.2'\n",
       "30. 'm2.3'\n",
       "31. 'm2.4'\n",
       "32. 'm20.3'\n",
       "33. 'm20.4'\n",
       "34. 'm21.1'\n",
       "35. 'm21.2'\n",
       "36. 'm21.3'\n",
       "37. 'm21.4'\n",
       "38. 'm22.1'\n",
       "39. 'm22.2'\n",
       "40. 'm22.3'\n",
       "41. 'm22.4'\n",
       "42. 'm23.1'\n",
       "43. 'm23.2'\n",
       "44. 'm23.3'\n",
       "45. 'm23.4'\n",
       "46. 'm24.2'\n",
       "47. 'm24.3'\n",
       "48. 'm24.4'\n",
       "49. 'm25.1'\n",
       "50. 'm25.2'\n",
       "51. 'm25.3'\n",
       "52. 'm26.1'\n",
       "53. 'm26.2'\n",
       "54. 'm26.3'\n",
       "55. 'm26.4'\n",
       "56. 'm27.1'\n",
       "57. 'm27.2'\n",
       "58. 'm27.3'\n",
       "59. 'm28.2'\n",
       "60. 'm28.3'\n",
       "61. 'm28.4'\n",
       "62. 'm3.2'\n",
       "63. 'm3.3'\n",
       "64. 'm3.4'\n",
       "65. 'm4.1'\n",
       "66. 'm4.2'\n",
       "67. 'm4.3'\n",
       "68. 'm4.4'\n",
       "69. 'm5.1'\n",
       "70. 'm5.3'\n",
       "71. 'm5.4'\n",
       "72. 'm6.1'\n",
       "73. 'm6.2'\n",
       "74. 'm6.3'\n",
       "75. 'm6.4'\n",
       "76. 'm7.1'\n",
       "77. 'm7.2'\n",
       "78. 'm7.3'\n",
       "79. 'm7.4'\n",
       "80. 'm8.1'\n",
       "81. 'm8.2'\n",
       "82. 'm8.3'\n",
       "83. 'm8.4'\n",
       "84. 'm9.1'\n",
       "85. 'm9.2'\n",
       "86. 'm9.3'\n",
       "87. 'm11.1'\n",
       "88. 'm10.1'\n",
       "89. 'm20.1'\n",
       "90. 'm24.1'\n",
       "91. 'm5.2'\n",
       "92. 'm20.2'\n",
       "93. 'm14.3'\n",
       "94. 'm14.4'\n",
       "95. 'm1.2'\n",
       "96. 'm1.3'\n",
       "97. 'm1.4'\n",
       "98. 'm10.4'\n",
       "99. 'm12.3'\n",
       "100. 'm12.4'\n",
       "101. 'm16.3'\n",
       "102. 'm16.4'\n",
       "103. 'm17.2'\n",
       "104. 'm17.3'\n",
       "105. 'm17.4'\n",
       "106. 'm18.4'\n",
       "107. 'm19.4'\n",
       "108. 'm25.4'\n",
       "109. 'm27.4'\n",
       "110. 'm28.1'\n",
       "111. 'm3.1'\n",
       "112. 'm9.4'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"m1.1\"  \"m10.2\" \"m10.3\" \"m11.2\" \"m11.3\" \"m11.4\" \"m12.1\" \"m12.2\" \"m13.1\"\n",
       " [10] \"m13.2\" \"m13.3\" \"m13.4\" \"m14.1\" \"m14.2\" \"m15.1\" \"m15.2\" \"m15.3\" \"m15.4\"\n",
       " [19] \"m16.1\" \"m16.2\" \"m17.1\" \"m18.1\" \"m18.2\" \"m18.3\" \"m19.1\" \"m19.2\" \"m19.3\"\n",
       " [28] \"m2.1\"  \"m2.2\"  \"m2.3\"  \"m2.4\"  \"m20.3\" \"m20.4\" \"m21.1\" \"m21.2\" \"m21.3\"\n",
       " [37] \"m21.4\" \"m22.1\" \"m22.2\" \"m22.3\" \"m22.4\" \"m23.1\" \"m23.2\" \"m23.3\" \"m23.4\"\n",
       " [46] \"m24.2\" \"m24.3\" \"m24.4\" \"m25.1\" \"m25.2\" \"m25.3\" \"m26.1\" \"m26.2\" \"m26.3\"\n",
       " [55] \"m26.4\" \"m27.1\" \"m27.2\" \"m27.3\" \"m28.2\" \"m28.3\" \"m28.4\" \"m3.2\"  \"m3.3\" \n",
       " [64] \"m3.4\"  \"m4.1\"  \"m4.2\"  \"m4.3\"  \"m4.4\"  \"m5.1\"  \"m5.3\"  \"m5.4\"  \"m6.1\" \n",
       " [73] \"m6.2\"  \"m6.3\"  \"m6.4\"  \"m7.1\"  \"m7.2\"  \"m7.3\"  \"m7.4\"  \"m8.1\"  \"m8.2\" \n",
       " [82] \"m8.3\"  \"m8.4\"  \"m9.1\"  \"m9.2\"  \"m9.3\"  \"m11.1\" \"m10.1\" \"m20.1\" \"m24.1\"\n",
       " [91] \"m5.2\"  \"m20.2\" \"m14.3\" \"m14.4\" \"m1.2\"  \"m1.3\"  \"m1.4\"  \"m10.4\" \"m12.3\"\n",
       "[100] \"m12.4\" \"m16.3\" \"m16.4\" \"m17.2\" \"m17.3\" \"m17.4\" \"m18.4\" \"m19.4\" \"m25.4\"\n",
       "[109] \"m27.4\" \"m28.1\" \"m3.1\"  \"m9.4\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(data_long$sample_id)  # only acs samples with timepoints"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51e7b1c9-d2cc-451c-adbd-6e1db0314a57",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "2489aedb-b0e5-4551-824d-32022ccce2af",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Filter on expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "88830225-91ed-4cbf-9e95-fc7f87fde415",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Filter out features with low expression (as in E2 script)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "3e943b19-a354-4993-89f4-cfa3c4808fd9",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long$expression = TRUE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "d26209ca-574b-44e5-a3d3-9e4fb09b7479",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long$expression[data_long$value == 0] = FALSE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "bdb68a11-5c3f-4f8a-9df7-df078e36f206",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'type'. You can override using the\n",
      "`.groups` argument.\n"
     ]
    }
   ],
   "source": [
    "expression_filter = data_long %>% group_by(type, variable) %>% summarise(perc_expression = sum(expression)  )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "e956950f-2cf0-4ce5-b4a1-50d07c754ec2",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_filter$perc_expression = expression_filter$perc_expression / length(unique(data_long$sample_id))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "dd591883-966d-4452-a069-6643efc06aca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "13289"
      ],
      "text/latex": [
       "13289"
      ],
      "text/markdown": [
       "13289"
      ],
      "text/plain": [
       "[1] 13289"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(expression_filter)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "ebe460c4-3b93-481e-b49c-464a01495ccb",
   "metadata": {},
   "outputs": [],
   "source": [
    "filtered_out = expression_filter[expression_filter$perc_expression <= 0.2,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "617706f1-e0b5-4967-9701-ab15051148fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_filter = expression_filter[expression_filter$perc_expression > 0.2,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "ea3e17bf-f2e1-4d5f-9386-4b03edc590aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "13282"
      ],
      "text/latex": [
       "13282"
      ],
      "text/markdown": [
       "13282"
      ],
      "text/plain": [
       "[1] 13282"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(expression_filter)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "d2049aa5-cbd2-4b79-9d5c-856755e7921e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 7 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>type</th><th scope=col>variable</th><th scope=col>perc_expression</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cytokine</td><td>IL16__IL16 </td><td>0.07142857</td></tr>\n",
       "\t<tr><td>cytokine</td><td>IL21__IL21 </td><td>0.11607143</td></tr>\n",
       "\t<tr><td>cytokine</td><td>IL23__IL23 </td><td>0.11607143</td></tr>\n",
       "\t<tr><td>cytokine</td><td>IL28A__IL28</td><td>0.08928571</td></tr>\n",
       "\t<tr><td>cytokine</td><td>IL33__IL33 </td><td>0.13392857</td></tr>\n",
       "\t<tr><td>cytokine</td><td>IL4__IL4   </td><td>0.19642857</td></tr>\n",
       "\t<tr><td>cytokine</td><td>TSLP__TSLP </td><td>0.15178571</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 7 × 3\n",
       "\\begin{tabular}{lll}\n",
       " type & variable & perc\\_expression\\\\\n",
       " <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t cytokine & IL16\\_\\_IL16  & 0.07142857\\\\\n",
       "\t cytokine & IL21\\_\\_IL21  & 0.11607143\\\\\n",
       "\t cytokine & IL23\\_\\_IL23  & 0.11607143\\\\\n",
       "\t cytokine & IL28A\\_\\_IL28 & 0.08928571\\\\\n",
       "\t cytokine & IL33\\_\\_IL33  & 0.13392857\\\\\n",
       "\t cytokine & IL4\\_\\_IL4    & 0.19642857\\\\\n",
       "\t cytokine & TSLP\\_\\_TSLP  & 0.15178571\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 7 × 3\n",
       "\n",
       "| type &lt;chr&gt; | variable &lt;chr&gt; | perc_expression &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| cytokine | IL16__IL16  | 0.07142857 |\n",
       "| cytokine | IL21__IL21  | 0.11607143 |\n",
       "| cytokine | IL23__IL23  | 0.11607143 |\n",
       "| cytokine | IL28A__IL28 | 0.08928571 |\n",
       "| cytokine | IL33__IL33  | 0.13392857 |\n",
       "| cytokine | IL4__IL4    | 0.19642857 |\n",
       "| cytokine | TSLP__TSLP  | 0.15178571 |\n",
       "\n"
      ],
      "text/plain": [
       "  type     variable    perc_expression\n",
       "1 cytokine IL16__IL16  0.07142857     \n",
       "2 cytokine IL21__IL21  0.11607143     \n",
       "3 cytokine IL23__IL23  0.11607143     \n",
       "4 cytokine IL28A__IL28 0.08928571     \n",
       "5 cytokine IL33__IL33  0.13392857     \n",
       "6 cytokine IL4__IL4    0.19642857     \n",
       "7 cytokine TSLP__TSLP  0.15178571     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "filtered_out # features that are filtered out because of too low expressionv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "89c6e156-d4f6-4a7b-8456-ee97d5e43dcd",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# filter the data\n",
    "data_long = merge(data_long, expression_filter[,c('type', 'variable')], by.x = c('type', 'variable'), by.y = c('type', 'variable'))   # filter the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "4d945e56-3ccf-43bc-9d3d-e1d910234fe0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'clinical_data'</li><li>'cytokine'</li><li>'neutrophil'</li><li>'proteomics'</li><li>'single_cell'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'clinical\\_data'\n",
       "\\item 'cytokine'\n",
       "\\item 'neutrophil'\n",
       "\\item 'proteomics'\n",
       "\\item 'single\\_cell'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'clinical_data'\n",
       "2. 'cytokine'\n",
       "3. 'neutrophil'\n",
       "4. 'proteomics'\n",
       "5. 'single_cell'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"clinical_data\" \"cytokine\"      \"neutrophil\"    \"proteomics\"   \n",
       "[5] \"single_cell\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(data_long$type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "1d60252e-15ac-4313-ab12-aab0dcf2ceae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1128092"
      ],
      "text/latex": [
       "1128092"
      ],
      "text/markdown": [
       "1128092"
      ],
      "text/plain": [
       "[1] 1128092"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(data_long)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "6e3a9573-dd28-4c93-b328-c640a59028af",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long$ident = paste0(data_long$type, '_0_', data_long$variable)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d35c4176-5c09-480e-b03f-de43b6b3ad75",
   "metadata": {},
   "source": [
    "### Extract gene format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "b4f883a3-6029-494e-b98a-34d624d190a5",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Extract gene codes from variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "e0c25d9c-f1e4-4a37-b0c0-51995cb353ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long$gene = str_replace(data_long$variable, '.*__', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "f8f45fe2-b433-4337-b511-a54e75450808",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long$gene[data_long$type == 'proteomics'] = str_replace(data_long$gene[data_long$type == 'proteomics'], '_.*', '')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0dd83fb4-7d0a-43d8-bf7b-fb2a46e35d52",
   "metadata": {},
   "source": [
    "### Data filtered for ligands and receptors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "ebfcee7e-5738-4cab-b43d-f09621524114",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Filter dataset on ligand and receptor variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "0adc55fc-0a2b-4364-9d61-db00ccf0ad8e",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long_lig_rec = merge(data_long, ligand_receptor_vars)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "77a7905c-4a37-4438-9c4b-2299987a87ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 7 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>type</th><th scope=col>interaction_type</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cytokine   </td><td>Ligand  </td><td>60</td></tr>\n",
       "\t<tr><td>neutrophil </td><td>Ligand  </td><td>24</td></tr>\n",
       "\t<tr><td>neutrophil </td><td>Receptor</td><td>43</td></tr>\n",
       "\t<tr><td>proteomics </td><td>Ligand  </td><td>53</td></tr>\n",
       "\t<tr><td>proteomics </td><td>Receptor</td><td>19</td></tr>\n",
       "\t<tr><td>single_cell</td><td>Ligand  </td><td>53</td></tr>\n",
       "\t<tr><td>single_cell</td><td>Receptor</td><td>75</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 7 × 3\n",
       "\\begin{tabular}{lll}\n",
       " type & interaction\\_type & n\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t cytokine    & Ligand   & 60\\\\\n",
       "\t neutrophil  & Ligand   & 24\\\\\n",
       "\t neutrophil  & Receptor & 43\\\\\n",
       "\t proteomics  & Ligand   & 53\\\\\n",
       "\t proteomics  & Receptor & 19\\\\\n",
       "\t single\\_cell & Ligand   & 53\\\\\n",
       "\t single\\_cell & Receptor & 75\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 7 × 3\n",
       "\n",
       "| type &lt;chr&gt; | interaction_type &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| cytokine    | Ligand   | 60 |\n",
       "| neutrophil  | Ligand   | 24 |\n",
       "| neutrophil  | Receptor | 43 |\n",
       "| proteomics  | Ligand   | 53 |\n",
       "| proteomics  | Receptor | 19 |\n",
       "| single_cell | Ligand   | 53 |\n",
       "| single_cell | Receptor | 75 |\n",
       "\n"
      ],
      "text/plain": [
       "  type        interaction_type n \n",
       "1 cytokine    Ligand           60\n",
       "2 neutrophil  Ligand           24\n",
       "3 neutrophil  Receptor         43\n",
       "4 proteomics  Ligand           53\n",
       "5 proteomics  Receptor         19\n",
       "6 single_cell Ligand           53\n",
       "7 single_cell Receptor         75"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# overview amount of ligands and receptors measured per dimension \n",
    "unique(data_long_lig_rec[,c('type', 'interaction_type', 'gene')]) %>% group_by(type, interaction_type) %>% count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "a4785708-163c-402c-8d8a-08d8d5555c0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "## generate a variable defining type: cell-type cluster or dimension"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "2e9fcade-c699-4646-a6e3-e6bccfb23b72",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long_lig_rec$cluster = data_long_lig_rec$type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "dd28dd33-1a4f-441c-9ff0-84b7271e1bfc",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long_lig_rec$cluster[data_long_lig_rec$type == 'single_cell'] = str_replace(data_long_lig_rec$variable[data_long_lig_rec$type == 'single_cell'] , '__.*', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "7b95d801-9a91-46c2-82f7-962036fe2254",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'proteomics'</li><li>'neutrophil'</li><li>'X6_Monocytes...CD14'</li><li>'X4_Monocytes...CD14'</li><li>'X7_Monocytes...CD14'</li><li>'X12_Monocytes...CD16_FCGR3A'</li><li>'X3_NK'</li><li>'X0_T.cell.CD4'</li><li>'X10_B.cell'</li><li>'X13_Dendritic'</li><li>'X5_T.cell.CD4'</li><li>'X9_Monocytes...CD16_FCGR3A'</li><li>'X11_T.cell.CD4'</li><li>'X1_T.cell.CD8'</li><li>'X2_T.cell.CD4'</li><li>'X8_B.cell'</li><li>'cytokine'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'proteomics'\n",
       "\\item 'neutrophil'\n",
       "\\item 'X6\\_Monocytes...CD14'\n",
       "\\item 'X4\\_Monocytes...CD14'\n",
       "\\item 'X7\\_Monocytes...CD14'\n",
       "\\item 'X12\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'X3\\_NK'\n",
       "\\item 'X0\\_T.cell.CD4'\n",
       "\\item 'X10\\_B.cell'\n",
       "\\item 'X13\\_Dendritic'\n",
       "\\item 'X5\\_T.cell.CD4'\n",
       "\\item 'X9\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'X11\\_T.cell.CD4'\n",
       "\\item 'X1\\_T.cell.CD8'\n",
       "\\item 'X2\\_T.cell.CD4'\n",
       "\\item 'X8\\_B.cell'\n",
       "\\item 'cytokine'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'proteomics'\n",
       "2. 'neutrophil'\n",
       "3. 'X6_Monocytes...CD14'\n",
       "4. 'X4_Monocytes...CD14'\n",
       "5. 'X7_Monocytes...CD14'\n",
       "6. 'X12_Monocytes...CD16_FCGR3A'\n",
       "7. 'X3_NK'\n",
       "8. 'X0_T.cell.CD4'\n",
       "9. 'X10_B.cell'\n",
       "10. 'X13_Dendritic'\n",
       "11. 'X5_T.cell.CD4'\n",
       "12. 'X9_Monocytes...CD16_FCGR3A'\n",
       "13. 'X11_T.cell.CD4'\n",
       "14. 'X1_T.cell.CD8'\n",
       "15. 'X2_T.cell.CD4'\n",
       "16. 'X8_B.cell'\n",
       "17. 'cytokine'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"proteomics\"                  \"neutrophil\"                 \n",
       " [3] \"X6_Monocytes...CD14\"         \"X4_Monocytes...CD14\"        \n",
       " [5] \"X7_Monocytes...CD14\"         \"X12_Monocytes...CD16_FCGR3A\"\n",
       " [7] \"X3_NK\"                       \"X0_T.cell.CD4\"              \n",
       " [9] \"X10_B.cell\"                  \"X13_Dendritic\"              \n",
       "[11] \"X5_T.cell.CD4\"               \"X9_Monocytes...CD16_FCGR3A\" \n",
       "[13] \"X11_T.cell.CD4\"              \"X1_T.cell.CD8\"              \n",
       "[15] \"X2_T.cell.CD4\"               \"X8_B.cell\"                  \n",
       "[17] \"cytokine\"                   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(data_long_lig_rec$cluster)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "9a581d85-11e0-4a7b-a3f0-532f858c6ca4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'proteomics'</li><li>'neutrophil'</li><li>'single_cell'</li><li>'cytokine'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'proteomics'\n",
       "\\item 'neutrophil'\n",
       "\\item 'single\\_cell'\n",
       "\\item 'cytokine'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'proteomics'\n",
       "2. 'neutrophil'\n",
       "3. 'single_cell'\n",
       "4. 'cytokine'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"proteomics\"  \"neutrophil\"  \"single_cell\" \"cytokine\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(data_long_lig_rec$type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "be0fb4a4-1639-4844-b8ab-ef1d08374d84",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>X</th><th scope=col>sample_id</th><th scope=col>value</th><th scope=col>config</th><th scope=col>expression</th><th scope=col>ident</th><th scope=col>interaction_type</th><th scope=col>cluster</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A2M</td><td>proteomics</td><td>A2M_P01023</td><td>1439457</td><td>m25.1</td><td>8.823516</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>proteomics_0_A2M_P01023</td><td>Ligand</td><td>proteomics</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A2M</td><td>proteomics</td><td>A2M_P01023</td><td>1439527</td><td>m4.3 </td><td>9.797027</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>proteomics_0_A2M_P01023</td><td>Ligand</td><td>proteomics</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & gene & type & variable & X & sample\\_id & value & config & expression & ident & interaction\\_type & cluster\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <chr> & <dbl> & <chr> & <lgl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & A2M & proteomics & A2M\\_P01023 & 1439457 & m25.1 & 8.823516 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & proteomics\\_0\\_A2M\\_P01023 & Ligand & proteomics\\\\\n",
       "\t2 & A2M & proteomics & A2M\\_P01023 & 1439527 & m4.3  & 9.797027 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & proteomics\\_0\\_A2M\\_P01023 & Ligand & proteomics\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 11\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | X &lt;int&gt; | sample_id &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; | expression &lt;lgl&gt; | ident &lt;chr&gt; | interaction_type &lt;chr&gt; | cluster &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | A2M | proteomics | A2M_P01023 | 1439457 | m25.1 | 8.823516 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | proteomics_0_A2M_P01023 | Ligand | proteomics |\n",
       "| 2 | A2M | proteomics | A2M_P01023 | 1439527 | m4.3  | 9.797027 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | proteomics_0_A2M_P01023 | Ligand | proteomics |\n",
       "\n"
      ],
      "text/plain": [
       "  gene type       variable   X       sample_id value   \n",
       "1 A2M  proteomics A2M_P01023 1439457 m25.1     8.823516\n",
       "2 A2M  proteomics A2M_P01023 1439527 m4.3      9.797027\n",
       "  config                           expression ident                  \n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       proteomics_0_A2M_P01023\n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       proteomics_0_A2M_P01023\n",
       "  interaction_type cluster   \n",
       "1 Ligand           proteomics\n",
       "2 Ligand           proteomics"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long_lig_rec,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2153557b-7a0a-4586-976a-6195d0f0c140",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "1ac46b83-0218-4cee-8740-526b4ab7b4f1",
   "metadata": {},
   "source": [
    "### Add information about percentage of cells expressing gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "782471a9-1e1b-4ec3-89c1-20a0ac68d508",
   "metadata": {},
   "outputs": [],
   "source": [
    "## add to the ligand receptor dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "1935e668-890d-4234-8cf4-a56b5e797af0",
   "metadata": {},
   "outputs": [],
   "source": [
    "cell_perc_cluster$cluster = make.names(cell_perc_cluster$cluster)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "4fdcb4c4-abd1-4f64-905f-302dc0e91506",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "69043"
      ],
      "text/latex": [
       "69043"
      ],
      "text/markdown": [
       "69043"
      ],
      "text/plain": [
       "[1] 69043"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(data_long_lig_rec)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "869492eb-1f28-4c31-8635-6779d74dfb42",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_long_lig_rec$X = NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "4dca2569-7a5f-4740-85fa-86f0b7803789",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th><th scope=col>gene</th><th scope=col>cluster</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AL627309.1</td><td>0.24327612</td><td>18</td><td>AL627309.1</td><td>X8_B.cell</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AL627309.4</td><td>0.05406136</td><td> 4</td><td>AL627309.4</td><td>X8_B.cell</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & X & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene & gene & cluster\\\\\n",
       "  & <chr> & <dbl> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & AL627309.1 & 0.24327612 & 18 & AL627309.1 & X8\\_B.cell\\\\\n",
       "\t2 & AL627309.4 & 0.05406136 &  4 & AL627309.4 & X8\\_B.cell\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 5\n",
       "\n",
       "| <!--/--> | X &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; | gene &lt;chr&gt; | cluster &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | AL627309.1 | 0.24327612 | 18 | AL627309.1 | X8_B.cell |\n",
       "| 2 | AL627309.4 | 0.05406136 |  4 | AL627309.4 | X8_B.cell |\n",
       "\n"
      ],
      "text/plain": [
       "  X          perc_cells_expressing_gene total_amount_cells_expressing_gene\n",
       "1 AL627309.1 0.24327612                 18                                \n",
       "2 AL627309.4 0.05406136                  4                                \n",
       "  gene       cluster  \n",
       "1 AL627309.1 X8_B.cell\n",
       "2 AL627309.4 X8_B.cell"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cell_perc_cluster,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "8778b21e-f755-4555-afc0-d102aabe82b0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>sample_id</th><th scope=col>value</th><th scope=col>config</th><th scope=col>expression</th><th scope=col>ident</th><th scope=col>interaction_type</th><th scope=col>cluster</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A2M</td><td>proteomics</td><td>A2M_P01023</td><td>m25.1</td><td>8.823516</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>proteomics_0_A2M_P01023</td><td>Ligand</td><td>proteomics</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A2M</td><td>proteomics</td><td>A2M_P01023</td><td>m4.3 </td><td>9.797027</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>proteomics_0_A2M_P01023</td><td>Ligand</td><td>proteomics</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 10\n",
       "\\begin{tabular}{r|llllllllll}\n",
       "  & gene & type & variable & sample\\_id & value & config & expression & ident & interaction\\_type & cluster\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <lgl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & A2M & proteomics & A2M\\_P01023 & m25.1 & 8.823516 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & proteomics\\_0\\_A2M\\_P01023 & Ligand & proteomics\\\\\n",
       "\t2 & A2M & proteomics & A2M\\_P01023 & m4.3  & 9.797027 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & proteomics\\_0\\_A2M\\_P01023 & Ligand & proteomics\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 10\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | sample_id &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; | expression &lt;lgl&gt; | ident &lt;chr&gt; | interaction_type &lt;chr&gt; | cluster &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | A2M | proteomics | A2M_P01023 | m25.1 | 8.823516 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | proteomics_0_A2M_P01023 | Ligand | proteomics |\n",
       "| 2 | A2M | proteomics | A2M_P01023 | m4.3  | 9.797027 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | proteomics_0_A2M_P01023 | Ligand | proteomics |\n",
       "\n"
      ],
      "text/plain": [
       "  gene type       variable   sample_id value   \n",
       "1 A2M  proteomics A2M_P01023 m25.1     8.823516\n",
       "2 A2M  proteomics A2M_P01023 m4.3      9.797027\n",
       "  config                           expression ident                  \n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       proteomics_0_A2M_P01023\n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       proteomics_0_A2M_P01023\n",
       "  interaction_type cluster   \n",
       "1 Ligand           proteomics\n",
       "2 Ligand           proteomics"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long_lig_rec,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "9bf63eae-93cc-48e5-9da8-b7d0bd2da112",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'proteomics'</li><li>'neutrophil'</li><li>'X6_Monocytes...CD14'</li><li>'X4_Monocytes...CD14'</li><li>'X7_Monocytes...CD14'</li><li>'X12_Monocytes...CD16_FCGR3A'</li><li>'X3_NK'</li><li>'X0_T.cell.CD4'</li><li>'X10_B.cell'</li><li>'X13_Dendritic'</li><li>'X5_T.cell.CD4'</li><li>'X9_Monocytes...CD16_FCGR3A'</li><li>'X11_T.cell.CD4'</li><li>'X1_T.cell.CD8'</li><li>'X2_T.cell.CD4'</li><li>'X8_B.cell'</li><li>'cytokine'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'proteomics'\n",
       "\\item 'neutrophil'\n",
       "\\item 'X6\\_Monocytes...CD14'\n",
       "\\item 'X4\\_Monocytes...CD14'\n",
       "\\item 'X7\\_Monocytes...CD14'\n",
       "\\item 'X12\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'X3\\_NK'\n",
       "\\item 'X0\\_T.cell.CD4'\n",
       "\\item 'X10\\_B.cell'\n",
       "\\item 'X13\\_Dendritic'\n",
       "\\item 'X5\\_T.cell.CD4'\n",
       "\\item 'X9\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'X11\\_T.cell.CD4'\n",
       "\\item 'X1\\_T.cell.CD8'\n",
       "\\item 'X2\\_T.cell.CD4'\n",
       "\\item 'X8\\_B.cell'\n",
       "\\item 'cytokine'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'proteomics'\n",
       "2. 'neutrophil'\n",
       "3. 'X6_Monocytes...CD14'\n",
       "4. 'X4_Monocytes...CD14'\n",
       "5. 'X7_Monocytes...CD14'\n",
       "6. 'X12_Monocytes...CD16_FCGR3A'\n",
       "7. 'X3_NK'\n",
       "8. 'X0_T.cell.CD4'\n",
       "9. 'X10_B.cell'\n",
       "10. 'X13_Dendritic'\n",
       "11. 'X5_T.cell.CD4'\n",
       "12. 'X9_Monocytes...CD16_FCGR3A'\n",
       "13. 'X11_T.cell.CD4'\n",
       "14. 'X1_T.cell.CD8'\n",
       "15. 'X2_T.cell.CD4'\n",
       "16. 'X8_B.cell'\n",
       "17. 'cytokine'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"proteomics\"                  \"neutrophil\"                 \n",
       " [3] \"X6_Monocytes...CD14\"         \"X4_Monocytes...CD14\"        \n",
       " [5] \"X7_Monocytes...CD14\"         \"X12_Monocytes...CD16_FCGR3A\"\n",
       " [7] \"X3_NK\"                       \"X0_T.cell.CD4\"              \n",
       " [9] \"X10_B.cell\"                  \"X13_Dendritic\"              \n",
       "[11] \"X5_T.cell.CD4\"               \"X9_Monocytes...CD16_FCGR3A\" \n",
       "[13] \"X11_T.cell.CD4\"              \"X1_T.cell.CD8\"              \n",
       "[15] \"X2_T.cell.CD4\"               \"X8_B.cell\"                  \n",
       "[17] \"cytokine\"                   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(data_long_lig_rec$cluster)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "b5640704-df29-4d66-9006-eb1c0debc903",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'X8_B.cell'</li><li>'X9_Monocytes...CD16_FCGR3A'</li><li>'X5_T.cell.CD4'</li><li>'X0_T.cell.CD4'</li><li>'X1_T.cell.CD8'</li><li>'X6_Monocytes...CD14'</li><li>'X2_T.cell.CD4'</li><li>'X4_Monocytes...CD14'</li><li>'X11_T.cell.CD4'</li><li>'X3_NK'</li><li>'X7_Monocytes...CD14'</li><li>'X13_Dendritic'</li><li>'X10_B.cell'</li><li>'X12_Monocytes...CD16_FCGR3A'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'X8\\_B.cell'\n",
       "\\item 'X9\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'X5\\_T.cell.CD4'\n",
       "\\item 'X0\\_T.cell.CD4'\n",
       "\\item 'X1\\_T.cell.CD8'\n",
       "\\item 'X6\\_Monocytes...CD14'\n",
       "\\item 'X2\\_T.cell.CD4'\n",
       "\\item 'X4\\_Monocytes...CD14'\n",
       "\\item 'X11\\_T.cell.CD4'\n",
       "\\item 'X3\\_NK'\n",
       "\\item 'X7\\_Monocytes...CD14'\n",
       "\\item 'X13\\_Dendritic'\n",
       "\\item 'X10\\_B.cell'\n",
       "\\item 'X12\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'X8_B.cell'\n",
       "2. 'X9_Monocytes...CD16_FCGR3A'\n",
       "3. 'X5_T.cell.CD4'\n",
       "4. 'X0_T.cell.CD4'\n",
       "5. 'X1_T.cell.CD8'\n",
       "6. 'X6_Monocytes...CD14'\n",
       "7. 'X2_T.cell.CD4'\n",
       "8. 'X4_Monocytes...CD14'\n",
       "9. 'X11_T.cell.CD4'\n",
       "10. 'X3_NK'\n",
       "11. 'X7_Monocytes...CD14'\n",
       "12. 'X13_Dendritic'\n",
       "13. 'X10_B.cell'\n",
       "14. 'X12_Monocytes...CD16_FCGR3A'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"X8_B.cell\"                   \"X9_Monocytes...CD16_FCGR3A\" \n",
       " [3] \"X5_T.cell.CD4\"               \"X0_T.cell.CD4\"              \n",
       " [5] \"X1_T.cell.CD8\"               \"X6_Monocytes...CD14\"        \n",
       " [7] \"X2_T.cell.CD4\"               \"X4_Monocytes...CD14\"        \n",
       " [9] \"X11_T.cell.CD4\"              \"X3_NK\"                      \n",
       "[11] \"X7_Monocytes...CD14\"         \"X13_Dendritic\"              \n",
       "[13] \"X10_B.cell\"                  \"X12_Monocytes...CD16_FCGR3A\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(cell_perc_cluster$cluster)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "51ee18fa-f610-44c4-8220-6a1bd90709ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "## merge the data\n",
    "data_long_lig_rec = merge(data_long_lig_rec, cell_perc_cluster, all.x = TRUE) # also keep cytokine, proteomics dimension where no cell expression is available"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "e4a5aac7-2226-4f13-9151-cf6ed7d778f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>cluster</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>sample_id</th><th scope=col>value</th><th scope=col>config</th><th scope=col>expression</th><th scope=col>ident</th><th scope=col>interaction_type</th><th scope=col>X</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A2M</td><td>proteomics</td><td>proteomics</td><td>A2M_P01023</td><td>m25.1</td><td>8.823516</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>proteomics_0_A2M_P01023</td><td>Ligand</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A2M</td><td>proteomics</td><td>proteomics</td><td>A2M_P01023</td><td>m4.3 </td><td>9.797027</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>proteomics_0_A2M_P01023</td><td>Ligand</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 13\n",
       "\\begin{tabular}{r|lllllllllllll}\n",
       "  & gene & cluster & type & variable & sample\\_id & value & config & expression & ident & interaction\\_type & X & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <lgl> & <chr> & <chr> & <chr> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & A2M & proteomics & proteomics & A2M\\_P01023 & m25.1 & 8.823516 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & proteomics\\_0\\_A2M\\_P01023 & Ligand & NA & NA & NA\\\\\n",
       "\t2 & A2M & proteomics & proteomics & A2M\\_P01023 & m4.3  & 9.797027 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & proteomics\\_0\\_A2M\\_P01023 & Ligand & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 13\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | cluster &lt;chr&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | sample_id &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; | expression &lt;lgl&gt; | ident &lt;chr&gt; | interaction_type &lt;chr&gt; | X &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | A2M | proteomics | proteomics | A2M_P01023 | m25.1 | 8.823516 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | proteomics_0_A2M_P01023 | Ligand | NA | NA | NA |\n",
       "| 2 | A2M | proteomics | proteomics | A2M_P01023 | m4.3  | 9.797027 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | proteomics_0_A2M_P01023 | Ligand | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  gene cluster    type       variable   sample_id value   \n",
       "1 A2M  proteomics proteomics A2M_P01023 m25.1     8.823516\n",
       "2 A2M  proteomics proteomics A2M_P01023 m4.3      9.797027\n",
       "  config                           expression ident                  \n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       proteomics_0_A2M_P01023\n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       proteomics_0_A2M_P01023\n",
       "  interaction_type X  perc_cells_expressing_gene\n",
       "1 Ligand           NA NA                        \n",
       "2 Ligand           NA NA                        \n",
       "  total_amount_cells_expressing_gene\n",
       "1 NA                                \n",
       "2 NA                                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long_lig_rec,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "35c967b1-d7ab-4c3a-ace7-7fad1a1ff49a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "ade437c8-39b7-4f46-8b61-71720b4070d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set dummy mean expression for proteomic/ cytokine data etc. where no mean cell expression values are available\n",
    "data_long_lig_rec$perc_cells_expressing_gene[is.na(data_long_lig_rec$perc_cells_expressing_gene)] = mean(data_long_lig_rec$perc_cells_expressing_gene, na.rm = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "982485cc-c310-47b3-9a8c-7c3a4d6aaba8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'proteomics'</li><li>'neutrophil'</li><li>'single_cell'</li><li>'cytokine'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'proteomics'\n",
       "\\item 'neutrophil'\n",
       "\\item 'single\\_cell'\n",
       "\\item 'cytokine'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'proteomics'\n",
       "2. 'neutrophil'\n",
       "3. 'single_cell'\n",
       "4. 'cytokine'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"proteomics\"  \"neutrophil\"  \"single_cell\" \"cytokine\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(data_long_lig_rec$type)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f2bbce85-8549-4b83-ad5f-a2c238c7a6fc",
   "metadata": {},
   "source": [
    "###  Aggregate expression across samples (mean expression etc.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "3ea9ca57-2819-48c1-8676-b3e1616d206d",
   "metadata": {},
   "outputs": [],
   "source": [
    "## for ligand- receptor data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "7701529a-4b65-4741-8704-7bf7fd3539b1",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'type', 'interaction_type', 'gene',\n",
      "'cluster', 'perc_cells_expressing_gene', 'total_amount_cells_expressing_gene'.\n",
      "You can override using the `.groups` argument.\n"
     ]
    }
   ],
   "source": [
    "### Calculate variance and mean expression per cell-type across samples for single cell data\n",
    "\n",
    "analyze_expr_cluster = data_long_lig_rec %>% group_by(type, interaction_type, gene, cluster, perc_cells_expressing_gene, total_amount_cells_expressing_gene, ident) %>% summarise(mean_expr = mean(value, na.rm = TRUE), var_expression = var(value, na.rm = TRUE))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "ee80fe61-1eea-4e55-923c-3bf097e95292",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Generate cell type columns\n",
    "analyze_expr_cluster$cell_type = str_replace(analyze_expr_cluster$cluster, 'X.*_', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "6fb55d7a-38ad-463b-967b-aa8f30cf1a46",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 2 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>type</th><th scope=col>interaction_type</th><th scope=col>gene</th><th scope=col>cluster</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th><th scope=col>ident</th><th scope=col>mean_expr</th><th scope=col>var_expression</th><th scope=col>cell_type</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cytokine</td><td>Ligand</td><td>CCL1 </td><td>cytokine</td><td>69.10381</td><td>NA</td><td>cytokine_0_I309__CCL1    </td><td>1.381370</td><td>0.2566531</td><td>cytokine</td></tr>\n",
       "\t<tr><td>cytokine</td><td>Ligand</td><td>CCL11</td><td>cytokine</td><td>69.10381</td><td>NA</td><td>cytokine_0_Eotaxin__CCL11</td><td>7.376516</td><td>2.8469700</td><td>cytokine</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 2 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " type & interaction\\_type & gene & cluster & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene & ident & mean\\_expr & var\\_expression & cell\\_type\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <dbl> & <int> & <chr> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t cytokine & Ligand & CCL1  & cytokine & 69.10381 & NA & cytokine\\_0\\_I309\\_\\_CCL1     & 1.381370 & 0.2566531 & cytokine\\\\\n",
       "\t cytokine & Ligand & CCL11 & cytokine & 69.10381 & NA & cytokine\\_0\\_Eotaxin\\_\\_CCL11 & 7.376516 & 2.8469700 & cytokine\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 2 × 10\n",
       "\n",
       "| type &lt;chr&gt; | interaction_type &lt;chr&gt; | gene &lt;chr&gt; | cluster &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; | ident &lt;chr&gt; | mean_expr &lt;dbl&gt; | var_expression &lt;dbl&gt; | cell_type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| cytokine | Ligand | CCL1  | cytokine | 69.10381 | NA | cytokine_0_I309__CCL1     | 1.381370 | 0.2566531 | cytokine |\n",
       "| cytokine | Ligand | CCL11 | cytokine | 69.10381 | NA | cytokine_0_Eotaxin__CCL11 | 7.376516 | 2.8469700 | cytokine |\n",
       "\n"
      ],
      "text/plain": [
       "  type     interaction_type gene  cluster  perc_cells_expressing_gene\n",
       "1 cytokine Ligand           CCL1  cytokine 69.10381                  \n",
       "2 cytokine Ligand           CCL11 cytokine 69.10381                  \n",
       "  total_amount_cells_expressing_gene ident                     mean_expr\n",
       "1 NA                                 cytokine_0_I309__CCL1     1.381370 \n",
       "2 NA                                 cytokine_0_Eotaxin__CCL11 7.376516 \n",
       "  var_expression cell_type\n",
       "1 0.2566531      cytokine \n",
       "2 2.8469700      cytokine "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(analyze_expr_cluster,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "7f2587e7-cee6-4c31-895d-7ed39d2d7ef2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cytokine'</li><li>'neutrophil'</li><li>'proteomics'</li><li>'Monocytes...CD14'</li><li>'T.cell.CD4'</li><li>'B.cell'</li><li>'FCGR3A'</li><li>'Dendritic'</li><li>'T.cell.CD8'</li><li>'NK'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cytokine'\n",
       "\\item 'neutrophil'\n",
       "\\item 'proteomics'\n",
       "\\item 'Monocytes...CD14'\n",
       "\\item 'T.cell.CD4'\n",
       "\\item 'B.cell'\n",
       "\\item 'FCGR3A'\n",
       "\\item 'Dendritic'\n",
       "\\item 'T.cell.CD8'\n",
       "\\item 'NK'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cytokine'\n",
       "2. 'neutrophil'\n",
       "3. 'proteomics'\n",
       "4. 'Monocytes...CD14'\n",
       "5. 'T.cell.CD4'\n",
       "6. 'B.cell'\n",
       "7. 'FCGR3A'\n",
       "8. 'Dendritic'\n",
       "9. 'T.cell.CD8'\n",
       "10. 'NK'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"cytokine\"         \"neutrophil\"       \"proteomics\"       \"Monocytes...CD14\"\n",
       " [5] \"T.cell.CD4\"       \"B.cell\"           \"FCGR3A\"           \"Dendritic\"       \n",
       " [9] \"T.cell.CD8\"       \"NK\"              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(analyze_expr_cluster$cell_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cf7283ea-9539-4e52-8e34-b378eec63a39",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "7efac83d-87b6-43d5-a4ac-37a087f4d6b5",
   "metadata": {},
   "source": [
    "# Ligand Receptor Data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4225d21f-ddaf-412c-bcf3-d3d503552069",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Define ligand and receptors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "de1533c6-c539-477d-8c80-0c29c0f10a24",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>cluster</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>sample_id</th><th scope=col>value</th><th scope=col>config</th><th scope=col>expression</th><th scope=col>ident</th><th scope=col>interaction_type</th><th scope=col>X</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A2M</td><td>proteomics</td><td>proteomics</td><td>A2M_P01023</td><td>m25.1</td><td>8.823516</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>proteomics_0_A2M_P01023</td><td>Ligand</td><td>NA</td><td>69.10381</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A2M</td><td>proteomics</td><td>proteomics</td><td>A2M_P01023</td><td>m4.3 </td><td>9.797027</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>proteomics_0_A2M_P01023</td><td>Ligand</td><td>NA</td><td>69.10381</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 13\n",
       "\\begin{tabular}{r|lllllllllllll}\n",
       "  & gene & cluster & type & variable & sample\\_id & value & config & expression & ident & interaction\\_type & X & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <lgl> & <chr> & <chr> & <chr> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & A2M & proteomics & proteomics & A2M\\_P01023 & m25.1 & 8.823516 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & proteomics\\_0\\_A2M\\_P01023 & Ligand & NA & 69.10381 & NA\\\\\n",
       "\t2 & A2M & proteomics & proteomics & A2M\\_P01023 & m4.3  & 9.797027 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & proteomics\\_0\\_A2M\\_P01023 & Ligand & NA & 69.10381 & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 13\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | cluster &lt;chr&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | sample_id &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; | expression &lt;lgl&gt; | ident &lt;chr&gt; | interaction_type &lt;chr&gt; | X &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | A2M | proteomics | proteomics | A2M_P01023 | m25.1 | 8.823516 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | proteomics_0_A2M_P01023 | Ligand | NA | 69.10381 | NA |\n",
       "| 2 | A2M | proteomics | proteomics | A2M_P01023 | m4.3  | 9.797027 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | proteomics_0_A2M_P01023 | Ligand | NA | 69.10381 | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  gene cluster    type       variable   sample_id value   \n",
       "1 A2M  proteomics proteomics A2M_P01023 m25.1     8.823516\n",
       "2 A2M  proteomics proteomics A2M_P01023 m4.3      9.797027\n",
       "  config                           expression ident                  \n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       proteomics_0_A2M_P01023\n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       proteomics_0_A2M_P01023\n",
       "  interaction_type X  perc_cells_expressing_gene\n",
       "1 Ligand           NA 69.10381                  \n",
       "2 Ligand           NA 69.10381                  \n",
       "  total_amount_cells_expressing_gene\n",
       "1 NA                                \n",
       "2 NA                                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long_lig_rec,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "482976fe-9fde-497b-aa0f-952b9ec58f4c",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Define Ligands and Receptors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "ecd17d9e-af13-4175-be76-156c2cf8eb13",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligands = unique(data_long_lig_rec$ident[data_long_lig_rec$interaction_type == 'Ligand'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "2115d5d7-db36-4946-a802-b311f6eef889",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "444"
      ],
      "text/latex": [
       "444"
      ],
      "text/markdown": [
       "444"
      ],
      "text/plain": [
       "[1] 444"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(ligands)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "b0203c53-2283-4a70-ab53-810c0d4e0187",
   "metadata": {},
   "outputs": [],
   "source": [
    "receptors = unique(data_long_lig_rec$ident[data_long_lig_rec$interaction_type == 'Receptor'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "14faf058-8aa4-461d-acb5-c12228f4c0e0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "367"
      ],
      "text/latex": [
       "367"
      ],
      "text/markdown": [
       "367"
      ],
      "text/plain": [
       "[1] 367"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(receptors)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d88a1d4d-d554-4d8b-9317-18cc7181c3b4",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Add nichenet interaction potential"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "b5308b81-87f5-47e2-9ae3-c6d6107048fb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Ligand</th><th scope=col>Receptor</th><th scope=col>amount_sources</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>A2M</td><td>LRP1    </td><td>1</td></tr>\n",
       "\t<tr><td>A2M</td><td>TNFRSF14</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 2 × 3\n",
       "\\begin{tabular}{lll}\n",
       " Ligand & Receptor & amount\\_sources\\\\\n",
       " <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t A2M & LRP1     & 1\\\\\n",
       "\t A2M & TNFRSF14 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 2 × 3\n",
       "\n",
       "| Ligand &lt;chr&gt; | Receptor &lt;chr&gt; | amount_sources &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| A2M | LRP1     | 1 |\n",
       "| A2M | TNFRSF14 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand Receptor amount_sources\n",
       "1 A2M    LRP1     1             \n",
       "2 A2M    TNFRSF14 1             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_receptor_interactions,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "38a05c1f-6155-4441-a5b5-c37585f4e114",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>from</th><th scope=col>to</th><th scope=col>weight</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>A1BG</td><td>ABCC6</td><td>0.4216439</td></tr>\n",
       "\t<tr><td>A1BG</td><td>ACE2 </td><td>0.1007411</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 2 × 3\n",
       "\\begin{tabular}{lll}\n",
       " from & to & weight\\\\\n",
       " <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t A1BG & ABCC6 & 0.4216439\\\\\n",
       "\t A1BG & ACE2  & 0.1007411\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 2 × 3\n",
       "\n",
       "| from &lt;chr&gt; | to &lt;chr&gt; | weight &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| A1BG | ABCC6 | 0.4216439 |\n",
       "| A1BG | ACE2  | 0.1007411 |\n",
       "\n"
      ],
      "text/plain": [
       "  from to    weight   \n",
       "1 A1BG ABCC6 0.4216439\n",
       "2 A1BG ACE2  0.1007411"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(weighted_networks$lr_sig,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "790487df-a457-4376-ad3a-7a45bc39e1d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# merge the two datasets\n",
    "ligand_receptor_interactions =merge( ligand_receptor_interactions, weighted_networks$lr_sig, by.x = c('Ligand', 'Receptor'), by.y = c('from','to'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "52b72696-4a7c-45c4-bbf2-6c32ef3b82cb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "11639"
      ],
      "text/latex": [
       "11639"
      ],
      "text/markdown": [
       "11639"
      ],
      "text/plain": [
       "[1] 11639"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_receptor_interactions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "5a671e36-a121-4654-a3dc-90334cce4a6d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Ligand</th><th scope=col>Receptor</th><th scope=col>amount_sources</th><th scope=col>weight</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A2M</td><td>LRP1    </td><td>1</td><td>0.4678977</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A2M</td><td>TNFRSF14</td><td>1</td><td>0.6990671</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & Ligand & Receptor & amount\\_sources & weight\\\\\n",
       "  & <chr> & <chr> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & A2M & LRP1     & 1 & 0.4678977\\\\\n",
       "\t2 & A2M & TNFRSF14 & 1 & 0.6990671\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 4\n",
       "\n",
       "| <!--/--> | Ligand &lt;chr&gt; | Receptor &lt;chr&gt; | amount_sources &lt;int&gt; | weight &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | A2M | LRP1     | 1 | 0.4678977 |\n",
       "| 2 | A2M | TNFRSF14 | 1 | 0.6990671 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand Receptor amount_sources weight   \n",
       "1 A2M    LRP1     1              0.4678977\n",
       "2 A2M    TNFRSF14 1              0.6990671"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_receptor_interactions,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "36033e6e-1ca2-4b40-92cc-f4149450897b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Add expression data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "c077fa5b-ae17-4fd7-a3b6-abc50cb8c559",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cytokine'</li><li>'neutrophil'</li><li>'proteomics'</li><li>'single_cell'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cytokine'\n",
       "\\item 'neutrophil'\n",
       "\\item 'proteomics'\n",
       "\\item 'single\\_cell'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cytokine'\n",
       "2. 'neutrophil'\n",
       "3. 'proteomics'\n",
       "4. 'single_cell'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"cytokine\"    \"neutrophil\"  \"proteomics\"  \"single_cell\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(analyze_expr_cluster$type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "e361930a-1928-48a0-8edb-ca9bd0429f1e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 2 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>type</th><th scope=col>interaction_type</th><th scope=col>gene</th><th scope=col>cluster</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th><th scope=col>ident</th><th scope=col>mean_expr</th><th scope=col>var_expression</th><th scope=col>cell_type</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>cytokine</td><td>Ligand</td><td>CCL1 </td><td>cytokine</td><td>69.10381</td><td>NA</td><td>cytokine_0_I309__CCL1    </td><td>1.381370</td><td>0.2566531</td><td>cytokine</td></tr>\n",
       "\t<tr><td>cytokine</td><td>Ligand</td><td>CCL11</td><td>cytokine</td><td>69.10381</td><td>NA</td><td>cytokine_0_Eotaxin__CCL11</td><td>7.376516</td><td>2.8469700</td><td>cytokine</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 2 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " type & interaction\\_type & gene & cluster & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene & ident & mean\\_expr & var\\_expression & cell\\_type\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <dbl> & <int> & <chr> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t cytokine & Ligand & CCL1  & cytokine & 69.10381 & NA & cytokine\\_0\\_I309\\_\\_CCL1     & 1.381370 & 0.2566531 & cytokine\\\\\n",
       "\t cytokine & Ligand & CCL11 & cytokine & 69.10381 & NA & cytokine\\_0\\_Eotaxin\\_\\_CCL11 & 7.376516 & 2.8469700 & cytokine\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 2 × 10\n",
       "\n",
       "| type &lt;chr&gt; | interaction_type &lt;chr&gt; | gene &lt;chr&gt; | cluster &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; | ident &lt;chr&gt; | mean_expr &lt;dbl&gt; | var_expression &lt;dbl&gt; | cell_type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| cytokine | Ligand | CCL1  | cytokine | 69.10381 | NA | cytokine_0_I309__CCL1     | 1.381370 | 0.2566531 | cytokine |\n",
       "| cytokine | Ligand | CCL11 | cytokine | 69.10381 | NA | cytokine_0_Eotaxin__CCL11 | 7.376516 | 2.8469700 | cytokine |\n",
       "\n"
      ],
      "text/plain": [
       "  type     interaction_type gene  cluster  perc_cells_expressing_gene\n",
       "1 cytokine Ligand           CCL1  cytokine 69.10381                  \n",
       "2 cytokine Ligand           CCL11 cytokine 69.10381                  \n",
       "  total_amount_cells_expressing_gene ident                     mean_expr\n",
       "1 NA                                 cytokine_0_I309__CCL1     1.381370 \n",
       "2 NA                                 cytokine_0_Eotaxin__CCL11 7.376516 \n",
       "  var_expression cell_type\n",
       "1 0.2566531      cytokine \n",
       "2 2.8469700      cytokine "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(analyze_expr_cluster,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "a443e8bf-7bf6-45fe-a294-b4339e14e55b",
   "metadata": {},
   "outputs": [],
   "source": [
    "analyze_expr_cluster$variable = paste0(analyze_expr_cluster$cluster, '_0_', analyze_expr_cluster$gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "1ba9b4ea-be0c-4e5e-9d3b-4c2818fdaa58",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "11639"
      ],
      "text/latex": [
       "11639"
      ],
      "text/markdown": [
       "11639"
      ],
      "text/plain": [
       "[1] 11639"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_receptor_interactions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "471e600a-eefb-402d-8c8b-c6592f6c4c99",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Add data for ligands\n",
    "data  = merge(ligand_receptor_interactions, analyze_expr_cluster[analyze_expr_cluster$interaction_type == 'Ligand',], by.x = c('Ligand'), by.y = c('gene'),  suffixes = c(\"\",\"_L\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "85ca594a-a361-4c85-bf34-5abd5daddd89",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Add data for Receptors\n",
    "data  = merge(data, analyze_expr_cluster[analyze_expr_cluster$interaction_type == 'Receptor',], by.x = c('Receptor'), by.y = c('gene'),  suffixes = c(\"\",\"_R\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "47c9edaf-8194-400b-a3ca-e0a9e8fb4a4a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 24</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Receptor</th><th scope=col>Ligand</th><th scope=col>amount_sources</th><th scope=col>weight</th><th scope=col>type</th><th scope=col>interaction_type</th><th scope=col>cluster</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th><th scope=col>ident</th><th scope=col>⋯</th><th scope=col>type_R</th><th scope=col>interaction_type_R</th><th scope=col>cluster_R</th><th scope=col>perc_cells_expressing_gene_R</th><th scope=col>total_amount_cells_expressing_gene_R</th><th scope=col>ident_R</th><th scope=col>mean_expr_R</th><th scope=col>var_expression_R</th><th scope=col>cell_type_R</th><th scope=col>variable_R</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ABCA7  </td><td>APOA1 </td><td>1</td><td>0.4572367</td><td>proteomics</td><td>Ligand</td><td>proteomics</td><td>69.10381</td><td>NA</td><td>proteomics_0_APOA1_P02647 </td><td>⋯</td><td>neutrophil</td><td>Receptor</td><td>neutrophil</td><td>69.10381</td><td>NA</td><td>neutrophil_0_ABCA7  </td><td>1.436823</td><td>0.8685307</td><td>neutrophil</td><td>neutrophil_0_ABCA7  </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ADIPOR1</td><td>ADIPOQ</td><td>1</td><td>0.7307022</td><td>proteomics</td><td>Ligand</td><td>proteomics</td><td>69.10381</td><td>NA</td><td>proteomics_0_ADIPOQ_Q15848</td><td>⋯</td><td>neutrophil</td><td>Receptor</td><td>neutrophil</td><td>69.10381</td><td>NA</td><td>neutrophil_0_ADIPOR1</td><td>2.384717</td><td>0.8461529</td><td>neutrophil</td><td>neutrophil_0_ADIPOR1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 24\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & Receptor & Ligand & amount\\_sources & weight & type & interaction\\_type & cluster & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene & ident & ⋯ & type\\_R & interaction\\_type\\_R & cluster\\_R & perc\\_cells\\_expressing\\_gene\\_R & total\\_amount\\_cells\\_expressing\\_gene\\_R & ident\\_R & mean\\_expr\\_R & var\\_expression\\_R & cell\\_type\\_R & variable\\_R\\\\\n",
       "  & <chr> & <chr> & <int> & <dbl> & <chr> & <chr> & <chr> & <dbl> & <int> & <chr> & ⋯ & <chr> & <chr> & <chr> & <dbl> & <int> & <chr> & <dbl> & <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & ABCA7   & APOA1  & 1 & 0.4572367 & proteomics & Ligand & proteomics & 69.10381 & NA & proteomics\\_0\\_APOA1\\_P02647  & ⋯ & neutrophil & Receptor & neutrophil & 69.10381 & NA & neutrophil\\_0\\_ABCA7   & 1.436823 & 0.8685307 & neutrophil & neutrophil\\_0\\_ABCA7  \\\\\n",
       "\t2 & ADIPOR1 & ADIPOQ & 1 & 0.7307022 & proteomics & Ligand & proteomics & 69.10381 & NA & proteomics\\_0\\_ADIPOQ\\_Q15848 & ⋯ & neutrophil & Receptor & neutrophil & 69.10381 & NA & neutrophil\\_0\\_ADIPOR1 & 2.384717 & 0.8461529 & neutrophil & neutrophil\\_0\\_ADIPOR1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 24\n",
       "\n",
       "| <!--/--> | Receptor &lt;chr&gt; | Ligand &lt;chr&gt; | amount_sources &lt;int&gt; | weight &lt;dbl&gt; | type &lt;chr&gt; | interaction_type &lt;chr&gt; | cluster &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; | ident &lt;chr&gt; | ⋯ ⋯ | type_R &lt;chr&gt; | interaction_type_R &lt;chr&gt; | cluster_R &lt;chr&gt; | perc_cells_expressing_gene_R &lt;dbl&gt; | total_amount_cells_expressing_gene_R &lt;int&gt; | ident_R &lt;chr&gt; | mean_expr_R &lt;dbl&gt; | var_expression_R &lt;dbl&gt; | cell_type_R &lt;chr&gt; | variable_R &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | ABCA7   | APOA1  | 1 | 0.4572367 | proteomics | Ligand | proteomics | 69.10381 | NA | proteomics_0_APOA1_P02647  | ⋯ | neutrophil | Receptor | neutrophil | 69.10381 | NA | neutrophil_0_ABCA7   | 1.436823 | 0.8685307 | neutrophil | neutrophil_0_ABCA7   |\n",
       "| 2 | ADIPOR1 | ADIPOQ | 1 | 0.7307022 | proteomics | Ligand | proteomics | 69.10381 | NA | proteomics_0_ADIPOQ_Q15848 | ⋯ | neutrophil | Receptor | neutrophil | 69.10381 | NA | neutrophil_0_ADIPOR1 | 2.384717 | 0.8461529 | neutrophil | neutrophil_0_ADIPOR1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Receptor Ligand amount_sources weight    type       interaction_type\n",
       "1 ABCA7    APOA1  1              0.4572367 proteomics Ligand          \n",
       "2 ADIPOR1  ADIPOQ 1              0.7307022 proteomics Ligand          \n",
       "  cluster    perc_cells_expressing_gene total_amount_cells_expressing_gene\n",
       "1 proteomics 69.10381                   NA                                \n",
       "2 proteomics 69.10381                   NA                                \n",
       "  ident                      ⋯ type_R     interaction_type_R cluster_R \n",
       "1 proteomics_0_APOA1_P02647  ⋯ neutrophil Receptor           neutrophil\n",
       "2 proteomics_0_ADIPOQ_Q15848 ⋯ neutrophil Receptor           neutrophil\n",
       "  perc_cells_expressing_gene_R total_amount_cells_expressing_gene_R\n",
       "1 69.10381                     NA                                  \n",
       "2 69.10381                     NA                                  \n",
       "  ident_R              mean_expr_R var_expression_R cell_type_R\n",
       "1 neutrophil_0_ABCA7   1.436823    0.8685307        neutrophil \n",
       "2 neutrophil_0_ADIPOR1 2.384717    0.8461529        neutrophil \n",
       "  variable_R          \n",
       "1 neutrophil_0_ABCA7  \n",
       "2 neutrophil_0_ADIPOR1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "id": "515c5f77-f33d-448b-92cc-49f5192cfe72",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Receptor'</li><li>'Ligand'</li><li>'amount_sources'</li><li>'weight'</li><li>'type'</li><li>'interaction_type'</li><li>'cluster'</li><li>'perc_cells_expressing_gene'</li><li>'total_amount_cells_expressing_gene'</li><li>'ident'</li><li>'mean_expr'</li><li>'var_expression'</li><li>'cell_type'</li><li>'variable'</li><li>'type_R'</li><li>'interaction_type_R'</li><li>'cluster_R'</li><li>'perc_cells_expressing_gene_R'</li><li>'total_amount_cells_expressing_gene_R'</li><li>'ident_R'</li><li>'mean_expr_R'</li><li>'var_expression_R'</li><li>'cell_type_R'</li><li>'variable_R'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Receptor'\n",
       "\\item 'Ligand'\n",
       "\\item 'amount\\_sources'\n",
       "\\item 'weight'\n",
       "\\item 'type'\n",
       "\\item 'interaction\\_type'\n",
       "\\item 'cluster'\n",
       "\\item 'perc\\_cells\\_expressing\\_gene'\n",
       "\\item 'total\\_amount\\_cells\\_expressing\\_gene'\n",
       "\\item 'ident'\n",
       "\\item 'mean\\_expr'\n",
       "\\item 'var\\_expression'\n",
       "\\item 'cell\\_type'\n",
       "\\item 'variable'\n",
       "\\item 'type\\_R'\n",
       "\\item 'interaction\\_type\\_R'\n",
       "\\item 'cluster\\_R'\n",
       "\\item 'perc\\_cells\\_expressing\\_gene\\_R'\n",
       "\\item 'total\\_amount\\_cells\\_expressing\\_gene\\_R'\n",
       "\\item 'ident\\_R'\n",
       "\\item 'mean\\_expr\\_R'\n",
       "\\item 'var\\_expression\\_R'\n",
       "\\item 'cell\\_type\\_R'\n",
       "\\item 'variable\\_R'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Receptor'\n",
       "2. 'Ligand'\n",
       "3. 'amount_sources'\n",
       "4. 'weight'\n",
       "5. 'type'\n",
       "6. 'interaction_type'\n",
       "7. 'cluster'\n",
       "8. 'perc_cells_expressing_gene'\n",
       "9. 'total_amount_cells_expressing_gene'\n",
       "10. 'ident'\n",
       "11. 'mean_expr'\n",
       "12. 'var_expression'\n",
       "13. 'cell_type'\n",
       "14. 'variable'\n",
       "15. 'type_R'\n",
       "16. 'interaction_type_R'\n",
       "17. 'cluster_R'\n",
       "18. 'perc_cells_expressing_gene_R'\n",
       "19. 'total_amount_cells_expressing_gene_R'\n",
       "20. 'ident_R'\n",
       "21. 'mean_expr_R'\n",
       "22. 'var_expression_R'\n",
       "23. 'cell_type_R'\n",
       "24. 'variable_R'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Receptor\"                            \n",
       " [2] \"Ligand\"                              \n",
       " [3] \"amount_sources\"                      \n",
       " [4] \"weight\"                              \n",
       " [5] \"type\"                                \n",
       " [6] \"interaction_type\"                    \n",
       " [7] \"cluster\"                             \n",
       " [8] \"perc_cells_expressing_gene\"          \n",
       " [9] \"total_amount_cells_expressing_gene\"  \n",
       "[10] \"ident\"                               \n",
       "[11] \"mean_expr\"                           \n",
       "[12] \"var_expression\"                      \n",
       "[13] \"cell_type\"                           \n",
       "[14] \"variable\"                            \n",
       "[15] \"type_R\"                              \n",
       "[16] \"interaction_type_R\"                  \n",
       "[17] \"cluster_R\"                           \n",
       "[18] \"perc_cells_expressing_gene_R\"        \n",
       "[19] \"total_amount_cells_expressing_gene_R\"\n",
       "[20] \"ident_R\"                             \n",
       "[21] \"mean_expr_R\"                         \n",
       "[22] \"var_expression_R\"                    \n",
       "[23] \"cell_type_R\"                         \n",
       "[24] \"variable_R\"                          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "cf52413f-7332-4e0b-89f1-8e7ca86bd238",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "5826"
      ],
      "text/latex": [
       "5826"
      ],
      "text/markdown": [
       "5826"
      ],
      "text/plain": [
       "[1] 5826"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "29b387e7-62ba-4524-a95d-cd6abeed4572",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'proteomics'</li><li>'neutrophil'</li><li>'single_cell'</li><li>'cytokine'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'proteomics'\n",
       "\\item 'neutrophil'\n",
       "\\item 'single\\_cell'\n",
       "\\item 'cytokine'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'proteomics'\n",
       "2. 'neutrophil'\n",
       "3. 'single_cell'\n",
       "4. 'cytokine'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"proteomics\"  \"neutrophil\"  \"single_cell\" \"cytokine\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(data$type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "id": "3044cb11-4e12-4354-8496-e28d8e6e8bb9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Remove same type links"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "fdcbeaeb-214f-4f7c-b707-e22d27c3981f",
   "metadata": {},
   "outputs": [],
   "source": [
    "data = data[data$cell_type != data$cell_type_R,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "f4fd0a82-e0ef-49d5-96a5-4f06e59f83fa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "4921"
      ],
      "text/latex": [
       "4921"
      ],
      "text/markdown": [
       "4921"
      ],
      "text/plain": [
       "[1] 4921"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "id": "f17e0a8b-9e45-48e5-b72e-0302f9f3f313",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'proteomics'</li><li>'neutrophil'</li><li>'single_cell'</li><li>'cytokine'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'proteomics'\n",
       "\\item 'neutrophil'\n",
       "\\item 'single\\_cell'\n",
       "\\item 'cytokine'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'proteomics'\n",
       "2. 'neutrophil'\n",
       "3. 'single_cell'\n",
       "4. 'cytokine'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"proteomics\"  \"neutrophil\"  \"single_cell\" \"cytokine\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(data$type)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "05d5e645-60c4-42da-a581-9578200fcd0e",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Define expressed ligands and potential ligand - target combinations based on expressed receptors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "14202f35-ea65-4cd2-8c31-f7a2594f4bff",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Define minimum amount of expression of receptor in receiving cell to be able to react to ligand"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "id": "61e16fac-4041-46d6-9942-b1946de77de0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Ligand</th><th scope=col>Receptor</th><th scope=col>amount_sources</th><th scope=col>weight</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A2M</td><td>LRP1    </td><td>1</td><td>0.4678977</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A2M</td><td>TNFRSF14</td><td>1</td><td>0.6990671</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & Ligand & Receptor & amount\\_sources & weight\\\\\n",
       "  & <chr> & <chr> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & A2M & LRP1     & 1 & 0.4678977\\\\\n",
       "\t2 & A2M & TNFRSF14 & 1 & 0.6990671\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 4\n",
       "\n",
       "| <!--/--> | Ligand &lt;chr&gt; | Receptor &lt;chr&gt; | amount_sources &lt;int&gt; | weight &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | A2M | LRP1     | 1 | 0.4678977 |\n",
       "| 2 | A2M | TNFRSF14 | 1 | 0.6990671 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand Receptor amount_sources weight   \n",
       "1 A2M    LRP1     1              0.4678977\n",
       "2 A2M    TNFRSF14 1              0.6990671"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_receptor_interactions,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "0f8f3c06-73da-492b-aa33-c83e36465e45",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th><th scope=col>gene</th><th scope=col>cluster</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AL627309.1</td><td>0.24327612</td><td>18</td><td>AL627309.1</td><td>X8_B.cell</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AL627309.4</td><td>0.05406136</td><td> 4</td><td>AL627309.4</td><td>X8_B.cell</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & X & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene & gene & cluster\\\\\n",
       "  & <chr> & <dbl> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & AL627309.1 & 0.24327612 & 18 & AL627309.1 & X8\\_B.cell\\\\\n",
       "\t2 & AL627309.4 & 0.05406136 &  4 & AL627309.4 & X8\\_B.cell\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 5\n",
       "\n",
       "| <!--/--> | X &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; | gene &lt;chr&gt; | cluster &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | AL627309.1 | 0.24327612 | 18 | AL627309.1 | X8_B.cell |\n",
       "| 2 | AL627309.4 | 0.05406136 |  4 | AL627309.4 | X8_B.cell |\n",
       "\n"
      ],
      "text/plain": [
       "  X          perc_cells_expressing_gene total_amount_cells_expressing_gene\n",
       "1 AL627309.1 0.24327612                 18                                \n",
       "2 AL627309.4 0.05406136                  4                                \n",
       "  gene       cluster  \n",
       "1 AL627309.1 X8_B.cell\n",
       "2 AL627309.4 X8_B.cell"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cell_perc_cluster,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "d60b6d7f-93e4-4337-88b1-0305c5fb5cce",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Set threshold to filter out receptors that are to low expressed\n",
    "cell_perc_cluster_filtered =  cell_perc_cluster[((cell_perc_cluster$perc_cells > 30) & (cell_perc_cluster$total_amount_cells_expressing_gene > 600)) | ((cell_perc_cluster$perc_cells > 10) & (cell_perc_cluster$total_amount_cells_expressing_gene > 1200)),]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "id": "bd54a0d2-ac92-49bf-900b-4f38e529d313",
   "metadata": {},
   "outputs": [],
   "source": [
    "potential_ligand_target_types_comb = merge(ligand_receptor_interactions, cell_perc_cluster_filtered, by.x = 'Receptor', by.y = 'gene')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "8892c6ea-aa4f-4514-9237-6e8648a94242",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Receptor</th><th scope=col>Ligand</th><th scope=col>amount_sources</th><th scope=col>weight</th><th scope=col>X</th><th scope=col>perc_cells_expressing_gene</th><th scope=col>total_amount_cells_expressing_gene</th><th scope=col>cluster</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ABCA1</td><td>APOA1</td><td>1</td><td>0.8194814</td><td>ABCA19</td><td>15.74701</td><td>2303</td><td>X4_Monocytes...CD14</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ABCA1</td><td>APOA1</td><td>1</td><td>0.8194814</td><td>ABCA17</td><td>16.46135</td><td>1844</td><td>X6_Monocytes...CD14</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Receptor & Ligand & amount\\_sources & weight & X & perc\\_cells\\_expressing\\_gene & total\\_amount\\_cells\\_expressing\\_gene & cluster\\\\\n",
       "  & <chr> & <chr> & <int> & <dbl> & <chr> & <dbl> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & ABCA1 & APOA1 & 1 & 0.8194814 & ABCA19 & 15.74701 & 2303 & X4\\_Monocytes...CD14\\\\\n",
       "\t2 & ABCA1 & APOA1 & 1 & 0.8194814 & ABCA17 & 16.46135 & 1844 & X6\\_Monocytes...CD14\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 8\n",
       "\n",
       "| <!--/--> | Receptor &lt;chr&gt; | Ligand &lt;chr&gt; | amount_sources &lt;int&gt; | weight &lt;dbl&gt; | X &lt;chr&gt; | perc_cells_expressing_gene &lt;dbl&gt; | total_amount_cells_expressing_gene &lt;int&gt; | cluster &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | ABCA1 | APOA1 | 1 | 0.8194814 | ABCA19 | 15.74701 | 2303 | X4_Monocytes...CD14 |\n",
       "| 2 | ABCA1 | APOA1 | 1 | 0.8194814 | ABCA17 | 16.46135 | 1844 | X6_Monocytes...CD14 |\n",
       "\n"
      ],
      "text/plain": [
       "  Receptor Ligand amount_sources weight    X      perc_cells_expressing_gene\n",
       "1 ABCA1    APOA1  1              0.8194814 ABCA19 15.74701                  \n",
       "2 ABCA1    APOA1  1              0.8194814 ABCA17 16.46135                  \n",
       "  total_amount_cells_expressing_gene cluster            \n",
       "1 2303                               X4_Monocytes...CD14\n",
       "2 1844                               X6_Monocytes...CD14"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(potential_ligand_target_types_comb,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "id": "742e810d-442f-4f78-a259-90b8b73e5f42",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Define all potential ligand receptor interactions based on expression\n",
    "potential_ligand_target_types_comb = unique(potential_ligand_target_types_comb[,c('Ligand', 'cluster')] )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "524ede29-4dc4-48f9-aa42-2fe7c1991182",
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames(potential_ligand_target_types_comb) = c('Ligand', 'cluster_R')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "id": "f8338a36-14f7-4430-b9eb-db2e060ffa83",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Ligand</th><th scope=col>cluster_R</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>APOA1</td><td>X4_Monocytes...CD14</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>APOA1</td><td>X6_Monocytes...CD14</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Ligand & cluster\\_R\\\\\n",
       "  & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & APOA1 & X4\\_Monocytes...CD14\\\\\n",
       "\t2 & APOA1 & X6\\_Monocytes...CD14\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 2\n",
       "\n",
       "| <!--/--> | Ligand &lt;chr&gt; | cluster_R &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| 1 | APOA1 | X4_Monocytes...CD14 |\n",
       "| 2 | APOA1 | X6_Monocytes...CD14 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand cluster_R          \n",
       "1 APOA1  X4_Monocytes...CD14\n",
       "2 APOA1  X6_Monocytes...CD14"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(potential_ligand_target_types_comb,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "4c061112-0e85-451d-9a8d-75d56732928d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'X4_Monocytes...CD14'</li><li>'X6_Monocytes...CD14'</li><li>'X2_T.cell.CD4'</li><li>'X1_T.cell.CD8'</li><li>'X7_Monocytes...CD14'</li><li>'X0_T.cell.CD4'</li><li>'X12_Monocytes...CD16_FCGR3A'</li><li>'X3_NK'</li><li>'X9_Monocytes...CD16_FCGR3A'</li><li>'X8_B.cell'</li><li>'X11_T.cell.CD4'</li><li>'X10_B.cell'</li><li>'X5_T.cell.CD4'</li><li>'X13_Dendritic'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'X4\\_Monocytes...CD14'\n",
       "\\item 'X6\\_Monocytes...CD14'\n",
       "\\item 'X2\\_T.cell.CD4'\n",
       "\\item 'X1\\_T.cell.CD8'\n",
       "\\item 'X7\\_Monocytes...CD14'\n",
       "\\item 'X0\\_T.cell.CD4'\n",
       "\\item 'X12\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'X3\\_NK'\n",
       "\\item 'X9\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'X8\\_B.cell'\n",
       "\\item 'X11\\_T.cell.CD4'\n",
       "\\item 'X10\\_B.cell'\n",
       "\\item 'X5\\_T.cell.CD4'\n",
       "\\item 'X13\\_Dendritic'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'X4_Monocytes...CD14'\n",
       "2. 'X6_Monocytes...CD14'\n",
       "3. 'X2_T.cell.CD4'\n",
       "4. 'X1_T.cell.CD8'\n",
       "5. 'X7_Monocytes...CD14'\n",
       "6. 'X0_T.cell.CD4'\n",
       "7. 'X12_Monocytes...CD16_FCGR3A'\n",
       "8. 'X3_NK'\n",
       "9. 'X9_Monocytes...CD16_FCGR3A'\n",
       "10. 'X8_B.cell'\n",
       "11. 'X11_T.cell.CD4'\n",
       "12. 'X10_B.cell'\n",
       "13. 'X5_T.cell.CD4'\n",
       "14. 'X13_Dendritic'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"X4_Monocytes...CD14\"         \"X6_Monocytes...CD14\"        \n",
       " [3] \"X2_T.cell.CD4\"               \"X1_T.cell.CD8\"              \n",
       " [5] \"X7_Monocytes...CD14\"         \"X0_T.cell.CD4\"              \n",
       " [7] \"X12_Monocytes...CD16_FCGR3A\" \"X3_NK\"                      \n",
       " [9] \"X9_Monocytes...CD16_FCGR3A\"  \"X8_B.cell\"                  \n",
       "[11] \"X11_T.cell.CD4\"              \"X10_B.cell\"                 \n",
       "[13] \"X5_T.cell.CD4\"               \"X13_Dendritic\"              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(potential_ligand_target_types_comb$cluster_R)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "id": "c89f95e5-12fc-404a-96a1-3f0b9121aa5c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 19 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Ligand</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ARTN   </td><td>1</td></tr>\n",
       "\t<tr><td>CNTN2  </td><td>1</td></tr>\n",
       "\t<tr><td>DEFB4A </td><td>1</td></tr>\n",
       "\t<tr><td>DEFB4B </td><td>1</td></tr>\n",
       "\t<tr><td>DHH    </td><td>1</td></tr>\n",
       "\t<tr><td>FLT3LG </td><td>1</td></tr>\n",
       "\t<tr><td>GPHA2  </td><td>1</td></tr>\n",
       "\t<tr><td>IHH    </td><td>1</td></tr>\n",
       "\t<tr><td>IL18   </td><td>1</td></tr>\n",
       "\t<tr><td>IL23A  </td><td>1</td></tr>\n",
       "\t<tr><td>LIPH   </td><td>1</td></tr>\n",
       "\t<tr><td>NRG1   </td><td>1</td></tr>\n",
       "\t<tr><td>NRTN   </td><td>1</td></tr>\n",
       "\t<tr><td>PSPN   </td><td>1</td></tr>\n",
       "\t<tr><td>SEMA3A </td><td>1</td></tr>\n",
       "\t<tr><td>SEMA3F </td><td>1</td></tr>\n",
       "\t<tr><td>SEMA6D </td><td>1</td></tr>\n",
       "\t<tr><td>TNFSF18</td><td>1</td></tr>\n",
       "\t<tr><td>TNFSF4 </td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 19 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Ligand & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t ARTN    & 1\\\\\n",
       "\t CNTN2   & 1\\\\\n",
       "\t DEFB4A  & 1\\\\\n",
       "\t DEFB4B  & 1\\\\\n",
       "\t DHH     & 1\\\\\n",
       "\t FLT3LG  & 1\\\\\n",
       "\t GPHA2   & 1\\\\\n",
       "\t IHH     & 1\\\\\n",
       "\t IL18    & 1\\\\\n",
       "\t IL23A   & 1\\\\\n",
       "\t LIPH    & 1\\\\\n",
       "\t NRG1    & 1\\\\\n",
       "\t NRTN    & 1\\\\\n",
       "\t PSPN    & 1\\\\\n",
       "\t SEMA3A  & 1\\\\\n",
       "\t SEMA3F  & 1\\\\\n",
       "\t SEMA6D  & 1\\\\\n",
       "\t TNFSF18 & 1\\\\\n",
       "\t TNFSF4  & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 19 × 2\n",
       "\n",
       "| Ligand &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| ARTN    | 1 |\n",
       "| CNTN2   | 1 |\n",
       "| DEFB4A  | 1 |\n",
       "| DEFB4B  | 1 |\n",
       "| DHH     | 1 |\n",
       "| FLT3LG  | 1 |\n",
       "| GPHA2   | 1 |\n",
       "| IHH     | 1 |\n",
       "| IL18    | 1 |\n",
       "| IL23A   | 1 |\n",
       "| LIPH    | 1 |\n",
       "| NRG1    | 1 |\n",
       "| NRTN    | 1 |\n",
       "| PSPN    | 1 |\n",
       "| SEMA3A  | 1 |\n",
       "| SEMA3F  | 1 |\n",
       "| SEMA6D  | 1 |\n",
       "| TNFSF18 | 1 |\n",
       "| TNFSF4  | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "   Ligand  n\n",
       "1  ARTN    1\n",
       "2  CNTN2   1\n",
       "3  DEFB4A  1\n",
       "4  DEFB4B  1\n",
       "5  DHH     1\n",
       "6  FLT3LG  1\n",
       "7  GPHA2   1\n",
       "8  IHH     1\n",
       "9  IL18    1\n",
       "10 IL23A   1\n",
       "11 LIPH    1\n",
       "12 NRG1    1\n",
       "13 NRTN    1\n",
       "14 PSPN    1\n",
       "15 SEMA3A  1\n",
       "16 SEMA3F  1\n",
       "17 SEMA6D  1\n",
       "18 TNFSF18 1\n",
       "19 TNFSF4  1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Below: ligands where receptors are only expressed in very low amount of targets \n",
    "unique(potential_ligand_target_types_comb[,c('Ligand', 'cluster_R')]) %>% group_by(Ligand) %>% count() %>% filter(n == 1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "0f48bbe4-f95b-40c7-bfd8-9f04d0635586",
   "metadata": {},
   "outputs": [],
   "source": [
    "## get all expressed ligands per cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "id": "49003b10-bc5a-45a8-9c06-0909157616ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "expressed_ligand_types = unique(data[,c('Ligand', 'cluster', 'mean_expr', 'type', 'variable', 'ident')])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8de04252-820d-4d43-893f-aa6f8f5ddda2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "dc74a9fe-4037-4e1b-b030-80964257adfb",
   "metadata": {},
   "source": [
    "# Ligand Target Data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "408f8793-3437-4325-b517-d589dc3cf3de",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Calculate ligand - target correlations (lagged targets)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "f0bf8c31-a78e-4ccd-b2a8-d185d659016b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Ligand</th><th scope=col>cluster</th><th scope=col>mean_expr</th><th scope=col>type</th><th scope=col>variable</th><th scope=col>ident</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>APOA1 </td><td>proteomics</td><td> 8.733489</td><td>proteomics</td><td>proteomics_0_APOA1 </td><td>proteomics_0_APOA1_P02647 </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ADIPOQ</td><td>proteomics</td><td>-1.154980</td><td>proteomics</td><td>proteomics_0_ADIPOQ</td><td>proteomics_0_ADIPOQ_Q15848</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & Ligand & cluster & mean\\_expr & type & variable & ident\\\\\n",
       "  & <chr> & <chr> & <dbl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & APOA1  & proteomics &  8.733489 & proteomics & proteomics\\_0\\_APOA1  & proteomics\\_0\\_APOA1\\_P02647 \\\\\n",
       "\t2 & ADIPOQ & proteomics & -1.154980 & proteomics & proteomics\\_0\\_ADIPOQ & proteomics\\_0\\_ADIPOQ\\_Q15848\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 6\n",
       "\n",
       "| <!--/--> | Ligand &lt;chr&gt; | cluster &lt;chr&gt; | mean_expr &lt;dbl&gt; | type &lt;chr&gt; | variable &lt;chr&gt; | ident &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | APOA1  | proteomics |  8.733489 | proteomics | proteomics_0_APOA1  | proteomics_0_APOA1_P02647  |\n",
       "| 2 | ADIPOQ | proteomics | -1.154980 | proteomics | proteomics_0_ADIPOQ | proteomics_0_ADIPOQ_Q15848 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand cluster    mean_expr type       variable           \n",
       "1 APOA1  proteomics  8.733489 proteomics proteomics_0_APOA1 \n",
       "2 ADIPOQ proteomics -1.154980 proteomics proteomics_0_ADIPOQ\n",
       "  ident                     \n",
       "1 proteomics_0_APOA1_P02647 \n",
       "2 proteomics_0_ADIPOQ_Q15848"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### expressed ligands per cluster\n",
    "head(expressed_ligand_types,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "id": "ae1073d9-f2e1-41f2-9cc3-68b353fb24d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Determine expressed ligands in single-cell data\n",
    "expressed_ligand_types$variable[expressed_ligand_types$type == 'single_cell'] = paste0(expressed_ligand_types$type[expressed_ligand_types$type == 'single_cell'], '_0_',expressed_ligand_types$cluster[expressed_ligand_types$type == 'single_cell'], '__', expressed_ligand_types$Ligand[expressed_ligand_types$type == 'single_cell'] )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "id": "0ad0a10a-42bd-40c2-b6d0-e5d30c7cf3d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "expressed_ligands = unique(expressed_ligand_types$ident)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "6a71d50b-d417-4283-a751-5f8aa4fd2af4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "343"
      ],
      "text/latex": [
       "343"
      ],
      "text/markdown": [
       "343"
      ],
      "text/plain": [
       "[1] 343"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(expressed_ligands)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "id": "938631c6-3a2f-4844-a2c3-05a59b0a6479",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Transform data_long to matrix format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "id": "dbf9a48b-08c1-471e-9abb-c2d5c923608d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>type</th><th scope=col>variable</th><th scope=col>X</th><th scope=col>sample_id</th><th scope=col>value</th><th scope=col>config</th><th scope=col>expression</th><th scope=col>ident</th><th scope=col>gene</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>clinical_data</td><td>CK</td><td>1557244</td><td>m1.1</td><td>7.467606</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>clinical_data_0_CK</td><td>CK</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>clinical_data</td><td>CK</td><td>1557245</td><td>m1.2</td><td>      NA</td><td>FALSE-FALSE-TRUE-0.2-FALSE-TRUE-</td><td>TRUE</td><td>clinical_data_0_CK</td><td>CK</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & type & variable & X & sample\\_id & value & config & expression & ident & gene\\\\\n",
       "  & <chr> & <chr> & <int> & <chr> & <dbl> & <chr> & <lgl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & clinical\\_data & CK & 1557244 & m1.1 & 7.467606 & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & clinical\\_data\\_0\\_CK & CK\\\\\n",
       "\t2 & clinical\\_data & CK & 1557245 & m1.2 &       NA & FALSE-FALSE-TRUE-0.2-FALSE-TRUE- & TRUE & clinical\\_data\\_0\\_CK & CK\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 9\n",
       "\n",
       "| <!--/--> | type &lt;chr&gt; | variable &lt;chr&gt; | X &lt;int&gt; | sample_id &lt;chr&gt; | value &lt;dbl&gt; | config &lt;chr&gt; | expression &lt;lgl&gt; | ident &lt;chr&gt; | gene &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | clinical_data | CK | 1557244 | m1.1 | 7.467606 | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | clinical_data_0_CK | CK |\n",
       "| 2 | clinical_data | CK | 1557245 | m1.2 |       NA | FALSE-FALSE-TRUE-0.2-FALSE-TRUE- | TRUE | clinical_data_0_CK | CK |\n",
       "\n"
      ],
      "text/plain": [
       "  type          variable X       sample_id value   \n",
       "1 clinical_data CK       1557244 m1.1      7.467606\n",
       "2 clinical_data CK       1557245 m1.2            NA\n",
       "  config                           expression ident              gene\n",
       "1 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       clinical_data_0_CK CK  \n",
       "2 FALSE-FALSE-TRUE-0.2-FALSE-TRUE- TRUE       clinical_data_0_CK CK  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(data_long,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "id": "e5b42efe-5ab0-43c8-a3a8-d1526dfefa15",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_data = data_long %>% dcast(sample_id ~ ident, value.var = 'value')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "id": "099dab7a-bf4a-4dba-9f18-0068c7642efe",
   "metadata": {},
   "outputs": [],
   "source": [
    "rownames(expression_data) = expression_data$sample_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "id": "62386c1c-2e61-4eef-ace9-ad42e732e1f9",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_data$sample_id = NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "id": "6bcaf00a-2aa3-47b5-8250-6d51291e71bc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 13282</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>clinical_data_0_CK</th><th scope=col>clinical_data_0_CK_MB</th><th scope=col>clinical_data_0_CRP</th><th scope=col>clinical_data_0_delta_ef_value</th><th scope=col>clinical_data_0_Troponin</th><th scope=col>cytokine_0_BCA1__CXCL13</th><th scope=col>cytokine_0_CTACK__CCL27</th><th scope=col>cytokine_0_EGF__EGF</th><th scope=col>cytokine_0_ENA78__CXCL5</th><th scope=col>cytokine_0_Eotaxin__CCL11</th><th scope=col>⋯</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>m1.1</th><td>7.467606</td><td>4.321928</td><td>3.137504</td><td>NA</td><td>0.75446</td><td>5.570766</td><td>9.747908</td><td>2.456806</td><td>6.335569</td><td>5.617357</td><td>⋯</td><td>2.201764</td><td>2.842767</td><td>2.610397</td><td>2.910977</td><td>2.461054</td><td>1.084743</td><td>0.7984776</td><td>0.4248882</td><td>1.451288</td><td>0.9409604</td></tr>\n",
       "\t<tr><th scope=row>m1.2</th><td>      NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>     NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>⋯</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>       NA</td><td>       NA</td><td>      NA</td><td>       NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 13282\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & clinical\\_data\\_0\\_CK & clinical\\_data\\_0\\_CK\\_MB & clinical\\_data\\_0\\_CRP & clinical\\_data\\_0\\_delta\\_ef\\_value & clinical\\_data\\_0\\_Troponin & cytokine\\_0\\_BCA1\\_\\_CXCL13 & cytokine\\_0\\_CTACK\\_\\_CCL27 & cytokine\\_0\\_EGF\\_\\_EGF & cytokine\\_0\\_ENA78\\_\\_CXCL5 & cytokine\\_0\\_Eotaxin\\_\\_CCL11 & ⋯ & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZEB2 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFAND5 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFAS1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36L1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36L2 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF106 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF207 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF706 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZYX\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tm1.1 & 7.467606 & 4.321928 & 3.137504 & NA & 0.75446 & 5.570766 & 9.747908 & 2.456806 & 6.335569 & 5.617357 & ⋯ & 2.201764 & 2.842767 & 2.610397 & 2.910977 & 2.461054 & 1.084743 & 0.7984776 & 0.4248882 & 1.451288 & 0.9409604\\\\\n",
       "\tm1.2 &       NA &       NA &       NA & NA &      NA &       NA &       NA &       NA &       NA &       NA & ⋯ &       NA &       NA &       NA &       NA &       NA &       NA &        NA &        NA &       NA &        NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 13282\n",
       "\n",
       "| <!--/--> | clinical_data_0_CK &lt;dbl&gt; | clinical_data_0_CK_MB &lt;dbl&gt; | clinical_data_0_CRP &lt;dbl&gt; | clinical_data_0_delta_ef_value &lt;dbl&gt; | clinical_data_0_Troponin &lt;dbl&gt; | cytokine_0_BCA1__CXCL13 &lt;dbl&gt; | cytokine_0_CTACK__CCL27 &lt;dbl&gt; | cytokine_0_EGF__EGF &lt;dbl&gt; | cytokine_0_ENA78__CXCL5 &lt;dbl&gt; | cytokine_0_Eotaxin__CCL11 &lt;dbl&gt; | ⋯ ⋯ | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| m1.1 | 7.467606 | 4.321928 | 3.137504 | NA | 0.75446 | 5.570766 | 9.747908 | 2.456806 | 6.335569 | 5.617357 | ⋯ | 2.201764 | 2.842767 | 2.610397 | 2.910977 | 2.461054 | 1.084743 | 0.7984776 | 0.4248882 | 1.451288 | 0.9409604 |\n",
       "| m1.2 |       NA |       NA |       NA | NA |      NA |       NA |       NA |       NA |       NA |       NA | ⋯ |       NA |       NA |       NA |       NA |       NA |       NA |        NA |        NA |       NA |        NA |\n",
       "\n"
      ],
      "text/plain": [
       "     clinical_data_0_CK clinical_data_0_CK_MB clinical_data_0_CRP\n",
       "m1.1 7.467606           4.321928              3.137504           \n",
       "m1.2       NA                 NA                    NA           \n",
       "     clinical_data_0_delta_ef_value clinical_data_0_Troponin\n",
       "m1.1 NA                             0.75446                 \n",
       "m1.2 NA                                  NA                 \n",
       "     cytokine_0_BCA1__CXCL13 cytokine_0_CTACK__CCL27 cytokine_0_EGF__EGF\n",
       "m1.1 5.570766                9.747908                2.456806           \n",
       "m1.2       NA                      NA                      NA           \n",
       "     cytokine_0_ENA78__CXCL5 cytokine_0_Eotaxin__CCL11 ⋯\n",
       "m1.1 6.335569                5.617357                  ⋯\n",
       "m1.2       NA                      NA                  ⋯\n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2\n",
       "m1.1 2.201764                                      \n",
       "m1.2       NA                                      \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5\n",
       "m1.1 2.842767                                        \n",
       "m1.2       NA                                        \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1\n",
       "m1.1 2.610397                                       \n",
       "m1.2       NA                                       \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36\n",
       "m1.1 2.910977                                       \n",
       "m1.2       NA                                       \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1\n",
       "m1.1 2.461054                                         \n",
       "m1.2       NA                                         \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2\n",
       "m1.1 1.084743                                         \n",
       "m1.2       NA                                         \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106\n",
       "m1.1 0.7984776                                       \n",
       "m1.2        NA                                       \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207\n",
       "m1.1 0.4248882                                       \n",
       "m1.2        NA                                       \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706\n",
       "m1.1 1.451288                                        \n",
       "m1.2       NA                                        \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX\n",
       "m1.1 0.9409604                                    \n",
       "m1.2        NA                                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(expression_data,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "id": "dc74d19d-d14c-40cb-8913-8d718d4b6334",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Calculate all correlations between ligands and targets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "id": "e122b3ea-d3a0-4924-b6b3-5ce0e8a21146",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 13282</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>clinical_data_0_CK</th><th scope=col>clinical_data_0_CK_MB</th><th scope=col>clinical_data_0_CRP</th><th scope=col>clinical_data_0_delta_ef_value</th><th scope=col>clinical_data_0_Troponin</th><th scope=col>cytokine_0_BCA1__CXCL13</th><th scope=col>cytokine_0_CTACK__CCL27</th><th scope=col>cytokine_0_EGF__EGF</th><th scope=col>cytokine_0_ENA78__CXCL5</th><th scope=col>cytokine_0_Eotaxin__CCL11</th><th scope=col>⋯</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>m1.1</th><td>7.467606</td><td>4.321928</td><td>3.137504</td><td>NA</td><td>0.75446</td><td>5.570766</td><td>9.747908</td><td>2.456806</td><td>6.335569</td><td>5.617357</td><td>⋯</td><td>2.201764</td><td>2.842767</td><td>2.610397</td><td>2.910977</td><td>2.461054</td><td>1.084743</td><td>0.7984776</td><td>0.4248882</td><td>1.451288</td><td>0.9409604</td></tr>\n",
       "\t<tr><th scope=row>m1.2</th><td>      NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>     NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>⋯</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>       NA</td><td>       NA</td><td>      NA</td><td>       NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 13282\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & clinical\\_data\\_0\\_CK & clinical\\_data\\_0\\_CK\\_MB & clinical\\_data\\_0\\_CRP & clinical\\_data\\_0\\_delta\\_ef\\_value & clinical\\_data\\_0\\_Troponin & cytokine\\_0\\_BCA1\\_\\_CXCL13 & cytokine\\_0\\_CTACK\\_\\_CCL27 & cytokine\\_0\\_EGF\\_\\_EGF & cytokine\\_0\\_ENA78\\_\\_CXCL5 & cytokine\\_0\\_Eotaxin\\_\\_CCL11 & ⋯ & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZEB2 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFAND5 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFAS1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36L1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36L2 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF106 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF207 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF706 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZYX\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tm1.1 & 7.467606 & 4.321928 & 3.137504 & NA & 0.75446 & 5.570766 & 9.747908 & 2.456806 & 6.335569 & 5.617357 & ⋯ & 2.201764 & 2.842767 & 2.610397 & 2.910977 & 2.461054 & 1.084743 & 0.7984776 & 0.4248882 & 1.451288 & 0.9409604\\\\\n",
       "\tm1.2 &       NA &       NA &       NA & NA &      NA &       NA &       NA &       NA &       NA &       NA & ⋯ &       NA &       NA &       NA &       NA &       NA &       NA &        NA &        NA &       NA &        NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 13282\n",
       "\n",
       "| <!--/--> | clinical_data_0_CK &lt;dbl&gt; | clinical_data_0_CK_MB &lt;dbl&gt; | clinical_data_0_CRP &lt;dbl&gt; | clinical_data_0_delta_ef_value &lt;dbl&gt; | clinical_data_0_Troponin &lt;dbl&gt; | cytokine_0_BCA1__CXCL13 &lt;dbl&gt; | cytokine_0_CTACK__CCL27 &lt;dbl&gt; | cytokine_0_EGF__EGF &lt;dbl&gt; | cytokine_0_ENA78__CXCL5 &lt;dbl&gt; | cytokine_0_Eotaxin__CCL11 &lt;dbl&gt; | ⋯ ⋯ | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| m1.1 | 7.467606 | 4.321928 | 3.137504 | NA | 0.75446 | 5.570766 | 9.747908 | 2.456806 | 6.335569 | 5.617357 | ⋯ | 2.201764 | 2.842767 | 2.610397 | 2.910977 | 2.461054 | 1.084743 | 0.7984776 | 0.4248882 | 1.451288 | 0.9409604 |\n",
       "| m1.2 |       NA |       NA |       NA | NA |      NA |       NA |       NA |       NA |       NA |       NA | ⋯ |       NA |       NA |       NA |       NA |       NA |       NA |        NA |        NA |       NA |        NA |\n",
       "\n"
      ],
      "text/plain": [
       "     clinical_data_0_CK clinical_data_0_CK_MB clinical_data_0_CRP\n",
       "m1.1 7.467606           4.321928              3.137504           \n",
       "m1.2       NA                 NA                    NA           \n",
       "     clinical_data_0_delta_ef_value clinical_data_0_Troponin\n",
       "m1.1 NA                             0.75446                 \n",
       "m1.2 NA                                  NA                 \n",
       "     cytokine_0_BCA1__CXCL13 cytokine_0_CTACK__CCL27 cytokine_0_EGF__EGF\n",
       "m1.1 5.570766                9.747908                2.456806           \n",
       "m1.2       NA                      NA                      NA           \n",
       "     cytokine_0_ENA78__CXCL5 cytokine_0_Eotaxin__CCL11 ⋯\n",
       "m1.1 6.335569                5.617357                  ⋯\n",
       "m1.2       NA                      NA                  ⋯\n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2\n",
       "m1.1 2.201764                                      \n",
       "m1.2       NA                                      \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5\n",
       "m1.1 2.842767                                        \n",
       "m1.2       NA                                        \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1\n",
       "m1.1 2.610397                                       \n",
       "m1.2       NA                                       \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36\n",
       "m1.1 2.910977                                       \n",
       "m1.2       NA                                       \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1\n",
       "m1.1 2.461054                                         \n",
       "m1.2       NA                                         \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2\n",
       "m1.1 1.084743                                         \n",
       "m1.2       NA                                         \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106\n",
       "m1.1 0.7984776                                       \n",
       "m1.2        NA                                       \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207\n",
       "m1.1 0.4248882                                       \n",
       "m1.2        NA                                       \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706\n",
       "m1.1 1.451288                                        \n",
       "m1.2       NA                                        \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX\n",
       "m1.1 0.9409604                                    \n",
       "m1.2        NA                                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(expression_data,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "id": "f49e0f2f-c067-4356-a56b-0eecf3696c43",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Add lagged variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "5eb3675a-dbed-401e-8f02-1ba15fec2d21",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_data_ligands = expression_data[,colnames(expression_data) %in% expressed_ligands]\n",
    "expression_data_targets = expression_data[,!colnames(expression_data) %in% expressed_ligands]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "3355a0b7-90ce-47c7-9d6e-b6151cfcd36c",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_data_targets$sample = rownames(expression_data_targets)\n",
    "expression_data_targets$tp = str_replace(expression_data_targets$sample, '.*\\\\.', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "06f54fb7-206a-42d7-a2b1-3f419cf1b05b",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_data_targets$tp_lag = expression_data_targets$tp\n",
    "expression_data_targets$tp_lag = str_replace(expression_data_targets$tp_lag, '1', '0')\n",
    "expression_data_targets$tp_lag = str_replace(expression_data_targets$tp_lag, '2', '1')\n",
    "expression_data_targets$tp_lag = str_replace(expression_data_targets$tp_lag, '3', '2')\n",
    "expression_data_targets$tp_lag = str_replace(expression_data_targets$tp_lag, '4', '3')\n",
    "expression_data_targets$sample = str_replace(expression_data_targets$sample, '\\\\..*', '')\n",
    "expression_data_targets$sample_lag = paste0(expression_data_targets$sample, '.', expression_data_targets$tp_lag)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "0f122745-7c16-4c12-aaff-107584ce6819",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_data_targets_lagged1 = expression_data_targets\n",
    "rownames(expression_data_targets_lagged1) = expression_data_targets_lagged1$sample_lag\n",
    "expression_data_targets_lagged1$sample = NULL\n",
    "expression_data_targets_lagged1$tp = NULL\n",
    "expression_data_targets_lagged1$tp_lag = NULL\n",
    "expression_data_targets_lagged1$sample_lag = NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "id": "6525b5ff-2b39-4aff-a7e7-b7b5d07c91c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 12943</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>clinical_data_0_CK</th><th scope=col>clinical_data_0_CK_MB</th><th scope=col>clinical_data_0_CRP</th><th scope=col>clinical_data_0_delta_ef_value</th><th scope=col>clinical_data_0_Troponin</th><th scope=col>cytokine_0_Eotaxin__CCL11</th><th scope=col>cytokine_0_Eotaxin2__CCL24</th><th scope=col>cytokine_0_FLT3L__FLT3LG</th><th scope=col>cytokine_0_GROalpha__CXCL1</th><th scope=col>cytokine_0_IL12p40__IL12B</th><th scope=col>⋯</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX</th><th scope=col>sample</th><th scope=col>tp</th><th scope=col>tp_lag</th><th scope=col>sample_lag</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>m1.1</th><td>7.467606</td><td>4.321928</td><td>3.137504</td><td>NA</td><td>0.75446</td><td>5.617357</td><td>9.013713</td><td>3.944858</td><td>2.147307</td><td>3.502076</td><td>⋯</td><td>2.461054</td><td>1.084743</td><td>0.7984776</td><td>0.4248882</td><td>1.451288</td><td>0.9409604</td><td>m1</td><td>1</td><td>0</td><td>m1.0</td></tr>\n",
       "\t<tr><th scope=row>m1.2</th><td>      NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>     NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>⋯</td><td>      NA</td><td>      NA</td><td>       NA</td><td>       NA</td><td>      NA</td><td>       NA</td><td>m1</td><td>2</td><td>1</td><td>m1.1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 12943\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & clinical\\_data\\_0\\_CK & clinical\\_data\\_0\\_CK\\_MB & clinical\\_data\\_0\\_CRP & clinical\\_data\\_0\\_delta\\_ef\\_value & clinical\\_data\\_0\\_Troponin & cytokine\\_0\\_Eotaxin\\_\\_CCL11 & cytokine\\_0\\_Eotaxin2\\_\\_CCL24 & cytokine\\_0\\_FLT3L\\_\\_FLT3LG & cytokine\\_0\\_GROalpha\\_\\_CXCL1 & cytokine\\_0\\_IL12p40\\_\\_IL12B & ⋯ & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36L1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36L2 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF106 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF207 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF706 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZYX & sample & tp & tp\\_lag & sample\\_lag\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\tm1.1 & 7.467606 & 4.321928 & 3.137504 & NA & 0.75446 & 5.617357 & 9.013713 & 3.944858 & 2.147307 & 3.502076 & ⋯ & 2.461054 & 1.084743 & 0.7984776 & 0.4248882 & 1.451288 & 0.9409604 & m1 & 1 & 0 & m1.0\\\\\n",
       "\tm1.2 &       NA &       NA &       NA & NA &      NA &       NA &       NA &       NA &       NA &       NA & ⋯ &       NA &       NA &        NA &        NA &       NA &        NA & m1 & 2 & 1 & m1.1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 12943\n",
       "\n",
       "| <!--/--> | clinical_data_0_CK &lt;dbl&gt; | clinical_data_0_CK_MB &lt;dbl&gt; | clinical_data_0_CRP &lt;dbl&gt; | clinical_data_0_delta_ef_value &lt;dbl&gt; | clinical_data_0_Troponin &lt;dbl&gt; | cytokine_0_Eotaxin__CCL11 &lt;dbl&gt; | cytokine_0_Eotaxin2__CCL24 &lt;dbl&gt; | cytokine_0_FLT3L__FLT3LG &lt;dbl&gt; | cytokine_0_GROalpha__CXCL1 &lt;dbl&gt; | cytokine_0_IL12p40__IL12B &lt;dbl&gt; | ⋯ ⋯ | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX &lt;dbl&gt; | sample &lt;chr&gt; | tp &lt;chr&gt; | tp_lag &lt;chr&gt; | sample_lag &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| m1.1 | 7.467606 | 4.321928 | 3.137504 | NA | 0.75446 | 5.617357 | 9.013713 | 3.944858 | 2.147307 | 3.502076 | ⋯ | 2.461054 | 1.084743 | 0.7984776 | 0.4248882 | 1.451288 | 0.9409604 | m1 | 1 | 0 | m1.0 |\n",
       "| m1.2 |       NA |       NA |       NA | NA |      NA |       NA |       NA |       NA |       NA |       NA | ⋯ |       NA |       NA |        NA |        NA |       NA |        NA | m1 | 2 | 1 | m1.1 |\n",
       "\n"
      ],
      "text/plain": [
       "     clinical_data_0_CK clinical_data_0_CK_MB clinical_data_0_CRP\n",
       "m1.1 7.467606           4.321928              3.137504           \n",
       "m1.2       NA                 NA                    NA           \n",
       "     clinical_data_0_delta_ef_value clinical_data_0_Troponin\n",
       "m1.1 NA                             0.75446                 \n",
       "m1.2 NA                                  NA                 \n",
       "     cytokine_0_Eotaxin__CCL11 cytokine_0_Eotaxin2__CCL24\n",
       "m1.1 5.617357                  9.013713                  \n",
       "m1.2       NA                        NA                  \n",
       "     cytokine_0_FLT3L__FLT3LG cytokine_0_GROalpha__CXCL1\n",
       "m1.1 3.944858                 2.147307                  \n",
       "m1.2       NA                       NA                  \n",
       "     cytokine_0_IL12p40__IL12B ⋯\n",
       "m1.1 3.502076                  ⋯\n",
       "m1.2       NA                  ⋯\n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1\n",
       "m1.1 2.461054                                         \n",
       "m1.2       NA                                         \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2\n",
       "m1.1 1.084743                                         \n",
       "m1.2       NA                                         \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106\n",
       "m1.1 0.7984776                                       \n",
       "m1.2        NA                                       \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207\n",
       "m1.1 0.4248882                                       \n",
       "m1.2        NA                                       \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706\n",
       "m1.1 1.451288                                        \n",
       "m1.2       NA                                        \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX sample tp tp_lag sample_lag\n",
       "m1.1 0.9409604                                     m1     1  0      m1.0      \n",
       "m1.2        NA                                     m1     2  1      m1.1      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(expression_data_targets,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "id": "6588cf2a-bad0-46b6-a5bc-d28cd39aa8c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 12939</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>clinical_data_0_CK</th><th scope=col>clinical_data_0_CK_MB</th><th scope=col>clinical_data_0_CRP</th><th scope=col>clinical_data_0_delta_ef_value</th><th scope=col>clinical_data_0_Troponin</th><th scope=col>cytokine_0_Eotaxin__CCL11</th><th scope=col>cytokine_0_Eotaxin2__CCL24</th><th scope=col>cytokine_0_FLT3L__FLT3LG</th><th scope=col>cytokine_0_GROalpha__CXCL1</th><th scope=col>cytokine_0_IL12p40__IL12B</th><th scope=col>⋯</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>m1.0</th><td>7.467606</td><td>4.321928</td><td>3.137504</td><td>NA</td><td>0.75446</td><td>5.617357</td><td>9.013713</td><td>3.944858</td><td>2.147307</td><td>3.502076</td><td>⋯</td><td>2.201764</td><td>2.842767</td><td>2.610397</td><td>2.910977</td><td>2.461054</td><td>1.084743</td><td>0.7984776</td><td>0.4248882</td><td>1.451288</td><td>0.9409604</td></tr>\n",
       "\t<tr><th scope=row>m1.1</th><td>      NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>     NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>⋯</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>       NA</td><td>       NA</td><td>      NA</td><td>       NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 12939\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & clinical\\_data\\_0\\_CK & clinical\\_data\\_0\\_CK\\_MB & clinical\\_data\\_0\\_CRP & clinical\\_data\\_0\\_delta\\_ef\\_value & clinical\\_data\\_0\\_Troponin & cytokine\\_0\\_Eotaxin\\_\\_CCL11 & cytokine\\_0\\_Eotaxin2\\_\\_CCL24 & cytokine\\_0\\_FLT3L\\_\\_FLT3LG & cytokine\\_0\\_GROalpha\\_\\_CXCL1 & cytokine\\_0\\_IL12p40\\_\\_IL12B & ⋯ & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZEB2 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFAND5 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFAS1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36L1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36L2 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF106 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF207 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF706 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZYX\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tm1.0 & 7.467606 & 4.321928 & 3.137504 & NA & 0.75446 & 5.617357 & 9.013713 & 3.944858 & 2.147307 & 3.502076 & ⋯ & 2.201764 & 2.842767 & 2.610397 & 2.910977 & 2.461054 & 1.084743 & 0.7984776 & 0.4248882 & 1.451288 & 0.9409604\\\\\n",
       "\tm1.1 &       NA &       NA &       NA & NA &      NA &       NA &       NA &       NA &       NA &       NA & ⋯ &       NA &       NA &       NA &       NA &       NA &       NA &        NA &        NA &       NA &        NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 12939\n",
       "\n",
       "| <!--/--> | clinical_data_0_CK &lt;dbl&gt; | clinical_data_0_CK_MB &lt;dbl&gt; | clinical_data_0_CRP &lt;dbl&gt; | clinical_data_0_delta_ef_value &lt;dbl&gt; | clinical_data_0_Troponin &lt;dbl&gt; | cytokine_0_Eotaxin__CCL11 &lt;dbl&gt; | cytokine_0_Eotaxin2__CCL24 &lt;dbl&gt; | cytokine_0_FLT3L__FLT3LG &lt;dbl&gt; | cytokine_0_GROalpha__CXCL1 &lt;dbl&gt; | cytokine_0_IL12p40__IL12B &lt;dbl&gt; | ⋯ ⋯ | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| m1.0 | 7.467606 | 4.321928 | 3.137504 | NA | 0.75446 | 5.617357 | 9.013713 | 3.944858 | 2.147307 | 3.502076 | ⋯ | 2.201764 | 2.842767 | 2.610397 | 2.910977 | 2.461054 | 1.084743 | 0.7984776 | 0.4248882 | 1.451288 | 0.9409604 |\n",
       "| m1.1 |       NA |       NA |       NA | NA |      NA |       NA |       NA |       NA |       NA |       NA | ⋯ |       NA |       NA |       NA |       NA |       NA |       NA |        NA |        NA |       NA |        NA |\n",
       "\n"
      ],
      "text/plain": [
       "     clinical_data_0_CK clinical_data_0_CK_MB clinical_data_0_CRP\n",
       "m1.0 7.467606           4.321928              3.137504           \n",
       "m1.1       NA                 NA                    NA           \n",
       "     clinical_data_0_delta_ef_value clinical_data_0_Troponin\n",
       "m1.0 NA                             0.75446                 \n",
       "m1.1 NA                                  NA                 \n",
       "     cytokine_0_Eotaxin__CCL11 cytokine_0_Eotaxin2__CCL24\n",
       "m1.0 5.617357                  9.013713                  \n",
       "m1.1       NA                        NA                  \n",
       "     cytokine_0_FLT3L__FLT3LG cytokine_0_GROalpha__CXCL1\n",
       "m1.0 3.944858                 2.147307                  \n",
       "m1.1       NA                       NA                  \n",
       "     cytokine_0_IL12p40__IL12B ⋯ single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2\n",
       "m1.0 3.502076                  ⋯ 2.201764                                      \n",
       "m1.1       NA                  ⋯       NA                                      \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5\n",
       "m1.0 2.842767                                        \n",
       "m1.1       NA                                        \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1\n",
       "m1.0 2.610397                                       \n",
       "m1.1       NA                                       \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36\n",
       "m1.0 2.910977                                       \n",
       "m1.1       NA                                       \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1\n",
       "m1.0 2.461054                                         \n",
       "m1.1       NA                                         \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2\n",
       "m1.0 1.084743                                         \n",
       "m1.1       NA                                         \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106\n",
       "m1.0 0.7984776                                       \n",
       "m1.1        NA                                       \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207\n",
       "m1.0 0.4248882                                       \n",
       "m1.1        NA                                       \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706\n",
       "m1.0 1.451288                                        \n",
       "m1.1       NA                                        \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX\n",
       "m1.0 0.9409604                                    \n",
       "m1.1        NA                                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(expression_data_targets_lagged1,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "ce0005fa-d87a-4570-800c-b0f233384b0e",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Merge lagged dataset by rownames\n",
    "expression_data_lagged1 = merge(expression_data_ligands, expression_data_targets_lagged1,  by = 'row.names')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "id": "08fe717d-e60a-48d7-bee5-6b65d72c6e32",
   "metadata": {},
   "outputs": [],
   "source": [
    "rownames(expression_data_lagged1 ) = expression_data_lagged1$Row.names\n",
    "expression_data_lagged1$Row.names = NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "id": "048c29dd-8227-49f6-b07c-e7030e9f8ce6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 13282</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cytokine_0_BCA1__CXCL13</th><th scope=col>cytokine_0_CTACK__CCL27</th><th scope=col>cytokine_0_EGF__EGF</th><th scope=col>cytokine_0_ENA78__CXCL5</th><th scope=col>cytokine_0_Eotaxin3__CCL26</th><th scope=col>cytokine_0_FGF2__FGF2</th><th scope=col>cytokine_0_Fractalkine__CX3CL1</th><th scope=col>cytokine_0_GCSF__CSF3</th><th scope=col>cytokine_0_GMCSF__CSF2</th><th scope=col>cytokine_0_I309__CCL1</th><th scope=col>⋯</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>m1.1</th><td>5.570766</td><td>9.747908</td><td>2.456806</td><td>6.335569</td><td>1.883621</td><td>5.74873</td><td>5.874305</td><td>3.277985</td><td> 0</td><td>1.090853</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>m1.2</th><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>      NA</td><td>     NA</td><td>      NA</td><td>      NA</td><td>NA</td><td>      NA</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 13282\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & cytokine\\_0\\_BCA1\\_\\_CXCL13 & cytokine\\_0\\_CTACK\\_\\_CCL27 & cytokine\\_0\\_EGF\\_\\_EGF & cytokine\\_0\\_ENA78\\_\\_CXCL5 & cytokine\\_0\\_Eotaxin3\\_\\_CCL26 & cytokine\\_0\\_FGF2\\_\\_FGF2 & cytokine\\_0\\_Fractalkine\\_\\_CX3CL1 & cytokine\\_0\\_GCSF\\_\\_CSF3 & cytokine\\_0\\_GMCSF\\_\\_CSF2 & cytokine\\_0\\_I309\\_\\_CCL1 & ⋯ & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZEB2 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFAND5 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFAS1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36L1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36L2 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF106 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF207 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF706 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZYX\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tm1.1 & 5.570766 & 9.747908 & 2.456806 & 6.335569 & 1.883621 & 5.74873 & 5.874305 & 3.277985 &  0 & 1.090853 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\tm1.2 &       NA &       NA &       NA &       NA &       NA &      NA &       NA &       NA & NA &       NA & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 13282\n",
       "\n",
       "| <!--/--> | cytokine_0_BCA1__CXCL13 &lt;dbl&gt; | cytokine_0_CTACK__CCL27 &lt;dbl&gt; | cytokine_0_EGF__EGF &lt;dbl&gt; | cytokine_0_ENA78__CXCL5 &lt;dbl&gt; | cytokine_0_Eotaxin3__CCL26 &lt;dbl&gt; | cytokine_0_FGF2__FGF2 &lt;dbl&gt; | cytokine_0_Fractalkine__CX3CL1 &lt;dbl&gt; | cytokine_0_GCSF__CSF3 &lt;dbl&gt; | cytokine_0_GMCSF__CSF2 &lt;dbl&gt; | cytokine_0_I309__CCL1 &lt;dbl&gt; | ⋯ ⋯ | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706 &lt;dbl&gt; | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| m1.1 | 5.570766 | 9.747908 | 2.456806 | 6.335569 | 1.883621 | 5.74873 | 5.874305 | 3.277985 |  0 | 1.090853 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| m1.2 |       NA |       NA |       NA |       NA |       NA |      NA |       NA |       NA | NA |       NA | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "     cytokine_0_BCA1__CXCL13 cytokine_0_CTACK__CCL27 cytokine_0_EGF__EGF\n",
       "m1.1 5.570766                9.747908                2.456806           \n",
       "m1.2       NA                      NA                      NA           \n",
       "     cytokine_0_ENA78__CXCL5 cytokine_0_Eotaxin3__CCL26 cytokine_0_FGF2__FGF2\n",
       "m1.1 6.335569                1.883621                   5.74873              \n",
       "m1.2       NA                      NA                        NA              \n",
       "     cytokine_0_Fractalkine__CX3CL1 cytokine_0_GCSF__CSF3\n",
       "m1.1 5.874305                       3.277985             \n",
       "m1.2       NA                             NA             \n",
       "     cytokine_0_GMCSF__CSF2 cytokine_0_I309__CCL1 ⋯\n",
       "m1.1  0                     1.090853              ⋯\n",
       "m1.2 NA                           NA              ⋯\n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2\n",
       "m1.1 NA                                            \n",
       "m1.2 NA                                            \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5\n",
       "m1.1 NA                                              \n",
       "m1.2 NA                                              \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1\n",
       "m1.1 NA                                             \n",
       "m1.2 NA                                             \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36\n",
       "m1.1 NA                                             \n",
       "m1.2 NA                                             \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1\n",
       "m1.1 NA                                               \n",
       "m1.2 NA                                               \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2\n",
       "m1.1 NA                                               \n",
       "m1.2 NA                                               \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106\n",
       "m1.1 NA                                              \n",
       "m1.2 NA                                              \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207\n",
       "m1.1 NA                                              \n",
       "m1.2 NA                                              \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706\n",
       "m1.1 NA                                              \n",
       "m1.2 NA                                              \n",
       "     single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX\n",
       "m1.1 NA                                           \n",
       "m1.2 NA                                           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(expression_data_lagged1,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "id": "7b8b3b50-c35a-4fda-b433-34ae69f05855",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "84"
      ],
      "text/latex": [
       "84"
      ],
      "text/markdown": [
       "84"
      ],
      "text/plain": [
       "[1] 84"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(expression_data_lagged1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "id": "4d7fa7e4-87da-42cf-be8b-7cd108fb9f92",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Calculate lagged correlations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "id": "5494afdd-6628-4711-9beb-739f43ad13a1",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n",
      "Warning message in stats::cor(x, y, use, method):\n",
      "“the standard deviation is zero”\n"
     ]
    }
   ],
   "source": [
    "ligand_target_correlations = cor(expression_data_lagged1[,colnames(expression_data_lagged1) %in% expressed_ligands], expression_data_lagged1[,!colnames(expression_data_lagged1) %in% expressed_ligands], method = 'spearman', use = 'pairwise.complete.obs') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "1aeb596e-d579-45c9-836a-929524e508b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 2 × 12939 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>clinical_data_0_CK</th><th scope=col>clinical_data_0_CK_MB</th><th scope=col>clinical_data_0_CRP</th><th scope=col>clinical_data_0_delta_ef_value</th><th scope=col>clinical_data_0_Troponin</th><th scope=col>cytokine_0_Eotaxin__CCL11</th><th scope=col>cytokine_0_Eotaxin2__CCL24</th><th scope=col>cytokine_0_FLT3L__FLT3LG</th><th scope=col>cytokine_0_GROalpha__CXCL1</th><th scope=col>cytokine_0_IL12p40__IL12B</th><th scope=col>⋯</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706</th><th scope=col>single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>cytokine_0_BCA1__CXCL13</th><td> 0.2267960</td><td> 0.23290413</td><td>-0.03414252</td><td>-0.2322271</td><td>0.21535575</td><td>-0.1236559</td><td>0.01003276</td><td>-0.22891364</td><td>-0.3385113</td><td>-0.2668651</td><td>⋯</td><td> 0.500991674</td><td> 0.1973448</td><td> 0.08517446</td><td> 0.06217439</td><td>0.120236931</td><td> 0.01181977</td><td>0.10062677</td><td> 0.18792976</td><td>0.09434661</td><td>-0.02692659</td></tr>\n",
       "\t<tr><th scope=row>cytokine_0_CTACK__CCL27</th><td>-0.1321796</td><td>-0.07370766</td><td>-0.21829420</td><td> 0.0271015</td><td>0.07871666</td><td>-0.1298483</td><td>0.13897056</td><td>-0.06036676</td><td> 0.1295516</td><td> 0.3087579</td><td>⋯</td><td>-0.001533783</td><td>-0.1684650</td><td>-0.20541298</td><td>-0.42644065</td><td>0.008964222</td><td>-0.17581575</td><td>0.04926877</td><td>-0.05945203</td><td>0.16225078</td><td>-0.10117309</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 2 × 12939 of type dbl\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & clinical\\_data\\_0\\_CK & clinical\\_data\\_0\\_CK\\_MB & clinical\\_data\\_0\\_CRP & clinical\\_data\\_0\\_delta\\_ef\\_value & clinical\\_data\\_0\\_Troponin & cytokine\\_0\\_Eotaxin\\_\\_CCL11 & cytokine\\_0\\_Eotaxin2\\_\\_CCL24 & cytokine\\_0\\_FLT3L\\_\\_FLT3LG & cytokine\\_0\\_GROalpha\\_\\_CXCL1 & cytokine\\_0\\_IL12p40\\_\\_IL12B & ⋯ & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZEB2 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFAND5 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFAS1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36L1 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZFP36L2 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF106 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF207 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZNF706 & single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ZYX\\\\\n",
       "\\hline\n",
       "\tcytokine\\_0\\_BCA1\\_\\_CXCL13 &  0.2267960 &  0.23290413 & -0.03414252 & -0.2322271 & 0.21535575 & -0.1236559 & 0.01003276 & -0.22891364 & -0.3385113 & -0.2668651 & ⋯ &  0.500991674 &  0.1973448 &  0.08517446 &  0.06217439 & 0.120236931 &  0.01181977 & 0.10062677 &  0.18792976 & 0.09434661 & -0.02692659\\\\\n",
       "\tcytokine\\_0\\_CTACK\\_\\_CCL27 & -0.1321796 & -0.07370766 & -0.21829420 &  0.0271015 & 0.07871666 & -0.1298483 & 0.13897056 & -0.06036676 &  0.1295516 &  0.3087579 & ⋯ & -0.001533783 & -0.1684650 & -0.20541298 & -0.42644065 & 0.008964222 & -0.17581575 & 0.04926877 & -0.05945203 & 0.16225078 & -0.10117309\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 2 × 12939 of type dbl\n",
       "\n",
       "| <!--/--> | clinical_data_0_CK | clinical_data_0_CK_MB | clinical_data_0_CRP | clinical_data_0_delta_ef_value | clinical_data_0_Troponin | cytokine_0_Eotaxin__CCL11 | cytokine_0_Eotaxin2__CCL24 | cytokine_0_FLT3L__FLT3LG | cytokine_0_GROalpha__CXCL1 | cytokine_0_IL12p40__IL12B | ⋯ | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706 | single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cytokine_0_BCA1__CXCL13 |  0.2267960 |  0.23290413 | -0.03414252 | -0.2322271 | 0.21535575 | -0.1236559 | 0.01003276 | -0.22891364 | -0.3385113 | -0.2668651 | ⋯ |  0.500991674 |  0.1973448 |  0.08517446 |  0.06217439 | 0.120236931 |  0.01181977 | 0.10062677 |  0.18792976 | 0.09434661 | -0.02692659 |\n",
       "| cytokine_0_CTACK__CCL27 | -0.1321796 | -0.07370766 | -0.21829420 |  0.0271015 | 0.07871666 | -0.1298483 | 0.13897056 | -0.06036676 |  0.1295516 |  0.3087579 | ⋯ | -0.001533783 | -0.1684650 | -0.20541298 | -0.42644065 | 0.008964222 | -0.17581575 | 0.04926877 | -0.05945203 | 0.16225078 | -0.10117309 |\n",
       "\n"
      ],
      "text/plain": [
       "                        clinical_data_0_CK clinical_data_0_CK_MB\n",
       "cytokine_0_BCA1__CXCL13  0.2267960          0.23290413          \n",
       "cytokine_0_CTACK__CCL27 -0.1321796         -0.07370766          \n",
       "                        clinical_data_0_CRP clinical_data_0_delta_ef_value\n",
       "cytokine_0_BCA1__CXCL13 -0.03414252         -0.2322271                    \n",
       "cytokine_0_CTACK__CCL27 -0.21829420          0.0271015                    \n",
       "                        clinical_data_0_Troponin cytokine_0_Eotaxin__CCL11\n",
       "cytokine_0_BCA1__CXCL13 0.21535575               -0.1236559               \n",
       "cytokine_0_CTACK__CCL27 0.07871666               -0.1298483               \n",
       "                        cytokine_0_Eotaxin2__CCL24 cytokine_0_FLT3L__FLT3LG\n",
       "cytokine_0_BCA1__CXCL13 0.01003276                 -0.22891364             \n",
       "cytokine_0_CTACK__CCL27 0.13897056                 -0.06036676             \n",
       "                        cytokine_0_GROalpha__CXCL1 cytokine_0_IL12p40__IL12B ⋯\n",
       "cytokine_0_BCA1__CXCL13 -0.3385113                 -0.2668651                ⋯\n",
       "cytokine_0_CTACK__CCL27  0.1295516                  0.3087579                ⋯\n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZEB2\n",
       "cytokine_0_BCA1__CXCL13  0.500991674                                  \n",
       "cytokine_0_CTACK__CCL27 -0.001533783                                  \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAND5\n",
       "cytokine_0_BCA1__CXCL13  0.1973448                                      \n",
       "cytokine_0_CTACK__CCL27 -0.1684650                                      \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFAS1\n",
       "cytokine_0_BCA1__CXCL13  0.08517446                                    \n",
       "cytokine_0_CTACK__CCL27 -0.20541298                                    \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36\n",
       "cytokine_0_BCA1__CXCL13  0.06217439                                    \n",
       "cytokine_0_CTACK__CCL27 -0.42644065                                    \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L1\n",
       "cytokine_0_BCA1__CXCL13 0.120236931                                      \n",
       "cytokine_0_CTACK__CCL27 0.008964222                                      \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZFP36L2\n",
       "cytokine_0_BCA1__CXCL13  0.01181977                                      \n",
       "cytokine_0_CTACK__CCL27 -0.17581575                                      \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF106\n",
       "cytokine_0_BCA1__CXCL13 0.10062677                                      \n",
       "cytokine_0_CTACK__CCL27 0.04926877                                      \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF207\n",
       "cytokine_0_BCA1__CXCL13  0.18792976                                     \n",
       "cytokine_0_CTACK__CCL27 -0.05945203                                     \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZNF706\n",
       "cytokine_0_BCA1__CXCL13 0.09434661                                      \n",
       "cytokine_0_CTACK__CCL27 0.16225078                                      \n",
       "                        single_cell_0_X9_Monocytes...CD16_FCGR3A__ZYX\n",
       "cytokine_0_BCA1__CXCL13 -0.02692659                                  \n",
       "cytokine_0_CTACK__CCL27 -0.10117309                                  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_target_correlations,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "88ae81b8-96d5-4fed-9a7d-a7f55c51df6b",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Transform to long format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "id": "f7fd06cc-1689-4038-83aa-420d6e2f7119",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations = melt(ligand_target_correlations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "d64d14aa-8de6-435d-aeba-7a246ce5c438",
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames(ligand_target_correlations) = c('Ligand', 'Target', 'correlation')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "dbb19653-a67b-4c21-b10d-3bcc9667532d",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Adjust and extract certain columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "id": "d99fd456-355e-40ad-9c1f-fcc15088ec1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Target_Gene = str_replace(ligand_target_correlations$Target, '.*_0_', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "345aa6c1-90b5-447c-a0e8-2cf6f6c1701b",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Target_Gene = str_replace(ligand_target_correlations$Target_Gene, '.*__', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "id": "12bd9b23-45c6-47de-bf05-d845c1959994",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Ligand_Gene = str_replace(ligand_target_correlations$Ligand, '.*_0_', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "id": "429f3bf9-d4cb-497d-ab13-380473e818e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Ligand_Gene = str_replace(ligand_target_correlations$Ligand_Gene, '.*__', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "id": "5dc6497b-6389-4f82-bfdc-bafe74421b29",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Target_Type = str_replace(ligand_target_correlations$Target, '_0_.*', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "id": "e28db9c1-7f65-443b-9eae-35c4a5b2325b",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Target_Type[ligand_target_correlations$Target_Type == 'single_cell']  = str_replace(ligand_target_correlations$Target[ligand_target_correlations$Target_Type == 'single_cell'] , '__.*', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "49e1e65e-9f94-4255-8a65-75e55e27120c",
   "metadata": {},
   "outputs": [],
   "source": [
    "###  Adapt target gene for proteomics\n",
    "ligand_target_correlations$Target_Gene[ligand_target_correlations$Target_Type == 'proteomics'] = str_replace(ligand_target_correlations$Target_Gene[ligand_target_correlations$Target_Type == 'proteomics'], '_.*', '')\n",
    "ligand_target_correlations$Ligand_Gene[ligand_target_correlations$Ligand_Type == 'proteomics'] = str_replace(ligand_target_correlations$Ligand_Gene[ligand_target_correlations$Ligand_Type == 'proteomics'], '_.*', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "id": "f93e38f2-6b42-4968-8986-b41ae12195b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Ligand_Type = str_replace(ligand_target_correlations$Ligand, '_0_.*', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "3c4c5166-477f-451a-a631-cc8703201ee3",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Ligand_Type[ligand_target_correlations$Ligand_Type == 'single_cell']  = str_replace(ligand_target_correlations$Ligand[ligand_target_correlations$Ligand_Type == 'single_cell'] , '__.*', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "id": "62fee583-2df2-4bbf-83d9-a1d86b1d02c1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>cytokine_0_BCA1__CXCL13</li><li>cytokine_0_CTACK__CCL27</li><li>cytokine_0_EGF__EGF</li><li>cytokine_0_ENA78__CXCL5</li><li>cytokine_0_Eotaxin3__CCL26</li><li>cytokine_0_FGF2__FGF2</li></ol>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'cytokine_0_BCA1__CXCL13'</li><li>'cytokine_0_CTACK__CCL27'</li><li>'cytokine_0_EGF__EGF'</li><li>'cytokine_0_ENA78__CXCL5'</li><li>'cytokine_0_Eotaxin3__CCL26'</li><li>'cytokine_0_FGF2__FGF2'</li><li>'cytokine_0_Fractalkine__CX3CL1'</li><li>'cytokine_0_GCSF__CSF3'</li><li>'cytokine_0_GMCSF__CSF2'</li><li>'cytokine_0_I309__CCL1'</li><li>'cytokine_0_IFNalpha2__IFNA2'</li><li>'cytokine_0_IFNgamma__IFNG'</li><li>'cytokine_0_IL10__IL10'</li><li>'cytokine_0_IL12p70__IL12A'</li><li>'cytokine_0_IL13__IL13'</li><li>'cytokine_0_IL15__IL15'</li><li>'cytokine_0_IL17A__IL17A'</li><li>'cytokine_0_IL17E_IL25__IL25'</li><li>'cytokine_0_IL17F__IL17F'</li><li>'cytokine_0_IL1alpha__IL1A'</li><li>'cytokine_0_IL1beta__IL1B'</li><li>'cytokine_0_IL1RA__IL1RN'</li><li>'cytokine_0_IL2__IL2'</li><li>'cytokine_0_IL22__IL22'</li><li>'cytokine_0_IL27__IL27'</li><li>'cytokine_0_IL5__IL5'</li><li>'cytokine_0_IL6__IL6'</li><li>'cytokine_0_IL7__IL7'</li><li>'cytokine_0_IL9__IL9'</li><li>'cytokine_0_IP10__CXCL10'</li><li>'cytokine_0_LIF__LIF'</li><li>'cytokine_0_MCP1__CCL2'</li><li>'cytokine_0_MCP2__CCL8'</li><li>'cytokine_0_MCP3__CCL7'</li><li>'cytokine_0_MCP4__CCL13'</li><li>'cytokine_0_MCSF__CSF1'</li><li>'cytokine_0_MDC__CCL22'</li><li>'cytokine_0_MIG_CXCL9__CXCL9'</li><li>'cytokine_0_MIP1alpha__CCL3'</li><li>'cytokine_0_MIP1beta__CCL4'</li><li>'cytokine_0_MIP1delta__CCL15'</li><li>'cytokine_0_RANTES__CCL5'</li><li>'cytokine_0_SCF__KITLG'</li><li>'cytokine_0_SDF1alpha_beta__CXCL12'</li><li>'cytokine_0_TARC__CCL17'</li><li>'cytokine_0_TNFalpha__TNF'</li><li>'cytokine_0_TNFbeta__LTA'</li><li>'cytokine_0_VEGFA__VEGFA'</li><li>'cytokine_0_X6CKine__CCL21'</li><li>'neutrophil_0_ADM'</li><li>'neutrophil_0_FBRS'</li><li>'neutrophil_0_GRN'</li><li>'neutrophil_0_HMGB1'</li><li>'neutrophil_0_HMGB2'</li><li>'neutrophil_0_IL16'</li><li>'neutrophil_0_MMP9'</li><li>'neutrophil_0_PGLYRP1'</li><li>'neutrophil_0_RTN4'</li><li>'neutrophil_0_SECTM1'</li><li>'neutrophil_0_TGFB1'</li><li>'neutrophil_0_TIMP1'</li><li>'neutrophil_0_TNFSF13B'</li><li>'proteomics_0_A2M_P01023'</li><li>'proteomics_0_ADIPOQ_Q15848'</li><li>'proteomics_0_AGT_P01019'</li><li>'proteomics_0_APOA1_P02647'</li><li>'proteomics_0_APOB_P04114'</li><li>'proteomics_0_APOE_P02649'</li><li>'proteomics_0_C3_M0R1Q1'</li><li>'proteomics_0_C3_P01024'</li><li>'proteomics_0_C5_P01031'</li><li>'proteomics_0_CAMP_J3KNB4'</li><li>'proteomics_0_CDH5_P33151'</li><li>'proteomics_0_CEACAM1_P13688'</li><li>'proteomics_0_CFH_P08603'</li><li>'proteomics_0_COL18A1_P39060'</li><li>'proteomics_0_COMP_G3XAP6'</li><li>'proteomics_0_DSG2_Q14126'</li><li>'proteomics_0_F2_P00734'</li><li>'proteomics_0_F9_P00740'</li><li>'proteomics_0_FGA_P02671'</li><li>'proteomics_0_FN1_H0Y4K8'</li><li>'proteomics_0_FN1_P02751'</li><li>'proteomics_0_GSTP1_P09211'</li><li>'proteomics_0_HSP90B1_P14625'</li><li>'proteomics_0_HSPG2_P98160'</li><li>'proteomics_0_ICAM1_P05362'</li><li>'proteomics_0_ICAM2_J3QQR8'</li><li>'proteomics_0_IGF2_P01344'</li><li>'proteomics_0_KNG1_P01042'</li><li>'proteomics_0_KNG1_P01042.2'</li><li>'proteomics_0_LTF_E7EQB2'</li><li>'proteomics_0_MMP9_P14780'</li><li>'proteomics_0_NID1_P14543'</li><li>'proteomics_0_PF4_P02776'</li><li>'proteomics_0_PLG_P00747'</li><li>'proteomics_0_PPBP_P02775'</li><li>'proteomics_0_SAA1_P0DJI8'</li><li>'proteomics_0_SELL_P14151'</li><li>'proteomics_0_SERPINA1_A0A024R6I7'</li><li>'proteomics_0_SERPINA1_P01009'</li><li>'proteomics_0_SERPINC1_P01008'</li><li>'proteomics_0_SERPING1_P05155'</li><li>'proteomics_0_TF_P02787'</li><li>'proteomics_0_THBS1_P07996'</li><li>'proteomics_0_TIMP1_P01033'</li><li>'proteomics_0_TTR_P02766'</li><li>'proteomics_0_VTN_P04004'</li><li>'single_cell_0_X0_T.cell.CD4__ANXA1'</li><li>'single_cell_0_X0_T.cell.CD4__ARF1'</li><li>'single_cell_0_X0_T.cell.CD4__CALM1'</li><li>'single_cell_0_X0_T.cell.CD4__CALR'</li><li>'single_cell_0_X0_T.cell.CD4__CD2'</li><li>'single_cell_0_X0_T.cell.CD4__HMGB1'</li><li>'single_cell_0_X0_T.cell.CD4__HMGB2'</li><li>'single_cell_0_X0_T.cell.CD4__HSP90B1'</li><li>'single_cell_0_X0_T.cell.CD4__ICAM3'</li><li>'single_cell_0_X0_T.cell.CD4__LTB'</li><li>'single_cell_0_X0_T.cell.CD4__MIF'</li><li>'single_cell_0_X0_T.cell.CD4__PTPRC'</li><li>'single_cell_0_X0_T.cell.CD4__SELL'</li><li>'single_cell_0_X1_T.cell.CD8__ANXA1'</li><li>'single_cell_0_X1_T.cell.CD8__ARF1'</li><li>'single_cell_0_X1_T.cell.CD8__CALM1'</li><li>'single_cell_0_X1_T.cell.CD8__CALR'</li><li>'single_cell_0_X1_T.cell.CD8__CCL4'</li><li>'single_cell_0_X1_T.cell.CD8__CCL5'</li><li>'single_cell_0_X1_T.cell.CD8__CD2'</li><li>'single_cell_0_X1_T.cell.CD8__GSTP1'</li><li>'single_cell_0_X1_T.cell.CD8__GZMB'</li><li>'single_cell_0_X1_T.cell.CD8__HMGB1'</li><li>'single_cell_0_X1_T.cell.CD8__HMGB2'</li><li>'single_cell_0_X1_T.cell.CD8__HSP90B1'</li><li>'single_cell_0_X1_T.cell.CD8__ICAM3'</li><li>'single_cell_0_X1_T.cell.CD8__LTB'</li><li>'single_cell_0_X1_T.cell.CD8__PTPRC'</li><li>'single_cell_0_X1_T.cell.CD8__TGFB1'</li><li>'single_cell_0_X10_B.cell__ANXA1'</li><li>'single_cell_0_X10_B.cell__ARF1'</li><li>'single_cell_0_X10_B.cell__CALM1'</li><li>'single_cell_0_X10_B.cell__CALR'</li><li>'single_cell_0_X10_B.cell__CD2'</li><li>'single_cell_0_X10_B.cell__GSTP1'</li><li>'single_cell_0_X10_B.cell__HMGB1'</li><li>'single_cell_0_X10_B.cell__HMGB2'</li><li>'single_cell_0_X10_B.cell__HSP90B1'</li><li>'single_cell_0_X10_B.cell__ICAM3'</li><li>'single_cell_0_X10_B.cell__LTB'</li><li>'single_cell_0_X10_B.cell__PTPRC'</li><li>'single_cell_0_X10_B.cell__SELL'</li><li>'single_cell_0_X10_B.cell__TGFB1'</li><li>'single_cell_0_X11_T.cell.CD4__ANXA1'</li><li>'single_cell_0_X11_T.cell.CD4__ARF1'</li><li>'single_cell_0_X11_T.cell.CD4__CALM1'</li><li>'single_cell_0_X11_T.cell.CD4__CALR'</li><li>'single_cell_0_X11_T.cell.CD4__CD2'</li><li>'single_cell_0_X11_T.cell.CD4__HMGB1'</li><li>'single_cell_0_X11_T.cell.CD4__HMGB2'</li><li>'single_cell_0_X11_T.cell.CD4__HSP90B1'</li><li>'single_cell_0_X11_T.cell.CD4__ICAM3'</li><li>'single_cell_0_X11_T.cell.CD4__LTB'</li><li>'single_cell_0_X11_T.cell.CD4__PTPRC'</li><li>'single_cell_0_X11_T.cell.CD4__SELL'</li><li>'single_cell_0_X11_T.cell.CD4__TGFB1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__ANXA1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__ARF1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__CALM1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__CALR'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__GSTP1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__HMGB1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__HMGB2'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__HSP90B1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__ICAM3'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__IL1B'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__LGALS3'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__LTB'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__NAMPT'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__PECAM1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__PSAP'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__PTPRC'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__RTN4'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__SERPINA1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__TGFB1'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A__TIMP1'</li><li>'single_cell_0_X13_Dendritic__ANXA1'</li><li>'single_cell_0_X13_Dendritic__CALM1'</li><li>'single_cell_0_X13_Dendritic__CALR'</li><li>'single_cell_0_X13_Dendritic__GRN'</li><li>'single_cell_0_X13_Dendritic__GSTP1'</li><li>'single_cell_0_X13_Dendritic__HMGB1'</li><li>'single_cell_0_X13_Dendritic__HSP90B1'</li><li>'single_cell_0_X13_Dendritic__NAMPT'</li><li>'single_cell_0_X13_Dendritic__PSAP'</li><li>'single_cell_0_X13_Dendritic__PTPRC'</li><li>'single_cell_0_X13_Dendritic__RTN4'</li><li>'single_cell_0_X13_Dendritic__TIMP1'</li><li>'single_cell_0_X2_T.cell.CD4__ANXA1'</li><li>'single_cell_0_X2_T.cell.CD4__ARF1'</li><li>'single_cell_0_X2_T.cell.CD4__CALM1'</li><li>'single_cell_0_X2_T.cell.CD4__CALR'</li><li>'single_cell_0_X2_T.cell.CD4__CD2'</li><li>'single_cell_0_X2_T.cell.CD4__HMGB1'</li><li>'single_cell_0_X2_T.cell.CD4__HMGB2'</li><li>'single_cell_0_X2_T.cell.CD4__HSP90B1'</li><li>'single_cell_0_X2_T.cell.CD4__LTB'</li><li>'single_cell_0_X2_T.cell.CD4__PTPRC'</li><li>'single_cell_0_X2_T.cell.CD4__SELL'</li><li>'single_cell_0_X3_NK__ANXA1'</li><li>'single_cell_0_X3_NK__ARF1'</li><li>'single_cell_0_X3_NK__CALM1'</li><li>'single_cell_0_X3_NK__CALR'</li><li>'single_cell_0_X3_NK__CCL4'</li><li>'single_cell_0_X3_NK__CCL5'</li><li>'single_cell_0_X3_NK__GSTP1'</li><li>'single_cell_0_X3_NK__GZMB'</li><li>'single_cell_0_X3_NK__HMGB1'</li><li>'single_cell_0_X3_NK__HMGB2'</li><li>'single_cell_0_X3_NK__HSP90B1'</li><li>'single_cell_0_X3_NK__PTPRC'</li><li>'single_cell_0_X3_NK__TGFB1'</li><li>'single_cell_0_X4_Monocytes...CD14__ADAM17'</li><li>'single_cell_0_X4_Monocytes...CD14__AIMP1'</li><li>'single_cell_0_X4_Monocytes...CD14__ALOX5AP'</li><li>'single_cell_0_X4_Monocytes...CD14__ANXA1'</li><li>'single_cell_0_X4_Monocytes...CD14__ARF1'</li><li>'single_cell_0_X4_Monocytes...CD14__CALM1'</li><li>'single_cell_0_X4_Monocytes...CD14__CALR'</li><li>'single_cell_0_X4_Monocytes...CD14__CCL5'</li><li>'single_cell_0_X4_Monocytes...CD14__COPA'</li><li>'single_cell_0_X4_Monocytes...CD14__CXCL2'</li><li>'single_cell_0_X4_Monocytes...CD14__GRN'</li><li>'single_cell_0_X4_Monocytes...CD14__GSTP1'</li><li>'single_cell_0_X4_Monocytes...CD14__HDGF'</li><li>'single_cell_0_X4_Monocytes...CD14__HMGB1'</li><li>'single_cell_0_X4_Monocytes...CD14__HMGB2'</li><li>'single_cell_0_X4_Monocytes...CD14__HSP90B1'</li><li>'single_cell_0_X4_Monocytes...CD14__ICAM1'</li><li>'single_cell_0_X4_Monocytes...CD14__ICAM3'</li><li>'single_cell_0_X4_Monocytes...CD14__IL1B'</li><li>'single_cell_0_X4_Monocytes...CD14__LGALS3'</li><li>'single_cell_0_X4_Monocytes...CD14__LRPAP1'</li><li>'single_cell_0_X4_Monocytes...CD14__LTB'</li><li>'single_cell_0_X4_Monocytes...CD14__MIF'</li><li>'single_cell_0_X4_Monocytes...CD14__PECAM1'</li><li>'single_cell_0_X4_Monocytes...CD14__PSAP'</li><li>'single_cell_0_X4_Monocytes...CD14__PSEN1'</li><li>'single_cell_0_X4_Monocytes...CD14__PTPRC'</li><li>'single_cell_0_X4_Monocytes...CD14__RTN4'</li><li>'single_cell_0_X4_Monocytes...CD14__SERPINA1'</li><li>'single_cell_0_X4_Monocytes...CD14__TGFB1'</li><li>'single_cell_0_X4_Monocytes...CD14__TIMP1'</li><li>'single_cell_0_X4_Monocytes...CD14__TNFSF10'</li><li>'single_cell_0_X4_Monocytes...CD14__TNFSF13B'</li><li>'single_cell_0_X5_T.cell.CD4__ANXA1'</li><li>'single_cell_0_X5_T.cell.CD4__CALM1'</li><li>'single_cell_0_X5_T.cell.CD4__CD2'</li><li>'single_cell_0_X5_T.cell.CD4__HMGB1'</li><li>'single_cell_0_X5_T.cell.CD4__HMGB2'</li><li>'single_cell_0_X5_T.cell.CD4__HSP90B1'</li><li>'single_cell_0_X5_T.cell.CD4__ICAM3'</li><li>'single_cell_0_X5_T.cell.CD4__LTB'</li><li>'single_cell_0_X5_T.cell.CD4__PTPRC'</li><li>'single_cell_0_X5_T.cell.CD4__SELL'</li><li>'single_cell_0_X6_Monocytes...CD14__ADAM17'</li><li>'single_cell_0_X6_Monocytes...CD14__ANXA1'</li><li>'single_cell_0_X6_Monocytes...CD14__ARF1'</li><li>'single_cell_0_X6_Monocytes...CD14__CALM1'</li><li>'single_cell_0_X6_Monocytes...CD14__CALR'</li><li>'single_cell_0_X6_Monocytes...CD14__CXCL2'</li><li>'single_cell_0_X6_Monocytes...CD14__GRN'</li><li>'single_cell_0_X6_Monocytes...CD14__GSTP1'</li><li>'single_cell_0_X6_Monocytes...CD14__HMGB1'</li><li>'single_cell_0_X6_Monocytes...CD14__HMGB2'</li><li>'single_cell_0_X6_Monocytes...CD14__HSP90B1'</li><li>'single_cell_0_X6_Monocytes...CD14__ICAM1'</li><li>'single_cell_0_X6_Monocytes...CD14__ICAM3'</li><li>'single_cell_0_X6_Monocytes...CD14__IL1B'</li><li>'single_cell_0_X6_Monocytes...CD14__LGALS3'</li><li>'single_cell_0_X6_Monocytes...CD14__LRPAP1'</li><li>'single_cell_0_X6_Monocytes...CD14__PECAM1'</li><li>'single_cell_0_X6_Monocytes...CD14__PSAP'</li><li>'single_cell_0_X6_Monocytes...CD14__PSEN1'</li><li>'single_cell_0_X6_Monocytes...CD14__PTPRC'</li><li>'single_cell_0_X6_Monocytes...CD14__RTN4'</li><li>'single_cell_0_X6_Monocytes...CD14__SERPINA1'</li><li>'single_cell_0_X6_Monocytes...CD14__TGFB1'</li><li>'single_cell_0_X6_Monocytes...CD14__THBS1'</li><li>'single_cell_0_X6_Monocytes...CD14__TIMP1'</li><li>'single_cell_0_X6_Monocytes...CD14__TNFSF10'</li><li>'single_cell_0_X6_Monocytes...CD14__TNFSF13B'</li><li>'single_cell_0_X7_Monocytes...CD14__ANXA1'</li><li>'single_cell_0_X7_Monocytes...CD14__ARF1'</li><li>'single_cell_0_X7_Monocytes...CD14__CALM1'</li><li>'single_cell_0_X7_Monocytes...CD14__CALR'</li><li>'single_cell_0_X7_Monocytes...CD14__GRN'</li><li>'single_cell_0_X7_Monocytes...CD14__GSTP1'</li><li>'single_cell_0_X7_Monocytes...CD14__HMGB1'</li><li>'single_cell_0_X7_Monocytes...CD14__HMGB2'</li><li>'single_cell_0_X7_Monocytes...CD14__HSP90B1'</li><li>'single_cell_0_X7_Monocytes...CD14__ICAM3'</li><li>'single_cell_0_X7_Monocytes...CD14__IL1B'</li><li>'single_cell_0_X7_Monocytes...CD14__LGALS3'</li><li>'single_cell_0_X7_Monocytes...CD14__LRPAP1'</li><li>'single_cell_0_X7_Monocytes...CD14__PECAM1'</li><li>'single_cell_0_X7_Monocytes...CD14__PSAP'</li><li>'single_cell_0_X7_Monocytes...CD14__PTPRC'</li><li>'single_cell_0_X7_Monocytes...CD14__RTN4'</li><li>'single_cell_0_X7_Monocytes...CD14__SERPINA1'</li><li>'single_cell_0_X7_Monocytes...CD14__TGFB1'</li><li>'single_cell_0_X7_Monocytes...CD14__TIMP1'</li><li>'single_cell_0_X7_Monocytes...CD14__TNFSF10'</li><li>'single_cell_0_X7_Monocytes...CD14__TNFSF13B'</li><li>'single_cell_0_X8_B.cell__CALM1'</li><li>'single_cell_0_X8_B.cell__CALR'</li><li>'single_cell_0_X8_B.cell__HMGB1'</li><li>'single_cell_0_X8_B.cell__HMGB2'</li><li>'single_cell_0_X8_B.cell__HSP90B1'</li><li>'single_cell_0_X8_B.cell__LTB'</li><li>'single_cell_0_X8_B.cell__PTPRC'</li><li>'single_cell_0_X8_B.cell__SELL'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__ANXA1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__ARF1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__CALM1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__CALR'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__CXCL16'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__GRN'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__GSTP1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__HMGB1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__HMGB2'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__HSP90B1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__ICAM3'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__IL1B'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__LGALS3'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__NAMPT'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__PECAM1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__PSAP'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__PTPRC'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__RTN4'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__SERPINA1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__TGFB1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__TIMP1'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFSF10'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFSF13B'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item cytokine\\_0\\_BCA1\\_\\_CXCL13\n",
       "\\item cytokine\\_0\\_CTACK\\_\\_CCL27\n",
       "\\item cytokine\\_0\\_EGF\\_\\_EGF\n",
       "\\item cytokine\\_0\\_ENA78\\_\\_CXCL5\n",
       "\\item cytokine\\_0\\_Eotaxin3\\_\\_CCL26\n",
       "\\item cytokine\\_0\\_FGF2\\_\\_FGF2\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'cytokine\\_0\\_BCA1\\_\\_CXCL13'\n",
       "\\item 'cytokine\\_0\\_CTACK\\_\\_CCL27'\n",
       "\\item 'cytokine\\_0\\_EGF\\_\\_EGF'\n",
       "\\item 'cytokine\\_0\\_ENA78\\_\\_CXCL5'\n",
       "\\item 'cytokine\\_0\\_Eotaxin3\\_\\_CCL26'\n",
       "\\item 'cytokine\\_0\\_FGF2\\_\\_FGF2'\n",
       "\\item 'cytokine\\_0\\_Fractalkine\\_\\_CX3CL1'\n",
       "\\item 'cytokine\\_0\\_GCSF\\_\\_CSF3'\n",
       "\\item 'cytokine\\_0\\_GMCSF\\_\\_CSF2'\n",
       "\\item 'cytokine\\_0\\_I309\\_\\_CCL1'\n",
       "\\item 'cytokine\\_0\\_IFNalpha2\\_\\_IFNA2'\n",
       "\\item 'cytokine\\_0\\_IFNgamma\\_\\_IFNG'\n",
       "\\item 'cytokine\\_0\\_IL10\\_\\_IL10'\n",
       "\\item 'cytokine\\_0\\_IL12p70\\_\\_IL12A'\n",
       "\\item 'cytokine\\_0\\_IL13\\_\\_IL13'\n",
       "\\item 'cytokine\\_0\\_IL15\\_\\_IL15'\n",
       "\\item 'cytokine\\_0\\_IL17A\\_\\_IL17A'\n",
       "\\item 'cytokine\\_0\\_IL17E\\_IL25\\_\\_IL25'\n",
       "\\item 'cytokine\\_0\\_IL17F\\_\\_IL17F'\n",
       "\\item 'cytokine\\_0\\_IL1alpha\\_\\_IL1A'\n",
       "\\item 'cytokine\\_0\\_IL1beta\\_\\_IL1B'\n",
       "\\item 'cytokine\\_0\\_IL1RA\\_\\_IL1RN'\n",
       "\\item 'cytokine\\_0\\_IL2\\_\\_IL2'\n",
       "\\item 'cytokine\\_0\\_IL22\\_\\_IL22'\n",
       "\\item 'cytokine\\_0\\_IL27\\_\\_IL27'\n",
       "\\item 'cytokine\\_0\\_IL5\\_\\_IL5'\n",
       "\\item 'cytokine\\_0\\_IL6\\_\\_IL6'\n",
       "\\item 'cytokine\\_0\\_IL7\\_\\_IL7'\n",
       "\\item 'cytokine\\_0\\_IL9\\_\\_IL9'\n",
       "\\item 'cytokine\\_0\\_IP10\\_\\_CXCL10'\n",
       "\\item 'cytokine\\_0\\_LIF\\_\\_LIF'\n",
       "\\item 'cytokine\\_0\\_MCP1\\_\\_CCL2'\n",
       "\\item 'cytokine\\_0\\_MCP2\\_\\_CCL8'\n",
       "\\item 'cytokine\\_0\\_MCP3\\_\\_CCL7'\n",
       "\\item 'cytokine\\_0\\_MCP4\\_\\_CCL13'\n",
       "\\item 'cytokine\\_0\\_MCSF\\_\\_CSF1'\n",
       "\\item 'cytokine\\_0\\_MDC\\_\\_CCL22'\n",
       "\\item 'cytokine\\_0\\_MIG\\_CXCL9\\_\\_CXCL9'\n",
       "\\item 'cytokine\\_0\\_MIP1alpha\\_\\_CCL3'\n",
       "\\item 'cytokine\\_0\\_MIP1beta\\_\\_CCL4'\n",
       "\\item 'cytokine\\_0\\_MIP1delta\\_\\_CCL15'\n",
       "\\item 'cytokine\\_0\\_RANTES\\_\\_CCL5'\n",
       "\\item 'cytokine\\_0\\_SCF\\_\\_KITLG'\n",
       "\\item 'cytokine\\_0\\_SDF1alpha\\_beta\\_\\_CXCL12'\n",
       "\\item 'cytokine\\_0\\_TARC\\_\\_CCL17'\n",
       "\\item 'cytokine\\_0\\_TNFalpha\\_\\_TNF'\n",
       "\\item 'cytokine\\_0\\_TNFbeta\\_\\_LTA'\n",
       "\\item 'cytokine\\_0\\_VEGFA\\_\\_VEGFA'\n",
       "\\item 'cytokine\\_0\\_X6CKine\\_\\_CCL21'\n",
       "\\item 'neutrophil\\_0\\_ADM'\n",
       "\\item 'neutrophil\\_0\\_FBRS'\n",
       "\\item 'neutrophil\\_0\\_GRN'\n",
       "\\item 'neutrophil\\_0\\_HMGB1'\n",
       "\\item 'neutrophil\\_0\\_HMGB2'\n",
       "\\item 'neutrophil\\_0\\_IL16'\n",
       "\\item 'neutrophil\\_0\\_MMP9'\n",
       "\\item 'neutrophil\\_0\\_PGLYRP1'\n",
       "\\item 'neutrophil\\_0\\_RTN4'\n",
       "\\item 'neutrophil\\_0\\_SECTM1'\n",
       "\\item 'neutrophil\\_0\\_TGFB1'\n",
       "\\item 'neutrophil\\_0\\_TIMP1'\n",
       "\\item 'neutrophil\\_0\\_TNFSF13B'\n",
       "\\item 'proteomics\\_0\\_A2M\\_P01023'\n",
       "\\item 'proteomics\\_0\\_ADIPOQ\\_Q15848'\n",
       "\\item 'proteomics\\_0\\_AGT\\_P01019'\n",
       "\\item 'proteomics\\_0\\_APOA1\\_P02647'\n",
       "\\item 'proteomics\\_0\\_APOB\\_P04114'\n",
       "\\item 'proteomics\\_0\\_APOE\\_P02649'\n",
       "\\item 'proteomics\\_0\\_C3\\_M0R1Q1'\n",
       "\\item 'proteomics\\_0\\_C3\\_P01024'\n",
       "\\item 'proteomics\\_0\\_C5\\_P01031'\n",
       "\\item 'proteomics\\_0\\_CAMP\\_J3KNB4'\n",
       "\\item 'proteomics\\_0\\_CDH5\\_P33151'\n",
       "\\item 'proteomics\\_0\\_CEACAM1\\_P13688'\n",
       "\\item 'proteomics\\_0\\_CFH\\_P08603'\n",
       "\\item 'proteomics\\_0\\_COL18A1\\_P39060'\n",
       "\\item 'proteomics\\_0\\_COMP\\_G3XAP6'\n",
       "\\item 'proteomics\\_0\\_DSG2\\_Q14126'\n",
       "\\item 'proteomics\\_0\\_F2\\_P00734'\n",
       "\\item 'proteomics\\_0\\_F9\\_P00740'\n",
       "\\item 'proteomics\\_0\\_FGA\\_P02671'\n",
       "\\item 'proteomics\\_0\\_FN1\\_H0Y4K8'\n",
       "\\item 'proteomics\\_0\\_FN1\\_P02751'\n",
       "\\item 'proteomics\\_0\\_GSTP1\\_P09211'\n",
       "\\item 'proteomics\\_0\\_HSP90B1\\_P14625'\n",
       "\\item 'proteomics\\_0\\_HSPG2\\_P98160'\n",
       "\\item 'proteomics\\_0\\_ICAM1\\_P05362'\n",
       "\\item 'proteomics\\_0\\_ICAM2\\_J3QQR8'\n",
       "\\item 'proteomics\\_0\\_IGF2\\_P01344'\n",
       "\\item 'proteomics\\_0\\_KNG1\\_P01042'\n",
       "\\item 'proteomics\\_0\\_KNG1\\_P01042.2'\n",
       "\\item 'proteomics\\_0\\_LTF\\_E7EQB2'\n",
       "\\item 'proteomics\\_0\\_MMP9\\_P14780'\n",
       "\\item 'proteomics\\_0\\_NID1\\_P14543'\n",
       "\\item 'proteomics\\_0\\_PF4\\_P02776'\n",
       "\\item 'proteomics\\_0\\_PLG\\_P00747'\n",
       "\\item 'proteomics\\_0\\_PPBP\\_P02775'\n",
       "\\item 'proteomics\\_0\\_SAA1\\_P0DJI8'\n",
       "\\item 'proteomics\\_0\\_SELL\\_P14151'\n",
       "\\item 'proteomics\\_0\\_SERPINA1\\_A0A024R6I7'\n",
       "\\item 'proteomics\\_0\\_SERPINA1\\_P01009'\n",
       "\\item 'proteomics\\_0\\_SERPINC1\\_P01008'\n",
       "\\item 'proteomics\\_0\\_SERPING1\\_P05155'\n",
       "\\item 'proteomics\\_0\\_TF\\_P02787'\n",
       "\\item 'proteomics\\_0\\_THBS1\\_P07996'\n",
       "\\item 'proteomics\\_0\\_TIMP1\\_P01033'\n",
       "\\item 'proteomics\\_0\\_TTR\\_P02766'\n",
       "\\item 'proteomics\\_0\\_VTN\\_P04004'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_CD2'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_MIF'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4\\_\\_SELL'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_CCL4'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_CCL5'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_CD2'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_GZMB'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_CD2'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_SELL'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_CD2'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_SELL'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_IL1B'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_LGALS3'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_NAMPT'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_PECAM1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_PSAP'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_RTN4'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_SERPINA1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A\\_\\_TIMP1'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_GRN'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_NAMPT'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_PSAP'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_RTN4'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic\\_\\_TIMP1'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_CD2'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4\\_\\_SELL'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_CCL4'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_CCL5'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_GZMB'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_ADAM17'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_AIMP1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_ALOX5AP'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_CCL5'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_COPA'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_CXCL2'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_GRN'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_HDGF'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_ICAM1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_IL1B'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_LGALS3'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_LRPAP1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_MIF'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_PECAM1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_PSAP'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_PSEN1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_RTN4'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_SERPINA1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_TIMP1'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_TNFSF10'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14\\_\\_TNFSF13B'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_CD2'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4\\_\\_SELL'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_ADAM17'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_CXCL2'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_GRN'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_ICAM1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_IL1B'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_LGALS3'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_LRPAP1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_PECAM1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_PSAP'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_PSEN1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_RTN4'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_SERPINA1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_THBS1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_TIMP1'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_TNFSF10'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14\\_\\_TNFSF13B'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_GRN'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_IL1B'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_LGALS3'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_LRPAP1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_PECAM1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_PSAP'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_RTN4'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_SERPINA1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_TIMP1'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_TNFSF10'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14\\_\\_TNFSF13B'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell\\_\\_LTB'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell\\_\\_SELL'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ANXA1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ARF1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_CALM1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_CALR'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_CXCL16'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_GRN'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_GSTP1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_HMGB1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_HMGB2'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_HSP90B1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_ICAM3'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_IL1B'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_LGALS3'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_NAMPT'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_PECAM1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_PSAP'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_PTPRC'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_RTN4'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_SERPINA1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_TGFB1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_TIMP1'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_TNFSF10'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A\\_\\_TNFSF13B'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. cytokine_0_BCA1__CXCL13\n",
       "2. cytokine_0_CTACK__CCL27\n",
       "3. cytokine_0_EGF__EGF\n",
       "4. cytokine_0_ENA78__CXCL5\n",
       "5. cytokine_0_Eotaxin3__CCL26\n",
       "6. cytokine_0_FGF2__FGF2\n",
       "\n",
       "\n",
       "\n",
       "**Levels**: 1. 'cytokine_0_BCA1__CXCL13'\n",
       "2. 'cytokine_0_CTACK__CCL27'\n",
       "3. 'cytokine_0_EGF__EGF'\n",
       "4. 'cytokine_0_ENA78__CXCL5'\n",
       "5. 'cytokine_0_Eotaxin3__CCL26'\n",
       "6. 'cytokine_0_FGF2__FGF2'\n",
       "7. 'cytokine_0_Fractalkine__CX3CL1'\n",
       "8. 'cytokine_0_GCSF__CSF3'\n",
       "9. 'cytokine_0_GMCSF__CSF2'\n",
       "10. 'cytokine_0_I309__CCL1'\n",
       "11. 'cytokine_0_IFNalpha2__IFNA2'\n",
       "12. 'cytokine_0_IFNgamma__IFNG'\n",
       "13. 'cytokine_0_IL10__IL10'\n",
       "14. 'cytokine_0_IL12p70__IL12A'\n",
       "15. 'cytokine_0_IL13__IL13'\n",
       "16. 'cytokine_0_IL15__IL15'\n",
       "17. 'cytokine_0_IL17A__IL17A'\n",
       "18. 'cytokine_0_IL17E_IL25__IL25'\n",
       "19. 'cytokine_0_IL17F__IL17F'\n",
       "20. 'cytokine_0_IL1alpha__IL1A'\n",
       "21. 'cytokine_0_IL1beta__IL1B'\n",
       "22. 'cytokine_0_IL1RA__IL1RN'\n",
       "23. 'cytokine_0_IL2__IL2'\n",
       "24. 'cytokine_0_IL22__IL22'\n",
       "25. 'cytokine_0_IL27__IL27'\n",
       "26. 'cytokine_0_IL5__IL5'\n",
       "27. 'cytokine_0_IL6__IL6'\n",
       "28. 'cytokine_0_IL7__IL7'\n",
       "29. 'cytokine_0_IL9__IL9'\n",
       "30. 'cytokine_0_IP10__CXCL10'\n",
       "31. 'cytokine_0_LIF__LIF'\n",
       "32. 'cytokine_0_MCP1__CCL2'\n",
       "33. 'cytokine_0_MCP2__CCL8'\n",
       "34. 'cytokine_0_MCP3__CCL7'\n",
       "35. 'cytokine_0_MCP4__CCL13'\n",
       "36. 'cytokine_0_MCSF__CSF1'\n",
       "37. 'cytokine_0_MDC__CCL22'\n",
       "38. 'cytokine_0_MIG_CXCL9__CXCL9'\n",
       "39. 'cytokine_0_MIP1alpha__CCL3'\n",
       "40. 'cytokine_0_MIP1beta__CCL4'\n",
       "41. 'cytokine_0_MIP1delta__CCL15'\n",
       "42. 'cytokine_0_RANTES__CCL5'\n",
       "43. 'cytokine_0_SCF__KITLG'\n",
       "44. 'cytokine_0_SDF1alpha_beta__CXCL12'\n",
       "45. 'cytokine_0_TARC__CCL17'\n",
       "46. 'cytokine_0_TNFalpha__TNF'\n",
       "47. 'cytokine_0_TNFbeta__LTA'\n",
       "48. 'cytokine_0_VEGFA__VEGFA'\n",
       "49. 'cytokine_0_X6CKine__CCL21'\n",
       "50. 'neutrophil_0_ADM'\n",
       "51. 'neutrophil_0_FBRS'\n",
       "52. 'neutrophil_0_GRN'\n",
       "53. 'neutrophil_0_HMGB1'\n",
       "54. 'neutrophil_0_HMGB2'\n",
       "55. 'neutrophil_0_IL16'\n",
       "56. 'neutrophil_0_MMP9'\n",
       "57. 'neutrophil_0_PGLYRP1'\n",
       "58. 'neutrophil_0_RTN4'\n",
       "59. 'neutrophil_0_SECTM1'\n",
       "60. 'neutrophil_0_TGFB1'\n",
       "61. 'neutrophil_0_TIMP1'\n",
       "62. 'neutrophil_0_TNFSF13B'\n",
       "63. 'proteomics_0_A2M_P01023'\n",
       "64. 'proteomics_0_ADIPOQ_Q15848'\n",
       "65. 'proteomics_0_AGT_P01019'\n",
       "66. 'proteomics_0_APOA1_P02647'\n",
       "67. 'proteomics_0_APOB_P04114'\n",
       "68. 'proteomics_0_APOE_P02649'\n",
       "69. 'proteomics_0_C3_M0R1Q1'\n",
       "70. 'proteomics_0_C3_P01024'\n",
       "71. 'proteomics_0_C5_P01031'\n",
       "72. 'proteomics_0_CAMP_J3KNB4'\n",
       "73. 'proteomics_0_CDH5_P33151'\n",
       "74. 'proteomics_0_CEACAM1_P13688'\n",
       "75. 'proteomics_0_CFH_P08603'\n",
       "76. 'proteomics_0_COL18A1_P39060'\n",
       "77. 'proteomics_0_COMP_G3XAP6'\n",
       "78. 'proteomics_0_DSG2_Q14126'\n",
       "79. 'proteomics_0_F2_P00734'\n",
       "80. 'proteomics_0_F9_P00740'\n",
       "81. 'proteomics_0_FGA_P02671'\n",
       "82. 'proteomics_0_FN1_H0Y4K8'\n",
       "83. 'proteomics_0_FN1_P02751'\n",
       "84. 'proteomics_0_GSTP1_P09211'\n",
       "85. 'proteomics_0_HSP90B1_P14625'\n",
       "86. 'proteomics_0_HSPG2_P98160'\n",
       "87. 'proteomics_0_ICAM1_P05362'\n",
       "88. 'proteomics_0_ICAM2_J3QQR8'\n",
       "89. 'proteomics_0_IGF2_P01344'\n",
       "90. 'proteomics_0_KNG1_P01042'\n",
       "91. 'proteomics_0_KNG1_P01042.2'\n",
       "92. 'proteomics_0_LTF_E7EQB2'\n",
       "93. 'proteomics_0_MMP9_P14780'\n",
       "94. 'proteomics_0_NID1_P14543'\n",
       "95. 'proteomics_0_PF4_P02776'\n",
       "96. 'proteomics_0_PLG_P00747'\n",
       "97. 'proteomics_0_PPBP_P02775'\n",
       "98. 'proteomics_0_SAA1_P0DJI8'\n",
       "99. 'proteomics_0_SELL_P14151'\n",
       "100. 'proteomics_0_SERPINA1_A0A024R6I7'\n",
       "101. 'proteomics_0_SERPINA1_P01009'\n",
       "102. 'proteomics_0_SERPINC1_P01008'\n",
       "103. 'proteomics_0_SERPING1_P05155'\n",
       "104. 'proteomics_0_TF_P02787'\n",
       "105. 'proteomics_0_THBS1_P07996'\n",
       "106. 'proteomics_0_TIMP1_P01033'\n",
       "107. 'proteomics_0_TTR_P02766'\n",
       "108. 'proteomics_0_VTN_P04004'\n",
       "109. 'single_cell_0_X0_T.cell.CD4__ANXA1'\n",
       "110. 'single_cell_0_X0_T.cell.CD4__ARF1'\n",
       "111. 'single_cell_0_X0_T.cell.CD4__CALM1'\n",
       "112. 'single_cell_0_X0_T.cell.CD4__CALR'\n",
       "113. 'single_cell_0_X0_T.cell.CD4__CD2'\n",
       "114. 'single_cell_0_X0_T.cell.CD4__HMGB1'\n",
       "115. 'single_cell_0_X0_T.cell.CD4__HMGB2'\n",
       "116. 'single_cell_0_X0_T.cell.CD4__HSP90B1'\n",
       "117. 'single_cell_0_X0_T.cell.CD4__ICAM3'\n",
       "118. 'single_cell_0_X0_T.cell.CD4__LTB'\n",
       "119. 'single_cell_0_X0_T.cell.CD4__MIF'\n",
       "120. 'single_cell_0_X0_T.cell.CD4__PTPRC'\n",
       "121. 'single_cell_0_X0_T.cell.CD4__SELL'\n",
       "122. 'single_cell_0_X1_T.cell.CD8__ANXA1'\n",
       "123. 'single_cell_0_X1_T.cell.CD8__ARF1'\n",
       "124. 'single_cell_0_X1_T.cell.CD8__CALM1'\n",
       "125. 'single_cell_0_X1_T.cell.CD8__CALR'\n",
       "126. 'single_cell_0_X1_T.cell.CD8__CCL4'\n",
       "127. 'single_cell_0_X1_T.cell.CD8__CCL5'\n",
       "128. 'single_cell_0_X1_T.cell.CD8__CD2'\n",
       "129. 'single_cell_0_X1_T.cell.CD8__GSTP1'\n",
       "130. 'single_cell_0_X1_T.cell.CD8__GZMB'\n",
       "131. 'single_cell_0_X1_T.cell.CD8__HMGB1'\n",
       "132. 'single_cell_0_X1_T.cell.CD8__HMGB2'\n",
       "133. 'single_cell_0_X1_T.cell.CD8__HSP90B1'\n",
       "134. 'single_cell_0_X1_T.cell.CD8__ICAM3'\n",
       "135. 'single_cell_0_X1_T.cell.CD8__LTB'\n",
       "136. 'single_cell_0_X1_T.cell.CD8__PTPRC'\n",
       "137. 'single_cell_0_X1_T.cell.CD8__TGFB1'\n",
       "138. 'single_cell_0_X10_B.cell__ANXA1'\n",
       "139. 'single_cell_0_X10_B.cell__ARF1'\n",
       "140. 'single_cell_0_X10_B.cell__CALM1'\n",
       "141. 'single_cell_0_X10_B.cell__CALR'\n",
       "142. 'single_cell_0_X10_B.cell__CD2'\n",
       "143. 'single_cell_0_X10_B.cell__GSTP1'\n",
       "144. 'single_cell_0_X10_B.cell__HMGB1'\n",
       "145. 'single_cell_0_X10_B.cell__HMGB2'\n",
       "146. 'single_cell_0_X10_B.cell__HSP90B1'\n",
       "147. 'single_cell_0_X10_B.cell__ICAM3'\n",
       "148. 'single_cell_0_X10_B.cell__LTB'\n",
       "149. 'single_cell_0_X10_B.cell__PTPRC'\n",
       "150. 'single_cell_0_X10_B.cell__SELL'\n",
       "151. 'single_cell_0_X10_B.cell__TGFB1'\n",
       "152. 'single_cell_0_X11_T.cell.CD4__ANXA1'\n",
       "153. 'single_cell_0_X11_T.cell.CD4__ARF1'\n",
       "154. 'single_cell_0_X11_T.cell.CD4__CALM1'\n",
       "155. 'single_cell_0_X11_T.cell.CD4__CALR'\n",
       "156. 'single_cell_0_X11_T.cell.CD4__CD2'\n",
       "157. 'single_cell_0_X11_T.cell.CD4__HMGB1'\n",
       "158. 'single_cell_0_X11_T.cell.CD4__HMGB2'\n",
       "159. 'single_cell_0_X11_T.cell.CD4__HSP90B1'\n",
       "160. 'single_cell_0_X11_T.cell.CD4__ICAM3'\n",
       "161. 'single_cell_0_X11_T.cell.CD4__LTB'\n",
       "162. 'single_cell_0_X11_T.cell.CD4__PTPRC'\n",
       "163. 'single_cell_0_X11_T.cell.CD4__SELL'\n",
       "164. 'single_cell_0_X11_T.cell.CD4__TGFB1'\n",
       "165. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__ANXA1'\n",
       "166. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__ARF1'\n",
       "167. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__CALM1'\n",
       "168. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__CALR'\n",
       "169. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__GSTP1'\n",
       "170. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__HMGB1'\n",
       "171. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__HMGB2'\n",
       "172. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__HSP90B1'\n",
       "173. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__ICAM3'\n",
       "174. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__IL1B'\n",
       "175. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__LGALS3'\n",
       "176. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__LTB'\n",
       "177. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__NAMPT'\n",
       "178. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__PECAM1'\n",
       "179. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__PSAP'\n",
       "180. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__PTPRC'\n",
       "181. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__RTN4'\n",
       "182. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__SERPINA1'\n",
       "183. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__TGFB1'\n",
       "184. 'single_cell_0_X12_Monocytes...CD16_FCGR3A__TIMP1'\n",
       "185. 'single_cell_0_X13_Dendritic__ANXA1'\n",
       "186. 'single_cell_0_X13_Dendritic__CALM1'\n",
       "187. 'single_cell_0_X13_Dendritic__CALR'\n",
       "188. 'single_cell_0_X13_Dendritic__GRN'\n",
       "189. 'single_cell_0_X13_Dendritic__GSTP1'\n",
       "190. 'single_cell_0_X13_Dendritic__HMGB1'\n",
       "191. 'single_cell_0_X13_Dendritic__HSP90B1'\n",
       "192. 'single_cell_0_X13_Dendritic__NAMPT'\n",
       "193. 'single_cell_0_X13_Dendritic__PSAP'\n",
       "194. 'single_cell_0_X13_Dendritic__PTPRC'\n",
       "195. 'single_cell_0_X13_Dendritic__RTN4'\n",
       "196. 'single_cell_0_X13_Dendritic__TIMP1'\n",
       "197. 'single_cell_0_X2_T.cell.CD4__ANXA1'\n",
       "198. 'single_cell_0_X2_T.cell.CD4__ARF1'\n",
       "199. 'single_cell_0_X2_T.cell.CD4__CALM1'\n",
       "200. 'single_cell_0_X2_T.cell.CD4__CALR'\n",
       "201. 'single_cell_0_X2_T.cell.CD4__CD2'\n",
       "202. 'single_cell_0_X2_T.cell.CD4__HMGB1'\n",
       "203. 'single_cell_0_X2_T.cell.CD4__HMGB2'\n",
       "204. 'single_cell_0_X2_T.cell.CD4__HSP90B1'\n",
       "205. 'single_cell_0_X2_T.cell.CD4__LTB'\n",
       "206. 'single_cell_0_X2_T.cell.CD4__PTPRC'\n",
       "207. 'single_cell_0_X2_T.cell.CD4__SELL'\n",
       "208. 'single_cell_0_X3_NK__ANXA1'\n",
       "209. 'single_cell_0_X3_NK__ARF1'\n",
       "210. 'single_cell_0_X3_NK__CALM1'\n",
       "211. 'single_cell_0_X3_NK__CALR'\n",
       "212. 'single_cell_0_X3_NK__CCL4'\n",
       "213. 'single_cell_0_X3_NK__CCL5'\n",
       "214. 'single_cell_0_X3_NK__GSTP1'\n",
       "215. 'single_cell_0_X3_NK__GZMB'\n",
       "216. 'single_cell_0_X3_NK__HMGB1'\n",
       "217. 'single_cell_0_X3_NK__HMGB2'\n",
       "218. 'single_cell_0_X3_NK__HSP90B1'\n",
       "219. 'single_cell_0_X3_NK__PTPRC'\n",
       "220. 'single_cell_0_X3_NK__TGFB1'\n",
       "221. 'single_cell_0_X4_Monocytes...CD14__ADAM17'\n",
       "222. 'single_cell_0_X4_Monocytes...CD14__AIMP1'\n",
       "223. 'single_cell_0_X4_Monocytes...CD14__ALOX5AP'\n",
       "224. 'single_cell_0_X4_Monocytes...CD14__ANXA1'\n",
       "225. 'single_cell_0_X4_Monocytes...CD14__ARF1'\n",
       "226. 'single_cell_0_X4_Monocytes...CD14__CALM1'\n",
       "227. 'single_cell_0_X4_Monocytes...CD14__CALR'\n",
       "228. 'single_cell_0_X4_Monocytes...CD14__CCL5'\n",
       "229. 'single_cell_0_X4_Monocytes...CD14__COPA'\n",
       "230. 'single_cell_0_X4_Monocytes...CD14__CXCL2'\n",
       "231. 'single_cell_0_X4_Monocytes...CD14__GRN'\n",
       "232. 'single_cell_0_X4_Monocytes...CD14__GSTP1'\n",
       "233. 'single_cell_0_X4_Monocytes...CD14__HDGF'\n",
       "234. 'single_cell_0_X4_Monocytes...CD14__HMGB1'\n",
       "235. 'single_cell_0_X4_Monocytes...CD14__HMGB2'\n",
       "236. 'single_cell_0_X4_Monocytes...CD14__HSP90B1'\n",
       "237. 'single_cell_0_X4_Monocytes...CD14__ICAM1'\n",
       "238. 'single_cell_0_X4_Monocytes...CD14__ICAM3'\n",
       "239. 'single_cell_0_X4_Monocytes...CD14__IL1B'\n",
       "240. 'single_cell_0_X4_Monocytes...CD14__LGALS3'\n",
       "241. 'single_cell_0_X4_Monocytes...CD14__LRPAP1'\n",
       "242. 'single_cell_0_X4_Monocytes...CD14__LTB'\n",
       "243. 'single_cell_0_X4_Monocytes...CD14__MIF'\n",
       "244. 'single_cell_0_X4_Monocytes...CD14__PECAM1'\n",
       "245. 'single_cell_0_X4_Monocytes...CD14__PSAP'\n",
       "246. 'single_cell_0_X4_Monocytes...CD14__PSEN1'\n",
       "247. 'single_cell_0_X4_Monocytes...CD14__PTPRC'\n",
       "248. 'single_cell_0_X4_Monocytes...CD14__RTN4'\n",
       "249. 'single_cell_0_X4_Monocytes...CD14__SERPINA1'\n",
       "250. 'single_cell_0_X4_Monocytes...CD14__TGFB1'\n",
       "251. 'single_cell_0_X4_Monocytes...CD14__TIMP1'\n",
       "252. 'single_cell_0_X4_Monocytes...CD14__TNFSF10'\n",
       "253. 'single_cell_0_X4_Monocytes...CD14__TNFSF13B'\n",
       "254. 'single_cell_0_X5_T.cell.CD4__ANXA1'\n",
       "255. 'single_cell_0_X5_T.cell.CD4__CALM1'\n",
       "256. 'single_cell_0_X5_T.cell.CD4__CD2'\n",
       "257. 'single_cell_0_X5_T.cell.CD4__HMGB1'\n",
       "258. 'single_cell_0_X5_T.cell.CD4__HMGB2'\n",
       "259. 'single_cell_0_X5_T.cell.CD4__HSP90B1'\n",
       "260. 'single_cell_0_X5_T.cell.CD4__ICAM3'\n",
       "261. 'single_cell_0_X5_T.cell.CD4__LTB'\n",
       "262. 'single_cell_0_X5_T.cell.CD4__PTPRC'\n",
       "263. 'single_cell_0_X5_T.cell.CD4__SELL'\n",
       "264. 'single_cell_0_X6_Monocytes...CD14__ADAM17'\n",
       "265. 'single_cell_0_X6_Monocytes...CD14__ANXA1'\n",
       "266. 'single_cell_0_X6_Monocytes...CD14__ARF1'\n",
       "267. 'single_cell_0_X6_Monocytes...CD14__CALM1'\n",
       "268. 'single_cell_0_X6_Monocytes...CD14__CALR'\n",
       "269. 'single_cell_0_X6_Monocytes...CD14__CXCL2'\n",
       "270. 'single_cell_0_X6_Monocytes...CD14__GRN'\n",
       "271. 'single_cell_0_X6_Monocytes...CD14__GSTP1'\n",
       "272. 'single_cell_0_X6_Monocytes...CD14__HMGB1'\n",
       "273. 'single_cell_0_X6_Monocytes...CD14__HMGB2'\n",
       "274. 'single_cell_0_X6_Monocytes...CD14__HSP90B1'\n",
       "275. 'single_cell_0_X6_Monocytes...CD14__ICAM1'\n",
       "276. 'single_cell_0_X6_Monocytes...CD14__ICAM3'\n",
       "277. 'single_cell_0_X6_Monocytes...CD14__IL1B'\n",
       "278. 'single_cell_0_X6_Monocytes...CD14__LGALS3'\n",
       "279. 'single_cell_0_X6_Monocytes...CD14__LRPAP1'\n",
       "280. 'single_cell_0_X6_Monocytes...CD14__PECAM1'\n",
       "281. 'single_cell_0_X6_Monocytes...CD14__PSAP'\n",
       "282. 'single_cell_0_X6_Monocytes...CD14__PSEN1'\n",
       "283. 'single_cell_0_X6_Monocytes...CD14__PTPRC'\n",
       "284. 'single_cell_0_X6_Monocytes...CD14__RTN4'\n",
       "285. 'single_cell_0_X6_Monocytes...CD14__SERPINA1'\n",
       "286. 'single_cell_0_X6_Monocytes...CD14__TGFB1'\n",
       "287. 'single_cell_0_X6_Monocytes...CD14__THBS1'\n",
       "288. 'single_cell_0_X6_Monocytes...CD14__TIMP1'\n",
       "289. 'single_cell_0_X6_Monocytes...CD14__TNFSF10'\n",
       "290. 'single_cell_0_X6_Monocytes...CD14__TNFSF13B'\n",
       "291. 'single_cell_0_X7_Monocytes...CD14__ANXA1'\n",
       "292. 'single_cell_0_X7_Monocytes...CD14__ARF1'\n",
       "293. 'single_cell_0_X7_Monocytes...CD14__CALM1'\n",
       "294. 'single_cell_0_X7_Monocytes...CD14__CALR'\n",
       "295. 'single_cell_0_X7_Monocytes...CD14__GRN'\n",
       "296. 'single_cell_0_X7_Monocytes...CD14__GSTP1'\n",
       "297. 'single_cell_0_X7_Monocytes...CD14__HMGB1'\n",
       "298. 'single_cell_0_X7_Monocytes...CD14__HMGB2'\n",
       "299. 'single_cell_0_X7_Monocytes...CD14__HSP90B1'\n",
       "300. 'single_cell_0_X7_Monocytes...CD14__ICAM3'\n",
       "301. 'single_cell_0_X7_Monocytes...CD14__IL1B'\n",
       "302. 'single_cell_0_X7_Monocytes...CD14__LGALS3'\n",
       "303. 'single_cell_0_X7_Monocytes...CD14__LRPAP1'\n",
       "304. 'single_cell_0_X7_Monocytes...CD14__PECAM1'\n",
       "305. 'single_cell_0_X7_Monocytes...CD14__PSAP'\n",
       "306. 'single_cell_0_X7_Monocytes...CD14__PTPRC'\n",
       "307. 'single_cell_0_X7_Monocytes...CD14__RTN4'\n",
       "308. 'single_cell_0_X7_Monocytes...CD14__SERPINA1'\n",
       "309. 'single_cell_0_X7_Monocytes...CD14__TGFB1'\n",
       "310. 'single_cell_0_X7_Monocytes...CD14__TIMP1'\n",
       "311. 'single_cell_0_X7_Monocytes...CD14__TNFSF10'\n",
       "312. 'single_cell_0_X7_Monocytes...CD14__TNFSF13B'\n",
       "313. 'single_cell_0_X8_B.cell__CALM1'\n",
       "314. 'single_cell_0_X8_B.cell__CALR'\n",
       "315. 'single_cell_0_X8_B.cell__HMGB1'\n",
       "316. 'single_cell_0_X8_B.cell__HMGB2'\n",
       "317. 'single_cell_0_X8_B.cell__HSP90B1'\n",
       "318. 'single_cell_0_X8_B.cell__LTB'\n",
       "319. 'single_cell_0_X8_B.cell__PTPRC'\n",
       "320. 'single_cell_0_X8_B.cell__SELL'\n",
       "321. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__ANXA1'\n",
       "322. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__ARF1'\n",
       "323. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__CALM1'\n",
       "324. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__CALR'\n",
       "325. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__CXCL16'\n",
       "326. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__GRN'\n",
       "327. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__GSTP1'\n",
       "328. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__HMGB1'\n",
       "329. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__HMGB2'\n",
       "330. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__HSP90B1'\n",
       "331. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__ICAM3'\n",
       "332. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__IL1B'\n",
       "333. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__LGALS3'\n",
       "334. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__NAMPT'\n",
       "335. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__PECAM1'\n",
       "336. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__PSAP'\n",
       "337. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__PTPRC'\n",
       "338. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__RTN4'\n",
       "339. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__SERPINA1'\n",
       "340. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__TGFB1'\n",
       "341. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__TIMP1'\n",
       "342. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFSF10'\n",
       "343. 'single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFSF13B'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] cytokine_0_BCA1__CXCL13    cytokine_0_CTACK__CCL27   \n",
       "[3] cytokine_0_EGF__EGF        cytokine_0_ENA78__CXCL5   \n",
       "[5] cytokine_0_Eotaxin3__CCL26 cytokine_0_FGF2__FGF2     \n",
       "343 Levels: cytokine_0_BCA1__CXCL13 ... single_cell_0_X9_Monocytes...CD16_FCGR3A__TNFSF13B"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(sort(unique(ligand_target_correlations$Ligand)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "id": "6782e731-c592-4e50-874a-25682a0037c1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'clinical_data'</li><li>'cytokine'</li><li>'neutrophil'</li><li>'proteomics'</li><li>'single_cell_0_X0_T.cell.CD4'</li><li>'single_cell_0_X1_T.cell.CD8'</li><li>'single_cell_0_X10_B.cell'</li><li>'single_cell_0_X11_T.cell.CD4'</li><li>'single_cell_0_X12_Monocytes...CD16_FCGR3A'</li><li>'single_cell_0_X13_Dendritic'</li><li>'single_cell_0_X2_T.cell.CD4'</li><li>'single_cell_0_X3_NK'</li><li>'single_cell_0_X4_Monocytes...CD14'</li><li>'single_cell_0_X5_T.cell.CD4'</li><li>'single_cell_0_X6_Monocytes...CD14'</li><li>'single_cell_0_X7_Monocytes...CD14'</li><li>'single_cell_0_X8_B.cell'</li><li>'single_cell_0_X9_Monocytes...CD16_FCGR3A'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'clinical\\_data'\n",
       "\\item 'cytokine'\n",
       "\\item 'neutrophil'\n",
       "\\item 'proteomics'\n",
       "\\item 'single\\_cell\\_0\\_X0\\_T.cell.CD4'\n",
       "\\item 'single\\_cell\\_0\\_X1\\_T.cell.CD8'\n",
       "\\item 'single\\_cell\\_0\\_X10\\_B.cell'\n",
       "\\item 'single\\_cell\\_0\\_X11\\_T.cell.CD4'\n",
       "\\item 'single\\_cell\\_0\\_X12\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'single\\_cell\\_0\\_X13\\_Dendritic'\n",
       "\\item 'single\\_cell\\_0\\_X2\\_T.cell.CD4'\n",
       "\\item 'single\\_cell\\_0\\_X3\\_NK'\n",
       "\\item 'single\\_cell\\_0\\_X4\\_Monocytes...CD14'\n",
       "\\item 'single\\_cell\\_0\\_X5\\_T.cell.CD4'\n",
       "\\item 'single\\_cell\\_0\\_X6\\_Monocytes...CD14'\n",
       "\\item 'single\\_cell\\_0\\_X7\\_Monocytes...CD14'\n",
       "\\item 'single\\_cell\\_0\\_X8\\_B.cell'\n",
       "\\item 'single\\_cell\\_0\\_X9\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'clinical_data'\n",
       "2. 'cytokine'\n",
       "3. 'neutrophil'\n",
       "4. 'proteomics'\n",
       "5. 'single_cell_0_X0_T.cell.CD4'\n",
       "6. 'single_cell_0_X1_T.cell.CD8'\n",
       "7. 'single_cell_0_X10_B.cell'\n",
       "8. 'single_cell_0_X11_T.cell.CD4'\n",
       "9. 'single_cell_0_X12_Monocytes...CD16_FCGR3A'\n",
       "10. 'single_cell_0_X13_Dendritic'\n",
       "11. 'single_cell_0_X2_T.cell.CD4'\n",
       "12. 'single_cell_0_X3_NK'\n",
       "13. 'single_cell_0_X4_Monocytes...CD14'\n",
       "14. 'single_cell_0_X5_T.cell.CD4'\n",
       "15. 'single_cell_0_X6_Monocytes...CD14'\n",
       "16. 'single_cell_0_X7_Monocytes...CD14'\n",
       "17. 'single_cell_0_X8_B.cell'\n",
       "18. 'single_cell_0_X9_Monocytes...CD16_FCGR3A'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"clinical_data\"                            \n",
       " [2] \"cytokine\"                                 \n",
       " [3] \"neutrophil\"                               \n",
       " [4] \"proteomics\"                               \n",
       " [5] \"single_cell_0_X0_T.cell.CD4\"              \n",
       " [6] \"single_cell_0_X1_T.cell.CD8\"              \n",
       " [7] \"single_cell_0_X10_B.cell\"                 \n",
       " [8] \"single_cell_0_X11_T.cell.CD4\"             \n",
       " [9] \"single_cell_0_X12_Monocytes...CD16_FCGR3A\"\n",
       "[10] \"single_cell_0_X13_Dendritic\"              \n",
       "[11] \"single_cell_0_X2_T.cell.CD4\"              \n",
       "[12] \"single_cell_0_X3_NK\"                      \n",
       "[13] \"single_cell_0_X4_Monocytes...CD14\"        \n",
       "[14] \"single_cell_0_X5_T.cell.CD4\"              \n",
       "[15] \"single_cell_0_X6_Monocytes...CD14\"        \n",
       "[16] \"single_cell_0_X7_Monocytes...CD14\"        \n",
       "[17] \"single_cell_0_X8_B.cell\"                  \n",
       "[18] \"single_cell_0_X9_Monocytes...CD16_FCGR3A\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(ligand_target_correlations$Target_Type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "fcd0583b-7e85-4d78-ab3c-fce11cf7ba5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$cluster = str_replace(ligand_target_correlations$Target_Type, 'single_cell_0_', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "id": "3cf25a82-61f2-4a07-8d3c-a547fbd16226",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'clinical_data'</li><li>'cytokine'</li><li>'neutrophil'</li><li>'proteomics'</li><li>'X0_T.cell.CD4'</li><li>'X1_T.cell.CD8'</li><li>'X10_B.cell'</li><li>'X11_T.cell.CD4'</li><li>'X12_Monocytes...CD16_FCGR3A'</li><li>'X13_Dendritic'</li><li>'X2_T.cell.CD4'</li><li>'X3_NK'</li><li>'X4_Monocytes...CD14'</li><li>'X5_T.cell.CD4'</li><li>'X6_Monocytes...CD14'</li><li>'X7_Monocytes...CD14'</li><li>'X8_B.cell'</li><li>'X9_Monocytes...CD16_FCGR3A'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'clinical\\_data'\n",
       "\\item 'cytokine'\n",
       "\\item 'neutrophil'\n",
       "\\item 'proteomics'\n",
       "\\item 'X0\\_T.cell.CD4'\n",
       "\\item 'X1\\_T.cell.CD8'\n",
       "\\item 'X10\\_B.cell'\n",
       "\\item 'X11\\_T.cell.CD4'\n",
       "\\item 'X12\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\item 'X13\\_Dendritic'\n",
       "\\item 'X2\\_T.cell.CD4'\n",
       "\\item 'X3\\_NK'\n",
       "\\item 'X4\\_Monocytes...CD14'\n",
       "\\item 'X5\\_T.cell.CD4'\n",
       "\\item 'X6\\_Monocytes...CD14'\n",
       "\\item 'X7\\_Monocytes...CD14'\n",
       "\\item 'X8\\_B.cell'\n",
       "\\item 'X9\\_Monocytes...CD16\\_FCGR3A'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'clinical_data'\n",
       "2. 'cytokine'\n",
       "3. 'neutrophil'\n",
       "4. 'proteomics'\n",
       "5. 'X0_T.cell.CD4'\n",
       "6. 'X1_T.cell.CD8'\n",
       "7. 'X10_B.cell'\n",
       "8. 'X11_T.cell.CD4'\n",
       "9. 'X12_Monocytes...CD16_FCGR3A'\n",
       "10. 'X13_Dendritic'\n",
       "11. 'X2_T.cell.CD4'\n",
       "12. 'X3_NK'\n",
       "13. 'X4_Monocytes...CD14'\n",
       "14. 'X5_T.cell.CD4'\n",
       "15. 'X6_Monocytes...CD14'\n",
       "16. 'X7_Monocytes...CD14'\n",
       "17. 'X8_B.cell'\n",
       "18. 'X9_Monocytes...CD16_FCGR3A'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"clinical_data\"               \"cytokine\"                   \n",
       " [3] \"neutrophil\"                  \"proteomics\"                 \n",
       " [5] \"X0_T.cell.CD4\"               \"X1_T.cell.CD8\"              \n",
       " [7] \"X10_B.cell\"                  \"X11_T.cell.CD4\"             \n",
       " [9] \"X12_Monocytes...CD16_FCGR3A\" \"X13_Dendritic\"              \n",
       "[11] \"X2_T.cell.CD4\"               \"X3_NK\"                      \n",
       "[13] \"X4_Monocytes...CD14\"         \"X5_T.cell.CD4\"              \n",
       "[15] \"X6_Monocytes...CD14\"         \"X7_Monocytes...CD14\"        \n",
       "[17] \"X8_B.cell\"                   \"X9_Monocytes...CD16_FCGR3A\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(ligand_target_correlations$cluster)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "id": "4e5c2d62-0ac1-4b8c-b360-dc03df6aea14",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Ligand</th><th scope=col>Target</th><th scope=col>correlation</th><th scope=col>Target_Gene</th><th scope=col>Ligand_Gene</th><th scope=col>Target_Type</th><th scope=col>Ligand_Type</th><th scope=col>cluster</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>cytokine_0_BCA1__CXCL13</td><td>clinical_data_0_CK</td><td> 0.2267960</td><td>CK</td><td>CXCL13</td><td>clinical_data</td><td>cytokine</td><td>clinical_data</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>cytokine_0_CTACK__CCL27</td><td>clinical_data_0_CK</td><td>-0.1321796</td><td>CK</td><td>CCL27 </td><td>clinical_data</td><td>cytokine</td><td>clinical_data</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Ligand & Target & correlation & Target\\_Gene & Ligand\\_Gene & Target\\_Type & Ligand\\_Type & cluster\\\\\n",
       "  & <fct> & <fct> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & cytokine\\_0\\_BCA1\\_\\_CXCL13 & clinical\\_data\\_0\\_CK &  0.2267960 & CK & CXCL13 & clinical\\_data & cytokine & clinical\\_data\\\\\n",
       "\t2 & cytokine\\_0\\_CTACK\\_\\_CCL27 & clinical\\_data\\_0\\_CK & -0.1321796 & CK & CCL27  & clinical\\_data & cytokine & clinical\\_data\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 8\n",
       "\n",
       "| <!--/--> | Ligand &lt;fct&gt; | Target &lt;fct&gt; | correlation &lt;dbl&gt; | Target_Gene &lt;chr&gt; | Ligand_Gene &lt;chr&gt; | Target_Type &lt;chr&gt; | Ligand_Type &lt;chr&gt; | cluster &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | cytokine_0_BCA1__CXCL13 | clinical_data_0_CK |  0.2267960 | CK | CXCL13 | clinical_data | cytokine | clinical_data |\n",
       "| 2 | cytokine_0_CTACK__CCL27 | clinical_data_0_CK | -0.1321796 | CK | CCL27  | clinical_data | cytokine | clinical_data |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand                  Target             correlation Target_Gene\n",
       "1 cytokine_0_BCA1__CXCL13 clinical_data_0_CK  0.2267960  CK         \n",
       "2 cytokine_0_CTACK__CCL27 clinical_data_0_CK -0.1321796  CK         \n",
       "  Ligand_Gene Target_Type   Ligand_Type cluster      \n",
       "1 CXCL13      clinical_data cytokine    clinical_data\n",
       "2 CCL27       clinical_data cytokine    clinical_data"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_target_correlations,2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "34fd691d-2cd4-40ec-94ae-f69f4ac37db7",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Filter on ligands with matching receptor in target"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "id": "df075c80-9948-425e-8729-52d9441124dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Ligand</th><th scope=col>cluster_R</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>APOA1</td><td>X4_Monocytes...CD14</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>APOA1</td><td>X6_Monocytes...CD14</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Ligand & cluster\\_R\\\\\n",
       "  & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & APOA1 & X4\\_Monocytes...CD14\\\\\n",
       "\t2 & APOA1 & X6\\_Monocytes...CD14\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 2\n",
       "\n",
       "| <!--/--> | Ligand &lt;chr&gt; | cluster_R &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| 1 | APOA1 | X4_Monocytes...CD14 |\n",
       "| 2 | APOA1 | X6_Monocytes...CD14 |\n",
       "\n"
      ],
      "text/plain": [
       "  Ligand cluster_R          \n",
       "1 APOA1  X4_Monocytes...CD14\n",
       "2 APOA1  X6_Monocytes...CD14"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(potential_ligand_target_types_comb,2) # dataset with available receptor for ligands in target types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "id": "7a76f257-44b3-48cf-b136-139563f302be",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "4438077"
      ],
      "text/latex": [
       "4438077"
      ],
      "text/markdown": [
       "4438077"
      ],
      "text/plain": [
       "[1] 4438077"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_target_correlations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "id": "9ad3e57a-784f-4e49-baea-3c9c058ea6e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "filter_matched_receptor = TRUE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "id": "d457c105-e8cd-4f72-b452-0c01a13febb4",
   "metadata": {},
   "outputs": [],
   "source": [
    "if(filter_matched_receptor == TRUE){\n",
    "    ### seperately add clinical, proteomics as target (as here no filtering based on expression is applied)\n",
    "    no_sc = ligand_target_correlations[ligand_target_correlations$Target_Type %in%  c('clinical_data', 'proteomics', 'neutrophil'),]\n",
    "    ### filter the ligands\n",
    "    ligand_target_correlations = merge(ligand_target_correlations, potential_ligand_target_types_comb[,c('Ligand', 'cluster_R')], by.x = c('Ligand_Gene', 'cluster'), by.y = c('Ligand', 'cluster_R'))\n",
    "    ### add other omics data\n",
    "    ligand_target_correlations = rbind(ligand_target_correlations, no_sc)\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "6d24c6e9-3487-441c-b8c3-ccd5b55ccba8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "3510462"
      ],
      "text/latex": [
       "3510462"
      ],
      "text/markdown": [
       "3510462"
      ],
      "text/plain": [
       "[1] 3510462"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_target_correlations)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c3d3f22-e7e4-4719-bb86-f67014b2c53c",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Add regulatory potential"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "2f838c30-dc0d-43cc-b492-5d41252b7738",
   "metadata": {},
   "outputs": [],
   "source": [
    "## add ligand target regulatory potential score from nichenet to the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "58e7451a-647d-47f7-aa51-a234494a6cdb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>target</th><th scope=col>ligand</th><th scope=col>regulatory_potential</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>A1BG    </td><td>CXCL1</td><td>0.0003534343</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>A1BG-AS1</td><td>CXCL1</td><td>0.0001650894</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & target & ligand & regulatory\\_potential\\\\\n",
       "  & <fct> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & A1BG     & CXCL1 & 0.0003534343\\\\\n",
       "\t2 & A1BG-AS1 & CXCL1 & 0.0001650894\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 3\n",
       "\n",
       "| <!--/--> | target &lt;fct&gt; | ligand &lt;fct&gt; | regulatory_potential &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | A1BG     | CXCL1 | 0.0003534343 |\n",
       "| 2 | A1BG-AS1 | CXCL1 | 0.0001650894 |\n",
       "\n"
      ],
      "text/plain": [
       "  target   ligand regulatory_potential\n",
       "1 A1BG     CXCL1  0.0003534343        \n",
       "2 A1BG-AS1 CXCL1  0.0001650894        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ligand_target_matrix_long,2)# regulatory potential scores from nichenet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "id": "319efddf-541e-4d57-9a96-163e018913f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Adjust gene names for clinical data\n",
    "ligand_target_correlations$Target_Gene[ligand_target_correlations$Target == 'clinical_data_0_CK'] = 'CKM'\n",
    "ligand_target_correlations$Target_Gene[ligand_target_correlations$Target == 'clinical_data_0_CK_MB'] = 'CKB'\n",
    "ligand_target_correlations$Target_Gene[ligand_target_correlations$Target == 'clinical_data_0_Troponin'] = 'TNNT2'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "7d5ebdbd-7480-4f48-9bac-f6a343d27d47",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Merge regulatory potential scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "id": "42a3fc61-d2ed-42e2-bbc8-199264d61a23",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations = merge(ligand_target_correlations, ligand_target_matrix_long, by.x = c('Ligand_Gene', 'Target_Gene'), by.y = c('ligand', 'target'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "id": "15ec68da-5550-4e2f-b0cc-062eeb2bf4b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "3302756"
      ],
      "text/latex": [
       "3302756"
      ],
      "text/markdown": [
       "3302756"
      ],
      "text/plain": [
       "[1] 3302756"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_target_correlations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "id": "b57c0ac0-a342-4a3f-a0f2-93d0dcc29d70",
   "metadata": {},
   "outputs": [],
   "source": [
    "## scale the regulatory potential score based on max value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "id": "510d1d90-4bbe-4fb9-98b7-29886d092e63",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$regulatory_potential_scaled = ligand_target_correlations$regulatory_potential/ max(ligand_target_correlations$regulatory_potential)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6bbcfe81-c047-41ce-890c-c0752bc26ac0",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Adjustments to ligand-target correlations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "e9658dd3-77e7-41fd-8395-9906316f760d",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Remove cytokines that have been calculated as targets (because not included in ligand list above)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "id": "66250797-eb55-4a8d-8a04-f17d9d16e1a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "3302756"
      ],
      "text/latex": [
       "3302756"
      ],
      "text/markdown": [
       "3302756"
      ],
      "text/plain": [
       "[1] 3302756"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_target_correlations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "id": "7fd55f4e-4e5a-4c9d-a1ed-420c1452b3b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations = ligand_target_correlations[ligand_target_correlations$Target_Type != 'cytokine',]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "id": "328490cd-82c4-482f-9bcb-8de1381d8c18",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "3302756"
      ],
      "text/latex": [
       "3302756"
      ],
      "text/markdown": [
       "3302756"
      ],
      "text/plain": [
       "[1] 3302756"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_target_correlations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "98cb0ba0-d953-4732-97d2-ccaef93a6dfb",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Only look at between cell-type communications --> remove same cell-type Links"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "id": "4f41e8de-a4d9-4169-84dd-ee046a17b098",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Target_Cell_Type = str_replace(ligand_target_correlations$Target_Type, '.*_X\\\\d_|.*_X\\\\d\\\\d_','')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "id": "347e8379-2463-48dc-be1c-a3f777e112a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations$Ligand_Cell_Type = str_replace(ligand_target_correlations$Ligand_Type, '.*_X\\\\d_|.*_X\\\\d\\\\d_','')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "id": "ae650eda-ac47-4b64-a8c0-3c47ecc11c03",
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_correlations = ligand_target_correlations[ligand_target_correlations$Target_Cell_Type != ligand_target_correlations$Ligand_Cell_Type,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "id": "8dfa3e4f-1719-4cf9-a02d-919a8c3435a8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "2709661"
      ],
      "text/latex": [
       "2709661"
      ],
      "text/markdown": [
       "2709661"
      ],
      "text/plain": [
       "[1] 2709661"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(ligand_target_correlations)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1db9326e-d1b9-48a9-bcfc-8c4dac9e3a8b",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Save the ligand target data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "id": "3e1a1849-eeb6-4df8-ae9b-b4af8dfef178",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'V_FINAL'"
      ],
      "text/latex": [
       "'V\\_FINAL'"
      ],
      "text/markdown": [
       "'V_FINAL'"
      ],
      "text/plain": [
       "[1] \"V_FINAL\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "id": "dbcd7789-494f-47ef-a866-43bf98debd8a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'acs'"
      ],
      "text/latex": [
       "'acs'"
      ],
      "text/markdown": [
       "'acs'"
      ],
      "text/plain": [
       "[1] \"acs\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "filter_samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "id": "0e56f093-2e64-4d0b-ad60-345341f11a61",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(ligand_target_correlations,gzfile(paste0(result_path, '/F-Analysis/' ,'F4_Ligand_Target_Data', name, filter_samples,'.gz')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "56127e13-c41f-4b1f-8e13-3ba82d70fd3b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.1"
  },
  "toc-autonumbering": true
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
